<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { color: #20794d; } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007ba5; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #007ba5; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #007ba5; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #20794d; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>C057: Selection of biologically relevant cells for the Cooney (C057) memory CD4+ T-cell data set</title>
  
  
  
  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2021-04-01"/>
  <meta property="article:created" itemprop="dateCreated" content="2021-04-01"/>
  <meta name="article:author" content="Peter Hickey"/>
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="C057: Selection of biologically relevant cells for the Cooney (C057) memory CD4+ T-cell data set"/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="C057"/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="C057: Selection of biologically relevant cells for the Cooney (C057) memory CD4+ T-cell data set"/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Reference-based analysis of lung single-cell sequencing reveals a transitional profibrotic macrophage;citation_publication_date=2019;citation_volume=20;citation_author=D. Aran;citation_author=A. P. Looney;citation_author=L. Liu;citation_author=E. Wu;citation_author=V. Fong;citation_author=A. Hsu;citation_author=S. Chak;citation_author=R. P. Naikawadi;citation_author=P. J. Wolters;citation_author=A. R. Abate;citation_author=A. J. Butte;citation_author=M. Bhattacharya"/>
  <meta name="citation_reference" content="citation_title=An expression atlas of human primary cells: Inference of gene function from coexpression networks;citation_publication_date=2013;citation_volume=14;citation_doi=10.1186/1471-2164-14-632;citation_author=Neil A. Mabbott;citation_author=J. K. Baillie;citation_author=Helen Brown;citation_author=Tom C. Freeman;citation_author=David A. Hume"/>
  <meta name="citation_reference" content="citation_title=BLUEPRINT: Mapping human blood cell epigenomes;citation_publication_date=2013;citation_volume=98;citation_doi=10.3324/haematol.2013.094243;citation_author=Joost H A Martens;citation_author=Hendrik G. Stunnenberg"/>
  <meta name="citation_reference" content="citation_title=An integrated encyclopedia of dna elements in the human genome;citation_publication_date=2012;citation_doi=10.1038/nature11247;citation_issn=14764687;citation_author=An integrated encyclopedia of dna elements in the human genome"/>
  <meta name="citation_reference" content="citation_title=A single-cell molecular map of mouse gastrulation and early organogenesis;citation_publication_date=2019;citation_volume=566;citation_doi=10.1038/s41586-019-0933-9;citation_author=Blanca Pijuan-Sala;citation_author=Jonathan A Griffiths;citation_author=Carolina Guibentif;citation_author=Tom W Hiscock;citation_author=Wajid Jawaid;citation_author=Fernando J Calero-Nieto;citation_author=Carla Mulas;citation_author=Ibarra-Soria Shankar;citation_author=Benjamin D Simons;citation_author=Jennifer Nichols;citation_author=John C Marioni;citation_author=Berthold Göttgens"/>
  <meta name="citation_reference" content="citation_title=Impact of genetic polymorphisms on human immune cell gene expression;citation_publication_date=2018;citation_volume=175;citation_doi=10.1016/j.cell.2018.10.022;citation_issn=0092-8674;citation_author=Benjamin J. Schmiedel;citation_author=Divya Singh;citation_author=Ariel Madrigal;citation_author=Alan G. Valdovino-Gonzalez;citation_author=Brandie M. White;citation_author=Jose Zapardiel-Gonzalo;citation_author=Brendan Ha;citation_author=Gokmen Altay;citation_author=Jason A. Greenbaum;citation_author=Graham McVicker;citation_author=Grégory Seumois;citation_author=Anjana Rao;citation_author=Mitchell Kronenberg;citation_author=Bjoern Peters;citation_author=Pandurangan Vijayanand"/>
  <meta name="citation_reference" content="citation_title=Densely interconnected transcriptional circuits control cell states in human hematopoiesis;citation_publication_date=2011;citation_volume=144;citation_doi=10.1016/j.cell.2011.01.004;citation_issn=1097-4172;citation_author=Noa Novershtern;citation_author=Aravind Subramanian;citation_author=Lee N. Lawton;citation_author=Raymond H. Mak;citation_author=W. Nicholas Haining;citation_author=Marie E. McConkey;citation_author=Naomi Habib;citation_author=Nir Yosef;citation_author=Cindy Y. Chang;citation_author=Tal Shay;citation_author=Garrett M. Frampton;citation_author=Adam C. B. Drake;citation_author=Ilya Leskov;citation_author=Bjorn Nilsson;citation_author=Fred Preffer;citation_author=David Dombkowski;citation_author=John W. Evans;citation_author=Ted Liefeld;citation_author=John S. Smutko;citation_author=Jianzhu Chen;citation_author=Nir Friedman;citation_author=Richard A. Young;citation_author=Todd R. Golub;citation_author=Aviv Regev;citation_author=Benjamin L. Ebert"/>
  <meta name="citation_reference" content="citation_title=RNA-seq signatures normalized by mRNA abundance allow absolute deconvolution of human immune cell types;citation_publication_date=2019;citation_volume=26;citation_doi=10.1016/j.celrep.2019.01.041;citation_issn=2211-1247;citation_author=Gianni Monaco;citation_author=Bernett Lee;citation_author=Weili Xu;citation_author=Seri Mustafah;citation_author=You Yi Hwang;citation_author=Christophe Carré;citation_author=Nicolas Burdin;citation_author=Lucian Visan;citation_author=Michele Ceccarelli;citation_author=Michael Poidinger;citation_author=Alfred Zippelius;citation_author=João Pedro de Magalhães;citation_author=Anis Larbi"/>
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","date","output","editor_options","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["Selection of biologically relevant cells for the Cooney (C057) memory CD4+ T-cell data set"]},{"type":"NULL"},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url","affiliation","affiliation_url"]}},"value":[{"type":"character","attributes":{},"value":["Peter Hickey"]},{"type":"character","attributes":{},"value":["https://peterhickey.org"]},{"type":"character","attributes":{},"value":["Single Cell Open Research Endeavour (SCORE), WEHI"]},{"type":"character","attributes":{},"value":["https://www.wehi.edu.au/people/shalin-naik/3310/score"]}]}]},{"type":"character","attributes":{},"value":["2021-04-01"]},{"type":"character","attributes":{},"value":["distill::distill_article"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["chunk_output_type"]}},"value":[{"type":"character","attributes":{},"value":["console"]}]},{"type":"character","attributes":{},"value":["ref.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }
  
  </style>
  
  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>
  
  <script type="application/javascript">
  
  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }
  
  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }
  
  function createFuseIndex() {
  
    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);
  
    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });
  
        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }
  
  window.document.addEventListener("DOMContentLoaded", function (event) {
  
    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;
  
    createFuseIndex()
      .then(function(fuse) {
  
        // make search box visible
        searchEl.classList.remove('hidden');
  
        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });
  
  });
  
  </script>
  
  <style type="text/css">
  
  .nav-search {
    font-size: x-small;
  }
  
  /* Algolioa Autocomplete */
  
  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }
  
  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }
  
  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }
  
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }
  
  </style>
  
  
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }
  
  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }
  
  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }
  
  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }
  
  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }
  
  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }
  
  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }
  
  d-article h3 {
    margin-top: 1.5rem;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  /* Tweak code blocks */
  
  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }
  
  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }
  
  d-article div.sourceCode {
    background-color: white;
  }
  
  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  d-article pre a {
    border-bottom: none;
  }
  
  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }
  
  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }
  
  @media(min-width: 768px) {
  
  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }
  
  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }
  
  d-article pre {
    font-size: 14px;
  }
  
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for d-contents */
  
  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }
  
  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }
  
  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }
  
  .d-contents li {
    list-style-type: none
  }
  
  .d-contents nav > ul {
    padding-left: 0;
  }
  
  .d-contents ul {
    padding-left: 1em
  }
  
  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }
  
  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }
  
  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }
  
  .d-contents nav > ul > li > a {
    font-weight: 600;
  }
  
  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }
  
  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }
  
  
  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }
  
  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }
  
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  /* Citations */
  
  d-article .citation {
    color: inherit;
    cursor: inherit;
  }
  
  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }
  
  /* Citation hover box */
  
  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }
  
  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  /* Include appendix styles here so they can be overridden */
  
  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }
  
  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }
  
  d-appendix h3 + * {
    margin-top: 1em;
  }
  
  d-appendix ol {
    padding: 0 0 0 15px;
  }
  
  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }
  
  d-appendix li {
    margin-bottom: 1em;
  }
  
  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  d-appendix > * {
    grid-column: text;
  }
  
  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }
  
  /* Include footnote styles here so they can be overridden */
  
  d-footnote-list {
    contain: layout style;
  }
  
  d-footnote-list > * {
    grid-column: text;
  }
  
  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }
  
  
  
  /* Anchor.js */
  
  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .custom p {
    margin-bottom: 0.5em;
  }
  
  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }
  
  /* Styles for posts lists (not auto-injected) */
  
  
  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }
  
  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }
  
  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }
  
  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }
  
  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }
  
  .post-preview-last {
    border-bottom: none !important;
  }
  
  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }
  
  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }
  
  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }
  
  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }
  
  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }
  
  .posts-list .metadata > * {
    display: inline-block;
  }
  
  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }
  
  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }
  
  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }
  
  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }
  
  .posts-list img {
    opacity: 1;
  }
  
  .posts-list img[data-src] {
    opacity: 0;
  }
  
  .posts-more {
    clear: both;
  }
  
  
  .posts-sidebar {
    font-size: 16px;
  }
  
  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }
  
  .sidebar-section {
    margin-bottom: 30px;
  }
  
  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }
  
  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }
  
  .categories li>a {
    border-bottom: none;
  }
  
  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }
  
  .categories .active {
    font-weight: 600;
  }
  
  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }
  
  
  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  .downlevel .posts-list .post-preview {
    color: inherit;
  }
  
  
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }
  
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();
  
    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refHtml = $('#ref-' + $(this).attr('data-cites')).html();
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-contents').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Selection of biologically relevant cells for the Cooney (C057) memory CD4+ T-cell data set","authors":[{"author":"Peter Hickey","authorURL":"https://peterhickey.org","affiliation":"Single Cell Open Research Endeavour (SCORE), WEHI","affiliationURL":"https://www.wehi.edu.au/people/shalin-naik/3310/score","orcidID":""}],"publishedDate":"2021-04-01T00:00:00.000+11:00","citationText":"Hickey, 2021"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">C057</a>
<a href="index.html">Introduction</a>
<a href="C057_Cooney.demultiplex.html">Demultiplexing</a>
<a href="C057_Cooney.preprocess.html">Preprocessing</a>
<a href="C057_Cooney.cell_selection.html">Selecting biologically relevant cells</a>
<a href="C057_Cooney.annotate.html">Annotating cells</a>
<a href="C057_Cooney.multi-sample_comparisons.html">Differential analyses</a>
<a href="C057_Cooney.repertoire-seq.html">Immune repertoire analysis</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Supplementary files
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="../output/CellRanger/">CellRanger reports</a>
<a href="../output/marker_genes/">Marker gene tables</a>
<a href="../output/DEGs/">Differentially expressed genes</a>
<a href="../output/glimma-plots/">Glimma plots</a>
</div>
</div>
</div>
<div class="nav-right">
<a href="https://github.com/WEHISCORE/C057_Cooney">
<i class="fa fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Selection of biologically relevant cells for the Cooney (C057) memory CD4+ T-cell data set</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->

</div>

<div class="d-byline">
  Peter Hickey <a href="https://peterhickey.org" class="uri">https://peterhickey.org</a> (Single Cell Open Research Endeavour (SCORE), WEHI)<a href="https://www.wehi.edu.au/people/shalin-naik/3310/score" class="uri">https://www.wehi.edu.au/people/shalin-naik/3310/score</a>
  
<br/>2021-04-01
</div>

<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#motivation">Motivation</a><ul>
<li><a href="#clustering">Clustering</a><ul>
<li><a href="#clusters-vs.-cell-types">Clusters vs. cell types</a></li>
</ul></li>
<li><a href="#graph-based-clustering">Graph-based clustering</a></li>
</ul></li>
<li><a href="#preparing-the-data">Preparing the data</a><ul>
<li><a href="#initial-clustering">Initial clustering</a></li>
</ul></li>
<li><a href="#ambiguous-droplets"><code>ambiguous</code> droplets</a><ul>
<li><a href="#motivation-1">Motivation</a></li>
<li><a href="#analysis">Analysis</a></li>
<li><a href="#summary">Summary</a></li>
<li><a href="#re-processing">Re-processing</a></li>
</ul></li>
<li><a href="#doublets">Doublets</a><ul>
<li><a href="#motivation-2">Motivation</a></li>
<li><a href="#analysis-1">Analysis</a><ul>
<li><a href="#labelled-doublets">‘Labelled’ doublets</a></li>
<li><a href="#unlabelled-dropets">‘Unlabelled’ dropets</a></li>
<li><a href="#checking-for-removal-of-biologically-relevant-subpopulations">Checking for removal of biologically relevant subpopulations</a></li>
</ul></li>
<li><a href="#summary-1">Summary</a></li>
<li><a href="#re-processing-1">Re-processing</a></li>
</ul></li>
<li><a href="#exclusion-of-unwanted-cell-types">Exclusion of unwanted cell types</a><ul>
<li><a href="#motivation-3">Motivation</a></li>
<li><a href="#unknown-unwanted-cell-types">‘Unknown’ unwanted cell types</a><ul>
<li><a href="#motivation-4">Motivation</a></li>
<li><a href="#analysis-2">Analysis</a></li>
<li><a href="#re-processing-2">Re-processing</a></li>
</ul></li>
<li><a href="#known-unwanted-cell-types">‘Known’ unwanted cell types</a><ul>
<li><a href="#motivation-5">Motivation</a></li>
<li><a href="#analysis-3">Analysis</a></li>
</ul></li>
</ul></li>
<li><a href="#summary-6">Summary</a><ul>
<li><a href="#re-processing-3">Re-processing</a><ul>
<li><a href="#exclusion-of-ribosomal-protein-and-mitochondrial-genes-from-hvgs">Exclusion of ribosomal protein and mitochondrial genes from HVGs</a></li>
<li><a href="#visualisation">Visualisation</a></li>
</ul></li>
</ul></li>
<li><a href="#concluding-remarks">Concluding remarks</a></li>
<li><a href="#additional-information">Additional information</a></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</nav>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>SingleCellExperiment</span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://here.r-lib.org/'>here</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='http://bioconductor.org/packages/scater/'>scater</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>scran</span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://ggplot2.tidyverse.org'>ggplot2</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://wilkelab.org/cowplot/'>cowplot</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='http://bioinf.wehi.edu.au/edgeR'>edgeR</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/hasaru-k/GlimmaV2'>Glimma</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/Bioconductor/BiocParallel'>BiocParallel</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://patchwork.data-imaginist.com'>patchwork</a></span><span class='op'>)</span>

<span class='kw'><a href='https://rdrr.io/r/base/source.html'>source</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"code"</span>, <span class='st'>"helper_functions.R"</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># NOTE: Using multiple cores seizes up my laptop. Can use more on unix box.</span>
<span class='fu'><a href='https://rdrr.io/r/base/options.html'>options</a></span><span class='op'>(</span><span class='st'>"mc.cores"</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Sys.info.html'>Sys.info</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>[[</span><span class='st'>"nodename"</span><span class='op'>]</span><span class='op'>]</span> <span class='op'>==</span> <span class='st'>"PC1331"</span>, <span class='fl'>2L</span>, <span class='fl'>8L</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/pkg/BiocParallel/man/register.html'>register</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html'>MulticoreParam</a></span><span class='op'>(</span>workers <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"mc.cores"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'>knitr</span><span class='fu'>::</span><span class='va'><a href='https://rdrr.io/pkg/knitr/man/opts_chunk.html'>opts_chunk</a></span><span class='op'>$</span><span class='fu'>set</span><span class='op'>(</span>fig.path <span class='op'>=</span> <span class='st'>"C057_Cooney.cell_selection_files/"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h1 id="motivation">Motivation</h1>
<p>scRNA-seq datasets generated with 10X will include droplets that are not relevant to the study, even after the initial quality control, which we don’t want to include in downstream analyses. In this section aim to filter out these ‘unwanted’ droplets and retain only those droplets containing ‘biologically relevant’ cells. Examples of unwanted droplets include:</p>
<ul>
<li>Droplets containing stripped nuclei</li>
<li>Droplets containing more than one cell (‘doublets’)</li>
<li>Droplets without phenotype data, such as droplets for which we can’t infer the sample identify based on cell hashing or genetic variation data</li>
<li>Droplets containing unwanted cell types, such as those that might sneak through a FACS or magnetic bead enrichment sample preparation</li>
</ul>
<p>Once we are confident that we have selected the biologically relevant cells, we will perform data integration (if necessary) and a further round of clustering in preparation for downstream analysis.</p>
<p>The removal of unwanted droplets is an iterative process where at each step we:</p>
<ol type="1">
<li>Identify cluster(s) enriched for unwanted droplets. The exact criteria used to define ‘unwanted’ will depend on the type of droplets we are trying to identify at each step.
<ol type="a">
<li>Perform diagnostic checks to ensure we aren’t discarding biologically relevant droplets.</li>
<li>Remove the unwanted cells.</li>
</ol></li>
<li>Re-process the remaining droplets.
<ol type="a">
<li>Identify HVGs.</li>
<li>Perform dimensionality reduction (PCA and UMAP).</li>
<li>Cluster droplets.</li>
</ol></li>
</ol>
<p>Clustering is a critical component of this process, so we discuss it in further detail in the next subsection.</p>
<h2 id="clustering">Clustering</h2>
<p>Clustering is an unsupervised learning procedure that is used in scRNA-seq data analysis to empirically define groups of cells with similar expression profiles. Its primary purpose is to summarize the data in a digestible format for human interpretation. This allows us to describe population heterogeneity in terms of discrete labels that are easily understood, rather than attempting to comprehend the high-dimensional manifold on which the cells truly reside. Clustering is thus a critical step for extracting biological insights from scRNA-seq data.</p>
<p>Clustering calculations are usually performed using the top PCs to take advantage of data compression and denoising.</p>
<h3 id="clusters-vs.-cell-types">Clusters vs. cell types</h3>
<p>It is worth stressing the distinction between clusters and cell types. The former is an empirical construct while the latter is a biological truth (albeit a vaguely defined one). For this reason, questions like “what is the true number of clusters?” are usually meaningless. We can define as many clusters as we like, with whatever algorithm we like - each clustering will represent its own partitioning of the high-dimensional expression space, and is as “real” as any other clustering.</p>
<p>A more relevant question is “how well do the clusters approximate the cell types?” Unfortunately, this is difficult to answer given the context-dependent interpretation of biological truth. Some analysts will be satisfied with resolution of the major cell types; other analysts may want resolution of subtypes; and others still may require resolution of different states (e.g., metabolic activity, stress) within those subtypes. Two clusterings can also be highly inconsistent yet both valid, simply partitioning the cells based on different aspects of biology. Indeed, asking for an unqualified “best” clustering is akin to asking for the best magnification on a microscope without any context.</p>
<p>It is helpful to realize that clustering, like a microscope, is simply a tool to explore the data. We can zoom in and out by changing the resolution of the clustering parameters, and we can experiment with different clustering algorithms to obtain alternative perspectives of the data. This iterative approach is entirely permissible for data exploration, which constitutes the majority of all scRNA-seq data analysis.</p>
<h2 id="graph-based-clustering">Graph-based clustering</h2>
<p>We build a shared nearest neighbour graph <span class="citation" data-cites="xu2015identification">(Xu and Su <a href="#ref-xu2015identification" role="doc-biblioref">2015</a>)</span> and use the Louvain algorithm to identify clusters. We would build the graph using the principal components.</p>
<h1 id="preparing-the-data">Preparing the data</h1>
<p>We start from the preprocessed <em>SingleCellExperiment</em> object created in <a href="C057_Cooney.preprocess.html">‘Preprocessing the Cooney (C057) memory CD4+ T-cell data set’</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"SCEs"</span>, <span class='st'>"C057_Cooney.preprocessed.SCE.rds"</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># data frames containing co-ordinates and factors for creating reduced </span>
<span class='co'># dimensionality plots.</span>
<span class='va'>umap_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html'>reducedDim</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMAP"</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>,
    y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html'>reducedDim</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMAP"</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"TRA"</span>, <span class='st'>"TRB"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># Some useful colours</span>
<span class='va'>sample_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sample_colours</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sample_colours</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>treatment_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>treatment_colours</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>treatment_colours</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># Some useful gene sets</span>
<span class='va'>mito_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/which.html'>which</a></span><span class='op'>(</span><span class='fu'>rowData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>$</span><span class='va'>CHR</span> <span class='op'>==</span> <span class='st'>"MT"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>ribo_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span><span class='op'>(</span><span class='st'>"^RP(S|L)"</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>, value <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='co'># NOTE: A more curated approach for identifying ribosomal protein genes </span>
<span class='co'>#       (https://github.com/Bioconductor/OrchestratingSingleCellAnalysis-base/blob/ae201bf26e3e4fa82d9165d8abf4f4dc4b8e5a68/feature-selection.Rmd#L376-L380)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://igordot.github.io/msigdbr'>msigdbr</a></span><span class='op'>)</span>
<span class='va'>c2_sets</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://igordot.github.io/msigdbr/reference/msigdbr.html'>msigdbr</a></span><span class='op'>(</span>species <span class='op'>=</span> <span class='st'>"Homo sapiens"</span>, category <span class='op'>=</span> <span class='st'>"C2"</span><span class='op'>)</span>
<span class='va'>ribo_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/sets.html'>union</a></span><span class='op'>(</span>
  <span class='va'>ribo_set</span>,
  <span class='va'>c2_sets</span><span class='op'>[</span><span class='va'>c2_sets</span><span class='op'>$</span><span class='va'>gs_name</span> <span class='op'>==</span> <span class='st'>"KEGG_RIBOSOME"</span>, <span class='op'>]</span><span class='op'>$</span><span class='va'>human_gene_symbol</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<h2 id="initial-clustering">Initial clustering</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>75699</span><span class='op'>)</span>
<span class='va'>snn_gr</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/buildSNNGraph.html'>buildSNNGraph</a></span><span class='op'>(</span><span class='va'>sce</span>, use.dimred <span class='op'>=</span> <span class='st'>"PCA"</span>, k <span class='op'>=</span> <span class='fl'>50</span><span class='op'>)</span>
<span class='va'>clusters</span> <span class='op'>&lt;-</span> <span class='fu'>igraph</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/igraph/man/cluster_louvain.html'>cluster_louvain</a></span><span class='op'>(</span><span class='va'>snn_gr</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>clusters</span><span class='op'>$</span><span class='va'>membership</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nlevels.html'>nlevels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span> <span class='op'>==</span> <span class='fl'>7</span><span class='op'>)</span>
<span class='va'>umap_df</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span>
<span class='va'>cluster_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'>Polychrome</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/Polychrome/man/palettes.html'>glasbey.colors</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nlevels.html'>nlevels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>[</span><span class='op'>-</span><span class='fl'>1</span><span class='op'>]</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster_colours</span> <span class='op'>&lt;-</span> <span class='va'>cluster_colours</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<p>There are 7 clusters detected, shown on the UMAP plot Figure <a href="#fig:clusterplot-umap">1</a> and broken down by experimental factors in Figure <a href="#fig:cluster-barplot">2</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://wilkelab.org/cowplot/reference/plot_grid.html'>plot_grid</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>cluster</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>Treatment</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>Sample</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sample_colours</span><span class='op'>)</span>,
  ncol <span class='op'>=</span> <span class='fl'>2</span>, 
  align <span class='op'>=</span> <span class='st'>"v"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:clusterplot-umap"></span>
<img src="C057_Cooney.cell_selection_files/clusterplot-umap-1.png" alt="UMAP plot, where each point represents a droplet and is coloured according to the legend." width="624" />
<p class="caption">
Figure 1: UMAP plot, where each point represents a droplet and is coloured according to the legend.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://wilkelab.org/cowplot/reference/plot_grid.html'>plot_grid</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cluster"</span>, <span class='st'>"Sample"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>Sample</span><span class='op'>)</span>,
      position <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/position_stack.html'>position_fill</a></span><span class='op'>(</span>reverse <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Frequency"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sample_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cluster"</span>, <span class='st'>"Treatment"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>Treatment</span><span class='op'>)</span>,
      position <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/position_stack.html'>position_fill</a></span><span class='op'>(</span>reverse <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Frequency"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='st'>"cluster"</span>, drop <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>cluster</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Number of droplets"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>,
  align <span class='op'>=</span> <span class='st'>"h"</span>,
  ncol <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:cluster-barplot"></span>
<img src="C057_Cooney.cell_selection_files/cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 2: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>Notably:</p>
<ul>
<li>Most of the clusters are highly treatment-specific.</li>
<li>One of the clusters is enriched for <code>ambiguous</code> droplets.</li>
</ul>
<h1 id="ambiguous-droplets"><code>ambiguous</code> droplets</h1>
<h2 id="motivation-1">Motivation</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>prop_ambiguous</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/tapply.html'>tapply</a></span><span class='op'>(</span>
  <span class='va'>sce</span><span class='op'>$</span><span class='va'>Sample</span> <span class='op'>==</span> <span class='st'>"ambiguous"</span>,
  <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span>,
  <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/sort.html'>sort</a></span><span class='op'>(</span><span class='va'>prop_ambiguous</span><span class='op'>)</span>
<span class='va'>prop_ambiguous_outliers</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/isOutlier.html'>isOutlier</a></span><span class='op'>(</span>
  <span class='va'>prop_ambiguous</span>,
  type <span class='op'>=</span> <span class='st'>"higher"</span>,
  <span class='co'># NOTE: Want the cluster(s) that are **way** out there.</span>
  nmads <span class='op'>=</span> <span class='fl'>5</span><span class='op'>)</span>
<span class='va'>ambiguous_enriched_clusters</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span><span class='op'>[</span><span class='va'>prop_ambiguous_outliers</span><span class='op'>]</span>
<span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/identical.html'>identical</a></span><span class='op'>(</span><span class='va'>ambiguous_enriched_clusters</span>, <span class='st'>"5"</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:cluster-barplot">2</a> showed that there is one cluster with an unusually large proportion of droplets being of <code>ambiguous</code> origin, namely cluster 5 (proportion = 0.82). These <code>ambiguous</code> droplets cannot be assigned to a sample of origin and so are not useful in downstream analyses. However, before we exclude them, we want to ensure we aren’t systematically excluding a biologically relevant cell population.</p>
<p>We would further like to know that excluding the <code>ambiguous</code> droplets isn’t systematically excluding droplets from a particular sample or treatment group. This is challenging because by definition most of these droplets are of <code>ambiguous</code> origin. However, we can check if the ‘labelled’ droplets in those clusters are enriched for particular samples.</p>
<h2 id="analysis">Analysis</h2>
<p>Figure <a href="#fig:ambiguous-enriched-clusters-umap">3</a> highlights droplets from these clusters on the UMAP plot, showing that they are somewhat distinct from the bulk of droplets from the dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plot_reddim.html'>plotUMAP</a></span><span class='op'>(</span><span class='va'>sce</span>, colour_by <span class='op'>=</span> <span class='st'>"cluster"</span>, point_size <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span>, name <span class='op'>=</span> <span class='st'>"cluster"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plot_reddim.html'>plotUMAP</a></span><span class='op'>(</span>
    <span class='va'>sce</span>,
    colour_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
      `ambiguous enriched cluster` <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>%in%</span> <span class='va'>ambiguous_enriched_clusters</span><span class='op'>)</span>,
    text_by <span class='op'>=</span> <span class='st'>"cluster"</span>,
    point_size <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:ambiguous-enriched-clusters-umap"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-enriched-clusters-umap-1.png" alt="UMAP plot highlighting droplets from clusters that are enriched for `ambiguous` droplets. Cluster labels are overlaid on cluster centroids." width="624" />
<p class="caption">
Figure 3: UMAP plot highlighting droplets from clusters that are enriched for <code>ambiguous</code> droplets. Cluster labels are overlaid on cluster centroids.
</p>
</div>
</div>
<p>Figure <a href="#fig:ambiguous-adt">4</a> shows that droplets from cluster 5 have on average very few antibody-derived reads (i.e. HTOs), which is why we were unable to assign them to a <code>Sample</code> when demultiplexing the HTO data.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce</span>, 
  <span class='st'>"altexps_Antibody Capture_sum"</span>, 
  x <span class='op'>=</span> <span class='st'>"cluster"</span>, 
  colour_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    `ambiguous enriched cluster` <span class='op'>=</span> 
      <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>%in%</span> <span class='va'>ambiguous_enriched_clusters</span><span class='op'>)</span>,
  point_size <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:ambiguous-adt"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-adt-1.png" alt="Number of antibody-derived reads by cluster, highlighting droplets from clusters that are enriched for `ambiguous` droplets (orange)." width="624" />
<p class="caption">
Figure 4: Number of antibody-derived reads by cluster, highlighting droplets from clusters that are enriched for <code>ambiguous</code> droplets (orange).
</p>
</div>
</div>
<p>Figure <a href="#fig:ambiguous-enriched-clusters-qc-plot">5</a> shows that the QC metrics of droplets from cluster 5 are also distinct from the other clusters, having smaller library sizes and fewer genes detected on average. It is noteworthy that droplets in cluster 5 do not have lower mitochondrial proportions; this is notable because droplets with small library sizes and small mitochondrial proportions are hallmarks of stripped nuclei (cells that have lost their cytoplasm before encapsulation in the droplets of the 10X Chromium machine) <span class="citation" data-cites="Pijuan-Sala2019-yz">(Pijuan-Sala et al. <a href="#ref-Pijuan-Sala2019-yz" role="doc-biblioref">2019</a>)</span>. We therefore conclude that these droplets are unlikely to contain stripped nuclei.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce</span>, 
  <span class='st'>"sum"</span>, 
  x <span class='op'>=</span> <span class='st'>"cluster"</span>, 
  colour_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    `ambiguous enriched cluster` <span class='op'>=</span> 
      <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>%in%</span> <span class='va'>ambiguous_enriched_clusters</span><span class='op'>)</span>,
  point_size <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce</span>, 
  <span class='st'>"detected"</span>, 
  x <span class='op'>=</span> <span class='st'>"cluster"</span>, 
  colour_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    `ambiguous enriched cluster` <span class='op'>=</span> 
      <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>%in%</span> <span class='va'>ambiguous_enriched_clusters</span><span class='op'>)</span>,
  point_size <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='va'>p3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  <span class='st'>"subsets_Mito_percent"</span>,
  x <span class='op'>=</span> <span class='st'>"cluster"</span>,
  colour_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    `ambiguous enriched cluster` <span class='op'>=</span> 
      <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>%in%</span> <span class='va'>ambiguous_enriched_clusters</span><span class='op'>)</span>,
  point_size <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='va'>p4</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  <span class='st'>"subsets_Ribo_percent"</span>,
  x <span class='op'>=</span> <span class='st'>"cluster"</span>,
  colour_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    `ambiguous enriched cluster` <span class='op'>=</span> 
      <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>%in%</span> <span class='va'>ambiguous_enriched_clusters</span><span class='op'>)</span>,
  point_size <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='va'>p1</span> <span class='op'>+</span> <span class='va'>p2</span> <span class='op'>+</span> <span class='va'>p3</span> <span class='op'>+</span> <span class='va'>p4</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>ncol <span class='op'>=</span> <span class='fl'>2</span>, guides <span class='op'>=</span> <span class='st'>"collect"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:ambiguous-enriched-clusters-qc-plot"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-enriched-clusters-qc-plot-1.png" alt="QC metrics of droplets by cluster, highlighting droplets from clusters that are enriched for `ambiguous` droplets (orange)." width="624" />
<p class="caption">
Figure 5: QC metrics of droplets by cluster, highlighting droplets from clusters that are enriched for <code>ambiguous</code> droplets (orange).
</p>
</div>
</div>
<p>Figure <a href="#fig:ambiguous-heatmap">6</a> shows the top-20 genes more frequently detected in cluster 5 compared to all the other clusters. Aside from that the fact that there aren’t many genes consistently expressed more frequently in droplets from cluster 5, it is notable that all such genes are T cell receptor genes (<em>TRA</em> and <em>TRB</em>).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>markers_up</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/findMarkers.html'>findMarkers</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span>,
  direction <span class='op'>=</span> <span class='st'>"up"</span>,
  pval.type <span class='op'>=</span> <span class='st'>"all"</span>,
  test.type <span class='op'>=</span> <span class='st'>"binom"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHeatmap.html'>plotHeatmap</a></span><span class='op'>(</span>
  <span class='va'>sce</span>, 
  <span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>markers_up</span><span class='op'>[[</span><span class='va'>ambiguous_enriched_clusters</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span>, <span class='fl'>20</span><span class='op'>)</span>,
  order_columns_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cluster"</span>, <span class='st'>"Sample"</span><span class='op'>)</span>,
  center <span class='op'>=</span> <span class='cn'>TRUE</span>,
  zlim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='op'>-</span><span class='fl'>3</span>, <span class='fl'>3</span><span class='op'>)</span>,
  color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/palettes.html'>hcl.colors</a></span><span class='op'>(</span><span class='fl'>101</span>, <span class='st'>"Blue-Red 3"</span><span class='op'>)</span>,
  cluster_rows <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:ambiguous-heatmap"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-heatmap-1.png" alt="Heatmap of genes more frequently detected in the cluster enriched for `ambiguous` droplets." width="624" />
<p class="caption">
Figure 6: Heatmap of genes more frequently detected in the cluster enriched for <code>ambiguous</code> droplets.
</p>
</div>
</div>
<h2 id="summary">Summary</h2>
<p>To summarise, the droplets in cluster 5 are enriched for:</p>
<ul>
<li><code>ambiguous</code> droplets we could not HTO-demultiplex because they had few antibody-derived reads.</li>
<li>Droplets with smaller library sizes and fewer genes detected than the majority of droplets.</li>
<li>Droplets with higher-than-average expression of a range of T cell receptor genes.</li>
</ul>
<p>Together, this evidence strongly suggests that cluster 5 mostly comprises droplets containing ambient RNA. We can therefore safely remove all <code>ambiguous</code> droplets from cluster 5 from further analysis (<span class="math inline">\(n =\)</span> 2,533 droplets).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>remove</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>==</span> <span class='va'>ambiguous_enriched_clusters</span> <span class='op'>&amp;</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>Sample</span> <span class='op'>==</span> <span class='st'>"ambiguous"</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>[</span>, <span class='op'>!</span><span class='va'>remove</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<p>We also opt to remove the <span class="math inline">\(n =\)</span> 563 other droplets from cluster 5 (i.e. those that were able be assigned to a sample during HTO-demultiplexing). The rationale for removing these is that the transcriptomes of these droplets are consistent with droplets containing ambient RNA and the ‘HTO signal’ of these droplets may also be an artefact of ambient RNA. These droplets likely do not contain ‘real’ cells and by excluding them we can study the ‘real’ cells of interest.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>remove</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>==</span> <span class='va'>ambiguous_enriched_clusters</span> <span class='op'>&amp;</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>Sample</span> <span class='op'>!=</span> <span class='st'>"ambiguous"</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>[</span>, <span class='op'>!</span><span class='va'>remove</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<p>Figures <a href="#fig:umap-no-ambiguous-enriched-cluster">7</a> and <a href="#fig:cluster-barplot-no-ambiguous-enriched-cluster">8</a> show that the remaining <code>ambiguous</code> droplets are scattered across the other clusters. We can conclude that the remaining <code>ambiguous</code> droplets do not comprise a biologically distinct subpopulation but are rather cells that simply failed HTO-labelling or HTO-demultiplexing.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>umap_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html'>reducedDim</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMAP"</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>,
    y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html'>reducedDim</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMAP"</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"TRA"</span>, <span class='st'>"TRB"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>bg</span> <span class='op'>&lt;-</span> <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span><span class='va'>umap_df</span>, <span class='op'>-</span><span class='va'>Sample</span><span class='op'>)</span>
<span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>bg</span>, colour <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/alpha.html'>alpha</a></span><span class='op'>(</span><span class='st'>"grey"</span>, <span class='fl'>0.5</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.125</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>Sample</span><span class='op'>)</span>, alpha <span class='op'>=</span> <span class='fl'>1</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sample_colours</span>, name <span class='op'>=</span> <span class='st'>"HTO"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sample_colours</span>, name <span class='op'>=</span> <span class='st'>"HTO"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_wrap.html'>facet_wrap</a></span><span class='op'>(</span><span class='op'>~</span> <span class='va'>Sample</span>, ncol <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:umap-no-ambiguous-enriched-cluster"></span>
<img src="C057_Cooney.cell_selection_files/umap-no-ambiguous-enriched-cluster-1.png" alt="UMAP plot of the dataset. Each point represents a droplets and is coloured by `Sample`. Each panel highlights droplets from a particular sample." width="624" />
<p class="caption">
Figure 7: UMAP plot of the dataset. Each point represents a droplets and is coloured by <code>Sample</code>. Each panel highlights droplets from a particular sample.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://wilkelab.org/cowplot/reference/plot_grid.html'>plot_grid</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cluster"</span>, <span class='st'>"Sample"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>Sample</span><span class='op'>)</span>,
      position <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/position_stack.html'>position_fill</a></span><span class='op'>(</span>reverse <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Frequency"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sample_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cluster"</span>, <span class='st'>"Treatment"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>Treatment</span><span class='op'>)</span>,
      position <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/position_stack.html'>position_fill</a></span><span class='op'>(</span>reverse <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Frequency"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='st'>"cluster"</span>, drop <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>cluster</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Number of droplets"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>,
  align <span class='op'>=</span> <span class='st'>"h"</span>,
  ncol <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:cluster-barplot-no-ambiguous-enriched-cluster"></span>
<img src="C057_Cooney.cell_selection_files/cluster-barplot-no-ambiguous-enriched-cluster-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 8: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>These <code>ambiguous</code> droplets cannot be assigned to a sample of origin and so are removed from further analysis.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>[</span>, <span class='va'>sce</span><span class='op'>$</span><span class='va'>Sample</span> <span class='op'>!=</span> <span class='st'>"ambiguous"</span><span class='op'>]</span>
<span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"TRA"</span>, <span class='st'>"TRB"</span><span class='op'>)</span><span class='op'>]</span> <span class='op'>&lt;-</span> 
  <span class='fu'><a href='https://rdrr.io/r/base/droplevels.html'>droplevels</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"TRA"</span>, <span class='st'>"TRB"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>sample_colours</span> <span class='op'>&lt;-</span> <span class='va'>sample_colours</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>Sample</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>treatment_colours</span> <span class='op'>&lt;-</span> <span class='va'>treatment_colours</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>Treatment</span><span class='op'>)</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<h2 id="re-processing">Re-processing</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>623</span><span class='op'>)</span>
<span class='va'>var_fit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/modelGeneVarByPoisson.html'>modelGeneVarByPoisson</a></span><span class='op'>(</span><span class='va'>sce</span>, block <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span>
<span class='va'>hvg</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/getTopHVGs.html'>getTopHVGs</a></span><span class='op'>(</span><span class='va'>var_fit</span>, var.threshold <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>11235</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/denoisePCA.html'>denoisePCA</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='va'>var_fit</span>, subset.row <span class='op'>=</span> <span class='va'>hvg</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>8875</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/runUMAP.html'>runUMAP</a></span><span class='op'>(</span><span class='va'>sce</span>, dimred <span class='op'>=</span> <span class='st'>"PCA"</span><span class='op'>)</span>
<span class='va'>umap_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html'>reducedDim</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMAP"</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>,
    y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html'>reducedDim</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMAP"</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"TRA"</span>, <span class='st'>"TRB"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>8111</span><span class='op'>)</span>
<span class='va'>snn_gr</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/buildSNNGraph.html'>buildSNNGraph</a></span><span class='op'>(</span><span class='va'>sce</span>, use.dimred <span class='op'>=</span> <span class='st'>"PCA"</span><span class='op'>)</span>
<span class='va'>clusters</span> <span class='op'>&lt;-</span> <span class='fu'>igraph</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/igraph/man/cluster_louvain.html'>cluster_louvain</a></span><span class='op'>(</span><span class='va'>snn_gr</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>clusters</span><span class='op'>$</span><span class='va'>membership</span><span class='op'>)</span>
<span class='va'>umap_df</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span>
<span class='va'>cluster_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'>Polychrome</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/Polychrome/man/palettes.html'>glasbey.colors</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nlevels.html'>nlevels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>[</span><span class='op'>-</span><span class='fl'>1</span><span class='op'>]</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster_colours</span> <span class='op'>&lt;-</span> <span class='va'>cluster_colours</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<p>There are 11 clusters detected, shown on the UMAP plot Figure <a href="#fig:no-ambiguous-droplets-clusterplot-umap">9</a> and broken down by experimental factors in Figure <a href="#fig:no-ambiguous-droplets-cluster-barplot">10</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://wilkelab.org/cowplot/reference/plot_grid.html'>plot_grid</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>cluster</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>Treatment</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>Sample</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sample_colours</span><span class='op'>)</span>,
  ncol <span class='op'>=</span> <span class='fl'>2</span>, 
  align <span class='op'>=</span> <span class='st'>"v"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:no-ambiguous-droplets-clusterplot-umap"></span>
<img src="C057_Cooney.cell_selection_files/no-ambiguous-droplets-clusterplot-umap-1.png" alt="UMAP plot, where each point represents a droplet and is coloured according to the legend." width="624" />
<p class="caption">
Figure 9: UMAP plot, where each point represents a droplet and is coloured according to the legend.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://wilkelab.org/cowplot/reference/plot_grid.html'>plot_grid</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cluster"</span>, <span class='st'>"Sample"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>Sample</span><span class='op'>)</span>,
      position <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/position_stack.html'>position_fill</a></span><span class='op'>(</span>reverse <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Frequency"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sample_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cluster"</span>, <span class='st'>"Treatment"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>Treatment</span><span class='op'>)</span>,
      position <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/position_stack.html'>position_fill</a></span><span class='op'>(</span>reverse <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Frequency"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='st'>"cluster"</span>, drop <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>cluster</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Number of droplets"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>,
  align <span class='op'>=</span> <span class='st'>"h"</span>,
  ncol <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:no-ambiguous-droplets-cluster-barplot"></span>
<img src="C057_Cooney.cell_selection_files/no-ambiguous-droplets-cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 10: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>Notably:</p>
<ul>
<li>Still, most of the clusters are highly treatment-specific.</li>
</ul>
<h1 id="doublets">Doublets</h1>
<h2 id="motivation-2">Motivation</h2>
<p>For multiplexed samples such as these, we can identify (most of) the doublet cells based on the droplets that have multiple labels. The idea here is that cells from the same sample are labelled in a unique manner and that cells from all samples are then mixed together and the multiplexed pool is subjected to scRNA-seq, avoiding batch effects and simplifying the logistics of processing a large number of samples. Moreover, most per-cell libraries are expected to contain one label that can be used to assign that cell to its sample of origin. Cell libraries containing two labels are thus likely to be doublets of cells from different samples.</p>
<h2 id="analysis-1">Analysis</h2>
<p>We first use the HTOs to identify ‘labelled’ doublets and then leverage these in to identify ‘unlabelled’ doublets in the gene expression space.</p>
<h3 id="labelled-doublets">‘Labelled’ doublets</h3>
<p>For this experiment, the sample labelling is done with HTOs. With 5 unique HTOs, there are <span class="math inline">\({5 \choose 2} = 10\)</span> ways a droplet could be labelled with 2 HTOs. The cell hashing data will not identify all doublets, most obviously those formed between cells from the same sample. It will also be near-impossible to identify those doublets formed between a cell labelled with all 5 HTOs (i.e. a cell from the <code>uninfected_3</code> sample) and another cell labelled with a single HTO.</p>
<p>We can easily identify the HTO-derived doublets (droplets with <code>Sample</code> labelled as <code>doublet</code>; n = 414, 4.3% of remaining droplets). We term these the ‘labelled’ droplets.</p>
<h3 id="unlabelled-dropets">‘Unlabelled’ dropets</h3>
<p>One obvious limitation of relying solely on the HTO-derived doublets is that doublets of cells marked with the same HTO or those doublets formed between a cell labelled with all 5 HTOs (i.e. a cell from the <code>uninfected_3</code> sample) and another cell labelled with a single HTO are not detected<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>.</p>
<p>To avoid this, we recover the remaining intra-sample doublets based on their similarity with ‘known’ doublets in gene expression space, i.e. ‘guilt by association’.</p>
<p>For each droplet, we calculate the proportion of its nearest neighbours that are ‘labelled’ doublets<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>. Intra-sample doublets should have values of this metric under the assumption that their gene expression profiles are similar to inter-sample doublets involving the same combination of cell states/types.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># NOTE: This is a hack to estimate the relative frequency of each </span>
<span class='co'>#       (labelled singlet) sample.</span>
<span class='va'>sample_freq</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>Sample</span><span class='op'>)</span><span class='op'>[</span>
  <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"infected_"</span>, <span class='fl'>1</span><span class='op'>:</span><span class='fl'>3</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"uninfected_"</span>, <span class='fl'>1</span><span class='op'>:</span><span class='fl'>3</span><span class='op'>)</span><span class='op'>)</span><span class='op'>]</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/plger/scDblFinder'>scDblFinder</a></span><span class='op'>)</span>
<span class='va'>doublets_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scDblFinder/man/recoverDoublets.html'>recoverDoublets</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  doublets <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>Sample</span> <span class='op'>==</span> <span class='st'>"doublet"</span>,
  use.dimred <span class='op'>=</span> <span class='st'>"PCA"</span>,
  samples <span class='op'>=</span> <span class='va'>sample_freq</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:doublet-umap">11</a> highlights the location of ‘labelled’ doublets and those predicted to be doublets on the UMAP plot.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>umap_df</span><span class='op'>$</span><span class='va'>known_doublet_proportion</span> <span class='op'>&lt;-</span> <span class='va'>doublets_df</span><span class='op'>$</span><span class='va'>proportion</span>
<span class='va'>umap_df</span><span class='op'>$</span><span class='va'>known_doublet</span> <span class='op'>&lt;-</span> <span class='va'>doublets_df</span><span class='op'>$</span><span class='va'>known</span>
<span class='va'>umap_df</span><span class='op'>$</span><span class='va'>predicted_doublet</span> <span class='op'>&lt;-</span> <span class='va'>doublets_df</span><span class='op'>$</span><span class='va'>predicted</span>
<span class='fu'><a href='https://wilkelab.org/cowplot/reference/plot_grid.html'>plot_grid</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>
        colour <span class='op'>=</span> <span class='va'>known_doublet</span>,
        alpha <span class='op'>=</span> <span class='va'>known_doublet</span>,
        size <span class='op'>=</span> <span class='va'>known_doublet</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"FALSE"</span> <span class='op'>=</span> <span class='st'>"grey"</span>, <span class='st'>"TRUE"</span> <span class='op'>=</span> <span class='st'>"red"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_alpha_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"FALSE"</span> <span class='op'>=</span> <span class='fl'>0.25</span>, <span class='st'>"TRUE"</span> <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_size_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"FALSE"</span> <span class='op'>=</span> <span class='fl'>0.125</span>, <span class='st'>"TRUE"</span> <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='cn'>FALSE</span>, size <span class='op'>=</span> <span class='cn'>FALSE</span>, alpha <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Known doublets"</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>known_doublet_proportion</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.125</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_viridis.html'>scale_colour_viridis_c</a></span><span class='op'>(</span>guide <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guide_colourbar.html'>guide_colorbar</a></span><span class='op'>(</span><span class='st'>""</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Doublet proportions"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>legend.position <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0.9</span>, <span class='fl'>0.9</span><span class='op'>)</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>
        colour <span class='op'>=</span> <span class='va'>predicted_doublet</span>,
        alpha <span class='op'>=</span> <span class='va'>predicted_doublet</span>,
        size <span class='op'>=</span> <span class='va'>predicted_doublet</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"FALSE"</span> <span class='op'>=</span> <span class='st'>"grey"</span>, <span class='st'>"TRUE"</span> <span class='op'>=</span> <span class='st'>"royalblue"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_alpha_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"FALSE"</span> <span class='op'>=</span> <span class='fl'>0.25</span>, <span class='st'>"TRUE"</span> <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_size_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"FALSE"</span> <span class='op'>=</span> <span class='fl'>0.125</span>, <span class='st'>"TRUE"</span> <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='cn'>FALSE</span>, size <span class='op'>=</span> <span class='cn'>FALSE</span>, alpha <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Predicted doublets"</span><span class='op'>)</span>,
  align <span class='op'>=</span> <span class='st'>"v"</span>,
  ncol <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:doublet-umap"></span>
<img src="C057_Cooney.cell_selection_files/doublet-umap-1.png" alt="UMAP plot, where each point is a cell and is colored by the whether or not it is a 'labelled' doublet (left), by the proportion of neighbouring droplets that are 'labelled' droplets (centre), and whether or not it is a predicted 'unlabelled' doublet (right)." width="624" />
<p class="caption">
Figure 11: UMAP plot, where each point is a cell and is colored by the whether or not it is a ‘labelled’ doublet (left), by the proportion of neighbouring droplets that are ‘labelled’ droplets (centre), and whether or not it is a predicted ‘unlabelled’ doublet (right).
</p>
</div>
</div>
<p>Reassuringly, more of the ‘unlabelled’ doublets come from the droplets currently labelled as originating from the <code>uninfected_3</code> sample than any other sample, which was labelled with all 5 HTOs and for which it was correspondingly difficult to identify ‘labelled’ doublets.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/sfirke/janitor'>janitor</a></span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/pkg/janitor/man/tabyl.html'>tabyl</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>predicted_doublet <span class='op'>=</span> <span class='va'>doublets_df</span><span class='op'>$</span><span class='va'>predicted</span>, Sample <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>Sample</span><span class='op'>)</span>,
  <span class='va'>Sample</span>, <span class='va'>predicted_doublet</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_totals.html'>adorn_totals</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"row"</span>, <span class='st'>"col"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_percentages.html'>adorn_percentages</a></span><span class='op'>(</span><span class='st'>"row"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_pct_formatting.html'>adorn_pct_formatting</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_ns.html'>adorn_ns</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_title.html'>adorn_title</a></span><span class='op'>(</span>placement <span class='op'>=</span> <span class='st'>"combined"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>caption <span class='op'>=</span> <span class='st'>"Number of predicted 'unlabelled' doublets (`TRUE` column) that were labelled as coming from each `Sample`. Percentages are with respect to the total number of droplets from eahc `Sample`."</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-9">Table 1: </span>Number of predicted ‘unlabelled’ doublets (<code>TRUE</code> column) that were labelled as coming from each <code>Sample</code>. Percentages are with respect to the total number of droplets from eahc <code>Sample</code>.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">Sample/predicted_doublet</th>
<th style="text-align: left;">FALSE</th>
<th style="text-align: left;">TRUE</th>
<th style="text-align: left;">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">infected_1</td>
<td style="text-align: left;">99.3% (1338)</td>
<td style="text-align: left;">0.7% (9)</td>
<td style="text-align: left;">100.0% (1347)</td>
</tr>
<tr class="even">
<td style="text-align: left;">infected_2</td>
<td style="text-align: left;">99.1% (1165)</td>
<td style="text-align: left;">0.9% (10)</td>
<td style="text-align: left;">100.0% (1175)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">infected_3</td>
<td style="text-align: left;">99.0% (1321)</td>
<td style="text-align: left;">1.0% (14)</td>
<td style="text-align: left;">100.0% (1335)</td>
</tr>
<tr class="even">
<td style="text-align: left;">uninfected_1</td>
<td style="text-align: left;">99.3% (1836)</td>
<td style="text-align: left;">0.7% (13)</td>
<td style="text-align: left;">100.0% (1849)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">uninfected_2</td>
<td style="text-align: left;">99.3% (2107)</td>
<td style="text-align: left;">0.7% (15)</td>
<td style="text-align: left;">100.0% (2122)</td>
</tr>
<tr class="even">
<td style="text-align: left;">uninfected_3</td>
<td style="text-align: left;">98.2% (1424)</td>
<td style="text-align: left;">1.8% (26)</td>
<td style="text-align: left;">100.0% (1450)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">doublet</td>
<td style="text-align: left;">100.0% (414)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">100.0% (414)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Total</td>
<td style="text-align: left;">99.1% (9605)</td>
<td style="text-align: left;">0.9% (87)</td>
<td style="text-align: left;">100.0% (9692)</td>
</tr>
</tbody>
</table>
</div>
<h3 id="checking-for-removal-of-biologically-relevant-subpopulations">Checking for removal of biologically relevant subpopulations</h3>
<p>We check if removing these droplets would remove a biologically relevant subpopulation by looking for systematic differences in gene expression between the ‘known’ or ‘predicted’ doublets and the retained cells. If the discarded pool is enriched for a certain cell type, we should observe increased expression of the corresponding marker genes.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>keep</span> <span class='op'>&lt;-</span> <span class='op'>!</span><span class='op'>(</span><span class='va'>doublets_df</span><span class='op'>$</span><span class='va'>known</span> <span class='op'>|</span> <span class='va'>doublets_df</span><span class='op'>$</span><span class='va'>predicted</span><span class='op'>)</span>
<span class='va'>lost</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/calculateAverage.html'>calculateAverage</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/assays.html'>counts</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='op'>!</span><span class='va'>keep</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>kept</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/calculateAverage.html'>calculateAverage</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/assays.html'>counts</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='va'>keep</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>logged</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/assays.html'>cpm</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span><span class='va'>lost</span>, <span class='va'>kept</span><span class='op'>)</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span>, prior.count <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
<span class='va'>logFC</span> <span class='op'>&lt;-</span> <span class='va'>logged</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span> <span class='op'>-</span> <span class='va'>logged</span><span class='op'>[</span>, <span class='fl'>2</span><span class='op'>]</span>
<span class='va'>abundance</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>rowMeans</a></span><span class='op'>(</span><span class='va'>logged</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:doublet-md-plot">12</a> shows the result of this analysis, highlighting that there are few genes with a large <span class="math inline">\(logFC\)</span> between ‘lost’ and ‘kept’ droplets.</p>
<aside>
An interactive version of Figure <a href="#fig:doublet-md-plot">12</a> is available from <a href="../output/glimma-plots/doublet-md-plot.html"><code>output/glimma-plots/doublet-md-plot.html</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>is_mito</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='va'>mito_set</span>
<span class='va'>is_ribo</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='va'>ribo_set</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span>
  <span class='va'>abundance</span>,
  <span class='va'>logFC</span>,
  xlab <span class='op'>=</span> <span class='st'>"Average count"</span>,
  ylab <span class='op'>=</span> <span class='st'>"Log-FC (lost/kept)"</span>,
  pch <span class='op'>=</span> <span class='fl'>16</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/points.html'>points</a></span><span class='op'>(</span>
  <span class='va'>abundance</span><span class='op'>[</span><span class='va'>is_mito</span><span class='op'>]</span>,
  <span class='va'>logFC</span><span class='op'>[</span><span class='va'>is_mito</span><span class='op'>]</span>,
  col <span class='op'>=</span> <span class='st'>"dodgerblue"</span>,
  pch <span class='op'>=</span> <span class='fl'>16</span>,
  cex <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/points.html'>points</a></span><span class='op'>(</span>
  <span class='va'>abundance</span><span class='op'>[</span><span class='va'>is_ribo</span><span class='op'>]</span>,
  <span class='va'>logFC</span><span class='op'>[</span><span class='va'>is_ribo</span><span class='op'>]</span>,
  col <span class='op'>=</span> <span class='st'>"orange"</span>,
  pch <span class='op'>=</span> <span class='fl'>16</span>,
  cex <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='op'>-</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span>, col <span class='op'>=</span> <span class='st'>"red"</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:doublet-md-plot"></span>
<img src="C057_Cooney.cell_selection_files/doublet-md-plot-1.png" alt="Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate $|logFC| = 1$" width="624" />
<p class="caption">
Figure 12: Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate <span class="math inline">\(|logFC| = 1\)</span>
</p>
</div>
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/Glimma/man/glMDPlot.html'>glMDPlot</a></span><span class='op'>(</span>
  x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>abundance <span class='op'>=</span> <span class='va'>abundance</span>, logFC <span class='op'>=</span> <span class='va'>logFC</span><span class='op'>)</span>,
  xval <span class='op'>=</span> <span class='st'>"abundance"</span>, 
  yval <span class='op'>=</span> <span class='st'>"logFC"</span>,
  counts <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span><span class='va'>lost</span>, <span class='va'>kept</span><span class='op'>)</span>,
    anno <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>rowData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span>, 
      <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>GeneID <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>, stringsAsFactors <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span><span class='op'>)</span>,
  display.columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Symbol"</span>, <span class='st'>"ID"</span>, <span class='st'>"CHR"</span><span class='op'>)</span>,
  groups <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"lost"</span>, <span class='st'>"kept"</span><span class='op'>)</span><span class='op'>)</span>,
  samples <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"lost"</span>, <span class='st'>"kept"</span><span class='op'>)</span>,
  status <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/unname.html'>unname</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>is_mito</span>, <span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>is_ribo</span>, <span class='op'>-</span><span class='fl'>1</span>, <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>,
  transform <span class='op'>=</span> <span class='cn'>FALSE</span>,
  main <span class='op'>=</span> <span class='st'>"lost vs. kept"</span>,
  side.ylab <span class='op'>=</span> <span class='st'>"Average count"</span>,
  cols <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"orange"</span>, <span class='st'>"black"</span>, <span class='st'>"dodgerBlue"</span><span class='op'>)</span>,
  path <span class='op'>=</span> <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output"</span><span class='op'>)</span>,
  html <span class='op'>=</span> <span class='st'>"doublet-md-plot"</span>,
  launch <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Another concern is whether the cells removed by this procedure are preferentially derive from particular experimental groups. Reassuringly, Figure <a href="#fig:doublet-barplot-highlighting-removed-droplets">13</a> shows that this is not the case; most of the removed droplets are ‘labelled’ doublets, as we expect from the above analysis.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>Sample <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>Sample</span>, keep <span class='op'>=</span> <span class='va'>keep</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>Sample</span>, fill <span class='op'>=</span> <span class='va'>keep</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Number of droplets"</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>9</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:doublet-barplot-highlighting-removed-droplets"></span>
<img src="C057_Cooney.cell_selection_files/doublet-barplot-highlighting-removed-droplets-1.png" alt="Droplets removed after excluding 'labelled' and 'unlabelled' predicted doublets, stratified by `Sample`." width="624" />
<p class="caption">
Figure 13: Droplets removed after excluding ‘labelled’ and ‘unlabelled’ predicted doublets, stratified by <code>Sample</code>.
</p>
</div>
</div>
<h2 id="summary-1">Summary</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>keep</span> <span class='op'>&lt;-</span> <span class='op'>!</span><span class='va'>doublets_df</span><span class='op'>$</span><span class='va'>known</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>[</span>, <span class='va'>keep</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<p>The quality of the doublet predictions is hard to assess in a homogeneous dataset such as this. We therefore opt to remove only the ‘labelled’ known doublets and retain the ‘unlabelled’ predicted doublets in the dataset. The removes 414 droplets, retaining 9,278 for further analysis. The dataset may still contain some ‘unlabelled’ doublets, but their number should be very small owing to the experimental design.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"TRA"</span>, <span class='st'>"TRB"</span><span class='op'>)</span><span class='op'>]</span> <span class='op'>&lt;-</span>
  <span class='fu'><a href='https://rdrr.io/r/base/droplevels.html'>droplevels</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"TRA"</span>, <span class='st'>"TRB"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>sample_colours</span> <span class='op'>&lt;-</span> <span class='va'>sample_colours</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>Sample</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>treatment_colours</span> <span class='op'>&lt;-</span> <span class='va'>treatment_colours</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>Treatment</span><span class='op'>)</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<h2 id="re-processing-1">Re-processing</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>181</span><span class='op'>)</span>
<span class='va'>var_fit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/modelGeneVarByPoisson.html'>modelGeneVarByPoisson</a></span><span class='op'>(</span><span class='va'>sce</span>, block <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span>
<span class='va'>hvg</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/getTopHVGs.html'>getTopHVGs</a></span><span class='op'>(</span><span class='va'>var_fit</span>, var.threshold <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>11235</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/denoisePCA.html'>denoisePCA</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='va'>var_fit</span>, subset.row <span class='op'>=</span> <span class='va'>hvg</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>8875</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/runUMAP.html'>runUMAP</a></span><span class='op'>(</span><span class='va'>sce</span>, dimred <span class='op'>=</span> <span class='st'>"PCA"</span><span class='op'>)</span>
<span class='va'>umap_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html'>reducedDim</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMAP"</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>,
    y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html'>reducedDim</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMAP"</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"TRA"</span>, <span class='st'>"TRB"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>8111</span><span class='op'>)</span>
<span class='va'>snn_gr</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/buildSNNGraph.html'>buildSNNGraph</a></span><span class='op'>(</span><span class='va'>sce</span>, use.dimred <span class='op'>=</span> <span class='st'>"PCA"</span><span class='op'>)</span>
<span class='va'>clusters</span> <span class='op'>&lt;-</span> <span class='fu'>igraph</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/igraph/man/cluster_louvain.html'>cluster_louvain</a></span><span class='op'>(</span><span class='va'>snn_gr</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>clusters</span><span class='op'>$</span><span class='va'>membership</span><span class='op'>)</span>
<span class='va'>umap_df</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span>
<span class='va'>cluster_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'>Polychrome</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/Polychrome/man/palettes.html'>glasbey.colors</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nlevels.html'>nlevels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>[</span><span class='op'>-</span><span class='fl'>1</span><span class='op'>]</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster_colours</span> <span class='op'>&lt;-</span> <span class='va'>cluster_colours</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<p>There are 11 clusters detected, shown on the UMAP plot Figure <a href="#fig:no-doublets-clusterplot-umap-clusterplot-umap">14</a> and broken down by experimental factors in Figure <a href="#fig:no-doublets-clusterplot-umap-cluster-barplot">15</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://wilkelab.org/cowplot/reference/plot_grid.html'>plot_grid</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>cluster</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>Treatment</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>Sample</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sample_colours</span><span class='op'>)</span>,
  ncol <span class='op'>=</span> <span class='fl'>2</span>, 
  align <span class='op'>=</span> <span class='st'>"v"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:no-doublets-clusterplot-umap-clusterplot-umap"></span>
<img src="C057_Cooney.cell_selection_files/no-doublets-clusterplot-umap-clusterplot-umap-1.png" alt="UMAP plot, where each point represents a droplet and is coloured according to the legend." width="624" />
<p class="caption">
Figure 14: UMAP plot, where each point represents a droplet and is coloured according to the legend.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://wilkelab.org/cowplot/reference/plot_grid.html'>plot_grid</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cluster"</span>, <span class='st'>"Sample"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>Sample</span><span class='op'>)</span>,
      position <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/position_stack.html'>position_fill</a></span><span class='op'>(</span>reverse <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Frequency"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sample_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cluster"</span>, <span class='st'>"Treatment"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>Treatment</span><span class='op'>)</span>,
      position <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/position_stack.html'>position_fill</a></span><span class='op'>(</span>reverse <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Frequency"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='st'>"cluster"</span>, drop <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>cluster</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Number of droplets"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>,
  align <span class='op'>=</span> <span class='st'>"h"</span>,
  ncol <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:no-doublets-clusterplot-umap-cluster-barplot"></span>
<img src="C057_Cooney.cell_selection_files/no-doublets-clusterplot-umap-cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 15: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>Notably:</p>
<ul>
<li>Still, most of the clusters are highly treatment-specific.</li>
<li>There is a small cluster of cells that is very distinct from the majority of cells.</li>
</ul>
<h1 id="exclusion-of-unwanted-cell-types">Exclusion of unwanted cell types</h1>
<h2 id="motivation-3">Motivation</h2>
<p>James enriched for CD4+ T-cells prior to performing scRNA-seq. However, an enrichment procedure is just that and unwanted cell types may sneak through.</p>
<p>We can categorise unwanted cell types as either:</p>
<ol type="1">
<li>‘Known’ unwanted cell types: Unwanted cell types that we did know to look for <em>a priori</em> (i.e. unwanted cell types we know may have snuck through the enrichment process)</li>
<li>‘Unknown’ unwanted cell types: Unwanted cell types that we didn’t know to look for <em>a priori</em> (i.e. unwanted cell types that we didn’t may have snuck through the enrichment process).</li>
</ol>
<p>We will use different analysis strategies to identify the ‘known’ and ‘unknown’ unwanted cell types. Once we have identified the unwanted cell types, we will exclude the unwanted these cells from further analysis.</p>
<h2 id="unknown-unwanted-cell-types">‘Unknown’ unwanted cell types</h2>
<h3 id="motivation-4">Motivation</h3>
<p>A conceptually straightforward annotation approach is to compare the single-cell expression profiles with previously annotated reference data sets. Labels can then be assigned to each cell in our uncharacterised test data set based on the most similar reference sample(s), for some definition of “similar”. Any published and labelled RNA-seq data set (bulk or single-cell) can be used as a reference, though its reliability depends greatly on the expertise of the original authors who assigned the labels in the first place.</p>
<p><em><a href="https://bioconductor.org/packages/3.12/SingleR">SingleR</a></em> is one such automatic annotation method for scRNA-seq data <span class="citation" data-cites="aran2019reference">(Aran et al. <a href="#ref-aran2019reference" role="doc-biblioref">2019</a>)</span>. Given a reference data set(s) of samples<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a> with known labels, it labels new cells from a test data set based on similarity to the reference set(s). Specifically, for each test cell:</p>
<ol type="1">
<li>We compute the Spearman correlation between its expression profile and that of each reference sample.</li>
<li>We define the per-label score as a fixed quantile (by default, 0.8) of the distribution of correlations.</li>
<li>We repeat this for all labels and we take the label with the highest score as the annotation for this cell.</li>
<li>Finally, we perform a fine-tuning step:</li>
</ol>
<ul>
<li>The reference data set is subsetted to only include labels with scores close to the maximum.</li>
<li>Scores are recomputed using only marker genes for the subset of labels.</li>
<li>This is iterated until one label remains.</li>
</ul>
<p>For visualization purposes we use the normalized scores, which are the scores linearly adjusted for each cell so that the smallest score is 0 and the largest score is 1. This is followed by cubing of the adjusted scores to improve dynamic range near 1. Automatic annotation provides a convenient way of transferring biological knowledge across data sets. In this manner, the burden of interpreting clusters and defining marker genes only has to be done once (i.e. for the reference set).</p>
<p><em><a href="https://bioconductor.org/packages/3.12/SingleR">SingleR</a></em> can annotate at both the cluster-level and at the cell-level. The trade-off between cluster-level and cell-level annotation is one of increased robustness to noise (cluster-level) vs. increased resolution (cell-level). Poor-quality assignments with ‘low’ scores are labelled as <code>NA</code>.</p>
<h4 id="available-references">Available references</h4>
<p><em><a href="https://bioconductor.org/packages/3.12/SingleR">SingleR</a></em> includes several in-built reference data sets, some of which are relevant for this study:</p>
<ul>
<li><code>HumanPrimaryCellAtlasData</code> (<code>HPCA</code>): The Human Primary Cell Atlas <span class="citation" data-cites="hpcaRef">(Mabbott et al. <a href="#ref-hpcaRef" role="doc-biblioref">2013</a>)</span></li>
<li><code>BlueprintEncodeData</code> (<code>BE</code>): Blueprint <span class="citation" data-cites="blueprintRef">(Martens and Stunnenberg <a href="#ref-blueprintRef" role="doc-biblioref">2013</a>)</span> and Encode <span class="citation" data-cites="encodeRef">(The ENCODE Project Consortium <a href="#ref-encodeRef" role="doc-biblioref">2012</a>)</span></li>
<li><code>DatabaseImmuneCellExpressionData</code> (<code>DICE</code>): The <a href="https://dice-database.org/">Database of Immune Cell Expression (DICE)</a> <span class="citation" data-cites="diceRef">(Schmiedel et al. <a href="#ref-diceRef" role="doc-biblioref">2018</a>)</span></li>
<li><code>NovershternHematopoieticData</code> (<code>NH</code>): The <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE24759">Novershtern Hematopoietic Cell Data (GSE24759)</a> <span class="citation" data-cites="nhRef">(Novershtern et al. <a href="#ref-nhRef" role="doc-biblioref">2011</a>)</span></li>
<li><code>MonacoImmuneData</code> (<code>MI</code>): The <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107011">Monaco Immune Cell Data (GSE107011)</a> <span class="citation" data-cites="monaco_immuneRef">(Monaco et al. <a href="#ref-monaco_immuneRef" role="doc-biblioref">2019</a>)</span>.</li>
</ul>
<p>These bulk RNA-seq and microarray data sets were obtained from pre-sorted cell populations, i.e., the cell labels of these samples were mostly derived based on the respective sorting/purification strategy, not via <em>in silico</em> prediction methods.</p>
<p>The characteristics of each data set are summarized below<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a>:</p>
<table>
<colgroup>
<col style="width: 21%" />
<col style="width: 11%" />
<col style="width: 15%" />
<col style="width: 25%" />
<col style="width: 14%" />
<col style="width: 11%" />
</colgroup>
<thead>
<tr class="header">
<th>Reference</th>
<th>Samples</th>
<th>Sample types</th>
<th>No. of main labels</th>
<th>No. of fine labels</th>
<th>Cell type focus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>HPCA</code></td>
<td>713</td>
<td>microarrays of sorted cell populations</td>
<td>37</td>
<td>157</td>
<td>Non-specific</td>
</tr>
<tr class="even">
<td><code>BE</code></td>
<td>259</td>
<td>RNA-seq</td>
<td>24</td>
<td>43</td>
<td>Non-specific</td>
</tr>
<tr class="odd">
<td><code>DICE</code></td>
<td>1561</td>
<td>RNA-seq</td>
<td>5</td>
<td>15</td>
<td>Immune</td>
</tr>
<tr class="even">
<td><code>NH</code></td>
<td>211</td>
<td>microarrays of sorted cell populations</td>
<td>17</td>
<td>38</td>
<td>Hematopoietic &amp; Immune</td>
</tr>
<tr class="odd">
<td><code>MI</code></td>
<td>114</td>
<td>RNA-seq</td>
<td>11</td>
<td>29</td>
<td>Immune</td>
</tr>
</tbody>
</table>
<p>The available sample types in each set can be viewed in the collapsible sections below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/LTLA/SingleR'>SingleR</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/LTLA/celldex'>celldex</a></span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<details>
<p><summary><code>HumanPrimaryCellAtlasData</code> Labels</summary></p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>hpca</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleR/man/datasets.html'>HumanPrimaryCellAtlasData</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='fu'>.adf</span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>hpca</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span> 
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/count.html'>count</a></span><span class='op'>(</span><span class='va'>label.main</span>, <span class='va'>label.fine</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/arrange.html'>arrange</a></span><span class='op'>(</span><span class='va'>label.main</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Astrocyte</td>
<td style="text-align: left;">Astrocyte:Embryonic_stem_cell-derived</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:CXCR4-_centrocyte</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:CXCR4+_centroblast</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:Germinal_center</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:immature</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:Memory</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:Naive</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:Plasma_cell</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">BM</td>
<td style="text-align: left;">BM</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BM &amp; Prog.</td>
<td style="text-align: left;">BM</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">Chondrocytes</td>
<td style="text-align: left;">Chondrocytes:MSC-derived</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CMP</td>
<td style="text-align: left;">CMP</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived</td>
<td style="text-align: right;">19</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:A._fumigatus_germ_tubes_6h</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:AEC-conditioned</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:AM580</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:anti-DC-SIGN_2h</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:antiCD40/VAF347</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:CD40L</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:Galectin-1</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:immature</td>
<td style="text-align: right;">20</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:LPS</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:mature</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:Poly(IC)</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:rosiglitazone</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:rosiglitazone/AGN193109</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:Schuler_treatment</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Embryonic_stem_cells</td>
<td style="text-align: left;">Embryonic_stem_cells</td>
<td style="text-align: right;">17</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:blood_vessel</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC</td>
<td style="text-align: right;">16</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:B._anthracis_LT</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:Borrelia_burgdorferi</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:FPV-infected</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:H5N1-infected</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:IFNg</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:IL-1b</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:PR8-infected</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:Serum_Amyloid_A</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:VEGF</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:lymphatic</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:lymphatic:KSHV</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:lymphatic:TNFa_48h</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Epithelial_cells</td>
<td style="text-align: left;">Epithelial_cells:bladder</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Epithelial_cells</td>
<td style="text-align: left;">Epithelial_cells:bronchial</td>
<td style="text-align: right;">10</td>
</tr>
<tr class="even">
<td style="text-align: left;">Erythroblast</td>
<td style="text-align: left;">Erythroblast</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Fibroblasts</td>
<td style="text-align: left;">Fibroblasts:breast</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fibroblasts</td>
<td style="text-align: left;">Fibroblasts:foreskin</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gametocytes</td>
<td style="text-align: left;">Gametocytes:oocyte</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gametocytes</td>
<td style="text-align: left;">Gametocytes:spermatocyte</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GMP</td>
<td style="text-align: left;">GMP</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Hepatocytes</td>
<td style="text-align: left;">Hepatocytes</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSC_-G-CSF</td>
<td style="text-align: left;">HSC_-G-CSF</td>
<td style="text-align: right;">10</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSC_CD34+</td>
<td style="text-align: left;">HSC_CD34+</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:adipose_stem_cell-derived:lentiviral</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:adipose_stem_cell-derived:minicircle-derived</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:adipose_stem_cells</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:CRL2097_foreskin</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:CRL2097_foreskin-derived:d20_hepatic_diff</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:CRL2097_foreskin-derived:undiff.</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:fibroblast-derived:Direct_del._reprog</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:fibroblast-derived:Retroviral_transf</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:fibroblasts</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:foreskin_fibrobasts</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:iPS:minicircle-derived</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_1lox-17Puro-10</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_1lox-17Puro-5</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_1lox-21Puro-20</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_1lox-21Puro-26</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_2lox-17</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_2lox-21</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_2lox-22</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_2lox-5</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_fibroblasts</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:skin_fibroblast</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:skin_fibroblast-derived</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IFNg</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IL19</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IL1b</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IL20</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IL22</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IL24</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IL26</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:KGF</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:Alveolar</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:Alveolar:B._anthacis_spores</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived</td>
<td style="text-align: right;">26</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:IFNa</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:IL-4/cntrl</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:IL-4/Dex/cntrl</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:IL-4/Dex/TGFb</td>
<td style="text-align: right;">10</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:IL-4/TGFb</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:M-CSF</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:M-CSF/IFNg</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:M-CSF/IFNg/Pam3Cys</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:M-CSF/Pam3Cys</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:S._aureus</td>
<td style="text-align: right;">15</td>
</tr>
<tr class="odd">
<td style="text-align: left;">MEP</td>
<td style="text-align: left;">MEP</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte</td>
<td style="text-align: right;">27</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:anti-FcgRIIB</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:CD14+</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:CD16-</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:CD16+</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:CXCL4</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:F._tularensis_novicida</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:leukotriene_D4</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:MCSF</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:S._typhimurium_flagellin</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">MSC</td>
<td style="text-align: left;">MSC</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Myelocyte</td>
<td style="text-align: left;">Myelocyte</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neuroepithelial_cell</td>
<td style="text-align: left;">Neuroepithelial_cell:ESC-derived</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neurons</td>
<td style="text-align: left;">Neurons:adrenal_medulla_cell_line</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neurons</td>
<td style="text-align: left;">Neurons:ES_cell-derived_neural_precursor</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neurons</td>
<td style="text-align: left;">Neurons:Schwann_cell</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophil</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophil:commensal_E._coli_MG1655</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophil:GM-CSF_IFNg</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophil:inflam</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophil:LPS</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophil:uropathogenic_E._coli_UTI89</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK_cell</td>
<td style="text-align: left;">NK_cell</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK_cell</td>
<td style="text-align: left;">NK_cell:CD56hiCD62L+</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK_cell</td>
<td style="text-align: left;">NK_cell:IL2</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Osteoblasts</td>
<td style="text-align: left;">Osteoblasts</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="even">
<td style="text-align: left;">Osteoblasts</td>
<td style="text-align: left;">Osteoblasts:BMP2</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Platelets</td>
<td style="text-align: left;">Platelets</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pre-B_cell_CD34-</td>
<td style="text-align: left;">Pre-B_cell_CD34-</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Pro-B_cell_CD34+</td>
<td style="text-align: left;">Pro-B_cell_CD34+</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pro-Myelocyte</td>
<td style="text-align: left;">Pro-Myelocyte</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Smooth_muscle_cells</td>
<td style="text-align: left;">Smooth_muscle_cells:bronchial</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Smooth_muscle_cells</td>
<td style="text-align: left;">Smooth_muscle_cells:bronchial:vit_D</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Smooth_muscle_cells</td>
<td style="text-align: left;">Smooth_muscle_cells:umbilical_vein</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Smooth_muscle_cells</td>
<td style="text-align: left;">Smooth_muscle_cells:vascular</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Smooth_muscle_cells</td>
<td style="text-align: left;">Smooth_muscle_cells:vascular:IL-17</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CCR10-CLA+1,25(OH)2_vit_D3/IL-12</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CCR10+CLA+1,25(OH)2_vit_D3/IL-12</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD4+</td>
<td style="text-align: right;">12</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD4+_central_memory</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD4+_effector_memory</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD4+_Naive</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD8+</td>
<td style="text-align: right;">16</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD8+_Central_memory</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD8+_effector_memory</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD8+_effector_memory_RA</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD8+_naive</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:effector</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:gamma-delta</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:Treg:Naive</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:adipose-derived_MSC_AM3</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:BM_MSC</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:BM_MSC:BMP2</td>
<td style="text-align: right;">12</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:BM_MSC:osteogenic</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:BM_MSC:TGFb3</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:CD326-CD56+</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:dental_pulp</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:iliac_MSC</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:lipoma-derived_MSC</td>
<td style="text-align: right;">2</td>
</tr>
</tbody>
</table>
</div>
</details>
<details>
<p><summary><code>BlueprintEncodeData</code> Labels</summary></p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>be</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleR/man/datasets.html'>BlueprintEncodeData</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='fu'>.adf</span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>be</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span> 
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/count.html'>count</a></span><span class='op'>(</span><span class='va'>label.main</span>, <span class='va'>label.fine</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/arrange.html'>arrange</a></span><span class='op'>(</span><span class='va'>label.main</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Adipocytes</td>
<td style="text-align: left;">Adipocytes</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">Adipocytes</td>
<td style="text-align: left;">Preadipocytes</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">B-cells</td>
<td style="text-align: left;">Class-switched memory B-cells</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B-cells</td>
<td style="text-align: left;">Memory B-cells</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">B-cells</td>
<td style="text-align: left;">naive B-cells</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B-cells</td>
<td style="text-align: left;">Plasma cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T-cells</td>
<td style="text-align: left;">CD4+ T-cells</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T-cells</td>
<td style="text-align: left;">CD4+ Tcm</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T-cells</td>
<td style="text-align: left;">CD4+ Tem</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T-cells</td>
<td style="text-align: left;">Tregs</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T-cells</td>
<td style="text-align: left;">CD8+ T-cells</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8+ T-cells</td>
<td style="text-align: left;">CD8+ Tcm</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T-cells</td>
<td style="text-align: left;">CD8+ Tem</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Chondrocytes</td>
<td style="text-align: left;">Chondrocytes</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial cells</td>
<td style="text-align: left;">Endothelial cells</td>
<td style="text-align: right;">18</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial cells</td>
<td style="text-align: left;">mv Endothelial cells</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Eosinophils</td>
<td style="text-align: left;">Eosinophils</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">Epithelial cells</td>
<td style="text-align: left;">Epithelial cells</td>
<td style="text-align: right;">18</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Erythrocytes</td>
<td style="text-align: left;">Erythrocytes</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fibroblasts</td>
<td style="text-align: left;">Fibroblasts</td>
<td style="text-align: right;">20</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">CLP</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">CMP</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">GMP</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">HSC</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">Megakaryocytes</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">MEP</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">MPP</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophages</td>
<td style="text-align: left;">Macrophages</td>
<td style="text-align: right;">18</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophages</td>
<td style="text-align: left;">Macrophages M1</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophages</td>
<td style="text-align: left;">Macrophages M2</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Melanocytes</td>
<td style="text-align: left;">Melanocytes</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mesangial cells</td>
<td style="text-align: left;">Mesangial cells</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Monocytes</td>
<td style="text-align: right;">16</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Myocytes</td>
<td style="text-align: left;">Myocytes</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neurons</td>
<td style="text-align: left;">Neurons</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: right;">23</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Pericytes</td>
<td style="text-align: left;">Pericytes</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Skeletal muscle</td>
<td style="text-align: left;">Skeletal muscle</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Smooth muscle</td>
<td style="text-align: left;">Smooth muscle</td>
<td style="text-align: right;">16</td>
</tr>
</tbody>
</table>
</div>
</details>
<details>
<p><summary><code>DatabaseImmuneCellExpressionData</code> Labels</summary></p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>dice</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleR/man/datasets.html'>DatabaseImmuneCellExpressionData</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='fu'>.adf</span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>dice</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span> 
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/count.html'>count</a></span><span class='op'>(</span><span class='va'>label.main</span>, <span class='va'>label.fine</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/arrange.html'>arrange</a></span><span class='op'>(</span><span class='va'>label.main</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">B cells, naive</td>
<td style="text-align: right;">106</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Monocytes, CD14+</td>
<td style="text-align: right;">106</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Monocytes, CD16+</td>
<td style="text-align: right;">105</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">105</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, memory TREG</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, naive</td>
<td style="text-align: right;">103</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, naive TREG</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, naive, stimulated</td>
<td style="text-align: right;">102</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, TFH</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, Th1</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, Th1_17</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, Th17</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, Th2</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD8+</td>
<td style="text-align: left;">T cells, CD8+, naive</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD8+</td>
<td style="text-align: left;">T cells, CD8+, naive, stimulated</td>
<td style="text-align: right;">102</td>
</tr>
</tbody>
</table>
</div>
</details>
<details>
<p><summary><code>MonacoImmuneData</code> Labels</summary></p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>mi</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleR/man/datasets.html'>MonacoImmuneData</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='fu'>.adf</span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>mi</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span> 
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/count.html'>count</a></span><span class='op'>(</span><span class='va'>label.main</span>, <span class='va'>label.fine</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/arrange.html'>arrange</a></span><span class='op'>(</span><span class='va'>label.main</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Exhausted B cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Naive B cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Non-switched memory B cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Plasmablasts</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Switched memory B cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Basophils</td>
<td style="text-align: left;">Low-density basophils</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Follicular helper T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Naive CD4 T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">T regulatory cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Terminal effector CD4 T cells</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Th1 cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Th1/Th17 cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Th17 cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Th2 cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">Central memory CD8 T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">Effector memory CD8 T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">Naive CD8 T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">Terminal effector CD8 T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Dendritic cells</td>
<td style="text-align: left;">Myeloid dendritic cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Dendritic cells</td>
<td style="text-align: left;">Plasmacytoid dendritic cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Classical monocytes</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Intermediate monocytes</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Non classical monocytes</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Low-density neutrophils</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">Natural killer cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Progenitors</td>
<td style="text-align: left;">Progenitor cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells</td>
<td style="text-align: left;">MAIT cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells</td>
<td style="text-align: left;">Non-Vd2 gd T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells</td>
<td style="text-align: left;">Vd2 gd T cells</td>
<td style="text-align: right;">4</td>
</tr>
</tbody>
</table>
</div>
</details>
<details>
<p><summary><code>NovershternHematopoieticData</code> Labels</summary></p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>nh</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleR/man/datasets.html'>NovershternHematopoieticData</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='fu'>.adf</span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>nh</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span> 
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/count.html'>count</a></span><span class='op'>(</span><span class='va'>label.main</span>, <span class='va'>label.fine</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/arrange.html'>arrange</a></span><span class='op'>(</span><span class='va'>label.main</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Early B cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Mature B cells</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Mature B cells class able to switch</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Mature B cells class switched</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Naive B cells</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Pro B cells</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Basophils</td>
<td style="text-align: left;">Basophils</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">CD4+ Central Memory</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">CD4+ Effector Memory</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Naive CD4+ T cells</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">CD8+ Central Memory</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">CD8+ Effector Memory</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">CD8+ Effector Memory RA</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">Naive CD8+ T cells</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CMPs</td>
<td style="text-align: left;">Common myeloid progenitors</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Dendritic cells</td>
<td style="text-align: left;">Myeloid Dendritic Cells</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Dendritic cells</td>
<td style="text-align: left;">Plasmacytoid Dendritic Cells</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">Eosinophils</td>
<td style="text-align: left;">Eosinophils</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Erythroid cells</td>
<td style="text-align: left;">Erythroid_CD34- CD71- GlyA+</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">Erythroid cells</td>
<td style="text-align: left;">Erythroid_CD34- CD71+ GlyA-</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Erythroid cells</td>
<td style="text-align: left;">Erythroid_CD34- CD71+ GlyA+</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">Erythroid cells</td>
<td style="text-align: left;">Erythroid_CD34- CD71lo GlyA+</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Erythroid cells</td>
<td style="text-align: left;">Erythroid_CD34+ CD71+ GlyA-</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">GMPs</td>
<td style="text-align: left;">Granulocyte/monocyte progenitors</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Granulocytes</td>
<td style="text-align: left;">Colony Forming Unit-Granulocytes</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">Granulocytes</td>
<td style="text-align: left;">Granulocytes (Neutrophilic Metamyelocytes)</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Granulocytes</td>
<td style="text-align: left;">Granulocytes (Neutrophils)</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSCs</td>
<td style="text-align: left;">Hematopoietic stem cells_CD133+ CD34dim</td>
<td style="text-align: right;">10</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSCs</td>
<td style="text-align: left;">Hematopoietic stem cells_CD38- CD34+</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Megakaryocytes</td>
<td style="text-align: left;">Colony Forming Unit-Megakaryocytic</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Megakaryocytes</td>
<td style="text-align: left;">Megakaryocytes</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">MEPs</td>
<td style="text-align: left;">Megakaryocyte/erythroid progenitors</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Colony Forming Unit-Monocytes</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Monocytes</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">Mature NK cells_CD56- CD16- CD3-</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">Mature NK cells_CD56- CD16+ CD3-</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">Mature NK cells_CD56+ CD16+ CD3-</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK T cells</td>
<td style="text-align: left;">NK T cells</td>
<td style="text-align: right;">4</td>
</tr>
</tbody>
</table>
</div>
</details>
<h3 id="analysis-2">Analysis</h3>
<p>For simplicity, I have opted to only use the ‘main’ labels from the <code>DICE</code> reference annotation and annotated the dataset at the cluster-level. The ‘main’ labels provide broad cell type labels appropriate for identifying clusters of an ‘unwanted’ cell type.</p>
<aside>
This reference was recommended to me by colleagues with experience in using <em><a href="https://bioconductor.org/packages/3.12/SingleR">SingleR</a></em> to annotate T-cells profiled with scRNA-seq.
</aside>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ref</span> <span class='op'>&lt;-</span> <span class='va'>dice</span>
<span class='va'>labels_main</span> <span class='op'>&lt;-</span> <span class='va'>ref</span><span class='op'>$</span><span class='va'>label.main</span>
<span class='co'># NOTE: This code doesn't necessarily generalise beyond the DICE main labels.</span>
<span class='va'>label_main_collapsed_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
    <span class='fu'>Polychrome</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/Polychrome/man/palettes.html'>glasbey.colors</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nlevels.html'>nlevels</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>labels_main</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>[</span><span class='op'>-</span><span class='fl'>1</span><span class='op'>]</span>,
    <span class='st'>"orange"</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>labels_main</span><span class='op'>)</span><span class='op'>)</span>, <span class='st'>"other"</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>pred_cluster_main</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleR/man/SingleR.html'>SingleR</a></span><span class='op'>(</span>
  test <span class='op'>=</span> <span class='va'>sce</span>,
  ref <span class='op'>=</span> <span class='va'>ref</span>,
  labels <span class='op'>=</span> <span class='va'>labels_main</span>,
  cluster <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span>,
  BPPARAM <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocParallel/man/register.html'>bpparam</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cluster_main</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>pred_cluster_main</span><span class='op'>$</span><span class='va'>pruned.labels</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cluster_main_collapsed</span> <span class='op'>&lt;-</span> <span class='fu'>.collapseLabel</span><span class='op'>(</span>
  <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cluster_main</span>,
  <span class='va'>sce</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>label_main_collapsed_colours</span> <span class='op'>&lt;-</span> <span class='va'>label_main_collapsed_colours</span><span class='op'>[</span>
  <span class='fu'><a href='https://rdrr.io/r/base/character.html'>as.character</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cluster_main</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>umap_df</span><span class='op'>$</span><span class='va'>label_cluster_main_collapsed</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cluster_main_collapsed</span>
</code></pre>
</div>
</details>
</div>
<p>The results are summarised in the table below.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/janitor/man/tabyl.html'>tabyl</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>label.main <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cluster_main</span>, cluster <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span>, 
  <span class='va'>cluster</span>, 
  <span class='va'>label.main</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>"Cluster-level assignments using the main labels of the DICE reference."</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-22">Table 2: </span>Cluster-level assignments using the main labels of the DICE reference.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">cluster</th>
<th style="text-align: right;">Monocytes</th>
<th style="text-align: right;">T cells, CD4+</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">916</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">699</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1775</td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">278</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">649</td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1033</td>
</tr>
<tr class="odd">
<td style="text-align: left;">7</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">142</td>
</tr>
<tr class="even">
<td style="text-align: left;">8</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1371</td>
</tr>
<tr class="odd">
<td style="text-align: left;">9</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1388</td>
</tr>
<tr class="even">
<td style="text-align: left;">10</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">880</td>
</tr>
<tr class="odd">
<td style="text-align: left;">11</td>
<td style="text-align: right;">147</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
<p>Figure <a href="#fig:umap-with-main-cluster-labels">16</a> overlays these cell type labels on the UMAP plot and shows that the aforementioned small, distinct cluster is labelled as <code>Monocytes</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>cluster</span><span class='op'>)</span>,
    alpha <span class='op'>=</span> <span class='fl'>1</span>,
    size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Clusters"</span><span class='op'>)</span>
<span class='va'>bg</span> <span class='op'>&lt;-</span> <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span><span class='va'>umap_df</span>, <span class='op'>-</span><span class='va'>label_cluster_main_collapsed</span><span class='op'>)</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>bg</span>, colour <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/alpha.html'>alpha</a></span><span class='op'>(</span><span class='st'>"grey"</span>, <span class='fl'>0.5</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.125</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>label_cluster_main_collapsed</span><span class='op'>)</span>,
    alpha <span class='op'>=</span> <span class='fl'>1</span>,
    size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>label_main_collapsed_colours</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>label_main_collapsed_colours</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_wrap.html'>facet_wrap</a></span><span class='op'>(</span><span class='op'>~</span> <span class='va'>label_cluster_main_collapsed</span>, ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"'Main' cluster-level label"</span><span class='op'>)</span>

<span class='va'>p1</span> <span class='op'>+</span> <span class='va'>p2</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>widths <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:umap-with-main-cluster-labels"></span>
<img src="C057_Cooney.cell_selection_files/umap-with-main-cluster-labels-1.png" alt="UMAP plot highlighting clusters (left) and 'main' cluster-level labels (right) where each panel highlights droplets from a particular label. Labels with &lt; 1% frequency are grouped together as `other`." width="624" />
<p class="caption">
Figure 16: UMAP plot highlighting clusters (left) and ‘main’ cluster-level labels (right) where each panel highlights droplets from a particular label. Labels with &lt; 1% frequency are grouped together as <code>other</code>.
</p>
</div>
</div>
<h5 id="diagnostic-plots">Diagnostic plots</h5>
<p>As a sanity check, we can examine the expression of the marker genes for the relevant cell type labels by plotting a heatmap of their expression in:</p>
<ol type="1">
<li>The reference dataset</li>
<li>Our dataset</li>
</ol>
<p>The value of (1) is that we can assess if we believe the genes are indeed good markers of the relevant cell type in the reference dataset. The value of (2) is that we can check that these genes are useful markers in our dataset (e.g., that they are reasonably well sampled in our data).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>monocyte_vs_cd4_markers</span> <span class='op'>&lt;-</span> <span class='fu'>metadata</span><span class='op'>(</span>
  <span class='va'>pred_cluster_main</span><span class='op'>)</span><span class='op'>$</span><span class='va'>de.genes</span><span class='op'>$</span><span class='va'>`Monocytes`</span><span class='op'>$</span><span class='va'>`T cells, CD4+`</span>
<span class='va'>cd4_vs_monocyte_markers</span> <span class='op'>&lt;-</span> <span class='fu'>metadata</span><span class='op'>(</span>
  <span class='va'>pred_cluster_main</span><span class='op'>)</span><span class='op'>$</span><span class='va'>de.genes</span><span class='op'>$</span><span class='va'>`T cells, CD4+`</span><span class='op'>$</span><span class='va'>`Monocytes`</span>
<span class='va'>markers</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>monocyte_vs_cd4_markers</span>, <span class='fl'>10</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>cd4_vs_monocyte_markers</span>, <span class='fl'>10</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Here, we specifically select (some of) the most strongly upregulated genes when comparing the <code>Monocytes</code> to the <code>T cells, CD4+</code> (<code>dice.Mono_vs_CD4</code>) and vice-versa (<code>dice.CD4_vs_Mono</code>)</p>
<p>Figure <a href="#fig:ref-dice-mono-and-cd4-markers">17</a> confirms that both the <code>dice.Mono_vs_CD4</code> and <code>dice.CD4_vs_Mono</code> marker genes distinguish these two cell types from one another in the <code>DICE</code> reference dataset. However, the <code>dice.CD4_vs_Mono</code> marker genes are also expressed in <code>T cells, CD8+</code> samples, highlighting that what are useful marker genes in one comparison are not necessarily in another comparison.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># NOTE: Have to remove column names from DICE to avoid an error.</span>
<span class='va'>tmp</span> <span class='op'>&lt;-</span> <span class='va'>dice</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>tmp</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_len</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>ncol</a></span><span class='op'>(</span><span class='va'>tmp</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHeatmap.html'>plotHeatmap</a></span><span class='op'>(</span>
  <span class='va'>tmp</span>,
  features <span class='op'>=</span> <span class='va'>markers</span>,
  colour_columns_by <span class='op'>=</span> <span class='st'>"label.main"</span>,
  center <span class='op'>=</span> <span class='cn'>TRUE</span>,
  symmetric <span class='op'>=</span> <span class='cn'>TRUE</span>,
  order_columns_by <span class='op'>=</span> <span class='st'>"label.main"</span>,
  cluster_rows <span class='op'>=</span> <span class='cn'>FALSE</span>,
  cluster_cols <span class='op'>=</span> <span class='cn'>FALSE</span>,
  annotation_row <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    markers <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"dice.Mono_vs_CD4"</span>, <span class='fl'>10</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"dice.CD4_vs_Mono"</span>, <span class='fl'>10</span><span class='op'>)</span><span class='op'>)</span>,
    row.names <span class='op'>=</span> <span class='va'>markers</span><span class='op'>)</span>,
  color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/palettes.html'>hcl.colors</a></span><span class='op'>(</span><span class='fl'>101</span>, <span class='st'>"Blue-Red 3"</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:ref-dice-mono-and-cd4-markers"></span>
<img src="C057_Cooney.cell_selection_files/ref-dice-mono-and-cd4-markers-1.png" alt="Heatmap of log-expression values in the `DICE` reference dataset for selected marker genes between the `Monocytes` and `T cells, CD4+` labels. Each column is a sample, each row a gene" width="624" />
<p class="caption">
Figure 17: Heatmap of log-expression values in the <code>DICE</code> reference dataset for selected marker genes between the <code>Monocytes</code> and <code>T cells, CD4+</code> labels. Each column is a sample, each row a gene
</p>
</div>
</div>
<p>Figure <a href="#fig:test-dice-mono-and-cd4-markers">18</a> shows that both the <code>dice.Mono_vs_CD4</code> and <code>dice.CD4_vs_Mono</code> marker genes marker genes are indeed very specific to their respective clusters in our dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># NOTE: Alternatively, could plot the cluster-level data.</span>
<span class='co'># tmp &lt;- logNormCounts(</span>
<span class='co'>#   sumCountsAcrossCells(</span>
<span class='co'>#     sce,</span>
<span class='co'>#     ids = colData(sce)[, c("cluster", "label_cluster_main")],</span>
<span class='co'>#     subset_row = markers), exprs_values = 1)</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>561</span><span class='op'>)</span>
<span class='va'>tmp</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span>
  <span class='va'>sce</span><span class='op'>[</span>, <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cluster_main</span> <span class='op'>==</span> <span class='st'>"Monocytes"</span><span class='op'>]</span>,
  <span class='va'>sce</span><span class='op'>[</span>, <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cluster_main</span> <span class='op'>!=</span> <span class='st'>"Monocytes"</span><span class='op'>]</span><span class='op'>[</span>
    , <span class='fu'><a href='https://rdrr.io/r/base/sample.html'>sample</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cluster_main</span> <span class='op'>!=</span> <span class='st'>"Monocytes"</span><span class='op'>)</span>,
      <span class='fl'>0.1</span> <span class='op'>*</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cluster_main</span> <span class='op'>!=</span> <span class='st'>"Monocytes"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHeatmap.html'>plotHeatmap</a></span><span class='op'>(</span>
  <span class='va'>tmp</span>,
  features <span class='op'>=</span> <span class='va'>markers</span>,
  colour_columns_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"label_cluster_main"</span>, <span class='st'>"cluster"</span><span class='op'>)</span>,
  center <span class='op'>=</span> <span class='cn'>TRUE</span>,
  symmetric <span class='op'>=</span> <span class='cn'>TRUE</span>,
  order_columns_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"label_cluster_main"</span>, <span class='st'>"cluster"</span><span class='op'>)</span>,
  cluster_rows <span class='op'>=</span> <span class='cn'>FALSE</span>,
  cluster_cols <span class='op'>=</span> <span class='cn'>FALSE</span>,
  annotation_row <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    markers <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"dice.Mono_vs_CD4"</span>, <span class='fl'>10</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"dice.CD4_vs_Mono"</span>, <span class='fl'>10</span><span class='op'>)</span><span class='op'>)</span>,
    row.names <span class='op'>=</span> <span class='va'>markers</span><span class='op'>)</span>,
  color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/palettes.html'>hcl.colors</a></span><span class='op'>(</span><span class='fl'>101</span>, <span class='st'>"Blue-Red 3"</span><span class='op'>)</span>,<span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:test-dice-mono-and-cd4-markers"></span>
<img src="C057_Cooney.cell_selection_files/test-dice-mono-and-cd4-markers-1.png" alt="Heatmap of log-expression values in our dataset at the cell-level for selected marker genes between the `Monocytes` and `T cells, CD4+` labels. Each column is a sample, each row a gene. For legibility, only a random 10% of non-`Monocytes` cells are shown." width="624" />
<p class="caption">
Figure 18: Heatmap of log-expression values in our dataset at the cell-level for selected marker genes between the <code>Monocytes</code> and <code>T cells, CD4+</code> labels. Each column is a sample, each row a gene. For legibility, only a random 10% of non-<code>Monocytes</code> cells are shown.
</p>
</div>
</div>
<h4 id="expression-of-t-cell-receptor-genes">Expression of T cell receptor genes</h4>
<p>Figure <a href="#fig:tcr-proportion">19</a> shows that cells in the ‘monocyte’ cluster have few counts from the T cell receptor genes, further evidence that the cells in this cluster are indeed not T cells.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>is_tcr</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/any.html'>any</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grepl</a></span><span class='op'>(</span><span class='st'>"T cell receptor"</span>, <span class='fu'>rowData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>$</span><span class='va'>NCBI.GENENAME</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/addPerCellQC.html'>addPerCellQC</a></span><span class='op'>(</span><span class='va'>sce</span>, subsets <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>tcr <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/which.html'>which</a></span><span class='op'>(</span><span class='va'>is_tcr</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  <span class='st'>"subsets_tcr_percent"</span>,
  x <span class='op'>=</span> <span class='st'>"cluster"</span>,
  colour_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    Monocyte_cluster <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cluster_main</span> <span class='op'>==</span> <span class='st'>"Monocytes"</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:tcr-proportion"></span>
<img src="C057_Cooney.cell_selection_files/tcr-proportion-1.png" alt="Proportion of counts from T cell receptor genes stratified by cluster." width="624" />
<p class="caption">
Figure 19: Proportion of counts from T cell receptor genes stratified by cluster.
</p>
</div>
</div>
<h4 id="checking-for-enrichment-of-cells-in-experimental-group">Checking for enrichment of cells in experimental group</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>keep</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cluster_main_collapsed</span> <span class='op'>!=</span> <span class='st'>"Monocytes"</span>
</code></pre>
</div>
</details>
</div>
<p>We again check whether the cells removed by this procedure are preferentially derive from particular experimental groups. Figure <a href="#fig:monocytes-highlighting-removed-droplets">20</a> shows that excluding these cells will preferentially exclude cells from the <code>Uninfected</code> samples.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>Sample <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>Sample</span>, keep <span class='op'>=</span> <span class='va'>keep</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>Sample</span>, fill <span class='op'>=</span> <span class='va'>keep</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Number of droplets"</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>9</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>axis.text.x <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_text</a></span><span class='op'>(</span>angle <span class='op'>=</span> <span class='fl'>45</span>, hjust <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:monocytes-highlighting-removed-droplets"></span>
<img src="C057_Cooney.cell_selection_files/monocytes-highlighting-removed-droplets-1.png" alt="Droplets removed after excluding droplets from clusters labelled as `Monocytes`, stratified by `Sample`." width="624" />
<p class="caption">
Figure 20: Droplets removed after excluding droplets from clusters labelled as <code>Monocytes</code>, stratified by <code>Sample</code>.
</p>
</div>
</div>
<h4 id="summary-2">Summary</h4>
<p>We opt to remove cells from clusters labelled as <code>Monocytes</code>. The removes 147 droplets, retaining 9131 for further analysis. This will provide better resolution for us to analyse the remaining cells, which are nominally the CD4+ T-cells of interest.</p>
<aside>
If it is biologically relevant that the <code>Monocytes</code> are only found in <code>Uninfected</code> then we would need to retain these cells for downstream analysis.
</aside>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>[</span>, <span class='va'>keep</span><span class='op'>]</span>
<span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"TRA"</span>, <span class='st'>"TRB"</span><span class='op'>)</span><span class='op'>]</span> <span class='op'>&lt;-</span>
  <span class='fu'><a href='https://rdrr.io/r/base/droplevels.html'>droplevels</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"TRA"</span>, <span class='st'>"TRB"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>sample_colours</span> <span class='op'>&lt;-</span> <span class='va'>sample_colours</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>Sample</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>treatment_colours</span> <span class='op'>&lt;-</span> <span class='va'>treatment_colours</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>Treatment</span><span class='op'>)</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<h3 id="re-processing-2">Re-processing</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>120</span><span class='op'>)</span>
<span class='va'>var_fit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/modelGeneVarByPoisson.html'>modelGeneVarByPoisson</a></span><span class='op'>(</span><span class='va'>sce</span>, block <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span>
<span class='va'>hvg</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/getTopHVGs.html'>getTopHVGs</a></span><span class='op'>(</span><span class='va'>var_fit</span>, var.threshold <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>11235</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/denoisePCA.html'>denoisePCA</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='va'>var_fit</span>, subset.row <span class='op'>=</span> <span class='va'>hvg</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>8875</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/runUMAP.html'>runUMAP</a></span><span class='op'>(</span><span class='va'>sce</span>, dimred <span class='op'>=</span> <span class='st'>"PCA"</span><span class='op'>)</span>
<span class='va'>umap_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html'>reducedDim</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMAP"</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>,
    y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html'>reducedDim</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMAP"</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"TRA"</span>, <span class='st'>"TRB"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>8111</span><span class='op'>)</span>
<span class='va'>snn_gr</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/buildSNNGraph.html'>buildSNNGraph</a></span><span class='op'>(</span><span class='va'>sce</span>, use.dimred <span class='op'>=</span> <span class='st'>"PCA"</span><span class='op'>)</span>
<span class='va'>clusters</span> <span class='op'>&lt;-</span> <span class='fu'>igraph</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/igraph/man/cluster_louvain.html'>cluster_louvain</a></span><span class='op'>(</span><span class='va'>snn_gr</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>clusters</span><span class='op'>$</span><span class='va'>membership</span><span class='op'>)</span>
<span class='va'>umap_df</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span>
<span class='va'>cluster_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'>Polychrome</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/Polychrome/man/palettes.html'>glasbey.colors</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nlevels.html'>nlevels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>[</span><span class='op'>-</span><span class='fl'>1</span><span class='op'>]</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster_colours</span> <span class='op'>&lt;-</span> <span class='va'>cluster_colours</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<p>There are 10 clusters detected, shown on the UMAP plot Figure <a href="#fig:no-monocytes-clusterplot-umap">21</a> and broken down by experimental factors in Figure <a href="#fig:no-monocytes-cluster-barplot">22</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://wilkelab.org/cowplot/reference/plot_grid.html'>plot_grid</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>cluster</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>Treatment</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>Sample</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sample_colours</span><span class='op'>)</span>,
  ncol <span class='op'>=</span> <span class='fl'>2</span>, 
  align <span class='op'>=</span> <span class='st'>"v"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:no-monocytes-clusterplot-umap"></span>
<img src="C057_Cooney.cell_selection_files/no-monocytes-clusterplot-umap-1.png" alt="UMAP plot of the updated data, where each point represents a droplet and is coloured according to the legend." width="624" />
<p class="caption">
Figure 21: UMAP plot of the updated data, where each point represents a droplet and is coloured according to the legend.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://wilkelab.org/cowplot/reference/plot_grid.html'>plot_grid</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cluster"</span>, <span class='st'>"Sample"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>Sample</span><span class='op'>)</span>,
      position <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/position_stack.html'>position_fill</a></span><span class='op'>(</span>reverse <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Frequency"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sample_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cluster"</span>, <span class='st'>"Treatment"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>Treatment</span><span class='op'>)</span>,
      position <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/position_stack.html'>position_fill</a></span><span class='op'>(</span>reverse <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Frequency"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='st'>"cluster"</span>, drop <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>cluster</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Number of droplets"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>,
  align <span class='op'>=</span> <span class='st'>"h"</span>,
  ncol <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:no-monocytes-cluster-barplot"></span>
<img src="C057_Cooney.cell_selection_files/no-monocytes-cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 22: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>Notably:</p>
<ul>
<li>Still, most of the clusters are highly treatment-specific.</li>
<li>Still, none of the clusters are highly sample-specific.</li>
</ul>
<h2 id="known-unwanted-cell-types">‘Known’ unwanted cell types</h2>
<h3 id="motivation-5">Motivation</h3>
<p>James indicated that there may be a subpopulation of CD8+ T-cells in this dataset. CD4+ and CD8+ T-cells are notoriously difficult to distinguish from one another using scRNA-seq data<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a>.</p>
<aside>
Specifically, James believes that there may be a population of CD8+ T-cells that upregulated the cell-surface expression of CD4 and hence were retained in the enrichment step.
</aside>
<h3 id="analysis-3">Analysis</h3>
<p>Supposing that there are CD8+ T-cells in this dataset, then there are several reasons why the approach taken in <a href="#unknown-unwanted-cell-types">‘Unknown’ unwanted cell types</a> did not label any cluster as CD8+ T-cells:</p>
<ol type="1">
<li>If the CD8+ T-cells were spread across several clusters, and in a minority within each of those clusters, then a cluster-level analysis would likely not label any cluster as CD8+ T-cells.</li>
<li>If the marker genes distinguishing CD4+ T-cells and CD8+ T-cells in the reference dataset are poorly chosen
<ol type="a">
<li>If these genes can’t even distinguish CD4+ T-cells and CD8+ T-cells in the reference dataset</li>
<li>If these genes are poorly sampled in our dataset.</li>
</ol></li>
</ol>
<p>We explore these issues in turn.</p>
<h4 id="cell-level-annotations">Cell-level annotations</h4>
<p>A cell-level annotation using <em><a href="https://bioconductor.org/packages/3.12/SingleR">SingleR</a></em> may allow us to identify CD8+ T-cells that are clustering with CD4+ T-cells rather than as a distinct cluster. For simplicity, I have again opted to only use the <code>DICE</code> reference annotation but now annotated the dataset at the cell-level using both the ‘main’ and ‘fine’ labels.</p>
<aside>
A reminder that the trade-off between cluster-level and cell-level annotation is one of increased robustness to noise (cluster-level) vs. increased resolution (cell-level). The story is similar for the trade-off between using ‘main’ and ‘fine’ labels.
</aside>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>pred_cell_main</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleR/man/SingleR.html'>SingleR</a></span><span class='op'>(</span>
  test <span class='op'>=</span> <span class='va'>sce</span>,
  ref <span class='op'>=</span> <span class='va'>ref</span>,
  labels <span class='op'>=</span> <span class='va'>labels_main</span>,
  BPPARAM <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocParallel/man/register.html'>bpparam</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_main</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>pred_cell_main</span><span class='op'>$</span><span class='va'>pruned.labels</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_main_collapsed</span> <span class='op'>&lt;-</span> <span class='fu'>.collapseLabel</span><span class='op'>(</span>
  <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_main</span>,
  <span class='va'>sce</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span>
<span class='va'>umap_df</span><span class='op'>$</span><span class='va'>label_cell_main_collapsed</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_main_collapsed</span>

<span class='va'>labels_fine</span> <span class='op'>&lt;-</span> <span class='va'>ref</span><span class='op'>$</span><span class='va'>label.fine</span>
<span class='co'># NOTE: This code doesn't necessarily generalise beyond the DICE main labels.</span>
<span class='va'>label_fine_collapsed_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
    <span class='fu'>Polychrome</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/Polychrome/man/palettes.html'>glasbey.colors</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nlevels.html'>nlevels</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>labels_fine</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>[</span><span class='op'>-</span><span class='fl'>1</span><span class='op'>]</span>,
    <span class='st'>"orange"</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>labels_fine</span><span class='op'>)</span><span class='op'>)</span>, <span class='st'>"other"</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>pred_cell_fine</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleR/man/SingleR.html'>SingleR</a></span><span class='op'>(</span>
  test <span class='op'>=</span> <span class='va'>sce</span>,
  ref <span class='op'>=</span> <span class='va'>ref</span>,
  labels <span class='op'>=</span> <span class='va'>labels_fine</span>,
  BPPARAM <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocParallel/man/register.html'>bpparam</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_fine</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>pred_cell_fine</span><span class='op'>$</span><span class='va'>pruned.labels</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_fine_collapsed</span> <span class='op'>&lt;-</span> <span class='fu'>.collapseLabel</span><span class='op'>(</span>
  <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_fine</span>,
  <span class='va'>sce</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>label_fine_collapsed_colours</span> <span class='op'>&lt;-</span> <span class='va'>label_fine_collapsed_colours</span><span class='op'>[</span>
  <span class='fu'><a href='https://rdrr.io/r/base/character.html'>as.character</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_fine_collapsed</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>umap_df</span><span class='op'>$</span><span class='va'>label_cell_fine_collapsed</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_fine_collapsed</span>
</code></pre>
</div>
</details>
</div>
<p>The results using the ‘main’ label is summarised in the table below. Almost all cells are labelled as <code>T cells, CD4+</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/janitor/man/tabyl.html'>tabyl</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>label.main <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_main</span>, cluster <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span>, 
  <span class='va'>label.main</span>,
  <span class='va'>cluster</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_totals.html'>adorn_totals</a></span><span class='op'>(</span><span class='st'>"col"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_percentages.html'>adorn_percentages</a></span><span class='op'>(</span><span class='st'>"col"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_pct_formatting.html'>adorn_pct_formatting</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_ns.html'>adorn_ns</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>"Cell-level assignments using the 'main' labels of the DICE reference."</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table style="width:100%;">
<caption><span id="tab:unnamed-chunk-28">Table 3: </span>Cell-level assignments using the ‘main’ labels of the DICE reference.</caption>
<colgroup>
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: left;">1</th>
<th style="text-align: left;">2</th>
<th style="text-align: left;">3</th>
<th style="text-align: left;">4</th>
<th style="text-align: left;">5</th>
<th style="text-align: left;">6</th>
<th style="text-align: left;">7</th>
<th style="text-align: left;">8</th>
<th style="text-align: left;">9</th>
<th style="text-align: left;">10</th>
<th style="text-align: left;">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (1)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (1)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.7% (8)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">3.4% (62)</td>
<td style="text-align: left;">1.7% (23)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">1.0% (94)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">100.0% (1035)</td>
<td style="text-align: left;">99.5% (218)</td>
<td style="text-align: left;">99.5% (426)</td>
<td style="text-align: left;">99.3% (140)</td>
<td style="text-align: left;">99.2% (1087)</td>
<td style="text-align: left;">100.0% (739)</td>
<td style="text-align: left;">96.5% (1736)</td>
<td style="text-align: left;">98.1% (1309)</td>
<td style="text-align: left;">99.9% (1468)</td>
<td style="text-align: left;">100.0% (871)</td>
<td style="text-align: left;">98.9% (9029)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD8+</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.5% (1)</td>
<td style="text-align: left;">0.2% (1)</td>
<td style="text-align: left;">0.7% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (5)</td>
</tr>
<tr class="even">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (1)</td>
</tr>
</tbody>
</table>
</div>
<p>The results using the ‘fine’ label is summarised in the table below. Consistent with the ‘main’ labels, almost all cells are labelled as a subtype of <code>T cells, CD4+</code> cells.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/janitor/man/tabyl.html'>tabyl</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>label.fine <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_fine</span>, cluster <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span>, 
  <span class='va'>label.fine</span>,
  <span class='va'>cluster</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_totals.html'>adorn_totals</a></span><span class='op'>(</span><span class='st'>"col"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_percentages.html'>adorn_percentages</a></span><span class='op'>(</span><span class='st'>"col"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_pct_formatting.html'>adorn_pct_formatting</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_ns.html'>adorn_ns</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>"Cell-level assignments using the 'fine' labels of the DICE reference."</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-29">Table 4: </span>Cell-level assignments using the ‘fine’ labels of the DICE reference.</caption>
<colgroup>
<col style="width: 19%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 6%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 7%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">label.fine</th>
<th style="text-align: left;">1</th>
<th style="text-align: left;">2</th>
<th style="text-align: left;">3</th>
<th style="text-align: left;">4</th>
<th style="text-align: left;">5</th>
<th style="text-align: left;">6</th>
<th style="text-align: left;">7</th>
<th style="text-align: left;">8</th>
<th style="text-align: left;">9</th>
<th style="text-align: left;">10</th>
<th style="text-align: left;">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells, naive</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (1)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes, CD16+</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (1)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.3% (3)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">2.1% (38)</td>
<td style="text-align: left;">0.7% (10)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.6% (52)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, memory TREG</td>
<td style="text-align: left;">23.6% (244)</td>
<td style="text-align: left;">4.6% (10)</td>
<td style="text-align: left;">32.0% (137)</td>
<td style="text-align: left;">46.1% (65)</td>
<td style="text-align: left;">44.4% (487)</td>
<td style="text-align: left;">68.1% (503)</td>
<td style="text-align: left;">48.9% (880)</td>
<td style="text-align: left;">80.1% (1068)</td>
<td style="text-align: left;">21.1% (310)</td>
<td style="text-align: left;">99.0% (862)</td>
<td style="text-align: left;">50.0% (4566)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, naive</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">2.1% (9)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.5% (7)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (18)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, naive TREG</td>
<td style="text-align: left;">0.2% (2)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.7% (3)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.1% (2)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (8)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, naive, stimulated</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">3.5% (15)</td>
<td style="text-align: left;">1.4% (2)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.1% (2)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (20)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, TFH</td>
<td style="text-align: left;">9.3% (96)</td>
<td style="text-align: left;">1.8% (4)</td>
<td style="text-align: left;">31.1% (133)</td>
<td style="text-align: left;">2.1% (3)</td>
<td style="text-align: left;">0.4% (4)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.7% (9)</td>
<td style="text-align: left;">14.8% (217)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">5.1% (469)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, Th1</td>
<td style="text-align: left;">2.5% (26)</td>
<td style="text-align: left;">61.6% (135)</td>
<td style="text-align: left;">3.0% (13)</td>
<td style="text-align: left;">19.1% (27)</td>
<td style="text-align: left;">12.9% (141)</td>
<td style="text-align: left;">5.7% (42)</td>
<td style="text-align: left;">43.3% (779)</td>
<td style="text-align: left;">11.6% (155)</td>
<td style="text-align: left;">3.1% (46)</td>
<td style="text-align: left;">0.7% (6)</td>
<td style="text-align: left;">15.0% (1370)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, Th1_17</td>
<td style="text-align: left;">0.7% (7)</td>
<td style="text-align: left;">31.1% (68)</td>
<td style="text-align: left;">0.2% (1)</td>
<td style="text-align: left;">2.1% (3)</td>
<td style="text-align: left;">8.9% (97)</td>
<td style="text-align: left;">0.4% (3)</td>
<td style="text-align: left;">0.8% (14)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.3% (5)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">2.2% (199)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, Th17</td>
<td style="text-align: left;">30.2% (313)</td>
<td style="text-align: left;">0.9% (2)</td>
<td style="text-align: left;">8.2% (35)</td>
<td style="text-align: left;">11.3% (16)</td>
<td style="text-align: left;">28.7% (315)</td>
<td style="text-align: left;">15.0% (111)</td>
<td style="text-align: left;">2.6% (46)</td>
<td style="text-align: left;">1.9% (26)</td>
<td style="text-align: left;">18.4% (271)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">12.4% (1135)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, Th2</td>
<td style="text-align: left;">32.9% (341)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">16.1% (69)</td>
<td style="text-align: left;">17.0% (24)</td>
<td style="text-align: left;">4.2% (46)</td>
<td style="text-align: left;">10.7% (79)</td>
<td style="text-align: left;">2.2% (39)</td>
<td style="text-align: left;">4.1% (55)</td>
<td style="text-align: left;">40.8% (599)</td>
<td style="text-align: left;">0.2% (2)</td>
<td style="text-align: left;">13.7% (1254)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD8+, naive</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (1)</td>
<td style="text-align: left;">0.7% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (2)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD8+, naive, stimulated</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (2)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.5% (5)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">2.3% (10)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (2)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (2)</td>
<td style="text-align: left;">0.4% (6)</td>
<td style="text-align: left;">0.6% (9)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.4% (34)</td>
</tr>
</tbody>
</table>
</div>
<p>This consistency is further emphasised when we cross-tabulate the ‘main’ and ‘fine’ labels.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/janitor/man/tabyl.html'>tabyl</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
  label.main <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_main</span>,
  label.fine <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_fine</span>,
  cluster <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span>,
  <span class='va'>label.fine</span>, <span class='va'>label.main</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_totals.html'>adorn_totals</a></span><span class='op'>(</span><span class='st'>"row"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_percentages.html'>adorn_percentages</a></span><span class='op'>(</span><span class='st'>"col"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_pct_formatting.html'>adorn_pct_formatting</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/janitor/man/adorn_ns.html'>adorn_ns</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>
    caption <span class='op'>=</span> <span class='st'>"Cross-tabulation of the 'main' and 'fine' assignments using the DICE reference."</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<table style="width:100%;">
<caption><span id="tab:unnamed-chunk-30">Table 5: </span>Cross-tabulation of the ‘main’ and ‘fine’ assignments using the DICE reference.</caption>
<colgroup>
<col style="width: 31%" />
<col style="width: 10%" />
<col style="width: 10%" />
<col style="width: 11%" />
<col style="width: 13%" />
<col style="width: 13%" />
<col style="width: 10%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">label.fine</th>
<th style="text-align: left;">B cells</th>
<th style="text-align: left;">Monocytes</th>
<th style="text-align: left;">NK cells</th>
<th style="text-align: left;">T cells, CD4+</th>
<th style="text-align: left;">T cells, CD8+</th>
<th style="text-align: left;">NA_</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells, naive</td>
<td style="text-align: left;">100.0% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes, CD16+</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">100.0% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">54.3% (51)</td>
<td style="text-align: left;">0.0% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, memory TREG</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">2.1% (2)</td>
<td style="text-align: left;">50.5% (4564)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, naive</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (18)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, naive TREG</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (7)</td>
<td style="text-align: left;">20.0% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, naive, stimulated</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (20)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, TFH</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">5.2% (469)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, Th1</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">43.6% (41)</td>
<td style="text-align: left;">14.7% (1327)</td>
<td style="text-align: left;">40.0% (2)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, Th1_17</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">2.2% (199)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, Th17</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">12.6% (1135)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, Th2</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">13.9% (1254)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD8+, naive</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">40.0% (2)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD8+, naive, stimulated</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">100.0% (1)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.4% (34)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Total</td>
<td style="text-align: left;">100.0% (1)</td>
<td style="text-align: left;">100.0% (1)</td>
<td style="text-align: left;">100.0% (94)</td>
<td style="text-align: left;">100.0% (9029)</td>
<td style="text-align: left;">100.0% (5)</td>
<td style="text-align: left;">100.0% (1)</td>
</tr>
</tbody>
</table>
</div>
<p>Figure <a href="#fig:umap-with-fine-cell-labels">23</a> overlays the ‘fine’ cell type labels on the UMAP plot.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>cluster</span><span class='op'>)</span>,
    alpha <span class='op'>=</span> <span class='fl'>1</span>,
    size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Clusters"</span><span class='op'>)</span>
<span class='va'>bg</span> <span class='op'>&lt;-</span> <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span><span class='va'>umap_df</span>, <span class='op'>-</span><span class='va'>label_cell_fine_collapsed</span><span class='op'>)</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>bg</span>, colour <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/alpha.html'>alpha</a></span><span class='op'>(</span><span class='st'>"grey"</span>, <span class='fl'>0.5</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.125</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>label_cell_fine_collapsed</span><span class='op'>)</span>,
    alpha <span class='op'>=</span> <span class='fl'>1</span>,
    size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>label_fine_collapsed_colours</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>label_fine_collapsed_colours</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>5</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_wrap.html'>facet_wrap</a></span><span class='op'>(</span><span class='op'>~</span> <span class='va'>label_cell_fine_collapsed</span>, ncol <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>+</span> 
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"'Fine' cell-level label"</span><span class='op'>)</span>
<span class='va'>p1</span> <span class='op'>+</span> <span class='va'>p2</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>widths <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:umap-with-fine-cell-labels"></span>
<img src="C057_Cooney.cell_selection_files/umap-with-fine-cell-labels-1.png" alt="UMAP plot. Each point represents a cluster and is coloured by the 'fine' cell-level label. Each panel highlights droplets from a particular predicted cluster type. Cluster labels with &lt; 1% frequency are grouped together as `other`." width="624" />
<p class="caption">
Figure 23: UMAP plot. Each point represents a cluster and is coloured by the ‘fine’ cell-level label. Each panel highlights droplets from a particular predicted cluster type. Cluster labels with &lt; 1% frequency are grouped together as <code>other</code>.
</p>
</div>
</div>
<p>Essentially none of the cells are labelled as CD8+ T-cells by this approach.</p>
<h6 id="diagnostic-plots-1">Diagnostic plots</h6>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://tibble.tidyverse.org/'>tibble</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://tidyr.tidyverse.org'>tidyr</a></span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/files2.html'>dir.create</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"output"</span>, <span class='st'>"marker_genes"</span>, <span class='st'>"SingleR"</span><span class='op'>)</span>, recursive <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>

<span class='va'>cell_main_de_genes</span> <span class='op'>&lt;-</span> <span class='fu'>SingleRDEGsAsTibble</span><span class='op'>(</span><span class='fu'>metadata</span><span class='op'>(</span><span class='va'>pred_cell_main</span><span class='op'>)</span><span class='op'>$</span><span class='va'>de.genes</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='va'>cell_main_de_genes</span>,
    <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span>
        <span class='st'>"output"</span>, 
        <span class='st'>"marker_genes"</span>, 
        <span class='st'>"SingleR"</span>, 
        <span class='st'>"SingleR_markers.DICE.label_cell_main.csv.gz"</span><span class='op'>)</span><span class='op'>)</span>,
    row.names <span class='op'>=</span> <span class='cn'>FALSE</span>,
    quote <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='va'>cell_fine_de_genes</span> <span class='op'>&lt;-</span> <span class='fu'>SingleRDEGsAsTibble</span><span class='op'>(</span><span class='fu'>metadata</span><span class='op'>(</span><span class='va'>pred_cell_fine</span><span class='op'>)</span><span class='op'>$</span><span class='va'>de.genes</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/write.table.html'>write.csv</a></span><span class='op'>(</span>
    <span class='va'>cell_fine_de_genes</span>,
    <span class='fu'><a href='https://rdrr.io/r/base/connections.html'>gzfile</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span>
        <span class='st'>"output"</span>, 
        <span class='st'>"marker_genes"</span>, 
        <span class='st'>"SingleR"</span>, 
        <span class='st'>"SingleR_markers.DICE.label_cell_fine.csv.gz"</span><span class='op'>)</span><span class='op'>)</span>,
    row.names <span class='op'>=</span> <span class='cn'>FALSE</span>,
    quote <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<aside>
The lists of genes from the <code>DICE</code> reference used to annotate these cells with <em><a href="https://bioconductor.org/packages/3.12/SingleR">SingleR</a></em> are available in <a href="../output/marker_genes/SingleR/"><code>output/marker_genes/SingleR/</code></a>. You may wish to look at the expression of these genes yourself to verify/understand the annotations.
</aside>
<p>As a sanity check, we again examine the expression of the marker genes for the relevant cell type labels by plotting a heatmap of their expression in:</p>
<ol type="1">
<li>The reference dataset</li>
<li>Our dataset</li>
</ol>
<p>The value of (1) is that we can assess if we believe the genes are indeed good markers of the relevant cell type in the reference dataset. The value of (2) is that we can check that these genes are useful markers in our dataset (e.g., that they are reasonably well sampled in our data).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>markers</span> <span class='op'>&lt;-</span> <span class='va'>cell_main_de_genes</span> <span class='op'>%&gt;%</span>
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span>
    <span class='op'>(</span><span class='va'>group1</span> <span class='op'>==</span> <span class='st'>"T cells, CD8+"</span> <span class='op'>&amp;</span> <span class='va'>group2</span> <span class='op'>==</span> <span class='st'>"T cells, CD4+"</span><span class='op'>)</span> <span class='op'>|</span>
      <span class='op'>(</span><span class='va'>group1</span> <span class='op'>==</span> <span class='st'>"T cells, CD4+"</span> <span class='op'>&amp;</span> <span class='va'>group2</span> <span class='op'>==</span> <span class='st'>"T cells, CD8+"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>rank</span> <span class='op'>&lt;=</span> <span class='fl'>20</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/pull.html'>pull</a></span><span class='op'>(</span><span class='va'>gene</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>Here, we specifically select (some of) the most strongly upregulated genes when comparing the <code>T cells, CD8+</code> to the <code>T cells, CD4+</code> (<code>dice.CD8_vs_CD4</code>) and vice-versa (<code>dice.CD4_vs_CD8</code>).</p>
<p>Figure <a href="#fig:ref-dice-cd8-and-cd4-markers">24</a> confirms that both the <code>dice.CD8_vs_CD4</code> and <code>dice.CD4_vs_CD8</code> marker genes distinguish these two cell types from one another in the <code>DICE</code> reference dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># NOTE: Have to remove column names from DICE to avoid an error.</span>
<span class='va'>tmp</span> <span class='op'>&lt;-</span> <span class='va'>dice</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>tmp</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_len</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>ncol</a></span><span class='op'>(</span><span class='va'>tmp</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHeatmap.html'>plotHeatmap</a></span><span class='op'>(</span>
  <span class='va'>tmp</span>,
  features <span class='op'>=</span> <span class='va'>markers</span>,
  colour_columns_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"label.main"</span>, <span class='st'>"label.fine"</span><span class='op'>)</span>,
  center <span class='op'>=</span> <span class='cn'>TRUE</span>,
  symmetric <span class='op'>=</span> <span class='cn'>TRUE</span>,
  order_columns_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"label.main"</span>, <span class='st'>"label.fine"</span><span class='op'>)</span>,
  cluster_rows <span class='op'>=</span> <span class='cn'>FALSE</span>,
  cluster_cols <span class='op'>=</span> <span class='cn'>FALSE</span>,
  annotation_row <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    markers <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"dice.CD8_vs_CD4"</span>, <span class='fl'>20</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"dice.CD4_vs_CD8"</span>, <span class='fl'>20</span><span class='op'>)</span><span class='op'>)</span>,
    row.names <span class='op'>=</span> <span class='va'>markers</span><span class='op'>)</span>,
  color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/palettes.html'>hcl.colors</a></span><span class='op'>(</span><span class='fl'>101</span>, <span class='st'>"Blue-Red 3"</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:ref-dice-cd8-and-cd4-markers"></span>
<img src="C057_Cooney.cell_selection_files/ref-dice-cd8-and-cd4-markers-1.png" alt="Heatmap of log-expression values in the `DICE` reference dataset for selected marker genes between the `Monocytes` and `T cells, CD4+` labels. Each column is a sample, each row a gene" width="624" />
<p class="caption">
Figure 24: Heatmap of log-expression values in the <code>DICE</code> reference dataset for selected marker genes between the <code>Monocytes</code> and <code>T cells, CD4+</code> labels. Each column is a sample, each row a gene
</p>
</div>
</div>
<p>Figure <a href="#fig:test-dice-cd8-and-cd4-markers">25</a> shows that very few of the <code>dice.CD8_vs_CD4</code> marker genes are expressed at in any cells in our dataset. In contrast, many of the cells express several of the <code>dice.CD4_vs_CD8</code> marker genes.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHeatmap.html'>plotHeatmap</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  features <span class='op'>=</span> <span class='va'>markers</span>,
  colour_columns_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"label_cell_main"</span>, <span class='st'>"label_cell_fine"</span>, <span class='st'>"cluster"</span><span class='op'>)</span>,
  center <span class='op'>=</span> <span class='cn'>TRUE</span>,
  symmetric <span class='op'>=</span> <span class='cn'>TRUE</span>,
  order_columns_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"label_cell_main"</span>, <span class='st'>"label_cell_fine"</span>, <span class='st'>"cluster"</span><span class='op'>)</span>,
  cluster_rows <span class='op'>=</span> <span class='cn'>FALSE</span>,
  cluster_cols <span class='op'>=</span> <span class='cn'>FALSE</span>,
  annotation_row <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    markers <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"dice.CD8_vs_CD4"</span>, <span class='fl'>20</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"dice.CD4_vs_CD8"</span>, <span class='fl'>20</span><span class='op'>)</span><span class='op'>)</span>,
    row.names <span class='op'>=</span> <span class='va'>markers</span><span class='op'>)</span>,
  color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/palettes.html'>hcl.colors</a></span><span class='op'>(</span><span class='fl'>101</span>, <span class='st'>"Blue-Red 3"</span><span class='op'>)</span>,
  fontsize <span class='op'>=</span> <span class='fl'>7</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:test-dice-cd8-and-cd4-markers"></span>
<img src="C057_Cooney.cell_selection_files/test-dice-cd8-and-cd4-markers-1.png" alt="Heatmap of log-expression values in our dataset at the cell-level for selected marker genes between the `T cells, CD8+` and `T cells, CD4+` labels. Each column is a sample, each row a gene." width="624" />
<p class="caption">
Figure 25: Heatmap of log-expression values in our dataset at the cell-level for selected marker genes between the <code>T cells, CD8+</code> and <code>T cells, CD4+</code> labels. Each column is a sample, each row a gene.
</p>
</div>
</div>
<p>Figure <a href="#fig:subset-test-dice-cd8-and-cd4-markers">26</a> shows that even amongst those cells not labelled as <code>T cells, CD4+</code>, very few of the <code>dice.CD8_vs_CD4</code> marker genes are expressed in our dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHeatmap.html'>plotHeatmap</a></span><span class='op'>(</span>
  <span class='va'>sce</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/which.html'>which</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>label_cell_main</span> <span class='op'>!=</span> <span class='st'>"T cells, CD4+"</span><span class='op'>)</span><span class='op'>]</span>,
  features <span class='op'>=</span> <span class='va'>markers</span>,
  colour_columns_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"label_cell_main"</span>, <span class='st'>"label_cell_fine"</span>, <span class='st'>"cluster"</span><span class='op'>)</span>,
  center <span class='op'>=</span> <span class='cn'>TRUE</span>,
  symmetric <span class='op'>=</span> <span class='cn'>TRUE</span>,
  order_columns_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"label_cell_main"</span>, <span class='st'>"label_cell_fine"</span>, <span class='st'>"cluster"</span><span class='op'>)</span>,
  cluster_rows <span class='op'>=</span> <span class='cn'>FALSE</span>,
  cluster_cols <span class='op'>=</span> <span class='cn'>FALSE</span>,
  annotation_row <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    markers <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"dice.CD8_vs_CD4"</span>, <span class='fl'>20</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>"dice.CD4_vs_CD8"</span>, <span class='fl'>20</span><span class='op'>)</span><span class='op'>)</span>,
    row.names <span class='op'>=</span> <span class='va'>markers</span><span class='op'>)</span>,
  color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/palettes.html'>hcl.colors</a></span><span class='op'>(</span><span class='fl'>101</span>, <span class='st'>"Blue-Red 3"</span><span class='op'>)</span>,
  fontsize <span class='op'>=</span> <span class='fl'>7</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:subset-test-dice-cd8-and-cd4-markers"></span>
<img src="C057_Cooney.cell_selection_files/subset-test-dice-cd8-and-cd4-markers-1.png" alt="Heatmap of log-expression values in our dataset at the cell-level for selected marker genes between the `T cells, CD8+` and `T cells, CD4+` labels. Each column is a sample, each row a gene. Only cells not labelled as `T cells, CD4+` are shown." width="624" />
<p class="caption">
Figure 26: Heatmap of log-expression values in our dataset at the cell-level for selected marker genes between the <code>T cells, CD8+</code> and <code>T cells, CD4+</code> labels. Each column is a sample, each row a gene. Only cells not labelled as <code>T cells, CD4+</code> are shown.
</p>
</div>
</div>
<h5 id="summary-3">Summary</h5>
<p>By performing a cell-level annotation with <em><a href="https://bioconductor.org/packages/3.12/SingleR">SingleR</a></em>, we have ruled out the possibility that the absence of CD8+ T-cells identified in <a href="#unknown-unwanted-cell-types">‘Unknown’ unwanted cell types</a> is due to this analysis being performed at the cluster-level. In fact, the cell-level analysis has provided stronger evidence that these cells are almost all various subtypes of CD4+ T-cells.</p>
<p>However, we also observed that the <code>dice.CD8_vs_CD4</code> marker genes are poorly sampled in our dataset, leaving open the possibility that a different choice of marker genes would allow us to distinguish the two cell types and identify any stray CD8+ T-cells. We pursue this approach in the next section.</p>
<h4 id="alternative-marker-gene-lists">Alternative marker gene lists</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>james_cd8_markers</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"GNLY"</span>, <span class='st'>"GZMA"</span>, <span class='st'>"GZMK"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>As an alternative to using the <code>DICE</code>-derived marker genes, we could use any other list of putative marker genes. James provided some genes that he believes are markers of the unwanted CD8+ cells, namely GNLY, GZMA, and GZMK. We look at the expression of these in both the <code>DICE</code> reference dataset and in our dataset.</p>
<h5 id="diagnostic-plots-2">Diagnostic plots</h5>
<p>Figure <a href="#fig:ref-james-cd8-markers">27</a> shows that these genes are arguably stronger markers of NK cells than they are of CD8+ cells. More importantly, these genes are highly expressed in several CD4+ T-cell subtypes in the <code>DICE</code> reference dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># NOTE: Have to remove column names from DICE to avoid an error.</span>
<span class='va'>tmp</span> <span class='op'>&lt;-</span> <span class='va'>dice</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>tmp</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_len</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>ncol</a></span><span class='op'>(</span><span class='va'>tmp</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHeatmap.html'>plotHeatmap</a></span><span class='op'>(</span>
  <span class='va'>tmp</span>,
  features <span class='op'>=</span> <span class='va'>james_cd8_markers</span>,
  colour_columns_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"label.main"</span>, <span class='st'>"label.fine"</span><span class='op'>)</span>,
  center <span class='op'>=</span> <span class='cn'>TRUE</span>,
  symmetric <span class='op'>=</span> <span class='cn'>TRUE</span>,
  cluster_rows <span class='op'>=</span> <span class='cn'>TRUE</span>,
  cluster_cols <span class='op'>=</span> <span class='cn'>TRUE</span>,
  color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/palettes.html'>hcl.colors</a></span><span class='op'>(</span><span class='fl'>101</span>, <span class='st'>"Blue-Red 3"</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:ref-james-cd8-markers"></span>
<img src="C057_Cooney.cell_selection_files/ref-james-cd8-markers-1.png" alt="Heatmap of log-expression values in the `DICE` reference dataset for 'CD8+ T-cell' marker genes provided by James. Each column is a sample, each row a gene" width="624" />
<p class="caption">
Figure 27: Heatmap of log-expression values in the <code>DICE</code> reference dataset for ‘CD8+ T-cell’ marker genes provided by James. Each column is a sample, each row a gene
</p>
</div>
</div>
<p>Consistent with this, Figure <a href="#fig:test-james-cd8-markers">28</a> shows high expression of these genes in cells labelled as various subtypes of CD4+ T-cells.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHeatmap.html'>plotHeatmap</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  features <span class='op'>=</span> <span class='va'>james_cd8_markers</span>,
  colour_columns_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"label_cell_main"</span>, <span class='st'>"label_cell_fine"</span>, <span class='st'>"cluster"</span><span class='op'>)</span>,
  center <span class='op'>=</span> <span class='cn'>TRUE</span>,
  symmetric <span class='op'>=</span> <span class='cn'>TRUE</span>,
  order_columns_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"label_cell_main"</span>, <span class='st'>"label_cell_fine"</span>, <span class='st'>"cluster"</span><span class='op'>)</span>,
  cluster_rows <span class='op'>=</span> <span class='cn'>TRUE</span>,
  cluster_cols <span class='op'>=</span> <span class='cn'>TRUE</span>,
  color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/palettes.html'>hcl.colors</a></span><span class='op'>(</span><span class='fl'>101</span>, <span class='st'>"Blue-Red 3"</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:test-james-cd8-markers"></span>
<img src="C057_Cooney.cell_selection_files/test-james-cd8-markers-1.png" alt="Heatmap of log-expression values in our dataset at the cell-level for 'CD8+ T-cell' marker genes provided by James. Each column is a sample, each row a gene." width="624" />
<p class="caption">
Figure 28: Heatmap of log-expression values in our dataset at the cell-level for ‘CD8+ T-cell’ marker genes provided by James. Each column is a sample, each row a gene.
</p>
</div>
</div>
<p>Furthermore, Figure <a href="#fig:test-james-cd8-markers">28</a> shows high expression of these genes for those cells labelled as <code>NK cells</code> and low expression of those same genes in the few cells labelled as <code>T cells, CD8</code>.</p>
<h5 id="summary-4">Summary</h5>
<p>There is no evidence in the <code>DICE</code> reference dataset that GNLY, GZMA, and GZMK are markers of CD8+ T-cells. To the contrary, the <code>DICE</code> dataset provides strong evidence that these are markers of various CD4+ T-cell subtypes and NK cells.</p>
<aside>
We could extend this analysis using a different set of ‘CD8+ T-cell’ marker genes, but I don’t think this is necessary or fruitful analysis direction.
</aside>
<h4 id="summary-5">Summary</h4>
<ul>
<li>We identified a small number of <code>Monocytes</code> that we excluded from further analysis.</li>
<li>We did not identify any of the suspected CD8+ T-cells.</li>
<li>Instead, we found strong evidence that almost all the remaining cells are various subtypes of CD4+ T-cells.</li>
</ul>
<h1 id="summary-6">Summary</h1>
<p>After extensive analysis we have now identified those droplets containing ‘biologically relevant’ cells (n = 9131). Specifically, we have selected for singlets that are CD4+ T-cells. We can verify that we have selected CD4+ T-cells by looking at the expression of known marker genes. Figure <a href="#fig:cd4-markers-umap">29</a> gives an example using some marker genes of <code>T cells, CD4+</code> cells from the <code>DICE</code> reference dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># NOTE: This code was written before I had SingleRDEGsAsTibble().</span>
<span class='va'>all_markers</span> <span class='op'>&lt;-</span> <span class='fu'>metadata</span><span class='op'>(</span><span class='va'>pred_cell_main</span><span class='op'>)</span><span class='op'>$</span><span class='va'>de.genes</span>
<span class='va'>cd4_markers</span> <span class='op'>&lt;-</span> <span class='va'>all_markers</span><span class='op'>[[</span><span class='st'>"T cells, CD4+"</span><span class='op'>]</span><span class='op'>]</span>
<span class='va'>cd4_markers_t</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/unlist.html'>unlist</a></span><span class='op'>(</span><span class='va'>cd4_markers</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># Three approaches to selecting markers:</span>
<span class='co'># </span>
<span class='co'># 1. Identify genes occurring in most pairwise comparisons</span>
<span class='co'>#      `markers &lt;- names(sort(cd4_markers_t, decreasing = TRUE))`</span>
<span class='co'># 2. Identify the markers genes that are most highly expressed (on average)</span>
<span class='co'>#    in our dataset.</span>
<span class='co'>#     `markers &lt;- names(sort(rowMeans(logcounts(sce)[names(cd4_markers_t), ]), decreasing = TRUE))`</span>
<span class='co'># 3. A compromise between (1) and (2), here selecting most highly expressed markers that occur in at least X pairwise comparisons</span>
<span class='co'>#     `markers &lt;- names(sort(rowMeans(logcounts(sce)[names(cd4_markers_t[cd4_markers_t &gt;= X]), ]), decreasing = TRUE))`</span>

<span class='co'># NOTE: Opted to go with (3) with X = 2</span>
<span class='va'>markers</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/sort.html'>sort</a></span><span class='op'>(</span>
    <span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>rowMeans</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/assays.html'>logcounts</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>cd4_markers_t</span><span class='op'>[</span><span class='va'>cd4_markers_t</span> <span class='op'>&gt;=</span> <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>, <span class='op'>]</span><span class='op'>)</span>,
    decreasing <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># NOTE: Have to remove `TRA` and `TRB` before create the dataframe.</span>
<span class='va'>tmp</span> <span class='op'>&lt;-</span> <span class='va'>sce</span>
<span class='fu'>colData</span><span class='op'>(</span><span class='va'>tmp</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'>colData</span><span class='op'>(</span><span class='va'>tmp</span><span class='op'>)</span><span class='op'>[</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>tmp</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"TRA"</span>, <span class='st'>"TRB"</span><span class='op'>)</span><span class='op'>]</span>
<span class='va'>p</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>markers</span>, <span class='fl'>9</span><span class='op'>)</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>marker</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>b</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/makePerCellDF.html'>makePerCellDF</a></span><span class='op'>(</span><span class='va'>tmp</span>, features <span class='op'>=</span> <span class='va'>marker</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
    <span class='fu'>tidyr</span><span class='fu'>::</span><span class='fu'><a href='https://tidyr.tidyverse.org/reference/pivot_longer.html'>pivot_longer</a></span><span class='op'>(</span>
      <span class='fu'>all_of</span><span class='op'>(</span><span class='va'>marker</span><span class='op'>)</span>,
      names_to <span class='op'>=</span> <span class='st'>"feature"</span>,
      values_to <span class='op'>=</span> <span class='st'>"logcounts"</span><span class='op'>)</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>b</span>, <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>UMAP.1</span>, y <span class='op'>=</span> <span class='va'>UMAP.2</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>logcounts</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>1</span><span class='op'>/</span><span class='fl'>9</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_viridis.html'>scale_colour_viridis_c</a></span><span class='op'>(</span>option <span class='op'>=</span> <span class='st'>"magma"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='va'>marker</span><span class='op'>)</span>
<span class='op'>}</span><span class='op'>)</span>
<span class='fu'><a href='https://wilkelab.org/cowplot/reference/plot_grid.html'>plot_grid</a></span><span class='op'>(</span>plotlist <span class='op'>=</span> <span class='va'>p</span>, ncol <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:cd4-markers-umap"></span>
<img src="C057_Cooney.cell_selection_files/cd4-markers-umap-1.png" alt="UMAP plot of the updated data, where each point represents a cell and is coloured according to expression of selected genes. The selected genes are markers of the `T cells, CD4+` cells in the `DICE` reference dataset that are also highly expressed in our dataset." width="624" />
<p class="caption">
Figure 29: UMAP plot of the updated data, where each point represents a cell and is coloured according to expression of selected genes. The selected genes are markers of the <code>T cells, CD4+</code> cells in the <code>DICE</code> reference dataset that are also highly expressed in our dataset.
</p>
</div>
</div>
<h2 id="re-processing-3">Re-processing</h2>
<p>We perform a final re-processing of the ‘biologically relevant’ cells. Notably, we opt at this stage to exclude ribosomal protein and mitochondrial genes from the list of highly variable genes, for reasons explained below.</p>
<h3 id="exclusion-of-ribosomal-protein-and-mitochondrial-genes-from-hvgs">Exclusion of ribosomal protein and mitochondrial genes from HVGs</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>1000</span><span class='op'>)</span>
<span class='va'>var_fit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/modelGeneVarByPoisson.html'>modelGeneVarByPoisson</a></span><span class='op'>(</span><span class='va'>sce</span>, block <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>batch</span><span class='op'>)</span>
<span class='va'>hvg</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/getTopHVGs.html'>getTopHVGs</a></span><span class='op'>(</span><span class='va'>var_fit</span>, var.threshold <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>We find that the most highly variable genes in this dataset are somewhat enriched for ribosomal protein genes<a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a> (n = 106) and mitochondrial genes (n = 13), with Figure <a href="#fig:perc-mito-and-ribo">30</a> showing that more than 20% of the top-200 HVGs are ribosomal protein or mitochondrial genes.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>is_mito</span> <span class='op'>&lt;-</span> <span class='va'>hvg</span> <span class='op'>%in%</span> <span class='va'>mito_set</span>
<span class='va'>is_ribo</span> <span class='op'>&lt;-</span> <span class='va'>hvg</span> <span class='op'>%in%</span> <span class='va'>ribo_set</span>

<span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span>
  <span class='fl'>100</span> <span class='op'>*</span> <span class='fu'><a href='https://rdrr.io/r/base/cumsum.html'>cumsum</a></span><span class='op'>(</span><span class='va'>is_mito</span><span class='op'>)</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_along</a></span><span class='op'>(</span><span class='va'>hvg</span><span class='op'>)</span>,
  log <span class='op'>=</span> <span class='st'>"x"</span>,
  ylim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>100</span><span class='op'>)</span>,
  ylab <span class='op'>=</span> <span class='st'>"%"</span>,
  xlab <span class='op'>=</span> <span class='st'>"Top-k genes"</span>,
  col <span class='op'>=</span> <span class='st'>"dodgerBlue"</span>,
  pch <span class='op'>=</span> <span class='fl'>16</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/points.html'>points</a></span><span class='op'>(</span><span class='fl'>100</span> <span class='op'>*</span> <span class='fu'><a href='https://rdrr.io/r/base/cumsum.html'>cumsum</a></span><span class='op'>(</span><span class='va'>is_ribo</span><span class='op'>)</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_along</a></span><span class='op'>(</span><span class='va'>hvg</span><span class='op'>)</span>, col <span class='op'>=</span> <span class='st'>"orange"</span>, pch <span class='op'>=</span> <span class='fl'>16</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/points.html'>points</a></span><span class='op'>(</span><span class='fl'>100</span> <span class='op'>*</span> <span class='fu'><a href='https://rdrr.io/r/base/cumsum.html'>cumsum</a></span><span class='op'>(</span><span class='va'>is_mito</span> <span class='op'>|</span> <span class='va'>is_ribo</span><span class='op'>)</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_along</a></span><span class='op'>(</span><span class='va'>hvg</span><span class='op'>)</span>, col <span class='op'>=</span> <span class='st'>"black"</span>, pch <span class='op'>=</span> <span class='fl'>16</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/legend.html'>legend</a></span><span class='op'>(</span>
  <span class='st'>"topright"</span>,
  pch <span class='op'>=</span> <span class='fl'>16</span>,
  col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"black"</span>, <span class='st'>"orange"</span>, <span class='st'>"dodgerBlue"</span><span class='op'>)</span>,
  legend <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Both"</span>, <span class='st'>"Ribosomal protein"</span>, <span class='st'>"Mitochondrial"</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:perc-mito-and-ribo"></span>
<img src="C057_Cooney.cell_selection_files/perc-mito-and-ribo-1.png" alt="Percentage of top-K HVGs that are mitochondrial or ribosomal protein genes." width="624" />
<p class="caption">
Figure 30: Percentage of top-K HVGs that are mitochondrial or ribosomal protein genes.
</p>
</div>
</div>
<p>Ribosomal protein and mitochondrial genes are of lesser biological relevance to this study, so we it was decided to exclude them from the HVGs. This means that these genes can no longer <strong>directly</strong> influence some of the subsequent steps in the analysis, including:</p>
<ul>
<li>Dimensionality reduction
<ul>
<li>Principal component analysis (PCA)</li>
<li>Uniform manifold approximation and projection (UMAP)</li>
</ul></li>
<li>Data integration
<ul>
<li>Mutual nearest neighbours (MNN)</li>
</ul></li>
<li>Clustering</li>
</ul>
<p>Although the exclusion of these genes from the HVGs prevents them from directly influencing these analyses, they may still <strong>indirectly</strong> influence these steps or otherwise be associated with their outcomes. For example, if there is a set of (non ribosomal protein) genes that are strongly associated with ribosomal protein gene expression, then we may still see a cluster associated with ribosomal protein gene expression.</p>
<p>Finally, and to emphasise, we have only excluded ribosomal protein and mitochondrial genes from the HVGs, i.e. we have not excluded them entirely from the dataset. In particular, this means that these genes may appear in downstream results (e.g., tables of cluster marker genes and differentially expressed genes).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>hvg</span> <span class='op'>&lt;-</span> <span class='va'>hvg</span><span class='op'>[</span><span class='op'>!</span><span class='op'>(</span><span class='va'>is_mito</span> <span class='op'>|</span> <span class='va'>is_ribo</span><span class='op'>)</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<h3 id="visualisation">Visualisation</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>11235</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/denoisePCA.html'>denoisePCA</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='va'>var_fit</span>, subset.row <span class='op'>=</span> <span class='va'>hvg</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>8875</span><span class='op'>)</span>
<span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/runUMAP.html'>runUMAP</a></span><span class='op'>(</span><span class='va'>sce</span>, dimred <span class='op'>=</span> <span class='st'>"PCA"</span><span class='op'>)</span>
<span class='va'>umap_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html'>reducedDim</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMAP"</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>,
    y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html'>reducedDim</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMAP"</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"TRA"</span>, <span class='st'>"TRB"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>8111</span><span class='op'>)</span>
<span class='va'>snn_gr</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scran/man/buildSNNGraph.html'>buildSNNGraph</a></span><span class='op'>(</span><span class='va'>sce</span>, use.dimred <span class='op'>=</span> <span class='st'>"PCA"</span><span class='op'>)</span>
<span class='va'>clusters</span> <span class='op'>&lt;-</span> <span class='fu'>igraph</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/igraph/man/cluster_louvain.html'>cluster_louvain</a></span><span class='op'>(</span><span class='va'>snn_gr</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>clusters</span><span class='op'>$</span><span class='va'>membership</span><span class='op'>)</span>
<span class='va'>umap_df</span><span class='op'>$</span><span class='va'>cluster</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span>
<span class='va'>cluster_colours</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span>
  <span class='fu'>Polychrome</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/Polychrome/man/palettes.html'>glasbey.colors</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nlevels.html'>nlevels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>[</span><span class='op'>-</span><span class='fl'>1</span><span class='op'>]</span>,
  <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster_colours</span> <span class='op'>&lt;-</span> <span class='va'>cluster_colours</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>]</span>
</code></pre>
</div>
</details>
</div>
<p>There are 10 clusters detected, shown on the UMAP plot Figure <a href="#fig:selected-cells-clusterplot-umap">31</a> and broken down by experimental factors in Figure <a href="#fig:selected-cells-cluster-barplot">32</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://wilkelab.org/cowplot/reference/plot_grid.html'>plot_grid</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>cluster</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>Treatment</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, y <span class='op'>=</span> <span class='va'>y</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>umap_df</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>Sample</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.25</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_colour_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sample_colours</span><span class='op'>)</span>,
  ncol <span class='op'>=</span> <span class='fl'>2</span>, 
  align <span class='op'>=</span> <span class='st'>"v"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:selected-cells-clusterplot-umap"></span>
<img src="C057_Cooney.cell_selection_files/selected-cells-clusterplot-umap-1.png" alt="UMAP plot of the updated data, where each point represents a droplet and is coloured according to the legend." width="624" />
<p class="caption">
Figure 31: UMAP plot of the updated data, where each point represents a droplet and is coloured according to the legend.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://wilkelab.org/cowplot/reference/plot_grid.html'>plot_grid</a></span><span class='op'>(</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cluster"</span>, <span class='st'>"Sample"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>Sample</span><span class='op'>)</span>,
      position <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/position_stack.html'>position_fill</a></span><span class='op'>(</span>reverse <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Frequency"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>sample_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cluster"</span>, <span class='st'>"Treatment"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span>
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>Treatment</span><span class='op'>)</span>,
      position <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/position_stack.html'>position_fill</a></span><span class='op'>(</span>reverse <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Frequency"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>treatment_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span>,
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span>, <span class='st'>"cluster"</span>, drop <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>cluster</span>, fill <span class='op'>=</span> <span class='va'>cluster</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/coord_flip.html'>coord_flip</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Number of droplets"</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_fill_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>cluster_colours</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>,
  align <span class='op'>=</span> <span class='st'>"h"</span>,
  ncol <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<div class="figure"><span id="fig:selected-cells-cluster-barplot"></span>
<img src="C057_Cooney.cell_selection_files/selected-cells-cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 32: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>Notably:</p>
<ul>
<li>Still, most of the clusters are highly treatment-specific.</li>
<li>Now, some of the clusters are somewhat sample-specific.</li>
</ul>
<h1 id="concluding-remarks">Concluding remarks</h1>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>saveRDS</a></span><span class='op'>(</span>
  <span class='va'>sce</span>,
  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"SCEs"</span>, <span class='st'>"C057_Cooney.cells_selected.SCE.rds"</span><span class='op'>)</span>,
  compress <span class='op'>=</span> <span class='st'>"xz"</span><span class='op'>)</span>
</code></pre>
</div>
</details>
</div>
<p>The processed <em>SingleCellExperiment</em> object is available (see <a href="../data/SCEs/C057_Cooney.cells_selected.SCE.rds"><code>data/SCEs/C057_Cooney.cells_selected.SCE.rds</code></a>). This will be used in downstream analyses, e.g., identifying cluster marker genes and refining the cell labels.</p>
<h1 id="additional-information" class="appendix">Additional information</h1>
<p>The following are available on request:</p>
<ul>
<li>Full CSV tables of any data presented.</li>
<li>PDF/PNG files of any static plots.</li>
</ul>
<h1 id="session-info" class="appendix">Session info</h1>
<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>Show code</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>sessioninfo</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/sessioninfo/man/session_info.html'>session_info</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
</details>
<pre><code>─ Session info ─────────────────────────────────────────────────────
 setting  value                       
 version  R version 4.0.3 (2020-10-10)
 os       CentOS Linux 7 (Core)       
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Australia/Melbourne         
 date     2021-04-01                  

─ Packages ─────────────────────────────────────────────────────────
 ! package                * version  date       lib source        
 P annotate                 1.68.0   2020-10-27 [?] Bioconductor  
 P AnnotationDbi            1.52.0   2020-10-27 [?] Bioconductor  
 P AnnotationHub            2.22.0   2020-10-27 [?] Bioconductor  
 P assertthat               0.2.1    2019-03-21 [?] CRAN (R 4.0.0)
 P beachmat                 2.6.4    2020-12-20 [?] Bioconductor  
 P beeswarm                 0.2.3    2016-04-25 [?] CRAN (R 4.0.0)
 P Biobase                * 2.50.0   2020-10-27 [?] Bioconductor  
   BiocFileCache            1.14.0   2020-10-27 [1] Bioconductor  
   BiocGenerics           * 0.36.0   2020-10-27 [1] Bioconductor  
 P BiocManager              1.30.10  2019-11-16 [?] CRAN (R 4.0.0)
 P BiocNeighbors            1.8.2    2020-12-07 [?] Bioconductor  
   BiocParallel           * 1.24.1   2020-11-06 [1] Bioconductor  
 P BiocSingular             1.6.0    2020-10-27 [?] Bioconductor  
   BiocStyle                2.18.1   2020-11-24 [1] Bioconductor  
   BiocVersion              3.12.0   2020-04-27 [1] Bioconductor  
 P bit                      4.0.4    2020-08-04 [?] CRAN (R 4.0.0)
 P bit64                    4.0.5    2020-08-30 [?] CRAN (R 4.0.0)
 P bitops                   1.0-6    2013-08-17 [?] CRAN (R 4.0.0)
 P blob                     1.2.1    2020-01-20 [?] CRAN (R 4.0.0)
 P bluster                  1.0.0    2020-10-27 [?] Bioconductor  
 P bslib                    0.2.4    2021-01-25 [?] CRAN (R 4.0.3)
 P cachem                   1.0.4    2021-02-13 [?] CRAN (R 4.0.3)
 P celldex                * 1.0.0    2020-10-29 [?] Bioconductor  
 P cli                      2.3.1    2021-02-23 [?] CRAN (R 4.0.3)
 P codetools                0.2-18   2020-11-04 [?] CRAN (R 4.0.3)
 P colorspace               2.0-0    2020-11-11 [?] CRAN (R 4.0.3)
 P cowplot                * 1.1.1    2020-12-30 [?] CRAN (R 4.0.3)
 P crayon                   1.4.1    2021-02-08 [?] CRAN (R 4.0.3)
 P curl                     4.3      2019-12-02 [?] CRAN (R 4.0.0)
 P data.table               1.14.0   2021-02-21 [?] CRAN (R 4.0.3)
 P DBI                      1.1.1    2021-01-15 [?] CRAN (R 4.0.3)
 P dbplyr                   2.1.0    2021-02-03 [?] CRAN (R 4.0.3)
 P DelayedArray             0.16.2   2021-02-26 [?] Bioconductor  
 P DelayedMatrixStats       1.12.3   2021-02-03 [?] Bioconductor  
 P DESeq2                   1.30.1   2021-02-19 [?] Bioconductor  
 P digest                   0.6.27   2020-10-24 [?] CRAN (R 4.0.2)
 P distill                  1.2      2021-01-13 [?] CRAN (R 4.0.3)
 P downlit                  0.2.1    2020-11-04 [?] CRAN (R 4.0.3)
 P dplyr                    1.0.4    2021-02-02 [?] CRAN (R 4.0.3)
 P dqrng                    0.2.1    2019-05-17 [?] CRAN (R 4.0.0)
 P edgeR                  * 3.32.1   2021-01-14 [?] Bioconductor  
 P ellipsis                 0.3.1    2020-05-15 [?] CRAN (R 4.0.0)
 P evaluate                 0.14     2019-05-28 [?] CRAN (R 4.0.0)
 P ExperimentHub            1.16.0   2020-10-27 [?] Bioconductor  
 P fansi                    0.4.2    2021-01-15 [?] CRAN (R 4.0.3)
 P farver                   2.1.0    2021-02-28 [?] CRAN (R 4.0.3)
 P fastmap                  1.1.0    2021-01-25 [?] CRAN (R 4.0.3)
 P genefilter               1.72.1   2021-01-21 [?] Bioconductor  
 P geneplotter              1.68.0   2020-10-27 [?] Bioconductor  
 P generics                 0.1.0    2020-10-31 [?] CRAN (R 4.0.3)
 P GenomeInfoDb           * 1.26.2   2020-12-08 [?] Bioconductor  
   GenomeInfoDbData         1.2.4    2021-02-04 [1] Bioconductor  
 P GenomicRanges          * 1.42.0   2020-10-27 [?] Bioconductor  
 P ggbeeswarm               0.6.0    2017-08-07 [?] CRAN (R 4.0.0)
 P ggplot2                * 3.3.3    2020-12-30 [?] CRAN (R 4.0.3)
 P Glimma                 * 2.0.0    2020-10-27 [?] Bioconductor  
 P glue                     1.4.2    2020-08-27 [?] CRAN (R 4.0.0)
 P gridExtra                2.3      2017-09-09 [?] CRAN (R 4.0.0)
 P gtable                   0.3.0    2019-03-25 [?] CRAN (R 4.0.0)
 P here                   * 1.0.1    2020-12-13 [?] CRAN (R 4.0.3)
 P highr                    0.8      2019-03-20 [?] CRAN (R 4.0.0)
 P htmltools                0.5.1.1  2021-01-22 [?] CRAN (R 4.0.3)
 P htmlwidgets              1.5.3    2020-12-10 [?] CRAN (R 4.0.3)
 P httpuv                   1.5.5    2021-01-13 [?] CRAN (R 4.0.3)
 P httr                     1.4.2    2020-07-20 [?] CRAN (R 4.0.0)
 P igraph                   1.2.6    2020-10-06 [?] CRAN (R 4.0.2)
 P interactiveDisplayBase   1.28.0   2020-10-27 [?] Bioconductor  
 P IRanges                * 2.24.1   2020-12-12 [?] Bioconductor  
 P irlba                    2.3.3    2019-02-05 [?] CRAN (R 4.0.0)
 P janitor                * 2.1.0    2021-01-05 [?] CRAN (R 4.0.3)
 P jquerylib                0.1.3    2020-12-17 [?] CRAN (R 4.0.3)
 P jsonlite                 1.7.2    2020-12-09 [?] CRAN (R 4.0.3)
 P knitr                    1.31     2021-01-27 [?] CRAN (R 4.0.3)
 P labeling                 0.4.2    2020-10-20 [?] CRAN (R 4.0.0)
 P later                    1.1.0.1  2020-06-05 [?] CRAN (R 4.0.0)
 P lattice                  0.20-41  2020-04-02 [?] CRAN (R 4.0.0)
 P lifecycle                1.0.0    2021-02-15 [?] CRAN (R 4.0.3)
   limma                  * 3.46.0   2020-10-27 [1] Bioconductor  
 P locfit                   1.5-9.4  2020-03-25 [?] CRAN (R 4.0.0)
 P lubridate                1.7.10   2021-02-26 [?] CRAN (R 4.0.3)
 P magrittr                 2.0.1    2020-11-17 [?] CRAN (R 4.0.3)
 P Matrix                   1.3-2    2021-01-06 [?] CRAN (R 4.0.3)
   MatrixGenerics         * 1.2.1    2021-01-30 [1] Bioconductor  
 P matrixStats            * 0.58.0   2021-01-29 [?] CRAN (R 4.0.3)
 P memoise                  2.0.0    2021-01-26 [?] CRAN (R 4.0.3)
 P mime                     0.10     2021-02-13 [?] CRAN (R 4.0.3)
 P msigdbr                * 7.2.1    2020-10-02 [?] CRAN (R 4.0.2)
 P munsell                  0.5.0    2018-06-12 [?] CRAN (R 4.0.0)
 P patchwork              * 1.1.1    2020-12-17 [?] CRAN (R 4.0.3)
 P pheatmap                 1.0.12   2019-01-04 [?] CRAN (R 4.0.0)
 P pillar                   1.5.0    2021-02-22 [?] CRAN (R 4.0.3)
 P pkgconfig                2.0.3    2019-09-22 [?] CRAN (R 4.0.0)
 P Polychrome               1.2.6    2020-11-11 [?] CRAN (R 4.0.3)
 P promises                 1.2.0.1  2021-02-11 [?] CRAN (R 4.0.3)
 P purrr                    0.3.4    2020-04-17 [?] CRAN (R 4.0.0)
 P R6                       2.5.0    2020-10-28 [?] CRAN (R 4.0.2)
 P rappdirs                 0.3.3    2021-01-31 [?] CRAN (R 4.0.3)
 P RColorBrewer             1.1-2    2014-12-07 [?] CRAN (R 4.0.0)
 P Rcpp                     1.0.6    2021-01-15 [?] CRAN (R 4.0.3)
 P RcppAnnoy                0.0.18   2020-12-15 [?] CRAN (R 4.0.3)
 P RCurl                    1.98-1.2 2020-04-18 [?] CRAN (R 4.0.0)
 P rlang                    0.4.10   2020-12-30 [?] CRAN (R 4.0.3)
 P rmarkdown                2.7      2021-02-19 [?] CRAN (R 4.0.3)
 P rprojroot                2.0.2    2020-11-15 [?] CRAN (R 4.0.3)
 P RSpectra                 0.16-0   2019-12-01 [?] CRAN (R 4.0.0)
 P RSQLite                  2.2.3    2021-01-24 [?] CRAN (R 4.0.3)
 P rsvd                     1.0.3    2020-02-17 [?] CRAN (R 4.0.0)
 P S4Vectors              * 0.28.1   2020-12-09 [?] Bioconductor  
 P sass                     0.3.1    2021-01-24 [?] CRAN (R 4.0.3)
 P scales                   1.1.1    2020-05-11 [?] CRAN (R 4.0.0)
 P scater                 * 1.18.6   2021-02-26 [?] Bioconductor  
 P scatterplot3d            0.3-41   2018-03-14 [?] CRAN (R 4.0.0)
 P scDblFinder            * 1.4.0    2020-10-27 [?] Bioconductor  
 P scran                  * 1.18.5   2021-02-04 [?] Bioconductor  
 P scuttle                  1.0.4    2020-12-17 [?] Bioconductor  
 P sessioninfo              1.1.1    2018-11-05 [?] CRAN (R 4.0.0)
 P shiny                    1.6.0    2021-01-25 [?] CRAN (R 4.0.3)
 P SingleCellExperiment   * 1.12.0   2020-10-27 [?] Bioconductor  
 P SingleR                * 1.4.1    2021-02-02 [?] Bioconductor  
 P snakecase                0.11.0   2019-05-25 [?] CRAN (R 4.0.0)
 P sparseMatrixStats        1.2.1    2021-02-02 [?] Bioconductor  
 P statmod                  1.4.35   2020-10-19 [?] CRAN (R 4.0.2)
 P stringi                  1.5.3    2020-09-09 [?] CRAN (R 4.0.0)
 P stringr                  1.4.0    2019-02-10 [?] CRAN (R 4.0.0)
 P SummarizedExperiment   * 1.20.0   2020-10-27 [?] Bioconductor  
 P survival                 3.2-7    2020-09-28 [?] CRAN (R 4.0.2)
 P tibble                 * 3.1.0    2021-02-25 [?] CRAN (R 4.0.3)
 P tidyr                  * 1.1.3    2021-03-03 [?] CRAN (R 4.0.3)
 P tidyselect               1.1.0    2020-05-11 [?] CRAN (R 4.0.0)
 P utf8                     1.1.4    2018-05-24 [?] CRAN (R 4.0.0)
 P uwot                     0.1.10   2020-12-15 [?] CRAN (R 4.0.3)
 P vctrs                    0.3.6    2020-12-17 [?] CRAN (R 4.0.3)
 P vipor                    0.4.5    2017-03-22 [?] CRAN (R 4.0.0)
 P viridis                  0.5.1    2018-03-29 [?] CRAN (R 4.0.0)
 P viridisLite              0.3.0    2018-02-01 [?] CRAN (R 4.0.0)
 P withr                    2.4.1    2021-01-26 [?] CRAN (R 4.0.3)
 P xfun                     0.21     2021-02-10 [?] CRAN (R 4.0.3)
 P xgboost                  1.3.2.1  2021-01-18 [?] CRAN (R 4.0.3)
 P XML                      3.99-0.5 2020-07-23 [?] CRAN (R 4.0.0)
 P xtable                   1.8-4    2019-04-21 [?] CRAN (R 4.0.0)
 P XVector                  0.30.0   2020-10-27 [?] Bioconductor  
 P yaml                     2.2.1    2020-02-01 [?] CRAN (R 4.0.0)
 P zlibbioc                 1.36.0   2020-10-27 [?] Bioconductor  

[1] /stornext/Projects/score/Analyses/C057_Cooney/renv/library/R-4.0/x86_64-pc-linux-gnu
[2] /tmp/RtmpNjHKzV/renv-system-library
[3] /stornext/System/data/apps/R/R-4.0.3/lib64/R/library

 P ── Loaded and on-disk path mismatch.</code></pre>
</div>
</details>
<div class="sourceCode" id="cb2"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<div id="refs" class="references" role="doc-bibliography">
<div id="ref-aran2019reference">
<p>Aran, D., A. P. Looney, L. Liu, E. Wu, V. Fong, A. Hsu, S. Chak, et al. 2019. “Reference-Based Analysis of Lung Single-Cell Sequencing Reveals a Transitional Profibrotic Macrophage.” <em>Nat. Immunol.</em> 20 (2): 163–72.</p>
</div>
<div id="ref-hpcaRef">
<p>Mabbott, Neil A., J. K. Baillie, Helen Brown, Tom C. Freeman, and David A. Hume. 2013. “An Expression Atlas of Human Primary Cells: Inference of Gene Function from Coexpression Networks.” <em>BMC Genomics</em> 14. <a href="https://doi.org/10.1186/1471-2164-14-632">https://doi.org/10.1186/1471-2164-14-632</a>.</p>
</div>
<div id="ref-blueprintRef">
<p>Martens, Joost H A, and Hendrik G. Stunnenberg. 2013. “BLUEPRINT: Mapping Human Blood Cell Epigenomes.” <em>Haematologica</em> 98: 1487–9. <a href="https://doi.org/10.3324/haematol.2013.094243">https://doi.org/10.3324/haematol.2013.094243</a>.</p>
</div>
<div id="ref-monaco_immuneRef">
<p>Monaco, Gianni, Bernett Lee, Weili Xu, Seri Mustafah, You Yi Hwang, Christophe Carré, Nicolas Burdin, et al. 2019. “RNA-Seq Signatures Normalized by mRNA Abundance Allow Absolute Deconvolution of Human Immune Cell Types.” <em>Cell Reports</em> 26 (6): 1627–1640.e7. <a href="https://doi.org/10.1016/j.celrep.2019.01.041">https://doi.org/10.1016/j.celrep.2019.01.041</a>.</p>
</div>
<div id="ref-nhRef">
<p>Novershtern, Noa, Aravind Subramanian, Lee N. Lawton, Raymond H. Mak, W. Nicholas Haining, Marie E. McConkey, Naomi Habib, et al. 2011. “Densely Interconnected Transcriptional Circuits Control Cell States in Human Hematopoiesis.” <em>Cell</em> 144 (2): 296–309. <a href="https://doi.org/10.1016/j.cell.2011.01.004">https://doi.org/10.1016/j.cell.2011.01.004</a>.</p>
</div>
<div id="ref-Pijuan-Sala2019-yz">
<p>Pijuan-Sala, Blanca, Jonathan A Griffiths, Carolina Guibentif, Tom W Hiscock, Wajid Jawaid, Fernando J Calero-Nieto, Carla Mulas, et al. 2019. “A Single-Cell Molecular Map of Mouse Gastrulation and Early Organogenesis.” <em>Nature</em> 566 (7745). <a href="https://doi.org/10.1038/s41586-019-0933-9">https://doi.org/10.1038/s41586-019-0933-9</a>.</p>
</div>
<div id="ref-diceRef">
<p>Schmiedel, Benjamin J., Divya Singh, Ariel Madrigal, Alan G. Valdovino-Gonzalez, Brandie M. White, Jose Zapardiel-Gonzalo, Brendan Ha, et al. 2018. “Impact of Genetic Polymorphisms on Human Immune Cell Gene Expression.” <em>Cell</em> 175 (6): 1701–1715.e16. <a href="https://doi.org/10.1016/j.cell.2018.10.022">https://doi.org/10.1016/j.cell.2018.10.022</a>.</p>
</div>
<div id="ref-encodeRef">
<p>The ENCODE Project Consortium. 2012. “An Integrated Encyclopedia of Dna Elements in the Human Genome.” <em>Nature</em>. <a href="https://doi.org/10.1038/nature11247">https://doi.org/10.1038/nature11247</a>.</p>
</div>
<div id="ref-xu2015identification">
<p>Xu, C., and Z. Su. 2015. “Identification of Cell Types from Single-Cell Transcriptomes Using a Novel Clustering Method.” <em>Bioinformatics</em> 31 (12): 1974–80.</p>
</div>
</div>
<section class="footnotes" role="doc-endnotes">
<hr />
<ol>
<li id="fn1" role="doc-endnote"><p>In a simple multiplexing experiment involving <span class="math inline">\(N\)</span> HTOs (here <span class="math inline">\(N=5\)</span>) with similar numbers of cells, we would expect around <span class="math inline">\(1 / N = 1 / 5 = 20\%\)</span> of all doublets to involve cells from the same sample. This may still be enough to cause the formation of misleading doublet clusters even after the majority of ‘labelled’ doublets are removed.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2" role="doc-endnote"><p>The use of experimentally derived doublet calls avoids any assumptions about the relative quantity of total RNA or the probability of doublet formation across different cell types.<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3" role="doc-endnote"><p>This can be individual cells or bulk samples.<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4" role="doc-endnote"><p>Table adapted from <a href="https://bioconductor.org/packages/release/bioc/vignettes/SingleR/inst/doc/SingleR.html#5_available_references">https://bioconductor.org/packages/release/bioc/vignettes/SingleR/inst/doc/SingleR.html#5_available_references</a><a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5" role="doc-endnote"><p>E.g., see <a href="https://osca.bioconductor.org/cell-type-annotation.html#assigning-cell-labels-from-reference-data">https://osca.bioconductor.org/cell-type-annotation.html#assigning-cell-labels-from-reference-data</a><a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn6" role="doc-endnote"><p>Here, ribosomal protein genes are defined as those gene symbols starting with “RPL” or “RPS” or otherwise found in the <a href="https://www.gsea-msigdb.org/gsea/msigdb/cards/KEGG_RIBOSOME.html">“KEGG_RIBOSOME” gene set published by MSigDB</a>, namely: FAU, MRPL13, RPL10, RPL10A, RPL10L, RPL11, RPL12, RPL13, RPL13A, RPL14, RPL15, RPL17, RPL17-C18orf32, RPL18, RPL18A, RPL19, RPL21, RPL22, RPL22L1, RPL23, RPL23A, RPL24, RPL26, RPL26L1, RPL27, RPL27A, RPL28, RPL29, RPL3, RPL30, RPL31, RPL32, RPL34, RPL34-AS1, RPL35, RPL35A, RPL36, RPL36A, RPL36A-HNRNPH2, RPL36AL, RPL37, RPL37A, RPL38, RPL39, RPL39L, RPL3L, RPL4, RPL41, RPL5, RPL6, RPL7, RPL7A, RPL7L1, RPL8, RPL9, RPLP0, RPLP1, RPLP2, RPS10, RPS10-NUDT3, RPS11, RPS12, RPS13, RPS14, RPS15, RPS15A, RPS16, RPS17, RPS18, RPS19, RPS19BP1, RPS2, RPS20, RPS21, RPS23, RPS24, RPS25, RPS26, RPS27, RPS27A, RPS27L, RPS28, RPS29, RPS3, RPS3A, RPS4X, RPS4Y1, RPS4Y2, RPS5, RPS6, RPS6KA1, RPS6KA2, RPS6KA3, RPS6KA4, RPS6KA5, RPS6KA6, RPS6KB1, RPS6KB2, RPS6KB2-AS1, RPS6KC1, RPS6KL1, RPS7, RPS8, RPS9, RPSA, RSL24D1, RSL24D1P11, UBA52<a href="#fnref6" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
