<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

  <!--radix_placeholder_meta_tags-->
  <title>C057: Selection of biologically relevant cells for the Cooney (C057) memory CD4+ T-cell data set</title>
  
  
  
  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2020-07-23"/>
  <meta property="article:created" itemprop="dateCreated" content="2020-07-23"/>
  <meta name="article:author" content="Peter Hickey"/>
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="C057: Selection of biologically relevant cells for the Cooney (C057) memory CD4+ T-cell data set"/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="C057"/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="C057: Selection of biologically relevant cells for the Cooney (C057) memory CD4+ T-cell data set"/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Reference-based analysis of lung single-cell sequencing reveals a transitional profibrotic macrophage;citation_publication_date=2019;citation_volume=20;citation_author=D. Aran;citation_author=A. P. Looney;citation_author=L. Liu;citation_author=E. Wu;citation_author=V. Fong;citation_author=A. Hsu;citation_author=S. Chak;citation_author=R. P. Naikawadi;citation_author=P. J. Wolters;citation_author=A. R. Abate;citation_author=A. J. Butte;citation_author=M. Bhattacharya"/>
  <meta name="citation_reference" content="citation_title=An expression atlas of human primary cells: Inference of gene function from coexpression networks;citation_publication_date=2013;citation_volume=14;citation_doi=10.1186/1471-2164-14-632;citation_author=Neil A. Mabbott;citation_author=J. K. Baillie;citation_author=Helen Brown;citation_author=Tom C. Freeman;citation_author=David A. Hume"/>
  <meta name="citation_reference" content="citation_title=BLUEPRINT: Mapping human blood cell epigenomes;citation_publication_date=2013;citation_volume=98;citation_doi=10.3324/haematol.2013.094243;citation_author=Joost H A Martens;citation_author=Hendrik G. Stunnenberg"/>
  <meta name="citation_reference" content="citation_title=An integrated encyclopedia of dna elements in the human genome;citation_publication_date=2012;citation_doi=10.1038/nature11247;citation_issn=14764687;citation_author=An integrated encyclopedia of dna elements in the human genome"/>
  <meta name="citation_reference" content="citation_title=Impact of genetic polymorphisms on human immune cell gene expression;citation_publication_date=2018;citation_volume=175;citation_doi=10.1016/j.cell.2018.10.022;citation_issn=0092-8674;citation_author=Benjamin J. Schmiedel;citation_author=Divya Singh;citation_author=Ariel Madrigal;citation_author=Alan G. Valdovino-Gonzalez;citation_author=Brandie M. White;citation_author=Jose Zapardiel-Gonzalo;citation_author=Brendan Ha;citation_author=Gokmen Altay;citation_author=Jason A. Greenbaum;citation_author=Graham McVicker;citation_author=Grégory Seumois;citation_author=Anjana Rao;citation_author=Mitchell Kronenberg;citation_author=Bjoern Peters;citation_author=Pandurangan Vijayanand"/>
  <meta name="citation_reference" content="citation_title=Densely interconnected transcriptional circuits control cell states in human hematopoiesis;citation_publication_date=2011;citation_volume=144;citation_doi=10.1016/j.cell.2011.01.004;citation_issn=1097-4172;citation_author=Noa Novershtern;citation_author=Aravind Subramanian;citation_author=Lee N. Lawton;citation_author=Raymond H. Mak;citation_author=W. Nicholas Haining;citation_author=Marie E. McConkey;citation_author=Naomi Habib;citation_author=Nir Yosef;citation_author=Cindy Y. Chang;citation_author=Tal Shay;citation_author=Garrett M. Frampton;citation_author=Adam C. B. Drake;citation_author=Ilya Leskov;citation_author=Bjorn Nilsson;citation_author=Fred Preffer;citation_author=David Dombkowski;citation_author=John W. Evans;citation_author=Ted Liefeld;citation_author=John S. Smutko;citation_author=Jianzhu Chen;citation_author=Nir Friedman;citation_author=Richard A. Young;citation_author=Todd R. Golub;citation_author=Aviv Regev;citation_author=Benjamin L. Ebert"/>
  <meta name="citation_reference" content="citation_title=RNA-seq signatures normalized by mRNA abundance allow absolute deconvolution of human immune cell types;citation_publication_date=2019;citation_volume=26;citation_doi=10.1016/j.celrep.2019.01.041;citation_issn=2211-1247;citation_author=Gianni Monaco;citation_author=Bernett Lee;citation_author=Weili Xu;citation_author=Seri Mustafah;citation_author=You Yi Hwang;citation_author=Christophe Carré;citation_author=Nicolas Burdin;citation_author=Lucian Visan;citation_author=Michele Ceccarelli;citation_author=Michael Poidinger;citation_author=Alfred Zippelius;citation_author=João Pedro de Magalhães;citation_author=Anis Larbi"/>
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","date","output","editor_options","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["Selection of biologically relevant cells for the Cooney (C057) memory CD4+ T-cell data set"]},{"type":"NULL"},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url","affiliation","affiliation_url"]}},"value":[{"type":"character","attributes":{},"value":["Peter Hickey"]},{"type":"character","attributes":{},"value":["https://peterhickey.org"]},{"type":"character","attributes":{},"value":["Single Cell Open Research Endeavour (SCORE), WEHI"]},{"type":"character","attributes":{},"value":["https://www.wehi.edu.au/people/shalin-naik/3310/score"]}]}]},{"type":"character","attributes":{},"value":["2020-07-23"]},{"type":"character","attributes":{},"value":["distill::distill_article"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["chunk_output_type"]}},"value":[{"type":"character","attributes":{},"value":["console"]}]},{"type":"character","attributes":{},"value":["ref.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #455a64;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative;}
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  </style>
  
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for table of contents */
  
  .d-toc {
    color: rgba(0,0,0,0.8);
    font-size: 0.8em;
    line-height: 1em;
  }
  
  .d-toc-header {
    font-size: 0.6rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.5);
    text-transform: uppercase;
    margin-top: 0;
    margin-bottom: 1.3em;
  }
  
  .d-toc a {
    border-bottom: none;
  }
  
  .d-toc ul {
    padding-left: 0;
  }
  
  .d-toc li>ul {
    padding-top: 0.8em;
    padding-left: 16px;
    margin-bottom: 0.6em;
  }
  
  .d-toc ul,
  .d-toc li {
    list-style-type: none;
  }
  
  .d-toc li {
    margin-bottom: 0.9em;
  }
  
  .d-toc-separator {
    margin-top: 20px;
    margin-bottom: 2em;
  }
  
  .d-article-with-toc {
    border-top: none;
    padding-top: 0;
  }
  
  
  
  /* Tweak code blocks (note that this CSS is repeated above in an injection
     into the d-code shadow dom) */
  
  d-code {
    overflow-x: auto !important;
  }
  
  pre.d-code code.d-code {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  pre.text-output {
  
    font-size: 12px;
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  @media(min-width: 768px) {
  
  d-code {
    overflow-x: visible !important;
  }
  
  pre.d-code code.d-code  {
      padding-left: 18px;
      font-size: 14px;
  }
  pre.text-output {
    font-size: 14px;
  }
  }
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .custom p {
    margin-bottom: 0.5em;
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // create d-bibliography
    var bibliography = $('<d-bibliography></d-bibliography>');
    $('#distill-bibliography').wrap(bibliography);
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace citations with <d-cite>
    $('.citation').each(function(i, val) {
      appendix = true;
      var cites = $(this).attr('data-cites').split(" ");
      var dt_cite = $('<d-cite></d-cite>');
      dt_cite.attr('key', cites.join());
      $(this).replaceWith(dt_cite);
    });
    // remove refs
    $('#refs').remove();
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-toc a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // replace code blocks with d-code
    $('pre>code').each(function(i, val) {
      var code = $(this);
      var pre = code.parent();
      var clz = "";
      var language = pre.attr('class');
      if (language) {
        // map unknown languages to "clike" (without this they just dissapear)
        if ($.inArray(language, ["bash", "clike", "css", "go", "html",
                                 "javascript", "js", "julia", "lua", "markdown",
                                 "markup", "mathml", "python", "svg", "xml"]) == -1)
          language = "clike";
        language = ' language="' + language + '"';
        var dt_code = $('<d-code block' + language + clz + '></d-code>');
        dt_code.text(code.text());
        pre.replaceWith(dt_code);
      } else {
        code.addClass('text-output').unwrap().changeElementType('pre');
      }
    });
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('d-code, pre.text-output, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // table of contents
      if (have_authors) // adjust border if we are in authors
        $('.d-toc').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // inject pre code styles (can't do this with a global stylesheet b/c a shadow root is used)
      $('d-code').each(function(i, val) {
        var style = document.createElement('style');
        style.innerHTML = 'pre code { padding-left: 10px; font-size: 12px; border-left: 2px solid rgba(0,0,0,0.1); } ' +
                          '@media(min-width: 768px) { pre code { padding-left: 18px; font-size: 14px; } }';
        if (this.shadowRoot)
          this.shadowRoot.appendChild(style);
      });
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-toc-header').remove();
    $('.d-toc').remove();
    $('.d-toc-separator').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Selection of biologically relevant cells for the Cooney (C057) memory CD4+ T-cell data set","authors":[{"author":"Peter Hickey","authorURL":"https://peterhickey.org","affiliation":"Single Cell Open Research Endeavour (SCORE), WEHI","affiliationURL":"https://www.wehi.edu.au/people/shalin-naik/3310/score"}],"publishedDate":"2020-07-23T00:00:00.000+10:00","citationText":"Hickey, 2020"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">C057</a>
<a href="index.html">Introduction</a>
<a href="C057_Cooney.demultiplex.html">Demultiplexing</a>
<a href="C057_Cooney.preprocess.html">Preprocessing</a>
<a href="C057_Cooney.cell_selection.html">Selecting biologically relevant cells</a>
<a href="C057_Cooney.annotate.html">Annotating cells</a>
<a href="C057_Cooney.multi-sample_comparisons.html">Differential analyses</a>
<a href="C057_Cooney.repertoire-seq.html">Immune repertoire analysis</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Supplementary files
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="../output/CellRanger/">CellRanger reports</a>
<a href="../output/marker_genes/">Marker gene tables</a>
<a href="../output/DEGs/">Differentially expressed genes</a>
<a href="../output/glimma-plots/">Glimma plots</a>
</div>
</div>
</div>
<div class="nav-right">
<a href="https://github.com/WEHISCORE/C057_Cooney">
<i class="fa fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Selection of biologically relevant cells for the Cooney (C057) memory CD4+ T-cell data set</h1>

</div>

<div class="d-byline">
  Peter Hickey <a href="https://peterhickey.org" class="uri">https://peterhickey.org</a> (Single Cell Open Research Endeavour (SCORE), WEHI)<a href="https://www.wehi.edu.au/people/shalin-naik/3310/score" class="uri">https://www.wehi.edu.au/people/shalin-naik/3310/score</a>
  
<br/>2020-07-23
</div>

<div class="d-article">
<h3 class="d-toc-header">Table of Contents</h3>
<nav class="d-toc" id="TOC">
<ul>
<li><a href="#motivation">Motivation</a></li>
<li><a href="#preparing-the-data">Preparing the data</a></li>
<li><a href="#clusters-enriched-for-ambiguous-droplets">Clusters enriched for <code>ambiguous</code> droplets</a></li>
<li><a href="#doublets">Doublets</a></li>
<li><a href="#selection-of-labelled-singlets">Selection of ‘labelled’ singlets</a></li>
<li><a href="#exclusion-of-unwanted-cell-types">Exclusion of unwanted cell types</a></li>
<li><a href="#summary-8">Summary</a></li>
<li><a href="#concluding-remarks">Concluding remarks</a></li>
<li><a href="#additional-information">Additional information</a></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</nav>
<hr class="d-toc-separator"/>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="motivation">Motivation</h1>
<p>scRNA-seq datasets generated with 10X will include droplets that are not relevant to the study, even after the initial quality control, which we don’t want to include in downstream analyses. In this section aim to filter out these ‘unwanted’ droplets and retain only those droplets containing ‘biologically relevant’ cells. Examples of unwanted droplets include:</p>
<ul>
<li>Droplets containing stripped nuclei</li>
<li>Droplets containing more than one cell (‘doublets’)</li>
<li>Droplets without phenotype data, such as droplets for which we can’t infer the sample identify based on cell hashing or genetic variation data</li>
<li>Droplets containing unwanted cell types, such as those that might sneak through a FACS or magnetic bead enrichment sample preparation</li>
</ul>
<p>Once we are confident that we have selected the biologically relevant cells, we will perform data integration (if necessary) and a further round of clustering in preparation for downstream analysis.</p>
<p>The removal of unwanted droplets is an iterative process where at each step we:</p>
<ol type="1">
<li>Identify cluster(s) enriched for unwanted droplets. The exact criteria used to define ‘unwanted’ will depend on the type of droplets we are trying to identify at each step.
<ol type="a">
<li>Perform diagnostic checks to ensure we aren’t discarding biologically relevant droplets.</li>
<li>Remove the unwanted cells.</li>
</ol></li>
<li>Re-process the remaining droplets.
<ol type="a">
<li>Identify HVGs.</li>
<li>Perform dimensionality reduction (PCA and UMAP).</li>
<li>Cluster droplets.</li>
</ol></li>
</ol>
<p>Clustering is a critical component of this process, so we discuss it in further detail in the next subsection.</p>
<h2 id="clustering">Clustering</h2>
<p>Clustering is an unsupervised learning procedure that is used in scRNA-seq data analysis to empirically define groups of cells with similar expression profiles. Its primary purpose is to summarize the data in a digestible format for human interpretation. This allows us to describe population heterogeneity in terms of discrete labels that are easily understood, rather than attempting to comprehend the high-dimensional manifold on which the cells truly reside. Clustering is thus a critical step for extracting biological insights from scRNA-seq data.</p>
<p>Clustering calculations are usually performed using the top PCs to take advantage of data compression and denoising.</p>
<h3 id="clusters-vs.-cell-types">Clusters vs. cell types</h3>
<p>It is worth stressing the distinction between clusters and cell types. The former is an empirical construct while the latter is a biological truth (albeit a vaguely defined one). For this reason, questions like “what is the true number of clusters?” are usually meaningless. We can define as many clusters as we like, with whatever algorithm we like - each clustering will represent its own partitioning of the high-dimensional expression space, and is as “real” as any other clustering.</p>
<p>A more relevant question is “how well do the clusters approximate the cell types?” Unfortunately, this is difficult to answer given the context-dependent interpretation of biological truth. Some analysts will be satisfied with resolution of the major cell types; other analysts may want resolution of subtypes; and others still may require resolution of different states (e.g., metabolic activity, stress) within those subtypes. Two clusterings can also be highly inconsistent yet both valid, simply partitioning the cells based on different aspects of biology. Indeed, asking for an unqualified “best” clustering is akin to asking for the best magnification on a microscope without any context.</p>
<p>It is helpful to realize that clustering, like a microscope, is simply a tool to explore the data. We can zoom in and out by changing the resolution of the clustering parameters, and we can experiment with different clustering algorithms to obtain alternative perspectives of the data. This iterative approach is entirely permissible for data exploration, which constitutes the majority of all scRNA-seq data analysis.</p>
<h2 id="graph-based-clustering">Graph-based clustering</h2>
<p>We build a shared nearest neighbour graph <span class="citation" data-cites="xu2015identification">(Xu and Su <a href="#ref-xu2015identification" role="doc-biblioref">2015</a>)</span> and use the Louvain algorithm to identify clusters. We would build the graph using the principal components.</p>
<h1 id="preparing-the-data">Preparing the data</h1>
<p>We start from the preprocessed <em>SingleCellExperiment</em> object created in <a href="C057_Cooney.preprocess.html">‘Preprocessing the Cooney (C057) memory CD4+ T-cell data set’</a>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="initial-clustering">Initial clustering</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>There are 10 clusters detected, shown on the UMAP plot Figure <a href="#fig:clusterplot-umap">1</a> and broken down by experimental factors in Figure <a href="#fig:cluster-barplot">2</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:clusterplot-umap"></span>
<img src="C057_Cooney.cell_selection_files/clusterplot-umap-1.png" alt="UMAP plot, where each point represents a droplet and is coloured according to the legend." width="624" />
<p class="caption">
Figure 1: UMAP plot, where each point represents a droplet and is coloured according to the legend.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:cluster-barplot"></span>
<img src="C057_Cooney.cell_selection_files/cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 2: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>Notably:</p>
<ul>
<li>Some of the clusters are highly treatment-specific.</li>
<li>Many of the clusters are enriched for <code>ambiguous</code> droplets.</li>
</ul>
<h1 id="clusters-enriched-for-ambiguous-droplets">Clusters enriched for <code>ambiguous</code> droplets</h1>
<p>This analysis is done in two parts: a <a href="#first-pass">First pass</a> and a <a href="#refinement">Refinement</a>.</p>
<h2 id="first-pass">First pass</h2>
<h3 id="motivation-1">Motivation</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>There are several clusters with unusually large proportions of droplets being of <code>ambiguous</code> origin, namely clusters 2, 7, 8, and 9 (proportions = 0.69, 0.84, 0.52, and 0.71, respectively).</p>
<p>These droplets cannot be assigned to a sample of origin and so are not useful in downstream analyses. However, before we exclude them, we want to ensure we aren’t systematically excluding a biologically relevant cell population.</p>
<p>We would further like to know that this isn’t systematically excluding droplets from a particular sample or treatment group. This is challenging because by definition most of these droplets are of <code>ambiguous</code> origin. However, we can check if the ‘labelled’ droplets in those clusters are enriched for particular samples.</p>
<h3 id="analysis">Analysis</h3>
<p>Figure <a href="#fig:ambiguous-enriched-clusters-umap">3</a> highlights droplets from these clusters on the UMAP plot, showing that they are largely distinct from the bulk of droplets from the dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:ambiguous-enriched-clusters-umap"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-enriched-clusters-umap-1.png" alt="UMAP plot highlighting droplets from clusters that are enriched for `ambiguous` droplets. Cluster labels are overlaid on cluster centroids." width="624" />
<p class="caption">
Figure 3: UMAP plot highlighting droplets from clusters that are enriched for <code>ambiguous</code> droplets. Cluster labels are overlaid on cluster centroids.
</p>
</div>
</div>
<p>Figure <a href="#fig:ambiguous-enriched-clusters-qc-plot">4</a> shows that the QC metrics of droplets from these clusters are distinct from the other clusters, having smaller library sizes and fewer genes detected (but no difference in mitochondrial proportions).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:ambiguous-enriched-clusters-qc-plot"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-enriched-clusters-qc-plot-1.png" alt="QC metrics of droplets by cluster, highlighting droplets from clusters that are enriched for `ambiguous` droplets (orange)." width="624" />
<p class="caption">
Figure 4: QC metrics of droplets by cluster, highlighting droplets from clusters that are enriched for <code>ambiguous</code> droplets (orange).
</p>
</div>
</div>
<p>This is particularly clear for clusters 2, 7, and 9, whereas cluster 8 comprises droplets with abnormal and typical QC metrics. Droplets with small library sizes are likely to contain ambient RNA rather than intact cells.</p>
<h4 id="checking-for-removal-of-biologically-relevant-subpopulations">Checking for removal of biologically relevant subpopulations</h4>
<p>We check for cell type loss by looking for systematic differences in gene expression between the discarded and retained droplets, doing this separately for each of the suspicious clusters.</p>
<h5 id="cluster-2">Cluster 2</h5>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:ambiguous-enriched-clusters-2-md-plot">5</a> shows the result of this analysis, highlighting that there are few genes with a large <span class="math inline">\(logFC\)</span> between ‘lost’ and ‘kept’ droplets.</p>
<aside>
An interactive version of Figure <a href="#fig:ambiguous-enriched-clusters-2-md-plot">5</a> is available from <a href="../output/glimma-plots/ambiguous-enriched-clusters-2-md-plot.html"><code>output/glimma-plots/ambiguous-enriched-clusters-2-md-plot.html</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:ambiguous-enriched-clusters-2-md-plot"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-enriched-clusters-2-md-plot-1.png" alt="Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate $|logFC| = 1$" width="624" />
<p class="caption">
Figure 5: Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate <span class="math inline">\(|logFC| = 1\)</span>
</p>
</div>
</div>
<p>If the discarded pool is enriched for a certain cell type, we should observe increased expression of the corresponding marker genes.</p>
<p>Another concern is whether the cells removed by this procedure are preferentially derive from particular experimental groups. Reassuringly, Figure <a href="#fig:ambiguous-enriched-clusters-2-barplot-highlighting-removed-droplets">6</a> shows that this is not the case; most of the removed droplets are <code>ambiguous</code>, as we expect from the above analysis.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:ambiguous-enriched-clusters-2-barplot-highlighting-removed-droplets"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-enriched-clusters-2-barplot-highlighting-removed-droplets-1.png" alt="Droplets retained after excluding droplets from cluster 2, stratified by `Sample`." width="624" />
<p class="caption">
Figure 6: Droplets retained after excluding droplets from cluster 2, stratified by <code>Sample</code>.
</p>
</div>
</div>
<h5 id="cluster-7">Cluster 7</h5>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:ambiguous-enriched-clusters-7-md-plot">7</a> shows the result of this analysis, highlighting that there are few genes with a large <span class="math inline">\(logFC\)</span> between ‘lost’ and ‘kept’ droplets.</p>
<aside>
An interactive version of Figure <a href="#fig:ambiguous-enriched-clusters-7-md-plot">7</a> is available from <a href="../output/glimma-plots/ambiguous-enriched-clusters-7-md-plot.html"><code>output/glimma-plots/ambiguous-enriched-clusters-7-md-plot.html</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:ambiguous-enriched-clusters-7-md-plot"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-enriched-clusters-7-md-plot-1.png" alt="Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate $|logFC| = 1$" width="624" />
<p class="caption">
Figure 7: Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate <span class="math inline">\(|logFC| = 1\)</span>
</p>
</div>
</div>
<p>If the discarded pool is enriched for a certain cell type, we should observe increased expression of the corresponding marker genes.</p>
<p>Another concern is whether the cells removed by this procedure are preferentially derive from particular experimental groups. Reassuringly, Figure <a href="#fig:ambiguous-enriched-clusters-7-barplot-highlighting-removed-droplets">8</a> shows that this is not the case; most of the removed droplets are <code>ambiguous</code>, as we expect from the above analysis.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:ambiguous-enriched-clusters-7-barplot-highlighting-removed-droplets"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-enriched-clusters-7-barplot-highlighting-removed-droplets-1.png" alt="Droplets retained after excluding droplets from cluster 5, stratified by `Sample`." width="624" />
<p class="caption">
Figure 8: Droplets retained after excluding droplets from cluster 5, stratified by <code>Sample</code>.
</p>
</div>
</div>
<h5 id="cluster-8">Cluster 8</h5>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:ambiguous-enriched-clusters-8-md-plot">9</a> shows the result of this analysis, highlighting that there are a number of genes with a large <span class="math inline">\(logFC\)</span> between ‘lost’ and ‘kept’ droplets.</p>
<aside>
An interactive version of Figure <a href="#fig:ambiguous-enriched-clusters-8-md-plot">9</a> is available from <a href="../output/glimma-plots/ambiguous-enriched-clusters-8-md-plot.html"><code>output/glimma-plots/ambiguous-enriched-clusters-8-md-plot.html</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:ambiguous-enriched-clusters-8-md-plot"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-enriched-clusters-8-md-plot-1.png" alt="Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate $|logFC| = 1$" width="624" />
<p class="caption">
Figure 9: Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate <span class="math inline">\(|logFC| = 1\)</span>
</p>
</div>
</div>
<p>If the discarded pool is enriched for a certain cell type, we should observe increased expression of the corresponding marker genes.</p>
<p>Another concern is whether the cells removed by this procedure are preferentially derive from particular experimental groups. Figure <a href="#fig:ambiguous-enriched-clusters-8-barplot-highlighting-removed-droplets">10</a> shows that excluding these droplets might preferentially exclude droplets from the <code>Uninfected</code> samples.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:ambiguous-enriched-clusters-8-barplot-highlighting-removed-droplets"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-enriched-clusters-8-barplot-highlighting-removed-droplets-1.png" alt="Droplets retained after excluding droplets from cluster 8, stratified by `Sample`." width="624" />
<p class="caption">
Figure 10: Droplets retained after excluding droplets from cluster 8, stratified by <code>Sample</code>.
</p>
</div>
</div>
<h5 id="cluster-9">Cluster 9</h5>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:ambiguous-enriched-clusters-9-md-plot">11</a> shows the result of this analysis, highlighting that there are few genes with a large <span class="math inline">\(logFC\)</span> between ‘lost’ and ‘kept’ droplets.</p>
<aside>
An interactive version of Figure <a href="#fig:ambiguous-enriched-clusters-9-md-plot">11</a> is available from <a href="../output/glimma-plots/ambiguous-enriched-clusters-9-md-plot.html"><code>output/glimma-plots/ambiguous-enriched-clusters-9-md-plot.html</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:ambiguous-enriched-clusters-9-md-plot"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-enriched-clusters-9-md-plot-1.png" alt="Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate $|logFC| = 1$" width="624" />
<p class="caption">
Figure 11: Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate <span class="math inline">\(|logFC| = 1\)</span>
</p>
</div>
</div>
<p>If the discarded pool is enriched for a certain cell type, we should observe increased expression of the corresponding marker genes.</p>
<p>Another concern is whether the cells removed by this procedure are preferentially derive from particular experimental groups. Reassuringly, Figure <a href="#fig:ambiguous-enriched-clusters-8-barplot-highlighting-removed-droplets">10</a> shows that this is not the case; most of the removed droplets are <code>ambiguous</code>, as we expect from the above analysis.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:ambiguous-enriched-clusters-10-barplot-highlighting-removed-droplets"></span>
<img src="C057_Cooney.cell_selection_files/ambiguous-enriched-clusters-10-barplot-highlighting-removed-droplets-1.png" alt="Droplets retained after excluding droplets from cluster 9, stratified by `Sample`." width="624" />
<p class="caption">
Figure 12: Droplets retained after excluding droplets from cluster 9, stratified by <code>Sample</code>.
</p>
</div>
</div>
<h4 id="summary">Summary</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>We conclude that droplets in clusters 2 and 7 do not contain intact cells, but rather comprises ambient RNA, and therefore remove all droplets in these clusters. The removes 2771 droplets, retaining 10634 for further analysis. We do not exclude droplets from cluster 8 and 9 because these look to comprise some distinct cell type(s) that may be more prevalent in the <code>Uninfected</code> samples.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="re-processing">Re-processing</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>There are 13 clusters detected, shown on the UMAP plot Figure <a href="#fig:no-ambiguous-enriched-clusters-clusterplot-umap">13</a> and broken down by experimental factors in Figure <a href="#fig:no-ambiguous-enriched-clusters-cluster-barplot">14</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:no-ambiguous-enriched-clusters-clusterplot-umap"></span>
<img src="C057_Cooney.cell_selection_files/no-ambiguous-enriched-clusters-clusterplot-umap-1.png" alt="UMAP plot, where each point represents a droplet and is coloured according to the legend." width="624" />
<p class="caption">
Figure 13: UMAP plot, where each point represents a droplet and is coloured according to the legend.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:no-ambiguous-enriched-clusters-cluster-barplot"></span>
<img src="C057_Cooney.cell_selection_files/no-ambiguous-enriched-clusters-cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 14: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>Notably:</p>
<ul>
<li>Still, some of the clusters are highly treatment-specific.</li>
<li><strong>Still, some of the clusters are enriched for <code>ambiguous</code> droplets.</strong></li>
</ul>
<p>We therefore repeat the process described in <a href="#first-pass">First pass</a>.</p>
<h2 id="refinement">Refinement</h2>
<h3 id="motivation-2">Motivation</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>There are still clusters with unusually large proportions of droplets being of <code>ambiguous</code> origin, namely clusters 2 and 8 (proportions = 0.82 and 0.73, respectively).</p>
<p>These droplets cannot be assigned to a sample of origin and so are not useful in downstream analyses. However, before we exclude them, we want to ensure we aren’t systematically excluding a biologically relevant cell population.</p>
<p>We would further like to know that this isn’t systematically excluding droplets from a particular sample or treatment group. This is challenging because by definition most of these droplets are of <code>ambiguous</code> origin. However, we can check if the ‘labelled’ droplets in those clusters are enriched for particular samples.</p>
<h3 id="analysis-1">Analysis</h3>
<p>Figure <a href="#fig:refined-ambiguous-enriched-clusters-umap">15</a> highlights droplets from these clusters on the UMAP plot, showing that they are largely distinct from the bulk of droplets from the dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:refined-ambiguous-enriched-clusters-umap"></span>
<img src="C057_Cooney.cell_selection_files/refined-ambiguous-enriched-clusters-umap-1.png" alt="UMAP plot highlighting droplets from clusters that are enriched for `ambiguous` droplets. Cluster labels are overlaid on cluster centroids." width="624" />
<p class="caption">
Figure 15: UMAP plot highlighting droplets from clusters that are enriched for <code>ambiguous</code> droplets. Cluster labels are overlaid on cluster centroids.
</p>
</div>
</div>
<p>Figure <a href="#fig:refined-ambiguous-enriched-clusters-qc-plot">16</a> shows that the QC metrics of droplets from these clusters are distinct from the other clusters, having smaller library sizes and fewer genes detected (but no difference in mitochondrial proportions).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:refined-ambiguous-enriched-clusters-qc-plot"></span>
<img src="C057_Cooney.cell_selection_files/refined-ambiguous-enriched-clusters-qc-plot-1.png" alt="QC metrics of droplets by cluster, highlighting droplets from clusters that are enriched for `ambiguous` droplets (orange)." width="624" />
<p class="caption">
Figure 16: QC metrics of droplets by cluster, highlighting droplets from clusters that are enriched for <code>ambiguous</code> droplets (orange).
</p>
</div>
</div>
<p>Unlike in the <a href="#first-pass">First pass</a>, we have now clearly identified clusters of droplets that have abnormal QC metrics (rather than a mixture of abnormal and typical values). Droplets with small library sizes are likely to contain ambient RNA rather than intact cells.</p>
<h4 id="checking-for-removal-of-biologically-relevant-subpopulations-1">Checking for removal of biologically relevant subpopulations</h4>
<p>We check for cell type loss by looking for systematic differences in gene expression between the discarded and retained droplets, doing this separately for each of the suspicious clusters.</p>
<h5 id="cluster-2-1">Cluster 2</h5>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:refined-ambiguous-enriched-clusters-2-md-plot">17</a> shows the result of this analysis, highlighting that there are a few genes with a large <span class="math inline">\(logFC\)</span> between ‘lost’ and ‘kept’ droplets.</p>
<aside>
An interactive version of Figure <a href="#fig:refined-ambiguous-enriched-clusters-2-md-plot">17</a> is available from <a href="../output/glimma-plots/refined-ambiguous-enriched-clusters-2-md-plot.html"><code>output/glimma-plots/refined-ambiguous-enriched-clusters-2-md-plot.html</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:refined-ambiguous-enriched-clusters-2-md-plot"></span>
<img src="C057_Cooney.cell_selection_files/refined-ambiguous-enriched-clusters-2-md-plot-1.png" alt="Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate $|logFC| = 1$" width="624" />
<p class="caption">
Figure 17: Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate <span class="math inline">\(|logFC| = 1\)</span>
</p>
</div>
</div>
<p>If the discarded pool is enriched for a certain cell type, we should observe increased expression of the corresponding marker genes. Although there is a subset of genes with large <span class="math inline">\(logFC\)</span>, Figure <a href="#fig:refined-ambiguous-enriched-clusters-2-markers">18</a> shows that these same genes are expressed in other clusters which means that we are unlikely to be systematically excluding a biologically relevant population of cells.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:refined-ambiguous-enriched-clusters-2-markers"></span>
<img src="C057_Cooney.cell_selection_files/refined-ambiguous-enriched-clusters-2-markers-1.png" alt="Expression of genes with top-9 largest $|logFC|$ when comparing the the discarded cells to the retained cell. Each of these genes is expressed at a higher (or comparable) level in droplets from at least one of the retained clusters." width="624" />
<p class="caption">
Figure 18: Expression of genes with top-9 largest <span class="math inline">\(|logFC|\)</span> when comparing the the discarded cells to the retained cell. Each of these genes is expressed at a higher (or comparable) level in droplets from at least one of the retained clusters.
</p>
</div>
</div>
<p>Another concern is whether the cells removed by this procedure are preferentially derive from particular experimental groups. Reassuringly, Figure <a href="#fig:refined-ambiguous-enriched-clusters-2-barplot-highlighting-removed-droplets">19</a> shows that this is not the case; most of the removed droplets are <code>ambiguous</code>, as we expect from the above analysis.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:refined-ambiguous-enriched-clusters-2-barplot-highlighting-removed-droplets"></span>
<img src="C057_Cooney.cell_selection_files/refined-ambiguous-enriched-clusters-2-barplot-highlighting-removed-droplets-1.png" alt="Droplets retained after excluding droplets from cluster 2, stratified by `Sample`." width="624" />
<p class="caption">
Figure 19: Droplets retained after excluding droplets from cluster 2, stratified by <code>Sample</code>.
</p>
</div>
</div>
<h5 id="cluster-8-1">Cluster 8</h5>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:refined-ambiguous-enriched-clusters-8-md-plot">20</a> shows the result of this analysis, highlighting that there are few genes with a large <span class="math inline">\(logFC\)</span> between ‘lost’ and ‘kept’ droplets and that the systematically downregulated genes are largely ribosomal protein genes.</p>
<aside>
An interactive version of Figure <a href="#fig:refined-ambiguous-enriched-clusters-8-md-plot">20</a> is available from <a href="../output/glimma-plots/refined-ambiguous-enriched-clusters-8-md-plot.html"><code>output/glimma-plots/refined-ambiguous-enriched-clusters-8-md-plot.html</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:refined-ambiguous-enriched-clusters-8-md-plot"></span>
<img src="C057_Cooney.cell_selection_files/refined-ambiguous-enriched-clusters-8-md-plot-1.png" alt="Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate $|logFC| = 1$" width="624" />
<p class="caption">
Figure 20: Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate <span class="math inline">\(|logFC| = 1\)</span>
</p>
</div>
</div>
<p>If the discarded pool is enriched for a certain cell type, we should observe increased expression of the corresponding marker genes. Although there is a subset of genes with large <span class="math inline">\(logFC\)</span>, Figure <a href="#fig:refined-ambiguous-enriched-clusters-8-markers">21</a> shows that these same genes are expressed in other clusters which means that we are unlikely to be systematically excluding a biologically relevant population of cells.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:refined-ambiguous-enriched-clusters-8-markers"></span>
<img src="C057_Cooney.cell_selection_files/refined-ambiguous-enriched-clusters-8-markers-1.png" alt="Expression of genes with top-9 largest $|logFC|$ when comparing the the discarded cells to the retained cell. Each of these genes is expressed at a higher (or comparable) level in droplets from at least one of the retained clusters." width="624" />
<p class="caption">
Figure 21: Expression of genes with top-9 largest <span class="math inline">\(|logFC|\)</span> when comparing the the discarded cells to the retained cell. Each of these genes is expressed at a higher (or comparable) level in droplets from at least one of the retained clusters.
</p>
</div>
</div>
<p>Another concern is whether the cells removed by this procedure are preferentially derive from particular experimental groups. Reassuringly, Figure <a href="#fig:refined-ambiguous-enriched-clusters-8-barplot-highlighting-removed-droplets">22</a> shows that this is not the case; most of the removed droplets are <code>ambiguous</code>, as we expect from the above analysis.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:refined-ambiguous-enriched-clusters-8-barplot-highlighting-removed-droplets"></span>
<img src="C057_Cooney.cell_selection_files/refined-ambiguous-enriched-clusters-8-barplot-highlighting-removed-droplets-1.png" alt="Droplets retained after excluding droplets from cluster 8, stratified by `Sample`." width="624" />
<p class="caption">
Figure 22: Droplets retained after excluding droplets from cluster 8, stratified by <code>Sample</code>.
</p>
</div>
</div>
<h4 id="summary-1">Summary</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>We conclude that droplets in clusters 2 and 8 do not contain intact cells, but rather comprises ambient RNA, and therefore remove all droplets in these clusters. The removes 435 droplets, retaining 10199 for further analysis.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="re-processing-1">Re-processing</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>There are 11 clusters detected, shown on the UMAP plot Figure <a href="#fig:no-refined-ambiguous-enriched-clusters-clusterplot-umap">23</a> and broken down by experimental factors in Figure <a href="#fig:no-refined-ambiguous-enriched-clusters-cluster-barplot">24</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:no-refined-ambiguous-enriched-clusters-clusterplot-umap"></span>
<img src="C057_Cooney.cell_selection_files/no-refined-ambiguous-enriched-clusters-clusterplot-umap-1.png" alt="UMAP plot, where each point represents a droplet and is coloured according to the legend." width="624" />
<p class="caption">
Figure 23: UMAP plot, where each point represents a droplet and is coloured according to the legend.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:no-refined-ambiguous-enriched-clusters-cluster-barplot"></span>
<img src="C057_Cooney.cell_selection_files/no-refined-ambiguous-enriched-clusters-cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 24: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>Notably:</p>
<ul>
<li>Still, some of the clusters are highly treatment-specific.</li>
<li><strong>Now, none of the clusters are enriched for <code>ambiguous</code> droplets.</strong></li>
</ul>
<h1 id="doublets">Doublets</h1>
<h2 id="motivation-3">Motivation</h2>
<p>For multiplexed samples such as these, we can identify (most of) the doublet cells based on the droplets that have multiple labels. The idea here is that cells from the same sample are labelled in a unique manner and that cells from all samples are then mixed together and the multiplexed pool is subjected to scRNA-seq, avoiding batch effects and simplifying the logistics of processing a large number of samples. Moreover, most per-cell libraries are expected to contain one label that can be used to assign that cell to its sample of origin. Cell libraries containing two labels are thus likely to be doublets of cells from different samples.</p>
<h2 id="analysis-2">Analysis</h2>
<p>We first use the HTOs to identify ‘labelled’ doublets and then leverage these in to identify ‘unlabelled’ doublets in the gene expression space.</p>
<h3 id="labelled-doublets">‘Labelled’ doublets</h3>
<p>For this experiment, the sample labelling is done with HTOs. With 5 unique HTOs, there are <span class="math inline">\({5 \choose 2} = 10\)</span> ways a droplet could be labelled with 2 HTOs. The cell hashing data will not identify all doublets, most obviously those formed between cells from the same sample. It will also be near impossible to identify those doublets formed between a cell labelled with all 5 HTOs (i.e. a cell from the <code>uninfected_3</code> sample) and another cell labelled with a single HTO.</p>
<p>We can easily identify the HTO-derived doublets (droplets with <code>Sample</code> labelled as <code>doublet</code>; n = 410, 4% of remaining droplets). We term these the ‘labelled’ droplets.</p>
<h3 id="unlabelled-dropets">‘Unlabelled’ dropets</h3>
<p>One obvious limitation of relying solely on the HTO-derived doublets is that doublets of cells marked with the same HTO or those doublets formed between a cell labelled with all 5 HTOs (i.e. a cell from the <code>uninfected_3</code> sample) and another cell labelled with a single HTO are not detected<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>.</p>
<p>To avoid this, we recover the remaining intra-sample doublets based on their similarity with ‘known’ doublets in gene expression space, i.e. ‘guilt by association’.</p>
<p>For each droplet, we calculate the proportion of its nearest neighbours that are ‘labelled’ doublets<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>. Intra-sample doublets should have values of this metric under the assumption that their gene expression profiles are similar to inter-sample doublets involving the same combination of cell states/types.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:doublet-umap">25</a> highlights the location of ‘labelled’ doublets and those predicted to be doublets on the UMAP plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:doublet-umap"></span>
<img src="C057_Cooney.cell_selection_files/doublet-umap-1.png" alt="UMAP plot, where each point is a cell and is colored by the whether or not it is a 'labelled' doublet (left), by the proportion of neighbouring droplets that are 'labelled' droplets (centre), and whether or not it is a predicted 'unlabelled' doublet (right)." width="624" />
<p class="caption">
Figure 25: UMAP plot, where each point is a cell and is colored by the whether or not it is a ‘labelled’ doublet (left), by the proportion of neighbouring droplets that are ‘labelled’ droplets (centre), and whether or not it is a predicted ‘unlabelled’ doublet (right).
</p>
</div>
</div>
<p>Reassuringly, most of the ‘unlabelled’ doublets come from the droplets currently labelled as originating from the <code>uninfected_3</code> sample, which was labelled with all 5 HTOs, or the <code>ambiguous</code> sample, for bth of which it was difficult to identify ‘labelled’ doublets.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-18">Table 1: </span>Number of predicted ‘unlabelled’ doublets (<code>TRUE</code> column) that were labelled as coming from each <code>Sample</code>. Percentages are with respect to the total number of droplets from eahc <code>Sample</code>.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">Sample/predicted_doublet</th>
<th style="text-align: left;">FALSE</th>
<th style="text-align: left;">TRUE</th>
<th style="text-align: left;">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">infected_1</td>
<td style="text-align: left;">99.5% (1334)</td>
<td style="text-align: left;">0.5% (7)</td>
<td style="text-align: left;">100.0% (1341)</td>
</tr>
<tr class="even">
<td style="text-align: left;">infected_2</td>
<td style="text-align: left;">99.4% (1168)</td>
<td style="text-align: left;">0.6% (7)</td>
<td style="text-align: left;">100.0% (1175)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">infected_3</td>
<td style="text-align: left;">99.3% (1320)</td>
<td style="text-align: left;">0.7% (9)</td>
<td style="text-align: left;">100.0% (1329)</td>
</tr>
<tr class="even">
<td style="text-align: left;">uninfected_1</td>
<td style="text-align: left;">99.7% (1829)</td>
<td style="text-align: left;">0.3% (5)</td>
<td style="text-align: left;">100.0% (1834)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">uninfected_2</td>
<td style="text-align: left;">99.6% (2082)</td>
<td style="text-align: left;">0.4% (9)</td>
<td style="text-align: left;">100.0% (2091)</td>
</tr>
<tr class="even">
<td style="text-align: left;">uninfected_3</td>
<td style="text-align: left;">98.8% (1423)</td>
<td style="text-align: left;">1.2% (18)</td>
<td style="text-align: left;">100.0% (1441)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">doublet</td>
<td style="text-align: left;">100.0% (410)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">100.0% (410)</td>
</tr>
<tr class="even">
<td style="text-align: left;">ambiguous</td>
<td style="text-align: left;">94.6% (547)</td>
<td style="text-align: left;">5.4% (31)</td>
<td style="text-align: left;">100.0% (578)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Total</td>
<td style="text-align: left;">99.2% (10113)</td>
<td style="text-align: left;">0.8% (86)</td>
<td style="text-align: left;">100.0% (10199)</td>
</tr>
</tbody>
</table>
</div>
<h3 id="checking-for-removal-of-biologically-relevant-subpopulations-2">Checking for removal of biologically relevant subpopulations</h3>
<p>We again check for cell type loss by looking for systematic differences in gene expression between the discarded and retained cells.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:doublet-md-plot">26</a> shows the result of this analysis, highlighting that there are few genes with a large <span class="math inline">\(logFC\)</span> between ‘lost’ and ‘kept’ droplets.</p>
<aside>
An interactive version of Figure <a href="#fig:doublet-md-plot">26</a> is available from <a href="../output/glimma-plots/doublet-md-plot.html"><code>output/glimma-plots/doublet-md-plot.html</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:doublet-md-plot"></span>
<img src="C057_Cooney.cell_selection_files/doublet-md-plot-1.png" alt="Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate $|logFC| = 1$" width="624" />
<p class="caption">
Figure 26: Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate <span class="math inline">\(|logFC| = 1\)</span>
</p>
</div>
</div>
<p>If the discarded pool is enriched for a certain cell type, we should observe increased expression of the corresponding marker genes.</p>
<p>Another concern is whether the cells removed by this procedure are preferentially derive from particular experimental groups. Reassuringly, Figure <a href="#fig:doublet-barplot-highlighting-removed-droplets">27</a> shows that this is not the case; most of the removed droplets are ‘labelled’ doublets, as we expect from the above analysis.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:doublet-barplot-highlighting-removed-droplets"></span>
<img src="C057_Cooney.cell_selection_files/doublet-barplot-highlighting-removed-droplets-1.png" alt="Droplets removed after excluding 'labelled' and 'unlabelled' predicted doublets, stratified by `Sample`." width="624" />
<p class="caption">
Figure 27: Droplets removed after excluding ‘labelled’ and ‘unlabelled’ predicted doublets, stratified by <code>Sample</code>.
</p>
</div>
</div>
<h4 id="summary-2">Summary</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>We opt to remove both the ‘labelled’ known doublets and ‘unlabelled’ predicted doublets from the dataset. The removes 496 droplets, retaining 9703 for further analysis. The dataset may still contain some ‘unlabelled’ doublets, but their number should be very small owing to the experimental design.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="re-processing-2">Re-processing</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>There are 9 clusters detected, shown on the UMAP plot Figure <a href="#fig:no-doublets-clusterplot-umap-clusterplot-umap">28</a> and broken down by experimental factors in Figure <a href="#fig:no-doublets-clusterplot-umap-cluster-barplot">29</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:no-doublets-clusterplot-umap-clusterplot-umap"></span>
<img src="C057_Cooney.cell_selection_files/no-doublets-clusterplot-umap-clusterplot-umap-1.png" alt="UMAP plot, where each point represents a droplet and is coloured according to the legend." width="624" />
<p class="caption">
Figure 28: UMAP plot, where each point represents a droplet and is coloured according to the legend.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:no-doublets-clusterplot-umap-cluster-barplot"></span>
<img src="C057_Cooney.cell_selection_files/no-doublets-clusterplot-umap-cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 29: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>Notably:</p>
<ul>
<li>Still, some of the clusters are highly treatment-specific.</li>
<li>Still, none of the clusters are enriched for <code>ambiguous</code> droplets.</li>
<li><strong>Now, none of the clusters are highly sample-specific.</strong></li>
</ul>
<h1 id="selection-of-labelled-singlets">Selection of ‘labelled’ singlets</h1>
<h2 id="motivation-4">Motivation</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>There is no cluster with a notably large number of <code>ambiguous</code> droplets. If there were, our concern would be that this could indicate a cell subpopulation (e.g., cell type) that is systematically failing to be labelled by one or more of the HTOs.</p>
<h2 id="analysis-3">Analysis</h2>
<p>Since there are no clusters enriched for <code>ambiguous</code> droplets, there is no cluster-level analysis to be performed. Instead, we simply remove the droplets labelled as <code>ambiguous</code>.</p>
<h3 id="checking-for-removal-of-biologically-relevant-subpopulations-3">Checking for removal of biologically relevant subpopulations</h3>
<p>We again check for cell type loss by looking for systematic differences in gene expression between the discarded and retained cells.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:labelled-singlets-md-plot">30</a> shows the result of this analysis, highlighting that there are few genes with a large <span class="math inline">\(logFC\)</span> between ‘lost’ and ‘kept’ droplets.</p>
<aside>
An interactive version of Figure <a href="#fig:labelled-singlets-md-plot">30</a> is available from <a href="../output/glimma-plots/labelled-singlets-md-plot.html"><code>output/glimma-plots/labelled-singlets-md-plot.html</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:labelled-singlets-md-plot"></span>
<img src="C057_Cooney.cell_selection_files/labelled-singlets-md-plot-1.png" alt="Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate $|logFC| = 1$" width="624" />
<p class="caption">
Figure 30: Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate <span class="math inline">\(|logFC| = 1\)</span>
</p>
</div>
</div>
<p>If the discarded pool is enriched for a certain cell type, we should observe increased expression of the corresponding marker genes.</p>
<p>Another concern is whether the cells removed by this procedure are preferentially derive from particular experimental groups. Reassuringly, Figure <a href="#fig:labelled-singlets-barplot-highlighting-removed-droplets">31</a> shows that this is not the case; all of the removed droplets are <code>ambiguous</code> droplets, as we expect from the above analysis.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:labelled-singlets-barplot-highlighting-removed-droplets"></span>
<img src="C057_Cooney.cell_selection_files/labelled-singlets-barplot-highlighting-removed-droplets-1.png" alt="Droplets removed after excluding `Negative` and `unassigned` droplets, stratified by `Sample`." width="624" />
<p class="caption">
Figure 31: Droplets removed after excluding <code>Negative</code> and <code>unassigned</code> droplets, stratified by <code>Sample</code>.
</p>
</div>
</div>
<h3 id="summary-3">Summary</h3>
<p>We can conclude that it is unlikely that our HTO labelling has systematically under-labelled a biologically relevant cell type. Furthermore, the <code>ambiguous</code> droplets are not useful in downstream analyses because we cannot match them to a treatment, and so we removed all <code>ambiguous</code> droplets from the data. The removes 547 droplets, retaining 9156 for further analysis.</p>
<p>At this point, our dataset only contains ‘labelled’ singlets, i.e. those droplets we are confident are single cells and that we can assign to a single sample. These are the cells we want to use in our downstream analysis.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="re-processing-3">Re-processing</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>There are 10 clusters detected, shown on the UMAP plot Figure <a href="#fig:labelled-singlets-clusterplot-umap">32</a> and broken down by experimental factors in Figure <a href="#fig:labelled-singlets-cluster-barplot">33</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:labelled-singlets-clusterplot-umap"></span>
<img src="C057_Cooney.cell_selection_files/labelled-singlets-clusterplot-umap-1.png" alt="UMAP plot of the updated data, where each point represents a droplet and is coloured according to the legend." width="624" />
<p class="caption">
Figure 32: UMAP plot of the updated data, where each point represents a droplet and is coloured according to the legend.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:labelled-singlets-cluster-barplot"></span>
<img src="C057_Cooney.cell_selection_files/labelled-singlets-cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 33: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>Notably:</p>
<ul>
<li>Still, some of the clusters are highly treatment-specific.</li>
<li>Still, none of the clusters are highly sample-specific.</li>
</ul>
<h1 id="exclusion-of-unwanted-cell-types">Exclusion of unwanted cell types</h1>
<h2 id="motivation-5">Motivation</h2>
<p>James enriched for CD4+ T-cells prior to performing scRNA-seq. However, an enrichment procedure is just that and unwanted cell types may sneak through.</p>
<p>We can categorise unwanted cell types as either:</p>
<ol type="1">
<li>‘Known’ unwanted cell types: Unwanted cell types that we did know to look for <em>a priori</em> (i.e. unwanted cell types we know may have snuck through the enrichment process)</li>
<li>‘Unknown’ unwanted cell types: Unwanted cell types that we didn’t know to look for <em>a priori</em> (i.e. unwanted cell types that we didn’t may have snuck through the enrichment process).</li>
</ol>
<p>We will use different analysis strategies to identify the ‘known’ and ‘unknown’ unwanted cell types. Once we have identified the unwanted cell types, we will exclude the unwanted these cells from further analysis.</p>
<h2 id="unknown-unwanted-cell-types">‘Unknown’ unwanted cell types</h2>
<h3 id="motivation-6">Motivation</h3>
<p>A conceptually straightforward annotation approach is to compare the single-cell expression profiles with previously annotated reference data sets. Labels can then be assigned to each cell in our uncharacterised test data set based on the most similar reference sample(s), for some definition of “similar”. Any published and labelled RNA-seq data set (bulk or single-cell) can be used as a reference, though its reliability depends greatly on the expertise of the original authors who assigned the labels in the first place.</p>
<p><em><a href="https://bioconductor.org/packages/3.11/SingleR">SingleR</a></em> is one such automatic annotation method for scRNA-seq data <span class="citation" data-cites="aran2019reference">(Aran et al. <a href="#ref-aran2019reference" role="doc-biblioref">2019</a>)</span>. Given a reference data set(s) of samples<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a> with known labels, it labels new cells from a test data set based on similarity to the reference set(s). Specifically, for each test cell:</p>
<ol type="1">
<li>We compute the Spearman correlation between its expression profile and that of each reference sample.</li>
<li>We define the per-label score as a fixed quantile (by default, 0.8) of the distribution of correlations.</li>
<li>We repeat this for all labels and we take the label with the highest score as the annotation for this cell.</li>
<li>Finally, we perform a fine-tuning step:</li>
</ol>
<ul>
<li>The reference data set is subsetted to only include labels with scores close to the maximum.</li>
<li>Scores are recomputed using only marker genes for the subset of labels.</li>
<li>This is iterated until one label remains.</li>
</ul>
<p>For visualization purposes we use the normalized scores, which are the scores linearly adjusted for each cell so that the smallest score is 0 and the largest score is 1. This is followed by cubing of the adjusted scores to improve dynamic range near 1.</p>
<p>Automatic annotation provides a convenient way of transferring biological knowledge across data sets. In this manner, the burden of interpreting clusters and defining marker genes only has to be done once (i.e. for the reference set).</p>
<p><em><a href="https://bioconductor.org/packages/3.11/SingleR">SingleR</a></em> can annotate at both the cluster-level and at the cell-level. The trade-off between cluster-level and cell-level annotation is one of increased robustness to noise (cluster-level) vs. increased resolution (cell-level). Poor-quality assignments with ‘low’ scores are labelled as <code>NA</code>.</p>
<h4 id="available-references">Available references</h4>
<p><em><a href="https://bioconductor.org/packages/3.11/SingleR">SingleR</a></em> includes several in-built reference data sets, some of which are relevant for this study:</p>
<ul>
<li><code>HumanPrimaryCellAtlasData</code> (<code>HPCA</code>): The Human Primary Cell Atlas <span class="citation" data-cites="hpcaRef">(Mabbott et al. <a href="#ref-hpcaRef" role="doc-biblioref">2013</a>)</span></li>
<li><code>BlueprintEncodeData</code> (<code>BE</code>): Blueprint <span class="citation" data-cites="blueprintRef">(Martens and Stunnenberg <a href="#ref-blueprintRef" role="doc-biblioref">2013</a>)</span> and Encode <span class="citation" data-cites="encodeRef">(The ENCODE Project Consortium <a href="#ref-encodeRef" role="doc-biblioref">2012</a>)</span></li>
<li><code>DatabaseImmuneCellExpressionData</code> (<code>DICE</code>): The <a href="https://dice-database.org/">Database of Immune Cell Expression (DICE)</a> <span class="citation" data-cites="diceRef">(Schmiedel et al. <a href="#ref-diceRef" role="doc-biblioref">2018</a>)</span></li>
<li><code>NovershternHematopoieticData</code> (<code>NH</code>): The <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE24759">Novershtern Hematopoietic Cell Data (GSE24759)</a> <span class="citation" data-cites="nhRef">(Novershtern et al. <a href="#ref-nhRef" role="doc-biblioref">2011</a>)</span></li>
<li><code>MonacoImmuneData</code> (<code>MI</code>): The <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107011">Monaco Immune Cell Data (GSE107011)</a> <span class="citation" data-cites="monaco_immuneRef">(Monaco et al. <a href="#ref-monaco_immuneRef" role="doc-biblioref">2019</a>)</span>.</li>
</ul>
<p>These bulk RNA-seq and microarray data sets were obtained from pre-sorted cell populations, i.e., the cell labels of these samples were mostly derived based on the respective sorting/purification strategy, not via <em>in silico</em> prediction methods.</p>
<p>The characteristics of each data set are summarized below<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a>:</p>
<table>
<colgroup>
<col style="width: 21%" />
<col style="width: 11%" />
<col style="width: 15%" />
<col style="width: 25%" />
<col style="width: 14%" />
<col style="width: 11%" />
</colgroup>
<thead>
<tr class="header">
<th>Reference</th>
<th>Samples</th>
<th>Sample types</th>
<th>No. of main labels</th>
<th>No. of fine labels</th>
<th>Cell type focus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>HPCA</code></td>
<td>713</td>
<td>microarrays of sorted cell populations</td>
<td>37</td>
<td>157</td>
<td>Non-specific</td>
</tr>
<tr class="even">
<td><code>BE</code></td>
<td>259</td>
<td>RNA-seq</td>
<td>24</td>
<td>43</td>
<td>Non-specific</td>
</tr>
<tr class="odd">
<td><code>DICE</code></td>
<td>1561</td>
<td>RNA-seq</td>
<td>5</td>
<td>15</td>
<td>Immune</td>
</tr>
<tr class="even">
<td><code>NH</code></td>
<td>211</td>
<td>microarrays of sorted cell populations</td>
<td>17</td>
<td>38</td>
<td>Hematopoietic &amp; Immune</td>
</tr>
<tr class="odd">
<td><code>MI</code></td>
<td>114</td>
<td>RNA-seq</td>
<td>11</td>
<td>29</td>
<td>Immune</td>
</tr>
</tbody>
</table>
<p>The available sample types in each set can be viewed in the collapsible sections below.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<details>
<p><summary><code>HumanPrimaryCellAtlasData</code> Labels</summary></p>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Astrocyte</td>
<td style="text-align: left;">Astrocyte:Embryonic_stem_cell-derived</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:CXCR4-_centrocyte</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:CXCR4+_centroblast</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:Germinal_center</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:immature</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:Memory</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:Naive</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B_cell</td>
<td style="text-align: left;">B_cell:Plasma_cell</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">BM</td>
<td style="text-align: left;">BM</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BM &amp; Prog.</td>
<td style="text-align: left;">BM</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">Chondrocytes</td>
<td style="text-align: left;">Chondrocytes:MSC-derived</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CMP</td>
<td style="text-align: left;">CMP</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived</td>
<td style="text-align: right;">19</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:A._fumigatus_germ_tubes_6h</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:AEC-conditioned</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:AM580</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:anti-DC-SIGN_2h</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:antiCD40/VAF347</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:CD40L</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:Galectin-1</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:immature</td>
<td style="text-align: right;">20</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:LPS</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:mature</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:Poly(IC)</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:rosiglitazone</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:rosiglitazone/AGN193109</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC:monocyte-derived:Schuler_treatment</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Embryonic_stem_cells</td>
<td style="text-align: left;">Embryonic_stem_cells</td>
<td style="text-align: right;">17</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:blood_vessel</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC</td>
<td style="text-align: right;">16</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:B._anthracis_LT</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:Borrelia_burgdorferi</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:FPV-infected</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:H5N1-infected</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:IFNg</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:IL-1b</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:PR8-infected</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:Serum_Amyloid_A</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:HUVEC:VEGF</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:lymphatic</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:lymphatic:KSHV</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: left;">Endothelial_cells:lymphatic:TNFa_48h</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Epithelial_cells</td>
<td style="text-align: left;">Epithelial_cells:bladder</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Epithelial_cells</td>
<td style="text-align: left;">Epithelial_cells:bronchial</td>
<td style="text-align: right;">10</td>
</tr>
<tr class="even">
<td style="text-align: left;">Erythroblast</td>
<td style="text-align: left;">Erythroblast</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Fibroblasts</td>
<td style="text-align: left;">Fibroblasts:breast</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fibroblasts</td>
<td style="text-align: left;">Fibroblasts:foreskin</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gametocytes</td>
<td style="text-align: left;">Gametocytes:oocyte</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gametocytes</td>
<td style="text-align: left;">Gametocytes:spermatocyte</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GMP</td>
<td style="text-align: left;">GMP</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Hepatocytes</td>
<td style="text-align: left;">Hepatocytes</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSC_-G-CSF</td>
<td style="text-align: left;">HSC_-G-CSF</td>
<td style="text-align: right;">10</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSC_CD34+</td>
<td style="text-align: left;">HSC_CD34+</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:adipose_stem_cell-derived:lentiviral</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:adipose_stem_cell-derived:minicircle-derived</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:adipose_stem_cells</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:CRL2097_foreskin</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:CRL2097_foreskin-derived:d20_hepatic_diff</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:CRL2097_foreskin-derived:undiff.</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:fibroblast-derived:Direct_del._reprog</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:fibroblast-derived:Retroviral_transf</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:fibroblasts</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:foreskin_fibrobasts</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:iPS:minicircle-derived</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_1lox-17Puro-10</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_1lox-17Puro-5</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_1lox-21Puro-20</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_1lox-21Puro-26</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_2lox-17</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_2lox-21</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_2lox-22</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_2lox-5</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:PDB_fibroblasts</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:skin_fibroblast</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: left;">iPS_cells:skin_fibroblast-derived</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IFNg</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IL19</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IL1b</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IL20</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IL22</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IL24</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:IL26</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes:KGF</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:Alveolar</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:Alveolar:B._anthacis_spores</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived</td>
<td style="text-align: right;">26</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:IFNa</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:IL-4/cntrl</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:IL-4/Dex/cntrl</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:IL-4/Dex/TGFb</td>
<td style="text-align: right;">10</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:IL-4/TGFb</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:M-CSF</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:M-CSF/IFNg</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:M-CSF/IFNg/Pam3Cys</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:M-CSF/Pam3Cys</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: left;">Macrophage:monocyte-derived:S._aureus</td>
<td style="text-align: right;">15</td>
</tr>
<tr class="odd">
<td style="text-align: left;">MEP</td>
<td style="text-align: left;">MEP</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte</td>
<td style="text-align: right;">27</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:anti-FcgRIIB</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:CD14+</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:CD16-</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:CD16+</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:CXCL4</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:F._tularensis_novicida</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:leukotriene_D4</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:MCSF</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: left;">Monocyte:S._typhimurium_flagellin</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">MSC</td>
<td style="text-align: left;">MSC</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Myelocyte</td>
<td style="text-align: left;">Myelocyte</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neuroepithelial_cell</td>
<td style="text-align: left;">Neuroepithelial_cell:ESC-derived</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neurons</td>
<td style="text-align: left;">Neurons:adrenal_medulla_cell_line</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neurons</td>
<td style="text-align: left;">Neurons:ES_cell-derived_neural_precursor</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neurons</td>
<td style="text-align: left;">Neurons:Schwann_cell</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophil</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophil:commensal_E._coli_MG1655</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophil:GM-CSF_IFNg</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophil:inflam</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophil:LPS</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophil:uropathogenic_E._coli_UTI89</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK_cell</td>
<td style="text-align: left;">NK_cell</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK_cell</td>
<td style="text-align: left;">NK_cell:CD56hiCD62L+</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK_cell</td>
<td style="text-align: left;">NK_cell:IL2</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Osteoblasts</td>
<td style="text-align: left;">Osteoblasts</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="even">
<td style="text-align: left;">Osteoblasts</td>
<td style="text-align: left;">Osteoblasts:BMP2</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Platelets</td>
<td style="text-align: left;">Platelets</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pre-B_cell_CD34-</td>
<td style="text-align: left;">Pre-B_cell_CD34-</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Pro-B_cell_CD34+</td>
<td style="text-align: left;">Pro-B_cell_CD34+</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pro-Myelocyte</td>
<td style="text-align: left;">Pro-Myelocyte</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Smooth_muscle_cells</td>
<td style="text-align: left;">Smooth_muscle_cells:bronchial</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Smooth_muscle_cells</td>
<td style="text-align: left;">Smooth_muscle_cells:bronchial:vit_D</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Smooth_muscle_cells</td>
<td style="text-align: left;">Smooth_muscle_cells:umbilical_vein</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Smooth_muscle_cells</td>
<td style="text-align: left;">Smooth_muscle_cells:vascular</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Smooth_muscle_cells</td>
<td style="text-align: left;">Smooth_muscle_cells:vascular:IL-17</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CCR10-CLA+1,25(OH)2_vit_D3/IL-12</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CCR10+CLA+1,25(OH)2_vit_D3/IL-12</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD4+</td>
<td style="text-align: right;">12</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD4+_central_memory</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD4+_effector_memory</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD4+_Naive</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD8+</td>
<td style="text-align: right;">16</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD8+_Central_memory</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD8+_effector_memory</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD8+_effector_memory_RA</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:CD8+_naive</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:effector</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:gamma-delta</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: left;">T_cell:Treg:Naive</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:adipose-derived_MSC_AM3</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:BM_MSC</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:BM_MSC:BMP2</td>
<td style="text-align: right;">12</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:BM_MSC:osteogenic</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:BM_MSC:TGFb3</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:CD326-CD56+</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:dental_pulp</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:iliac_MSC</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: left;">Tissue_stem_cells:lipoma-derived_MSC</td>
<td style="text-align: right;">2</td>
</tr>
</tbody>
</table>
</div>
</details>
<details>
<p><summary><code>BlueprintEncodeData</code> Labels</summary></p>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Adipocytes</td>
<td style="text-align: left;">Adipocytes</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">Adipocytes</td>
<td style="text-align: left;">Preadipocytes</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">B-cells</td>
<td style="text-align: left;">Class-switched memory B-cells</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B-cells</td>
<td style="text-align: left;">Memory B-cells</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">B-cells</td>
<td style="text-align: left;">naive B-cells</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B-cells</td>
<td style="text-align: left;">Plasma cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T-cells</td>
<td style="text-align: left;">CD4+ T-cells</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T-cells</td>
<td style="text-align: left;">CD4+ Tcm</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T-cells</td>
<td style="text-align: left;">CD4+ Tem</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T-cells</td>
<td style="text-align: left;">Tregs</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T-cells</td>
<td style="text-align: left;">CD8+ T-cells</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8+ T-cells</td>
<td style="text-align: left;">CD8+ Tcm</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T-cells</td>
<td style="text-align: left;">CD8+ Tem</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Chondrocytes</td>
<td style="text-align: left;">Chondrocytes</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: left;">DC</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial cells</td>
<td style="text-align: left;">Endothelial cells</td>
<td style="text-align: right;">18</td>
</tr>
<tr class="even">
<td style="text-align: left;">Endothelial cells</td>
<td style="text-align: left;">mv Endothelial cells</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Eosinophils</td>
<td style="text-align: left;">Eosinophils</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">Epithelial cells</td>
<td style="text-align: left;">Epithelial cells</td>
<td style="text-align: right;">18</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Erythrocytes</td>
<td style="text-align: left;">Erythrocytes</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fibroblasts</td>
<td style="text-align: left;">Fibroblasts</td>
<td style="text-align: right;">20</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">CLP</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">CMP</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">GMP</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">HSC</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">Megakaryocytes</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">MEP</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSC</td>
<td style="text-align: left;">MPP</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophages</td>
<td style="text-align: left;">Macrophages</td>
<td style="text-align: right;">18</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophages</td>
<td style="text-align: left;">Macrophages M1</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Macrophages</td>
<td style="text-align: left;">Macrophages M2</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Melanocytes</td>
<td style="text-align: left;">Melanocytes</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mesangial cells</td>
<td style="text-align: left;">Mesangial cells</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Monocytes</td>
<td style="text-align: right;">16</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Myocytes</td>
<td style="text-align: left;">Myocytes</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neurons</td>
<td style="text-align: left;">Neurons</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: right;">23</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Pericytes</td>
<td style="text-align: left;">Pericytes</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Skeletal muscle</td>
<td style="text-align: left;">Skeletal muscle</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Smooth muscle</td>
<td style="text-align: left;">Smooth muscle</td>
<td style="text-align: right;">16</td>
</tr>
</tbody>
</table>
</div>
</details>
<details>
<p><summary><code>DatabaseImmuneCellExpressionData</code> Labels</summary></p>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">B cells, naive</td>
<td style="text-align: right;">106</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Monocytes, CD14+</td>
<td style="text-align: right;">106</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Monocytes, CD16+</td>
<td style="text-align: right;">105</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">105</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, memory TREG</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, naive</td>
<td style="text-align: right;">103</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, naive TREG</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, naive, stimulated</td>
<td style="text-align: right;">102</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, TFH</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, Th1</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, Th1_17</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, Th17</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">T cells, CD4+, Th2</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD8+</td>
<td style="text-align: left;">T cells, CD8+, naive</td>
<td style="text-align: right;">104</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD8+</td>
<td style="text-align: left;">T cells, CD8+, naive, stimulated</td>
<td style="text-align: right;">102</td>
</tr>
</tbody>
</table>
</div>
</details>
<details>
<p><summary><code>MonacoImmuneData</code> Labels</summary></p>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Exhausted B cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Naive B cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Non-switched memory B cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Plasmablasts</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Switched memory B cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Basophils</td>
<td style="text-align: left;">Low-density basophils</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Follicular helper T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Naive CD4 T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">T regulatory cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Terminal effector CD4 T cells</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Th1 cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Th1/Th17 cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Th17 cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Th2 cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">Central memory CD8 T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">Effector memory CD8 T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">Naive CD8 T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">Terminal effector CD8 T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Dendritic cells</td>
<td style="text-align: left;">Myeloid dendritic cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Dendritic cells</td>
<td style="text-align: left;">Plasmacytoid dendritic cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Classical monocytes</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Intermediate monocytes</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Non classical monocytes</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: left;">Low-density neutrophils</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">Natural killer cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Progenitors</td>
<td style="text-align: left;">Progenitor cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells</td>
<td style="text-align: left;">MAIT cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells</td>
<td style="text-align: left;">Non-Vd2 gd T cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells</td>
<td style="text-align: left;">Vd2 gd T cells</td>
<td style="text-align: right;">4</td>
</tr>
</tbody>
</table>
</div>
</details>
<details>
<p><summary><code>NovershternHematopoieticData</code> Labels</summary></p>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Early B cells</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Mature B cells</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Mature B cells class able to switch</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Mature B cells class switched</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Naive B cells</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">Pro B cells</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Basophils</td>
<td style="text-align: left;">Basophils</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">CD4+ Central Memory</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">CD4+ Effector Memory</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: left;">Naive CD4+ T cells</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">CD8+ Central Memory</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">CD8+ Effector Memory</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">CD8+ Effector Memory RA</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: left;">Naive CD8+ T cells</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CMPs</td>
<td style="text-align: left;">Common myeloid progenitors</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Dendritic cells</td>
<td style="text-align: left;">Myeloid Dendritic Cells</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Dendritic cells</td>
<td style="text-align: left;">Plasmacytoid Dendritic Cells</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">Eosinophils</td>
<td style="text-align: left;">Eosinophils</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Erythroid cells</td>
<td style="text-align: left;">Erythroid_CD34- CD71- GlyA+</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">Erythroid cells</td>
<td style="text-align: left;">Erythroid_CD34- CD71+ GlyA-</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Erythroid cells</td>
<td style="text-align: left;">Erythroid_CD34- CD71+ GlyA+</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">Erythroid cells</td>
<td style="text-align: left;">Erythroid_CD34- CD71lo GlyA+</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Erythroid cells</td>
<td style="text-align: left;">Erythroid_CD34+ CD71+ GlyA-</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">GMPs</td>
<td style="text-align: left;">Granulocyte/monocyte progenitors</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Granulocytes</td>
<td style="text-align: left;">Colony Forming Unit-Granulocytes</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">Granulocytes</td>
<td style="text-align: left;">Granulocytes (Neutrophilic Metamyelocytes)</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Granulocytes</td>
<td style="text-align: left;">Granulocytes (Neutrophils)</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSCs</td>
<td style="text-align: left;">Hematopoietic stem cells_CD133+ CD34dim</td>
<td style="text-align: right;">10</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSCs</td>
<td style="text-align: left;">Hematopoietic stem cells_CD38- CD34+</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Megakaryocytes</td>
<td style="text-align: left;">Colony Forming Unit-Megakaryocytic</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Megakaryocytes</td>
<td style="text-align: left;">Megakaryocytes</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">MEPs</td>
<td style="text-align: left;">Megakaryocyte/erythroid progenitors</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Colony Forming Unit-Monocytes</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">Monocytes</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">Mature NK cells_CD56- CD16- CD3-</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">Mature NK cells_CD56- CD16+ CD3-</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">Mature NK cells_CD56+ CD16+ CD3-</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK T cells</td>
<td style="text-align: left;">NK T cells</td>
<td style="text-align: right;">4</td>
</tr>
</tbody>
</table>
</div>
</details>
<h3 id="analysis-4">Analysis</h3>
<p>For simplicity, I have opted to only use the ‘main’ labels from the <code>DICE</code> reference annotation and annotated the dataset at the cluster-level. The ‘main’ labels provide broad cell type labels appropriate for identifying clusters of an ‘unwanted’ cell type.</p>
<aside>
This reference was recommended to me by colleagues with experience in using <em><a href="https://bioconductor.org/packages/3.11/SingleR">SingleR</a></em> to annotate T-cells profiled with scRNA-seq.
</aside>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The results are summarised in the table below.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-35">Table 2: </span>Cluster-level assignments using the main labels of the DICE reference.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">cluster</th>
<th style="text-align: right;">Monocytes</th>
<th style="text-align: right;">T cells, CD4+</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">928</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: right;">146</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1267</td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">636</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">130</td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">114</td>
</tr>
<tr class="odd">
<td style="text-align: left;">7</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1879</td>
</tr>
<tr class="even">
<td style="text-align: left;">8</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1827</td>
</tr>
<tr class="odd">
<td style="text-align: left;">9</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1359</td>
</tr>
<tr class="even">
<td style="text-align: left;">10</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">870</td>
</tr>
</tbody>
</table>
</div>
<p>Figure <a href="#fig:umap-with-main-cluster-labels">34</a> overlays these cell type labels on the UMAP plot and shows that the aforementioned small, distinct cluster is labelled as <code>Monocytes</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:umap-with-main-cluster-labels"></span>
<img src="C057_Cooney.cell_selection_files/umap-with-main-cluster-labels-1.png" alt="UMAP plot highlighting clusters (left) and 'main' cluster-level labels (right) where each panel highlights droplets from a particular label. Labels with &lt; 1% frequency are grouped together as `other`." width="624" />
<p class="caption">
Figure 34: UMAP plot highlighting clusters (left) and ‘main’ cluster-level labels (right) where each panel highlights droplets from a particular label. Labels with &lt; 1% frequency are grouped together as <code>other</code>.
</p>
</div>
</div>
<h5 id="diagnostic-plots">Diagnostic plots</h5>
<p>As a sanity check, we can examine the expression of the marker genes for the relevant cell type labels by plotting a heatmap of their expression in:</p>
<ol type="1">
<li>The reference dataset</li>
<li>Our dataset</li>
</ol>
<p>The value of (1) is that we can assess if we believe the genes are indeed good markers of the relevant cell type in the reference dataset. The value of (2) is that we can check that these genes are useful markers in our dataset (e.g., that they are reasonably well sampled in our data).</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Here, we specifically select (some of) the most strongly upregulated genes when comparing the <code>Monocytes</code> to the <code>T cells, CD4+</code> (<code>dice.Mono_vs_CD4</code>) and vice-versa (<code>dice.CD4_vs_Mono</code>)</p>
<p>Figure <a href="#fig:ref-dice-mono-and-cd4-markers">35</a> confirms that both the <code>dice.Mono_vs_CD4</code> and <code>dice.CD4_vs_Mono</code> marker genes distinguish these two cell types from one another in the <code>DICE</code> reference dataset. However, the <code>dice.CD4_vs_Mono</code> marker genes are also expressed in <code>T cells, CD8+</code> samples, highlighting that what are useful marker genes in one comparison are not necessarily in another comparison.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:ref-dice-mono-and-cd4-markers"></span>
<img src="C057_Cooney.cell_selection_files/ref-dice-mono-and-cd4-markers-1.png" alt="Heatmap of log-expression values in the `DICE` reference dataset for selected marker genes between the `Monocytes` and `T cells, CD4+` labels. Each column is a sample, each row a gene" width="624" />
<p class="caption">
Figure 35: Heatmap of log-expression values in the <code>DICE</code> reference dataset for selected marker genes between the <code>Monocytes</code> and <code>T cells, CD4+</code> labels. Each column is a sample, each row a gene
</p>
</div>
</div>
<p>Figure <a href="#fig:test-dice-mono-and-cd4-markers">36</a> shows that both the <code>dice.Mono_vs_CD4</code> and <code>dice.CD4_vs_Mono</code> marker genes marker genes are indeed very specific to their respective clusters in our dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:test-dice-mono-and-cd4-markers"></span>
<img src="C057_Cooney.cell_selection_files/test-dice-mono-and-cd4-markers-1.png" alt="Heatmap of log-expression values in our dataset at the cell-level for selected marker genes between the `Monocytes` and `T cells, CD4+` labels. Each column is a sample, each row a gene. For legibility, only a random 10% of non-`Monocytes` cells are shown." width="624" />
<p class="caption">
Figure 36: Heatmap of log-expression values in our dataset at the cell-level for selected marker genes between the <code>Monocytes</code> and <code>T cells, CD4+</code> labels. Each column is a sample, each row a gene. For legibility, only a random 10% of non-<code>Monocytes</code> cells are shown.
</p>
</div>
</div>
<h4 id="checking-for-enrichment-of-cells-in-experimental-group">Checking for enrichment of cells in experimental group</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>We again check whether the cells removed by this procedure are preferentially derive from particular experimental groups. Figure <a href="#fig:monocytes-highlighting-removed-droplets">37</a> shows that excluding these cells will preferentially exclude cells from the <code>Uninfected</code> samples.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:monocytes-highlighting-removed-droplets"></span>
<img src="C057_Cooney.cell_selection_files/monocytes-highlighting-removed-droplets-1.png" alt="Droplets removed after excluding droplets from clusters labelled as `Monocytes`, stratified by `Sample`." width="624" />
<p class="caption">
Figure 37: Droplets removed after excluding droplets from clusters labelled as <code>Monocytes</code>, stratified by <code>Sample</code>.
</p>
</div>
</div>
<h4 id="summary-4">Summary</h4>
<p>We opt to remove cells from clusters labelled as <code>Monocytes</code>. The removes 146 droplets, retaining 9010 for further analysis. This will provide better resolution for us to analyse the remaining cells, which are nominally the CD4+ T-cells of interest.</p>
<aside>
If it is biologically relevant that the <code>Monocytes</code> are only found in <code>Uninfected</code> then we would need to retain these cells for downstream analysis.
</aside>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="re-processing-4">Re-processing</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>There are 10 clusters detected, shown on the UMAP plot Figure <a href="#fig:no-monocytes-clusterplot-umap">38</a> and broken down by experimental factors in Figure <a href="#fig:no-monocytes-cluster-barplot">39</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:no-monocytes-clusterplot-umap"></span>
<img src="C057_Cooney.cell_selection_files/no-monocytes-clusterplot-umap-1.png" alt="UMAP plot of the updated data, where each point represents a droplet and is coloured according to the legend." width="624" />
<p class="caption">
Figure 38: UMAP plot of the updated data, where each point represents a droplet and is coloured according to the legend.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:no-monocytes-cluster-barplot"></span>
<img src="C057_Cooney.cell_selection_files/no-monocytes-cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 39: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>Notably:</p>
<ul>
<li>Still, some of the clusters are highly treatment-specific.</li>
<li>Still, none of the clusters are highly sample-specific.</li>
</ul>
<h2 id="known-unwanted-cell-types">‘Known’ unwanted cell types</h2>
<h3 id="motivation-7">Motivation</h3>
<p>James indicated that there may be a subpopulation of CD8+ T-cells in this dataset. CD4+ and CD8+ T-cells are notoriously difficult to distinguish from one another using scRNA-seq data<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a>.</p>
<aside>
Specifically, James believes that there may be a population of CD8+ T-cells that upregulated the cell-surface expression of CD4 and hence were retained in the enrichment step.
</aside>
<h3 id="analysis-5">Analysis</h3>
<p>Supposing that there are CD8+ T-cells in this dataset, then there are several reasons why the approach taken in <a href="#unknown-unwanted-cell-types">‘Unknown’ unwanted cell types</a> did not label any cluster as CD8+ T-cells:</p>
<ol type="1">
<li>If the CD8+ T-cells were spread across several clusters, and in a minority within each of those clusters, then a cluster-level analysis would likely not label any cluster as CD8+ T-cells.</li>
<li>If the marker genes distinguishing CD4+ T-cells and CD8+ T-cells in the reference dataset are poorly chosen
<ol type="a">
<li>If these genes can’t even distinguish CD4+ T-cells and CD8+ T-cells in the reference dataset</li>
<li>If these genes are poorly sampled in our dataset.</li>
</ol></li>
</ol>
<p>We explore these issues in turn.</p>
<h4 id="cell-level-annotations">Cell-level annotations</h4>
<p>A cell-level annotation using <em><a href="https://bioconductor.org/packages/3.11/SingleR">SingleR</a></em> may allow us to identify CD8+ T-cells that are clustering with CD4+ T-cells rather than as a distinct cluster. For simplicity, I have again opted to only use the <code>DICE</code> reference annotation but now annotated the dataset at the cell-level using both the ‘main’ and ‘fine’ labels.</p>
<aside>
A reminder that the trade-off between cluster-level and cell-level annotation is one of increased robustness to noise (cluster-level) vs. increased resolution (cell-level). The story is similar for the trade-off between using ‘main’ and ‘fine’ labels.
</aside>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The results using the ‘main’ label is summarised in the table below. Almost all cells are labelled as <code>T cells, CD4+</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<table style="width:100%;">
<caption><span id="tab:unnamed-chunk-41">Table 3: </span>Cell-level assignments using the ‘main’ labels of the DICE reference.</caption>
<colgroup>
<col style="width: 9%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 7%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 8%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: left;">1</th>
<th style="text-align: left;">2</th>
<th style="text-align: left;">3</th>
<th style="text-align: left;">4</th>
<th style="text-align: left;">5</th>
<th style="text-align: left;">6</th>
<th style="text-align: left;">7</th>
<th style="text-align: left;">8</th>
<th style="text-align: left;">9</th>
<th style="text-align: left;">10</th>
<th style="text-align: left;">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (2)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (1)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.7% (7)</td>
<td style="text-align: left;">1.7% (23)</td>
<td style="text-align: left;">3.5% (63)</td>
<td style="text-align: left;">0.3% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">1.1% (95)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: left;">100.0% (900)</td>
<td style="text-align: left;">100.0% (735)</td>
<td style="text-align: left;">99.9% (1526)</td>
<td style="text-align: left;">99.5% (404)</td>
<td style="text-align: left;">100.0% (137)</td>
<td style="text-align: left;">99.2% (1025)</td>
<td style="text-align: left;">98.1% (1309)</td>
<td style="text-align: left;">96.4% (1734)</td>
<td style="text-align: left;">99.3% (284)</td>
<td style="text-align: left;">99.9% (853)</td>
<td style="text-align: left;">98.9% (8907)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD8+</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.3% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (4)</td>
</tr>
<tr class="even">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (1)</td>
</tr>
</tbody>
</table>
</div>
<p>The results using the ‘fine’ label is summarised in the table below. Consistent with the ‘main’ labels, almost all cells are labelled as a subtype of <code>T cells, CD4+</code> cells.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-42">Table 4: </span>Cell-level assignments using the ‘fine’ labels of the DICE reference.</caption>
<colgroup>
<col style="width: 19%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 6%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 7%" />
<col style="width: 7%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">label.fine</th>
<th style="text-align: left;">1</th>
<th style="text-align: left;">2</th>
<th style="text-align: left;">3</th>
<th style="text-align: left;">4</th>
<th style="text-align: left;">5</th>
<th style="text-align: left;">6</th>
<th style="text-align: left;">7</th>
<th style="text-align: left;">8</th>
<th style="text-align: left;">9</th>
<th style="text-align: left;">10</th>
<th style="text-align: left;">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells, naive</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (2)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes, CD16+</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (1)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.3% (3)</td>
<td style="text-align: left;">0.7% (10)</td>
<td style="text-align: left;">2.1% (38)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.6% (52)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, memory TREG</td>
<td style="text-align: left;">21.0% (189)</td>
<td style="text-align: left;">64.5% (474)</td>
<td style="text-align: left;">20.9% (319)</td>
<td style="text-align: left;">33.3% (135)</td>
<td style="text-align: left;">48.9% (67)</td>
<td style="text-align: left;">45.8% (473)</td>
<td style="text-align: left;">80.1% (1069)</td>
<td style="text-align: left;">49.3% (887)</td>
<td style="text-align: left;">6.3% (18)</td>
<td style="text-align: left;">98.8% (844)</td>
<td style="text-align: left;">49.7% (4475)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, naive</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.5% (7)</td>
<td style="text-align: left;">2.2% (9)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (18)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, naive TREG</td>
<td style="text-align: left;">0.2% (2)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (2)</td>
<td style="text-align: left;">0.5% (2)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (7)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, naive, stimulated</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (2)</td>
<td style="text-align: left;">3.9% (16)</td>
<td style="text-align: left;">1.5% (2)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (20)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, TFH</td>
<td style="text-align: left;">9.7% (87)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">15.2% (232)</td>
<td style="text-align: left;">30.5% (124)</td>
<td style="text-align: left;">1.5% (2)</td>
<td style="text-align: left;">0.4% (4)</td>
<td style="text-align: left;">0.7% (9)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">1.4% (4)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">5.2% (465)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, Th1</td>
<td style="text-align: left;">2.0% (18)</td>
<td style="text-align: left;">5.4% (40)</td>
<td style="text-align: left;">3.1% (47)</td>
<td style="text-align: left;">3.2% (13)</td>
<td style="text-align: left;">19.0% (26)</td>
<td style="text-align: left;">11.4% (118)</td>
<td style="text-align: left;">11.8% (158)</td>
<td style="text-align: left;">43.3% (779)</td>
<td style="text-align: left;">56.3% (161)</td>
<td style="text-align: left;">0.6% (5)</td>
<td style="text-align: left;">15.1% (1365)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, Th1_17</td>
<td style="text-align: left;">0.7% (6)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.3% (5)</td>
<td style="text-align: left;">0.2% (1)</td>
<td style="text-align: left;">2.2% (3)</td>
<td style="text-align: left;">6.4% (66)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.6% (11)</td>
<td style="text-align: left;">35.0% (100)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">2.2% (194)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, Th17</td>
<td style="text-align: left;">31.8% (286)</td>
<td style="text-align: left;">17.6% (129)</td>
<td style="text-align: left;">18.8% (287)</td>
<td style="text-align: left;">8.4% (34)</td>
<td style="text-align: left;">12.4% (17)</td>
<td style="text-align: left;">29.8% (308)</td>
<td style="text-align: left;">1.9% (26)</td>
<td style="text-align: left;">2.1% (37)</td>
<td style="text-align: left;">1.0% (3)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">12.5% (1128)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, Th2</td>
<td style="text-align: left;">34.0% (306)</td>
<td style="text-align: left;">12.2% (90)</td>
<td style="text-align: left;">40.3% (615)</td>
<td style="text-align: left;">14.8% (60)</td>
<td style="text-align: left;">14.6% (20)</td>
<td style="text-align: left;">5.6% (58)</td>
<td style="text-align: left;">4.2% (56)</td>
<td style="text-align: left;">2.4% (43)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (2)</td>
<td style="text-align: left;">13.9% (1250)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD8+, naive</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (1)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD8+, naive, stimulated</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (2)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.6% (5)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.7% (10)</td>
<td style="text-align: left;">2.2% (9)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (2)</td>
<td style="text-align: left;">0.1% (2)</td>
<td style="text-align: left;">0.1% (2)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.3% (30)</td>
</tr>
</tbody>
</table>
</div>
<p>This consistency is further emphasised when we cross-tabulate the ‘main’ and ‘fine’ labels.</p>
<div class="layout-chunk" data-layout="l-body">
<table style="width:100%;">
<caption><span id="tab:unnamed-chunk-43">Table 5: </span>Cross-tabulation of the ‘main’ and ‘fine’ assignments using the DICE reference.</caption>
<colgroup>
<col style="width: 31%" />
<col style="width: 10%" />
<col style="width: 10%" />
<col style="width: 11%" />
<col style="width: 13%" />
<col style="width: 13%" />
<col style="width: 10%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">label.fine</th>
<th style="text-align: left;">B cells</th>
<th style="text-align: left;">Monocytes</th>
<th style="text-align: left;">NK cells</th>
<th style="text-align: left;">T cells, CD4+</th>
<th style="text-align: left;">T cells, CD8+</th>
<th style="text-align: left;">NA_</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells, naive</td>
<td style="text-align: left;">100.0% (2)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes, CD16+</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">100.0% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">53.7% (51)</td>
<td style="text-align: left;">0.0% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, memory TREG</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">2.1% (2)</td>
<td style="text-align: left;">50.2% (4473)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, naive</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (18)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, naive TREG</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.1% (6)</td>
<td style="text-align: left;">25.0% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, naive, stimulated</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.2% (20)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, TFH</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">5.2% (465)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, Th1</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">44.2% (42)</td>
<td style="text-align: left;">14.8% (1321)</td>
<td style="text-align: left;">50.0% (2)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, Th1_17</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">2.2% (194)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, Th17</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">12.7% (1128)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, Th2</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">14.0% (1250)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD8+, naive</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">25.0% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD8+, naive, stimulated</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (1)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">100.0% (1)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.3% (30)</td>
<td style="text-align: left;">0.0% (0)</td>
<td style="text-align: left;">0.0% (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Total</td>
<td style="text-align: left;">100.0% (2)</td>
<td style="text-align: left;">100.0% (1)</td>
<td style="text-align: left;">100.0% (95)</td>
<td style="text-align: left;">100.0% (8907)</td>
<td style="text-align: left;">100.0% (4)</td>
<td style="text-align: left;">100.0% (1)</td>
</tr>
</tbody>
</table>
</div>
<p>Figure <a href="#fig:umap-with-fine-cell-labels">40</a> overlays the ‘fine’ cell type labels on the UMAP plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:umap-with-fine-cell-labels"></span>
<img src="C057_Cooney.cell_selection_files/umap-with-fine-cell-labels-1.png" alt="UMAP plot. Each point represents a cluster and is coloured by the 'fine' cell-level label. Each panel highlights droplets from a particular predicted cluster type. Cluster labels with &lt; 1% frequency are grouped together as `other`." width="624" />
<p class="caption">
Figure 40: UMAP plot. Each point represents a cluster and is coloured by the ‘fine’ cell-level label. Each panel highlights droplets from a particular predicted cluster type. Cluster labels with &lt; 1% frequency are grouped together as <code>other</code>.
</p>
</div>
</div>
<p>Essentially none of the cells are labelled as CD8+ T-cells by this approach.</p>
<h6 id="diagnostic-plots-1">Diagnostic plots</h6>
<div class="layout-chunk" data-layout="l-body">

</div>
<aside>
The lists of genes from the <code>DICE</code> reference used to annotate these cells with <em><a href="https://bioconductor.org/packages/3.11/SingleR">SingleR</a></em> are available in <a href="../output/marker_genes/SingleR/"><code>output/marker_genes/SingleR/</code></a>. You may wish to look at the expression of these genes yourself to verify/understand the annotations.
</aside>
<p>As a sanity check, we again examine the expression of the marker genes for the relevant cell type labels by plotting a heatmap of their expression in:</p>
<ol type="1">
<li>The reference dataset</li>
<li>Our dataset</li>
</ol>
<p>The value of (1) is that we can assess if we believe the genes are indeed good markers of the relevant cell type in the reference dataset. The value of (2) is that we can check that these genes are useful markers in our dataset (e.g., that they are reasonably well sampled in our data).</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Here, we specifically select (some of) the most strongly upregulated genes when comparing the <code>T cells, CD8+</code> to the <code>T cells, CD4+</code> (<code>dice.CD8_vs_CD4</code>) and vice-versa (<code>dice.CD4_vs_CD8</code>).</p>
<p>Figure <a href="#fig:ref-dice-cd8-and-cd4-markers">41</a> confirms that both the <code>dice.CD8_vs_CD4</code> and <code>dice.CD4_vs_CD8</code> marker genes distinguish these two cell types from one another in the <code>DICE</code> reference dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:ref-dice-cd8-and-cd4-markers"></span>
<img src="C057_Cooney.cell_selection_files/ref-dice-cd8-and-cd4-markers-1.png" alt="Heatmap of log-expression values in the `DICE` reference dataset for selected marker genes between the `Monocytes` and `T cells, CD4+` labels. Each column is a sample, each row a gene" width="624" />
<p class="caption">
Figure 41: Heatmap of log-expression values in the <code>DICE</code> reference dataset for selected marker genes between the <code>Monocytes</code> and <code>T cells, CD4+</code> labels. Each column is a sample, each row a gene
</p>
</div>
</div>
<p>Figure <a href="#fig:test-dice-cd8-and-cd4-markers">42</a> shows that very few of the <code>dice.CD8_vs_CD4</code> marker genes are expressed at in any cells in our dataset. In contrast, many of the cells express several of the <code>dice.CD4_vs_CD8</code> marker genes.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:test-dice-cd8-and-cd4-markers"></span>
<img src="C057_Cooney.cell_selection_files/test-dice-cd8-and-cd4-markers-1.png" alt="Heatmap of log-expression values in our dataset at the cell-level for selected marker genes between the `T cells, CD8+` and `T cells, CD4+` labels. Each column is a sample, each row a gene." width="624" />
<p class="caption">
Figure 42: Heatmap of log-expression values in our dataset at the cell-level for selected marker genes between the <code>T cells, CD8+</code> and <code>T cells, CD4+</code> labels. Each column is a sample, each row a gene.
</p>
</div>
</div>
<p>Figure <a href="#fig:subset-test-dice-cd8-and-cd4-markers">43</a> shows that even amongst those cells not labelled as <code>T cells, CD4+</code>, very few of the <code>dice.CD8_vs_CD4</code> marker genes are expressed in our dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:subset-test-dice-cd8-and-cd4-markers"></span>
<img src="C057_Cooney.cell_selection_files/subset-test-dice-cd8-and-cd4-markers-1.png" alt="Heatmap of log-expression values in our dataset at the cell-level for selected marker genes between the `T cells, CD8+` and `T cells, CD4+` labels. Each column is a sample, each row a gene. Only cells not labelled as `T cells, CD4+` are shown." width="624" />
<p class="caption">
Figure 43: Heatmap of log-expression values in our dataset at the cell-level for selected marker genes between the <code>T cells, CD8+</code> and <code>T cells, CD4+</code> labels. Each column is a sample, each row a gene. Only cells not labelled as <code>T cells, CD4+</code> are shown.
</p>
</div>
</div>
<h5 id="summary-5">Summary</h5>
<p>By performing a cell-level annotation with <em><a href="https://bioconductor.org/packages/3.11/SingleR">SingleR</a></em>, we have ruled out the possibility that the absence of CD8+ T-cells identified in <a href="#unknown-unwanted-cell-types">‘Unknown’ unwanted cell types</a> is due to this analysis being performed at the cluster-level. In fact, the cell-level analysis has provided stronger evidence that these cells are almost all various subtypes of CD4+ T-cells.</p>
<p>However, we also observed that the <code>dice.CD8_vs_CD4</code> marker genes are poorly sampled in our dataset, leaving open the possibility that a different choice of marker genes would allow us to distinguish the two cell types and identify any stray CD8+ T-cells. We pursue this approach in the next section.</p>
<h4 id="alternative-marker-gene-lists">Alternative marker gene lists</h4>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>As an alternative to using the <code>DICE</code>-derived marker genes, we could use any other list of putative marker genes. James provided some genes that he believes are markers of the unwanted CD8+ cells, namely GNLY, GZMA, and GZMK. We look at the expression of these in both the <code>DICE</code> reference dataset and in our dataset.</p>
<h5 id="diagnostic-plots-2">Diagnostic plots</h5>
<p>Figure <a href="#fig:ref-james-cd8-markers">44</a> shows that these genes are arguably stronger markers of NK cells than they are of CD8+ cells. More importantly, these genes are highly expressed in several CD4+ T-cell subtypes in the <code>DICE</code> reference dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:ref-james-cd8-markers"></span>
<img src="C057_Cooney.cell_selection_files/ref-james-cd8-markers-1.png" alt="Heatmap of log-expression values in the `DICE` reference dataset for 'CD8+ T-cell' marker genes provided by James. Each column is a sample, each row a gene" width="624" />
<p class="caption">
Figure 44: Heatmap of log-expression values in the <code>DICE</code> reference dataset for ‘CD8+ T-cell’ marker genes provided by James. Each column is a sample, each row a gene
</p>
</div>
</div>
<p>Consistent with this, Figure <a href="#fig:test-james-cd8-markers">45</a> shows high expression of these genes in cells labelled as various subtypes of CD4+ T-cells.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:test-james-cd8-markers"></span>
<img src="C057_Cooney.cell_selection_files/test-james-cd8-markers-1.png" alt="Heatmap of log-expression values in our dataset at the cell-level for 'CD8+ T-cell' marker genes provided by James. Each column is a sample, each row a gene." width="624" />
<p class="caption">
Figure 45: Heatmap of log-expression values in our dataset at the cell-level for ‘CD8+ T-cell’ marker genes provided by James. Each column is a sample, each row a gene.
</p>
</div>
</div>
<p>Furthermore, Figure <a href="#fig:test-james-cd8-markers">45</a> shows high expression of these genes for those cells labelled as <code>NK cells</code> and low expression of those same genes in the few cells labelled as <code>T cells, CD8</code>.</p>
<h5 id="summary-6">Summary</h5>
<p>There is no evidence in the <code>DICE</code> reference dataset that GNLY, GZMA, and GZMK are markers of CD8+ T-cells. To the contrary, the <code>DICE</code> dataset provides strong evidence that these are markers of various CD4+ T-cell subtypes and NK cells.</p>
<aside>
We could extend this analysis using a different set of ‘CD8+ T-cell’ marker genes, but I don’t think this is necessary or fruitful analysis direction.
</aside>
<h4 id="summary-7">Summary</h4>
<ul>
<li>We identified a small number of <code>Monocytes</code> that we excluded from further analysis.</li>
<li>We did not identify any of the suspected CD8+ T-cells.</li>
<li>Instead, we found strong evidence that almost all the remaining cells are various subtypes of CD4+ T-cells.</li>
</ul>
<h1 id="summary-8">Summary</h1>
<p>After extensive analysis we have now identified those droplets containing ‘biologically relevant’ cells (n = 9010). Specifically, we have selected for singlets that are CD4+ T-cells. We can verify that we have selected CD4+ T-cells by looking at the expression of known marker genes.</p>
<h3 id="exclusion-of-ribosomal-protein-and-mitochondrial-genes-from-hvgs">Exclusion of ribosomal protein and mitochondrial genes from HVGs</h3>
<p>Figure <a href="#fig:cd4-markers-umap">46</a> gives an example using some marker genes of <code>T cells, CD4+</code> cells from the <code>DICE</code> reference dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:cd4-markers-umap"></span>
<img src="C057_Cooney.cell_selection_files/cd4-markers-umap-1.png" alt="UMAP plot of the updated data, where each point represents a cell and is coloured according to expression of selected genes. The selected genes are markers of the `T cells, CD4+` cells in the `DICE` reference dataset that are also highly expressed in our dataset." width="624" />
<p class="caption">
Figure 46: UMAP plot of the updated data, where each point represents a cell and is coloured according to expression of selected genes. The selected genes are markers of the <code>T cells, CD4+</code> cells in the <code>DICE</code> reference dataset that are also highly expressed in our dataset.
</p>
</div>
</div>
<h2 id="re-processing-5">Re-processing</h2>
<p>We perform a final re-processing of the ‘biologically relevant’ cells. Notably, we opt at this stage to exclude ribosomal protein and mitochondrial genes from the list of highly variable genes, for reasons example below.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>We find that the most highly variable genes in this dataset are somewhat enriched for ribosomal protein genes<a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a> (n = 106) and mitochondrial genes (n = 13), with Figure <a href="#fig:perc-mito-and-ribo">47</a> showing that more than 20% of the top-200 HVGs are ribosomal protein or mitochondrial genes.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:perc-mito-and-ribo"></span>
<img src="C057_Cooney.cell_selection_files/perc-mito-and-ribo-1.png" alt="Percentage of top-K HVGs that are mitochondrial or ribosomal protein genes." width="624" />
<p class="caption">
Figure 47: Percentage of top-K HVGs that are mitochondrial or ribosomal protein genes.
</p>
</div>
</div>
<p>Ribosomal protein and mitochondrial genes are of lesser biological relevance to this study, so we it was decided to exclude them from the HVGs. This means that these genes can no longer <strong>directly</strong> influence some of the subsequent steps in the analysis, including:</p>
<ul>
<li>Dimensionality reduction
<ul>
<li>Principal component analysis (PCA)</li>
<li>Uniform manifold approximation and projection (UMAP)</li>
</ul></li>
<li>Data integration
<ul>
<li>Mutual nearest neighbours (MNN)</li>
</ul></li>
<li>Clustering</li>
</ul>
<p>Although the exclusion of these genes from the HVGs prevents them from directly influencing these analyses, they may still <strong>indirectly</strong> influence these steps or otherwise be associated with their outcomes. For example, if there is a set of (non ribosomal protein) genes that are strongly associated with ribosomal protein gene expression, then we may still see a cluster associated with ribosomal protein gene expression.</p>
<p>Finally, and to emphasise, we have only excluded ribosomal protein and mitochondrial genes from the HVGs, i.e. we have not excluded them entirely from the dataset. In particular, this means that these genes may appear in downstream results (e.g., tables of cluster marker genes and differentially expressed genes).</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="visualisation">Visualisation</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>There are 8 clusters detected, shown on the UMAP plot Figure <a href="#fig:selected-cells-clusterplot-umap">48</a> and broken down by experimental factors in Figure <a href="#fig:selected-cells-cluster-barplot">49</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:selected-cells-clusterplot-umap"></span>
<img src="C057_Cooney.cell_selection_files/selected-cells-clusterplot-umap-1.png" alt="UMAP plot of the updated data, where each point represents a droplet and is coloured according to the legend." width="624" />
<p class="caption">
Figure 48: UMAP plot of the updated data, where each point represents a droplet and is coloured according to the legend.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:selected-cells-cluster-barplot"></span>
<img src="C057_Cooney.cell_selection_files/selected-cells-cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 49: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>Notably:</p>
<ul>
<li>Still, some of the clusters are highly treatment-specific.</li>
<li>Still, none of the clusters are highly sample-specific.</li>
</ul>
<h1 id="concluding-remarks">Concluding remarks</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The processed <em>SingleCellExperiment</em> object is available (see <a href="../data/SCEs/C057_Cooney.cells_selected.SCE.rds"><code>data/SCEs/C057_Cooney.cells_selected.SCE.rds</code></a>). This will be used in downstream analyses, e.g., identifying cluster marker genes and refining the cell labels.</p>
<h1 id="additional-information" class="appendix">Additional information</h1>
<p>The following are available on request:</p>
<ul>
<li>Full CSV tables of any data presented.</li>
<li>PDF/PNG files of any static plots.</li>
</ul>
<h1 id="session-info" class="appendix">Session info</h1>
<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
─ Session info ─────────────────────────────────────────────────────
 setting  value                       
 version  R version 4.0.0 (2020-04-24)
 os       CentOS Linux 7 (Core)       
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Australia/Melbourne         
 date     2020-07-23                  

─ Packages ─────────────────────────────────────────────────────────
 ! package                * version  date       lib source        
 P AnnotationDbi            1.50.0   2020-04-27 [?] Bioconductor  
 P AnnotationHub            2.20.0   2020-04-27 [?] Bioconductor  
 P assertthat               0.2.1    2019-03-21 [?] CRAN (R 4.0.0)
 P backports                1.1.8    2020-06-17 [?] CRAN (R 4.0.0)
 P beeswarm                 0.2.3    2016-04-25 [?] CRAN (R 4.0.0)
 P Biobase                * 2.48.0   2020-04-27 [?] Bioconductor  
 P BiocFileCache            1.12.0   2020-04-27 [?] Bioconductor  
 P BiocGenerics           * 0.34.0   2020-04-27 [?] Bioconductor  
 P BiocManager              1.30.10  2019-11-16 [?] CRAN (R 4.0.0)
 P BiocNeighbors            1.6.0    2020-04-27 [?] Bioconductor  
 P BiocParallel           * 1.22.0   2020-04-27 [?] Bioconductor  
 P BiocSingular             1.4.0    2020-04-27 [?] Bioconductor  
 P BiocStyle                2.16.0   2020-04-27 [?] Bioconductor  
 P BiocVersion              3.11.1   2019-11-13 [?] Bioconductor  
 P bit                      1.1-15.2 2020-02-10 [?] CRAN (R 4.0.0)
 P bit64                    0.9-7    2017-05-08 [?] CRAN (R 4.0.0)
 P bitops                   1.0-6    2013-08-17 [?] CRAN (R 4.0.0)
 P blob                     1.2.1    2020-01-20 [?] CRAN (R 4.0.0)
 P cli                      2.0.2    2020-02-28 [?] CRAN (R 4.0.0)
 P codetools                0.2-16   2018-12-24 [3] CRAN (R 4.0.0)
 P colorspace               1.4-1    2019-03-18 [?] CRAN (R 4.0.0)
 P cowplot                * 1.0.0    2019-07-11 [?] CRAN (R 4.0.0)
 P crayon                   1.3.4    2017-09-16 [?] CRAN (R 4.0.0)
 P curl                     4.3      2019-12-02 [?] CRAN (R 4.0.0)
 P DBI                      1.1.0    2019-12-15 [?] CRAN (R 4.0.0)
 P dbplyr                   1.4.4    2020-05-27 [?] CRAN (R 4.0.0)
 P DelayedArray           * 0.14.0   2020-04-27 [?] Bioconductor  
 P DelayedMatrixStats       1.10.0   2020-04-27 [?] Bioconductor  
 P digest                   0.6.25   2020-02-23 [?] CRAN (R 4.0.0)
 P distill                  0.8      2020-06-04 [?] CRAN (R 4.0.0)
 P dplyr                    1.0.0    2020-05-29 [?] CRAN (R 4.0.0)
 P dqrng                    0.2.1    2019-05-17 [?] CRAN (R 4.0.0)
 P edgeR                  * 3.30.3   2020-06-02 [?] Bioconductor  
 P ellipsis                 0.3.1    2020-05-15 [?] CRAN (R 4.0.0)
 P evaluate                 0.14     2019-05-28 [?] CRAN (R 4.0.0)
 P ExperimentHub            1.14.0   2020-04-27 [?] Bioconductor  
 P fansi                    0.4.1    2020-01-08 [?] CRAN (R 4.0.0)
 P farver                   2.0.3    2020-01-16 [?] CRAN (R 4.0.0)
 P fastmap                  1.0.1    2019-10-08 [?] CRAN (R 4.0.0)
 P generics                 0.0.2    2018-11-29 [?] CRAN (R 4.0.0)
 P GenomeInfoDb           * 1.24.2   2020-06-15 [?] Bioconductor  
 P GenomeInfoDbData         1.2.3    2020-05-18 [?] Bioconductor  
 P GenomicRanges          * 1.40.0   2020-04-27 [?] Bioconductor  
 P ggbeeswarm               0.6.0    2017-08-07 [?] CRAN (R 4.0.0)
 P ggplot2                * 3.3.2    2020-06-19 [?] CRAN (R 4.0.0)
 P Glimma                 * 1.16.0   2020-04-27 [?] Bioconductor  
 P glue                     1.4.1    2020-05-13 [?] CRAN (R 4.0.0)
 P gridExtra                2.3      2017-09-09 [?] CRAN (R 4.0.0)
 P gtable                   0.3.0    2019-03-25 [?] CRAN (R 4.0.0)
 P here                   * 0.1      2017-05-28 [?] CRAN (R 4.0.0)
 P highr                    0.8      2019-03-20 [?] CRAN (R 4.0.0)
 P htmltools                0.5.0    2020-06-16 [?] CRAN (R 4.0.0)
 P httpuv                   1.5.4    2020-06-06 [?] CRAN (R 4.0.0)
 P httr                     1.4.1    2019-08-05 [?] CRAN (R 4.0.0)
 P igraph                   1.2.5    2020-03-19 [?] CRAN (R 4.0.0)
 P interactiveDisplayBase   1.26.3   2020-06-02 [?] Bioconductor  
 P IRanges                * 2.22.2   2020-05-21 [?] Bioconductor  
 P irlba                    2.3.3    2019-02-05 [?] CRAN (R 4.0.0)
 P janitor                * 2.0.1    2020-04-12 [?] CRAN (R 4.0.0)
 P jsonlite                 1.6.1    2020-02-02 [?] CRAN (R 4.0.0)
 P knitr                    1.29     2020-06-23 [?] CRAN (R 4.0.0)
 P labeling                 0.3      2014-08-23 [?] CRAN (R 4.0.0)
 P later                    1.1.0.1  2020-06-05 [?] CRAN (R 4.0.0)
 P lattice                  0.20-41  2020-04-02 [?] CRAN (R 4.0.0)
 P lifecycle                0.2.0    2020-03-06 [?] CRAN (R 4.0.0)
 P limma                  * 3.44.3   2020-06-12 [?] Bioconductor  
 P locfit                   1.5-9.4  2020-03-25 [?] CRAN (R 4.0.0)
 P lubridate                1.7.9    2020-06-08 [?] CRAN (R 4.0.0)
 P magrittr                 1.5      2014-11-22 [?] CRAN (R 4.0.0)
 P Matrix                   1.2-18   2019-11-27 [?] CRAN (R 4.0.0)
 P matrixStats            * 0.56.0   2020-03-13 [?] CRAN (R 4.0.0)
 P memoise                  1.1.0    2017-04-21 [?] CRAN (R 4.0.0)
 P mime                     0.9      2020-02-04 [?] CRAN (R 4.0.0)
 P msigdbr                * 7.1.1    2020-05-14 [?] CRAN (R 4.0.0)
 P munsell                  0.5.0    2018-06-12 [?] CRAN (R 4.0.0)
 P patchwork              * 1.0.1    2020-06-22 [?] CRAN (R 4.0.0)
 P pheatmap                 1.0.12   2019-01-04 [?] CRAN (R 4.0.0)
 P pillar                   1.4.4    2020-05-05 [?] CRAN (R 4.0.0)
 P pkgconfig                2.0.3    2019-09-22 [?] CRAN (R 4.0.0)
 P Polychrome               1.2.5    2020-03-29 [?] CRAN (R 4.0.0)
 P promises                 1.1.1    2020-06-09 [?] CRAN (R 4.0.0)
 P purrr                    0.3.4    2020-04-17 [?] CRAN (R 4.0.0)
 P R6                       2.4.1    2019-11-12 [?] CRAN (R 4.0.0)
 P rappdirs                 0.3.1    2016-03-28 [?] CRAN (R 4.0.0)
 P RColorBrewer             1.1-2    2014-12-07 [?] CRAN (R 4.0.0)
 P Rcpp                     1.0.4.6  2020-04-09 [?] CRAN (R 4.0.0)
 P RcppAnnoy                0.0.16   2020-03-08 [?] CRAN (R 4.0.0)
 P RCurl                    1.98-1.2 2020-04-18 [?] CRAN (R 4.0.0)
 P rlang                    0.4.6    2020-05-02 [?] CRAN (R 4.0.0)
 P rmarkdown                2.3      2020-06-18 [?] CRAN (R 4.0.0)
 P rprojroot                1.3-2    2018-01-03 [?] CRAN (R 4.0.0)
 P RSpectra                 0.16-0   2019-12-01 [?] CRAN (R 4.0.0)
 P RSQLite                  2.2.0    2020-01-07 [?] CRAN (R 4.0.0)
 P rsvd                     1.0.3    2020-02-17 [?] CRAN (R 4.0.0)
 P S4Vectors              * 0.26.1   2020-05-16 [?] Bioconductor  
 P scales                   1.1.1    2020-05-11 [?] CRAN (R 4.0.0)
 P scater                 * 1.16.1   2020-05-28 [?] Bioconductor  
 P scatterplot3d            0.3-41   2018-03-14 [?] CRAN (R 4.0.0)
 P scran                  * 1.16.0   2020-04-27 [?] Bioconductor  
 P sessioninfo              1.1.1    2018-11-05 [?] CRAN (R 4.0.0)
 P shiny                    1.5.0    2020-06-23 [?] CRAN (R 4.0.0)
 P SingleCellExperiment   * 1.10.1   2020-04-28 [?] Bioconductor  
 P SingleR                * 1.2.4    2020-05-25 [?] Bioconductor  
 P snakecase                0.11.0   2019-05-25 [?] CRAN (R 4.0.0)
 P statmod                  1.4.34   2020-02-17 [?] CRAN (R 4.0.0)
 P stringi                  1.4.6    2020-02-17 [?] CRAN (R 4.0.0)
 P stringr                  1.4.0    2019-02-10 [?] CRAN (R 4.0.0)
 P SummarizedExperiment   * 1.18.1   2020-04-30 [?] Bioconductor  
 P tibble                 * 3.0.1    2020-04-20 [?] CRAN (R 4.0.0)
 P tidyr                  * 1.1.0    2020-05-20 [?] CRAN (R 4.0.0)
 P tidyselect               1.1.0    2020-05-11 [?] CRAN (R 4.0.0)
 P uwot                     0.1.8    2020-03-16 [?] CRAN (R 4.0.0)
 P vctrs                    0.3.1    2020-06-05 [?] CRAN (R 4.0.0)
 P vipor                    0.4.5    2017-03-22 [?] CRAN (R 4.0.0)
 P viridis                  0.5.1    2018-03-29 [?] CRAN (R 4.0.0)
 P viridisLite              0.3.0    2018-02-01 [?] CRAN (R 4.0.0)
 P withr                    2.2.0    2020-04-20 [?] CRAN (R 4.0.0)
 P xfun                     0.15     2020-06-21 [?] CRAN (R 4.0.0)
 P xtable                   1.8-4    2019-04-21 [?] CRAN (R 4.0.0)
 P XVector                  0.28.0   2020-04-27 [?] Bioconductor  
 P yaml                     2.2.1    2020-02-01 [?] CRAN (R 4.0.0)
 P zlibbioc                 1.34.0   2020-04-27 [?] Bioconductor  

[1] /stornext/General/data/user_managed/grpu_mritchie_1/hickey/SCORE/C057_Cooney/renv/library/R-4.0/x86_64-pc-linux-gnu
[2] /tmp/RtmpXEXo65/renv-system-library
[3] /stornext/System/data/apps/R/R-4.0.0/lib64/R/library

 P ── Loaded and on-disk path mismatch.</code></pre>
</div>
</details>
<div id="refs" class="references" role="doc-bibliography">
<div id="ref-aran2019reference">
<p>Aran, D., A. P. Looney, L. Liu, E. Wu, V. Fong, A. Hsu, S. Chak, et al. 2019. “Reference-Based Analysis of Lung Single-Cell Sequencing Reveals a Transitional Profibrotic Macrophage.” <em>Nat. Immunol.</em> 20 (2): 163–72.</p>
</div>
<div id="ref-hpcaRef">
<p>Mabbott, Neil A., J. K. Baillie, Helen Brown, Tom C. Freeman, and David A. Hume. 2013. “An Expression Atlas of Human Primary Cells: Inference of Gene Function from Coexpression Networks.” <em>BMC Genomics</em> 14. <a href="https://doi.org/10.1186/1471-2164-14-632">https://doi.org/10.1186/1471-2164-14-632</a>.</p>
</div>
<div id="ref-blueprintRef">
<p>Martens, Joost H A, and Hendrik G. Stunnenberg. 2013. “BLUEPRINT: Mapping Human Blood Cell Epigenomes.” <em>Haematologica</em> 98: 1487–9. <a href="https://doi.org/10.3324/haematol.2013.094243">https://doi.org/10.3324/haematol.2013.094243</a>.</p>
</div>
<div id="ref-monaco_immuneRef">
<p>Monaco, Gianni, Bernett Lee, Weili Xu, Seri Mustafah, You Yi Hwang, Christophe Carré, Nicolas Burdin, et al. 2019. “RNA-Seq Signatures Normalized by mRNA Abundance Allow Absolute Deconvolution of Human Immune Cell Types.” <em>Cell Reports</em> 26 (6): 1627–1640.e7. <a href="https://doi.org/10.1016/j.celrep.2019.01.041">https://doi.org/10.1016/j.celrep.2019.01.041</a>.</p>
</div>
<div id="ref-nhRef">
<p>Novershtern, Noa, Aravind Subramanian, Lee N. Lawton, Raymond H. Mak, W. Nicholas Haining, Marie E. McConkey, Naomi Habib, et al. 2011. “Densely Interconnected Transcriptional Circuits Control Cell States in Human Hematopoiesis.” <em>Cell</em> 144 (2): 296–309. <a href="https://doi.org/10.1016/j.cell.2011.01.004">https://doi.org/10.1016/j.cell.2011.01.004</a>.</p>
</div>
<div id="ref-diceRef">
<p>Schmiedel, Benjamin J., Divya Singh, Ariel Madrigal, Alan G. Valdovino-Gonzalez, Brandie M. White, Jose Zapardiel-Gonzalo, Brendan Ha, et al. 2018. “Impact of Genetic Polymorphisms on Human Immune Cell Gene Expression.” <em>Cell</em> 175 (6): 1701–1715.e16. <a href="https://doi.org/10.1016/j.cell.2018.10.022">https://doi.org/10.1016/j.cell.2018.10.022</a>.</p>
</div>
<div id="ref-encodeRef">
<p>The ENCODE Project Consortium. 2012. “An Integrated Encyclopedia of Dna Elements in the Human Genome.” <em>Nature</em>. <a href="https://doi.org/10.1038/nature11247">https://doi.org/10.1038/nature11247</a>.</p>
</div>
<div id="ref-xu2015identification">
<p>Xu, C., and Z. Su. 2015. “Identification of Cell Types from Single-Cell Transcriptomes Using a Novel Clustering Method.” <em>Bioinformatics</em> 31 (12): 1974–80.</p>
</div>
</div>
<section class="footnotes" role="doc-endnotes">
<hr />
<ol>
<li id="fn1" role="doc-endnote"><p>In a simple multiplexing experiment involving <span class="math inline">\(N\)</span> HTOs (here <span class="math inline">\(N=5\)</span>) with similar numbers of cells, we would expect around <span class="math inline">\(1 / N = 1 / 5 = 20\%\)</span> of all doublets to involve cells from the same sample. This may still be enough to cause the formation of misleading doublet clusters even after the majority of ‘labelled’ doublets are removed.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2" role="doc-endnote"><p>The use of experimentally derived doublet calls avoids any assumptions about the relative quantity of total RNA or the probability of doublet formation across different cell types.<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3" role="doc-endnote"><p>This can be individual cells or bulk samples.<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4" role="doc-endnote"><p>Table adapted from <a href="https://bioconductor.org/packages/release/bioc/vignettes/SingleR/inst/doc/SingleR.html#5_available_references">https://bioconductor.org/packages/release/bioc/vignettes/SingleR/inst/doc/SingleR.html#5_available_references</a><a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5" role="doc-endnote"><p>E.g., see <a href="https://osca.bioconductor.org/cell-type-annotation.html#assigning-cell-labels-from-reference-data">https://osca.bioconductor.org/cell-type-annotation.html#assigning-cell-labels-from-reference-data</a><a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn6" role="doc-endnote"><p>Here, ribosomal protein genes are defined as those gene symbols starting with “RPL” or “RPS” or otherwise found in the <a href="https://www.gsea-msigdb.org/gsea/msigdb/cards/KEGG_RIBOSOME.html">“KEGG_RIBOSOME” gene set published by MSigDB</a>, namely: FAU, MRPL13, RPL10, RPL10A, RPL10L, RPL11, RPL12, RPL13, RPL13A, RPL14, RPL15, RPL17, RPL17-C18orf32, RPL18, RPL18A, RPL19, RPL21, RPL22, RPL22L1, RPL23, RPL23A, RPL24, RPL26, RPL26L1, RPL27, RPL27A, RPL28, RPL29, RPL3, RPL30, RPL31, RPL32, RPL34, RPL34-AS1, RPL35, RPL35A, RPL36, RPL36A, RPL36A-HNRNPH2, RPL36AL, RPL37, RPL37A, RPL38, RPL39, RPL39L, RPL3L, RPL4, RPL41, RPL5, RPL6, RPL7, RPL7A, RPL7L1, RPL8, RPL9, RPLP0, RPLP1, RPLP2, RPS10, RPS10-NUDT3, RPS11, RPS12, RPS13, RPS14, RPS15, RPS15A, RPS16, RPS17, RPS18, RPS19, RPS19BP1, RPS2, RPS20, RPS21, RPS23, RPS24, RPS25, RPS26, RPS27, RPS27A, RPS27L, RPS28, RPS29, RPS3, RPS3A, RPS4X, RPS4Y1, RPS4Y2, RPS5, RPS6, RPS6KA1, RPS6KA2, RPS6KA3, RPS6KA4, RPS6KA5, RPS6KA6, RPS6KB1, RPS6KB2, RPS6KB2-AS1, RPS6KC1, RPS6KL1, RPS7, RPS8, RPS9, RPSA, RSL24D1, RSL24D1P11, UBA52<a href="#fnref6" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom"></div>
<script id="distill-bibliography" type="text/bibtex">
@Article{stegle2015computational,
    Author="Stegle, O.  and Teichmann, S. A.  and Marioni, J. C. ",
    Title="Computational and analytical challenges in single-cell transcriptomics",
    Journal="Nat. Rev. Genet.",
    Year="2015",
    Volume="16",
    Number="3",
    Pages="133--145",
    Month="Mar",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/25628217}
}

@Article{islam2014quantitative,
    Author="Islam, S.  and Zeisel, A.  and Joost, S.  and La Manno, G.  and Zajac, P.  and Kasper, M.  and Lonnerberg, P.  and Linnarsson, S. ",
    Title="Quantitative single-cell RNA-seq with unique molecular identifiers",
    Journal="Nat. Methods",
    Year="2014",
    Volume="11",
    Number="2",
    Pages="163--166",
    Month="Feb"
}

@Article{huber2015orchestrating,
    Author="Huber, W.  and Carey, V. J.  and Gentleman, R.  and Anders, S.  and Carlson, M.  and Carvalho, B. S.  and Bravo, H. C.  and Davis, S.  and Gatto, L.  and Girke, T.  and Gottardo, R.  and Hahne, F.  and Hansen, K. D.  and Irizarry, R. A.  and Lawrence, M.  and Love, M. I.  and MacDonald, J.  and Obenchain, V.  and Oles, A. K.  and Pag\`es, H.  and Reyes, A.  and Shannon, P.  and Smyth, G. K.  and Tenenbaum, D.  and Waldron, L.  and Morgan, M. ",
    Title="{{O}rchestrating high-throughput genomic analysis with {B}ioconductor}",
    Journal="Nat. Methods",
    Year="2015",
    Volume="12",
    Number="2",
    Pages="115--121",
    Month="Feb"
}

@Article{brennecke2013accounting,
    Author="Brennecke, P.  and Anders, S.  and Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Zhang, X.  and Proserpio, V.  and Baying, B.  and Benes, V.  and Teichmann, S. A.  and Marioni, J. C.  and Heisler, M. G. ",
    Title="{{A}ccounting for technical noise in single-cell {R}{N}{A}-seq experiments}",
    Journal="Nat. Methods",
    Year="2013",
    Volume="10",
    Number="11",
    Pages="1093--1095",
    Month="Nov"
}

@Article{marinov2014singlecell,
    Author="Marinov, G. K.  and Williams, B. A.  and McCue, K.  and Schroth, G. P.  and Gertz, J.  and Myers, R. M.  and Wold, B. J. ",
    Title="{{F}rom single-cell to cell-pool transcriptomes: stochasticity in gene expression and {R}{N}{A} splicing}",
    Journal="Genome Res.",
    Year="2014",
    Volume="24",
    Number="3",
    Pages="496--510",
    Month="Mar"
}

@Article{vallejos2015basics,
    Author="Vallejos, C. A.  and Marioni, J. C.  and Richardson, S. ",
    Title="{{B}{A}{S}i{C}{S}: Bayesian analysis of single-cell sequencing data}",
    Journal="PLoS Comput. Biol.",
    Year="2015",
    Volume="11",
    Number="6",
    Pages="e1004333",
    Month="Jun"
}

@Article{trapnell2014dynamics,
    Author="Trapnell, C.  and Cacchiarelli, D.  and Grimsby, J.  and Pokharel, P.  and Li, S.  and Morse, M.  and Lennon, N. J.  and Livak, K. J.  and Mikkelsen, T. S.  and Rinn, J. L. ",
    Title="{{T}he dynamics and regulators of cell fate decisions are revealed by pseudotemporal ordering of single cells}",
    Journal="Nat. Biotechnol.",
    Year="2014",
    Volume="32",
    Number="4",
    Pages="381--386",
    Month="Apr"
}

@Article{kolod2015singlecell,
    Author="Ko\l{}odziejczyk, A. A.  and Kim, J. K.  and Tsang, J. C.  and Ilicic, T.  and Henriksson, J.  and Natarajan, K. N.  and Tuck, A. C.  and Gao, X.  and Buhler, M.  and Liu, P.  and Marioni, J. C.  and Teichmann, S. A. ",
    Title="{{S}ingle cell {R}{N}{A}-sequencing of pluripotent states unlocks modular transcriptional variation}",
    Journal="Cell Stem Cell",
    Year="2015",
    Volume="17",
    Number="4",
    Pages="471--485",
    Month="Oct"
}

@Article{anders2010differential,
    Author="Anders, S.  and Huber, W. ",
    Title="Differential expression analysis for sequence count data",
    Journal="Genome Biol.",
    Year="2010",
    Volume="11",
    Number="10",
    Pages="R106",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20979621}
}

@Article{robinson2010scaling,
    Author="Robinson, M. D.  and Oshlack, A. ",
    Title="A scaling normalization method for differential expression analysis of RNA-seq data",
    Journal="Genome Biol.",
    Year="2010",
    Volume="11",
    Number="3",
    Pages="R25",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20196867}
}

@Article{islam2011characterization,
    Author="Islam, S.  and Kjallquist, U.  and Moliner, A.  and Zajac, P.  and Fan, J. B.  and Lonnerberg, P.  and Linnarsson, S. ",
    Title="{{C}haracterization of the single-cell transcriptional landscape by highly multiplex {R}{N}{A}-seq}",
    Journal="Genome Res.",
    Year="2011",
    Volume="21",
    Number="7",
    Pages="1160--1167",
    Month="Jul"
}

@Article{wilson2015combined,
    Author="Wilson, N. K.  and Kent, D. G.  and Buettner, F.  and Shehata, M.  and Macaulay, I. C.  and Calero-Nieto, F. J.  and Sanchez Castillo, M.  and Oedekoven, C. A.  and Diamanti, E.  and Schulte, R.  and Ponting, C. P.  and Voet, T.  and Caldas, C.  and Stingl, J.  and Green, A. R.  and Theis, F. J.  and Gottgens, B. ",
    Title="{{C}ombined single-cell functional and gene expression analysis resolves heterogeneity within stem cell populations}",
    Journal="Cell Stem Cell",
    Year="2015",
    Volume="16",
    Number="6",
    Pages="712--724",
    Month="Jun"
}

@Article{picelli2014fulllength,
    Author="Picelli, S.  and Faridani, O. R.  and Bjorklund, A. K.  and Winberg, G.  and Sagasser, S.  and Sandberg, R. ",
    Title="{{F}ull-length {R}{N}{A}-seq from single cells using {S}mart-seq2}",
    Journal="Nat Protoc",
    Year="2014",
    Volume="9",
    Number="1",
    Pages="171--181",
    Month="Jan"
}

@Article{pollen2014lowcoverage,
    Author="Pollen, A. A.  and Nowakowski, T. J.  and Shuga, J.  and Wang, X.  and Leyrat, A. A.  and Lui, J. H.  and Li, N.  and Szpankowski, L.  and Fowler, B.  and Chen, P.  and Ramalingam, N.  and Sun, G.  and Thu, M.  and Norris, M.  and Lebofsky, R.  and Toppani, D.  and Kemp, D. W.  and Wong, M.  and Clerkson, B.  and Jones, B. N.  and Wu, S.  and Knutsson, L.  and Alvarado, B.  and Wang, J.  and Weaver, L. S.  and May, A. P.  and Jones, R. C.  and Unger, M. A.  and Kriegstein, A. R.  and West, J. A. ",
    Title="{{L}ow-coverage single-cell m{R}{N}{A} sequencing reveals cellular heterogeneity and activated signaling pathways in developing cerebral cortex}",
    Journal="Nat. Biotechnol.",
    Year="2014",
    Volume="32",
    Number="10",
    Pages="1053--1058",
    Month="Oct"
}

@Article{love2015rnaseq,
    Author="Love, M. I.  and Anders, S.  and Kim, V.  and Huber, W. ",
    Title="{{R}{N}{A}-{S}eq workflow: gene-level exploratory analysis and differential expression}",
    Journal="F1000Res",
    Year="2015",
    Volume="4",
    Pages="1070"
}

@Article{liao2014featurecounts,
    Author="Liao, Y.  and Smyth, G. K.  and Shi, W. ",
    Title="{feature{C}ounts: an efficient general purpose program for assigning sequence reads to genomic features}",
    Journal="Bioinformatics",
    Year="2014",
    Volume="30",
    Number="7",
    Pages="923--930",
    Month="Apr"
}

@Article{liao2013subread,
    Author="Liao, Y.  and Smyth, G. K.  and Shi, W. ",
    Title="{{T}he {S}ubread aligner: fast, accurate and scalable read mapping by seed-and-vote}",
    Journal="Nucleic Acids Res.",
    Year="2013",
    Volume="41",
    Number="10",
    Pages="e108",
    Month="May"
}

@Article{law2014voom,
    Author="Law, C. W.  and Chen, Y.  and Shi, W.  and Smyth, G. K. ",
    Title="voom: Precision weights unlock linear model analysis tools for RNA-seq read counts",
    Journal="Genome Biol.",
    Year="2014",
    Volume="15",
    Number="2",
    Pages="R29",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/24485249}
}

@Article{love2014moderated,
    Author="Love, M. I.  and Huber, W.  and Anders, S. ",
    Title="Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2",
    Journal="Genome Biol.",
    Year="2014",
    Volume="15",
    Number="12",
    Pages="550",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/25516281}
}

@article{van2008visualizing,
    title={Visualizing data using t-SNE},
    author={Van der Maaten, L. and Hinton, G.},
    journal={J. Mach. Learn. Res.},
    volume={9},
    number={2579-2605},
    pages={85},
    year={2008},
    URL={http://www.jmlr.org/papers/v9/vandermaaten08a.html}
}

@Article{scialdone2015computational,
    Author="Scialdone, A.  and Natarajan, K. N.  and Saraiva, L. R.  and Proserpio, V.  and Teichmann, S. A.  and Stegle, O.  and Marioni, J. C.  and Buettner, F. ",
    Title="Computational assignment of cell-cycle stage from single-cell transcriptome data",
    Journal="Methods",
    Year="2015",
    Volume="85",
    Pages="54--61",
    Month="Sep",
    URL = {https://www.ncbi.nlm.nih.gov/pubmed/26142758},
}

@Article{kim2015characterizing,
    Author="Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Illicic, T.  and Teichmann, S. A.  and Marioni, J. C. ",
    Title="{{C}haracterizing noise structure in single-cell {R}{N}{A}-seq distinguishes genuine from technical stochastic allelic expression}",
    Journal="Nat. Commun.",
    Year="2015",
    Volume="6",
    Pages="8687"
}

@Article{mccarthy2012differential,
    Author="McCarthy, D. J.  and Chen, Y.  and Smyth, G. K. ",
    Title="{{D}ifferential expression analysis of multifactor {R}{N}{A}-{S}eq experiments with respect to biological variation}",
    Journal="Nucleic Acids Res.",
    Year="2012",
    Volume="40",
    Number="10",
    Pages="4288--4297",
    Month="May"
}

@Article{angel1991role,
    Author="Angel, P.  and Karin, M. ",
    Title="{{T}he role of {J}un, {F}os and the {A}{P}-1 complex in cell-proliferation and transformation}",
    Journal="Biochim. Biophys. Acta",
    Year="1991",
    Volume="1072",
    Number="2-3",
    Pages="129--157",
    Month="Dec"
}

@Article{langfelder2008defining,
    Author="Langfelder, P.  and Zhang, B.  and Horvath, S. ",
    Title="{{D}efining clusters from a hierarchical cluster tree: the {D}ynamic {T}ree {C}ut package for {R}}",
    Journal="Bioinformatics",
    Year="2008",
    Volume="24",
    Number="5",
    Pages="719--720",
    Month="Mar"
}

@Article{ilicic2016classification,
    Author="Ilicic, T.  and Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Bagger, F. O.  and McCarthy, D. J.  and Marioni, J. C.  and Teichmann, S. A. ",
    Title="{{C}lassification of low quality cells from single-cell {R}{N}{A}-seq data}",
    Journal="Genome Biol.",
    Year="2016",
    Volume="17",
    Number="1",
    Pages="29"
}

@Article{heng2008immunological,
    Author="Heng, T. S.  and Painter, M. W.  and Elpek, K.  and Lukacs-Kornek, V.  and Mauermann, N.  and Turley, S. J.  and Koller, D.  and Kim, F. S.  and Wagers, A. J.  and Asinovski, N.  and Davis, S.  and Fassett, M.  and Feuerer, M.  and Gray, D. H.  and Haxhinasto, S.  and Hill, J. A.  and Hyatt, G.  and Laplace, C.  and Leatherbee, K.  and Mathis, D.  and Benoist, C.  and Jianu, R.  and Laidlaw, D. H.  and Best, J. A.  and Knell, J.  and Goldrath, A. W.  and Jarjoura, J.  and Sun, J. C.  and Zhu, Y.  and Lanier, L. L.  and Ergun, A.  and Li, Z.  and Collins, J. J.  and Shinton, S. A.  and Hardy, R. R.  and Friedline, R.  and Sylvia, K.  and Kang, J. ",
    Title="{{T}he {I}mmunological {G}enome {P}roject: networks of gene expression in immune cells}",
    Journal="Nat. Immunol.",
    Year="2008",
    Volume="9",
    Number="10",
    Pages="1091--1094",
    Month="Oct"
}

@Article{zeisel2015brain,
    Author="Zeisel, A.  and Munoz-Manchado, A. B.  and Codeluppi, S.  and Lonnerberg, P.  and La Manno, G.  and Jureus, A.  and Marques, S.  and Munguba, H.  and He, L.  and Betsholtz, C.  and Rolny, C.  and Castelo-Branco, G.  and Hjerling-Leffler, J.  and Linnarsson, S. ",
    Title="{{B}rain structure. {C}ell types in the mouse cortex and hippocampus revealed by single-cell {R}{N}{A}-seq}",
    Journal="Science",
    Year="2015",
    Volume="347",
    Number="6226",
    Pages="1138--1142",
    Month="Mar"
}

@Article{robinson2010edgeR,
    Author="Robinson, M. D.  and McCarthy, D. J.  and Smyth, G. K. ",
    Title="{edge{R}: a {B}ioconductor package for differential expression analysis of digital gene expression data}",
    Journal="Bioinformatics",
    Year="2010",
    Volume="26",
    Number="1",
    Pages="139--140",
    Month="Jan"
}

@Article{ritchie2015limma,
    Author="Ritchie, M. E.  and Phipson, B.  and Wu, D.  and Hu, Y.  and Law, C. W.  and Shi, W.  and Smyth, G. K. ",
    Title="{limma powers differential expression analyses for {R}{N}{A}-sequencing and microarray studies}",
    Journal="Nucleic Acids Res.",
    Year="2015",
    Volume="43",
    Number="7",
    Pages="e47",
    Month="Apr"
}

@Article{julia2015sincell,
    Author="Julia, M.  and Telenti, A.  and Rausell, A. ",
    Title="{{S}incell: an {R}/{B}ioconductor package for statistical assessment of cell-state hierarchies from single-cell {R}{N}{A}-seq}",
    Journal="Bioinformatics",
    Year="2015",
    Volume="31",
    Number="20",
    Pages="3380--3382",
    Month="Oct"
}

@Article{buettner2015computational,
    Author="Buettner, F.  and Natarajan, K. N.  and Casale, F. P.  and Proserpio, V.  and Scialdone, A.  and Theis, F. J.  and Teichmann, S. A.  and Marioni, J. C.  and Stegle, O. ",
    Title="{{C}omputational analysis of cell-to-cell heterogeneity in single-cell {R}{N}{A}-sequencing data reveals hidden subpopulations of cells}",
    Journal="Nat. Biotechnol.",
    Year="2015",
    Volume="33",
    Number="2",
    Pages="155--160",
    Month="Feb"
}

@Article{mahata2014singlecell,
    Author="Mahata, B.  and Zhang, X.  and Ko\l{}odziejczyk, A. A.  and Proserpio, V.  and Haim-Vilmovsky, L.  and Taylor, A. E.  and Hebenstreit, D.  and Dingler, F. A.  and Moignard, V.  and Gottgens, B.  and Arlt, W.  and McKenzie, A. N.  and Teichmann, S. A. ",
    Title="{{S}ingle-cell {R}{N}{A} sequencing reveals {T} helper cells synthesizing steroids de novo to contribute to immune homeostasis}",
    Journal="Cell Rep.",
    Year="2014",
    Volume="7",
    Number="4",
    Pages="1130--1142",
    Month="May"
}

@Article{phipson2010permutation,
    Author="Phipson, B.  and Smyth, G. K. ",
    Title="{{P}ermutation {P}-values should never be zero: calculating exact {P}-values when permutations are randomly drawn}",
    Journal="Stat. Appl. Genet. Mol. Biol.",
    Year="2010",
    Volume="9",
    Pages="Article 39"
}

@incollection{chen2014differential,
    title="{Differential expression analysis of complex RNA-seq experiments using edgeR}",
    author={Chen, Y. and Lun, A. T. L. and Smyth, G. K.},
    booktitle={Statistical analysis of next generation sequencing data},
    editor={Datta, S. and Nettleton, D.},
    pages={51--74},
    year={2014},
    publisher={Springer}
}

@article{hicks2015widespread,
    author = {Hicks, S. C. and Teng, M. and Irizarry, R. A.},
    title = "On the widespread and critical impact of systematic bias and batch effects in single-cell RNA-Seq data",
    year = {2015},
    doi = {10.1101/025528},
    publisher = {Cold Spring Harbor Labs Journals},
    URL = {http://biorxiv.org/content/early/2015/08/25/025528},
    eprint = {http://biorxiv.org/content/early/2015/08/25/025528.full.pdf},
    journal = {bioRxiv}
}

@Article{leng2015oscope,
    Author="Leng, N.  and Chu, L. F.  and Barry, C.  and Li, Y.  and Choi, J.  and Li, X.  and Jiang, P.  and Stewart, R. M.  and Thomson, J. A.  and Kendziorski, C. ",
    Title="{{O}scope identifies oscillatory genes in unsynchronized single-cell {R}{N}{A}-seq experiments}",
    Journal="Nat. Methods",
    Year="2015",
    Volume="12",
    Number="10",
    Pages="947--950",
    Month="Oct"
}

@Article{fan2016characterizing,
    Author="Fan, J.  and Salathia, N.  and Liu, R.  and Kaeser, G. E.  and Yung, Y. C.  and Herman, J. L.  and Kaper, F.  and Fan, J. B.  and Zhang, K.  and Chun, J.  and Kharchenko, P. V. ",
    Title="{{C}haracterizing transcriptional heterogeneity through pathway and gene set overdispersion analysis}",
    Journal="Nat. Methods",
    Year="2016",
    Volume="13",
    Number="3",
    Pages="241--244",
    Month="Mar"
}

@Article{lund2012detecting,
    Author="Lund, S. P.  and Nettleton, D.  and McCarthy, D. J.  and Smyth, G. K. ",
    Title="{{D}etecting differential expression in {R}{N}{A}-sequence data using quasi-likelihood with shrunken dispersion estimates}",
    Journal="Stat Appl Genet Mol Biol",
    Year="2012",
    Volume="11",
    Number="5",
    Pages=" "
}

@Article{angerer2016destiny,
    Author="Angerer, P.  and Haghverdi, L.  and Buttner, M.  and Theis, F. J.  and Marr, C.  and Buettner, F. ",
    Title="{destiny: diffusion maps for large-scale single-cell data in {R}}",
    Journal="Bioinformatics",
    Year="2016",
    Volume="32",
    Number="8",
    Pages="1241--1243",
}

@Article{lun2016pooling,
    Author="Lun, A. T.  and Bach, K.  and Marioni, J. C. ",
    Title="Pooling across cells to normalize single-cell RNA sequencing data with many zero counts}",
    Journal="Genome Biol.",
    Year="2016",
    Volume="17",
    Pages="75",
    Month="Apr",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/27122128}
}

@Article{zhu2006gata3,
    Author="Zhu, J. and Yamane, H. and Cote-Sierra, J. and Guo, L. and Paul, W. E.",
    Title="{{G}{A}{T}{A}-3 promotes {T}h2 responses through three different mechanisms: induction of {T}h2 cytokine production, selective growth of {T}h2 cells and inhibition of {T}h1 cell-specific factors}",
    Journal="Cell Res.",
    Year="2006",
    Volume="16",
    Number="1",
    Pages="3--10",
    Month="Jan"
}

@article{berger1996bioequivalence,
    author = "Berger, R. L. and Hsu, J. C.",
    doi = "10.1214/ss/1032280304",
    fjournal = "Statistical Science",
    journal = "Statist. Sci.",
    month = "11",
    number = "4",
    pages = "283--319",
    publisher = "The Institute of Mathematical Statistics",
    title = "Bioequivalence trials, intersection-union tests and equivalence confidence sets",
    url = "http://dx.doi.org/10.1214/ss/1032280304",
    volume = "11",
    year = "1996"
}

@Article{santos2015cyclebase,
    Author="Santos, A.  and Wernersson, R.  and Jensen, L. J. ",
    Title="{{C}yclebase 3.0: a multi-organism database on cell-cycle regulation and phenotypes}",
    Journal="Nucleic Acids Res.",
    Year="2015",
    Volume="43",
    Number="Database issue",
    Pages="D1140--1144",
    Month="Jan"
}

@article {tung2016batch,
    Author="Tung, P. Y.  and Blischak, J. D.  and Hsiao, C. J.  and Knowles, D. A.  and Burnett, J. E.  and Pritchard, J. K.  and Gilad, Y. ",
    Title="Batch effects and the effective design of single-cell gene expression studies",
    Journal="Sci. Rep.",
    Year="2017",
    Volume="7",
    Pages="39921",
    Month="Jan"
}

@Article{bray2016near,
    Author="Bray, N. L.  and Pimentel, H.  and Melsted, P.  and Pachter, L. ",
    Title="{{N}ear-optimal probabilistic {R}{N}{A}-seq quantification}",
    Journal="Nat. Biotechnol.",
    Year="2016",
    Volume="34",
    Number="5",
    Pages="525--527",
    Month="May"
}

@article{patro2015accurate,
    author = {Patro, Rob and Duggal, Geet and Kingsford, Carl},
    title = {Accurate, fast, and model-aware transcript expression quantification with Salmon},
    year = {2015},
    doi = {10.1101/021592},
    publisher = {Cold Spring Harbor Labs Journals},
    journal = {bioRxiv}
}

@Article{ji2016tscan,
    Author="Ji, Z.  and Ji, H. ",
    Title="{{T}{S}{C}{A}{N}: {P}seudo-time reconstruction and evaluation in single-cell {R}{N}{A}-seq analysis}",
    Journal="Nucleic Acids Res.",
    Year="2016",
    Volume="44",
    Number="13",
    Pages="e117",
    Month="Jul"
}

@Article{klein2015droplet,
    Author="Klein, A. M.  and Mazutis, L.  and Akartuna, I.  and Tallapragada, N.  and Veres, A.  and Li, V.  and Peshkin, L.  and Weitz, D. A.  and Kirschner, M. W. ",
    Title="{{D}roplet barcoding for single-cell transcriptomics applied to embryonic stem cells}",
    Journal="Cell",
    Year="2015",
    Volume="161",
    Number="5",
    Pages="1187--1201",
    Month="May"
}

@Article{macosko2015highly,
    Author="Macosko, E. Z.  and Basu, A.  and Satija, R.  and Nemesh, J.  and Shekhar, K.  and Goldman, M.  and Tirosh, I.  and Bialas, A. R.  and Kamitaki, N.  and Martersteck, E. M.  and Trombetta, J. J.  and Weitz, D. A.  and Sanes, J. R.  and Shalek, A. K.  and Regev, A.  and McCarroll, S. A. ",
    Title="{{H}ighly parallel genome-wide expression profiling of individual cells using nanoliter droplets}",
    Journal="Cell",
    Year="2015",
    Volume="161",
    Number="5",
    Pages="1202--1214",
    Month="May"
}

@Article{bourgon2010independent,
    Author="Bourgon, R.  and Gentleman, R.  and Huber, W. ",
    Title="Independent filtering increases detection power for high-throughput experiments",
    Journal="Proc. Natl. Acad. Sci. U.S.A.",
    Year="2010",
    Volume="107",
    Number="21",
    Pages="9546--9551",
    Month="May",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20460310}
}

@Article{conboy2007cell,
    Author="Conboy, C. M.  and Spyrou, C.  and Thorne, N. P.  and Wade, E. J.  and Barbosa-Morais, N. L.  and Wilson, M. D.  and Bhattacharjee, A.  and Young, R. A.  and Tavare, S.  and Lees, J. A.  and Odom, D. T. ",
    Title="Cell cycle genes are the evolutionarily conserved targets of the E2F4 transcription factor",
    Journal="PLoS ONE",
    Year="2007",
    Volume="2",
    Number="10",
    Pages="e1061",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/17957245}
}

@Article{bertoli2013control,
    Author="Bertoli, C.  and Skotheim, J. M.  and de Bruin, R. A. ",
    Title="Control of cell cycle transcription during G1 and S phases",
    Journal="Nat. Rev. Mol. Cell Biol.",
    Year="2013",
    Volume="14",
    Number="8",
    Pages="518--528",
    Month="Aug",
    URL={{C}ontrol of cell cycle transcription during {G}1 and {S} phases}
}

@Article{chen2016from,
    Author="Chen, Y.  and Lun, A. T.  and Smyth, G. K. ",
    Title="{{F}rom reads to genes to pathways: differential expression analysis of {R}{N}{A}-{S}eq experiments using {R}subread and the edge{R} quasi-likelihood pipeline}",
    Journal="F1000Res",
    Year="2016",
    Volume="5",
    Pages="1438"
}

@Article{coller2006new,
    Author="Coller, H. A.  and Sang, L.  and Roberts, J. M. ",
    Title="{{A} new description of cellular quiescence}",
    Journal="PLoS Biol.",
    Year="2006",
    Volume="4",
    Number="3",
    Pages="e83",
    Month="Mar"
}

@Article{mccarthy2017scater,
    Author="McCarthy, D. J.  and Campbell, K. R.  and Lun, A. T.  and Wills, Q. F. ",
    Title="{{S}cater: pre-processing, quality control, normalization and visualization of single-cell {R}{N}{A}-seq data in {R}}",
    Journal="Bioinformatics",
    Year="2017",
    Volume="33",
    Number="8",
    Pages="1179--1186",
    Month="Apr"
}

@article{simes1986improved,
    author = {Simes, R. J.},
    title = {An improved Bonferroni procedure for multiple tests of significance},
    volume = {73},
    number = {3},
    pages = {751--754},
    year = {1986},
    journal = {Biometrika}
}

@Article{xu2015identification,
    Author="Xu, C.  and Su, Z. ",
    Title="Identification of cell types from single-cell transcriptomes using a novel clustering method",
    Journal="Bioinformatics",
    Year="2015",
    Volume="31",
    Number="12",
    Pages="1974--1980",
    Month="Jun"
}

@Article{lun2017assessing,
    Author="Lun, A. T. L.  and Calero-Nieto, F. J.  and Haim-Vilmovsky, L.  and Gottgens, B.  and Marioni, J. C. ",
    Title="Assessing the reliability of spike-in normalization for analyses of single-cell RNA sequencing data",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="11",
    Pages="1795--1806",
    Month="Nov"
}

@Article{yang2016comparative,
    Author="Yang, Z.  and Algesheimer, R.  and Tessone, C. J. ",
    Title="{A comparative analysis of community detection algorithms on artificial networks}",
    Journal="Sci. Rep.",
    Year="2016",
    Volume="6",
    Pages="30750",
    Month="Aug"
}

@Article{smith2017umitools,
    Author="Smith, T.  and Heger, A.  and Sudbery, I. ",
    Title="{{U}{M}{I}-tools: modeling sequencing errors in {U}nique {M}olecular {I}dentifiers to improve quantification accuracy}",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="3",
    Pages="491--499",
    Month="Mar"
}

@Article{tasic2016adult,
    Author="Tasic, B.  and Menon, V.  and Nguyen, T. N.  and Kim, T. K.  and Jarsky, T.  and Yao, Z.  and Levi, B.  and Gray, L. T.  and Sorensen, S. A.  and Dolbeare, T.  and Bertagnolli, D.  and Goldy, J.  and Shapovalova, N.  and Parry, S.  and Lee, C.  and Smith, K.  and Bernard, A.  and Madisen, L.  and Sunkin, S. M.  and Hawrylycz, M.  and Koch, C.  and Zeng, H. ",
    Title="{{A}dult mouse cortical cell taxonomy revealed by single cell transcriptomics}",
    Journal="Nat. Neurosci.",
    Year="2016",
    Volume="19",
    Number="2",
    Pages="335--346",
    Month="Feb"
}

@Article{wajapeyee2010senescence,
    Author="Wajapeyee, N.  and Wang, S. Z.  and Serra, R. W.  and Solomon, P. D.  and Nagarajan, A.  and Zhu, X.  and Green, M. R. ",
    Title="{{S}enescence induction in human fibroblasts and hematopoietic progenitors by leukemogenic fusion proteins}",
    Journal="Blood",
    Year="2010",
    Volume="115",
    Number="24",
    Pages="5057--5060",
    Month="Jun"
}

@Article{zeng2012largescale,
    Author="Zeng, H.  and Shen, E. H.  and Hohmann, J. G.  and Oh, S. W.  and Bernard, A.  and Royall, J. J.  and Glattfelder, K. J.  and Sunkin, S. M.  and Morris, J. A.  and Guillozet-Bongaarts, A. L.  and Smith, K. A.  and Ebbert, A. J.  and Swanson, B.  and Kuan, L.  and Page, D. T.  and Overly, C. C.  and Lein, E. S.  and Hawrylycz, M. J.  and Hof, P. R.  and Hyde, T. M.  and Kleinman, J. E.  and Jones, A. R. ",
    Title="{{L}arge-scale cellular-resolution gene profiling in human neocortex reveals species-specific molecular signatures}",
    Journal="Cell",
    Year="2012",
    Volume="149",
    Number="2",
    Pages="483--496",
    Month="Apr"
}

@Article{grun2016denovo,
    Author="Grun, D.  and Muraro, M. J.  and Boisset, J. C.  and Wiebrands, K.  and Lyubimova, A.  and Dharmadhikari, G.  and van den Born, M.  and van Es, J.  and Jansen, E.  and Clevers, H.  and de Koning, E. J. P.  and van Oudenaarden, A. ",
    Title="{{D}e {N}ovo {P}rediction of {S}tem {C}ell {I}dentity using {S}ingle-{C}ell {T}ranscriptome {D}ata}",
    Journal="Cell Stem Cell",
    Year="2016",
    Volume="19",
    Number="2",
    Pages="266--277",
    Month="08"
}

@article{haghverdi2018batch,
    Author="Haghverdi, L.  and Lun, A. T. L.  and Morgan, M. D.  and Marioni, J. C. ",
    Title="Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors",
    Journal="Nat. Biotechnol.",
    Year="2018",
    Month="Apr"
}

@Article{muraro2016singlecell,
    Author="Muraro, M. J.  and Dharmadhikari, G.  and Grun, D.  and Groen, N.  and Dielen, T.  and Jansen, E.  and van Gurp, L.  and Engelse, M. A.  and Carlotti, F.  and de Koning, E. J.  and van Oudenaarden, A. ",
    Title="{{A} {S}ingle-{C}ell {T}ranscriptome {A}tlas of the {H}uman {P}ancreas}",
    Journal="Cell Syst",
    Year="2016",
    Volume="3",
    Number="4",
    Pages="385--394",
    Month="Oct"
}

@Article{segerstolpe2016singlecell,
    Author="Segerstolpe, A.  and Palasantza, A.  and Eliasson, P.  and Andersson, E. M.  and Andreasson, A. C.  and Sun, X.  and Picelli, S.  and Sabirsh, A.  and Clausen, M.  and Bjursell, M. K.  and Smith, D. M.  and Kasper, M.  and Ammala, C.  and Sandberg, R. ",
    Title="{{S}ingle-{C}ell {T}ranscriptome {P}rofiling of {H}uman {P}ancreatic {I}slets in {H}ealth and {T}ype 2 {D}iabetes}",
    Journal="Cell Metab.",
    Year="2016",
    Volume="24",
    Number="4",
    Pages="593--607",
    Month="Oct"
}

@Article{horn1965rationale,
    author="Horn, J. L.",
    title="A rationale and test for the number of factors in factor analysis",
    journal="Psychometrika",
    year="1965",
    month="Jun",
    day="01",
    volume="30",
    number="2",
    pages="179--185",
    issn="1860-0980",
    doi="10.1007/BF02289447",
    url="https://doi.org/10.1007/BF02289447"
}

@Article{buja1992remarks,
    Author="Buja, A.  and Eyuboglu, N. ",
    Title="{{R}emarks on {P}arallel {A}nalysis}",
    Journal="Multivariate Behav Res",
    Year="1992",
    Volume="27",
    Number="4",
    Pages="509--540",
    Month="Oct"
}

@Article{zheng2017massively,
    Author="Zheng, G. X.  and Terry, J. M.  and Belgrader, P.  and Ryvkin, P.  and Bent, Z. W.  and Wilson, R.  and Ziraldo, S. B.  and Wheeler, T. D.  and McDermott, G. P.  and Zhu, J.  and Gregory, M. T.  and Shuga, J.  and Montesclaros, L.  and Underwood, J. G.  and Masquelier, D. A.  and Nishimura, S. Y.  and Schnall-Levin, M.  and Wyatt, P. W.  and Hindson, C. M.  and Bharadwaj, R.  and Wong, A.  and Ness, K. D.  and Beppu, L. W.  and Deeg, H. J.  and McFarland, C.  and Loeb, K. R.  and Valente, W. J.  and Ericson, N. G.  and Stevens, E. A.  and Radich, J. P.  and Mikkelsen, T. S.  and Hindson, B. J.  and Bielas, J. H. ",
    Title="{{M}assively parallel digital transcriptional profiling of single cells}",
    Journal="Nat Commun",
    Year="2017",
    Volume="8",
    Pages="14049",
    Month="Jan"
}

@article{lun2018distinguishing,
    author = {Lun, A. and Riesenfeld, S. and Andrews, T. and Dao, T. P. and Gomes, T. and {{participants in the 1\textsuperscript{st} Human Cell Atlas Jamboree}} and Marioni, J.},
    title = {Distinguishing cells from empty droplets in droplet-based single-cell RNA sequencing data},
    year = {2018},
    doi = {10.1101/234872},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2018/04/04/234872},
    eprint = {https://www.biorxiv.org/content/early/2018/04/04/234872.full.pdf},
    journal = {bioRxiv}
}

@article {wolf2017graph,
    author = {Wolf, F. Alexander and Hamey, Fiona and Plass, Mireya and Solana, Jordi and Dahlin, Joakim S. and Gottgens, Berthold and Rajewsky, Nikolaus and Simon, Lukas and Theis, Fabian J.},
    title = {Graph abstraction reconciles clustering with trajectory inference through a topology preserving map of single cells},
    year = {2017},
    doi = {10.1101/208819},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2017/10/25/208819},
    eprint = {https://www.biorxiv.org/content/early/2017/10/25/208819.full.pdf},
    journal = {bioRxiv}
}

@Article{lawlor2017singlecell,
    Author="Lawlor, N.  and George, J.  and Bolisetty, M.  and Kursawe, R.  and Sun, L.  and Sivakamasundari, V.  and Kycia, I.  and Robson, P.  and Stitzel, M. L. ",
    Title="{{S}ingle-cell transcriptomes identify human islet cell signatures and reveal cell-type-specific expression changes in type 2 diabetes}",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="2",
    Pages="208--222",
    Month="02"
}

@Article{guimaraes2016patterns,
   Author="Guimaraes, J. C.  and Zavolan, M. ",
   Title="{{P}atterns of ribosomal protein expression specify normal and malignant human cells}",
   Journal="Genome Biol.",
   Year="2016",
   Volume="17",
   Number="1",
   Pages="236",
   Month="11"
}

@Article{glare2002betaactin,
   Author="Glare, E. M.  and Divjak, M.  and Bailey, M. J.  and Walters, E. H. ",
   Title="{beta-{A}ctin and {G}{A}{P}{D}{H} housekeeping gene expression in asthmatic airways is variable and not suitable for normalising m{R}{N}{A} levels}",
   Journal="Thorax",
   Year="2002",
   Volume="57",
   Number="9",
   Pages="765--770",
   Month="Sep"
}

@Article{nazari2015gapdh,
   Author="Nazari, F.  and Parham, A.  and Maleki, A. F. ",
   Title="{{G}{A}{P}{D}{H}, \beta-actin and \beta-microglobulin, as three common reference genes, are not reliable for gene expression studies in equine adipose- and marrow-derived mesenchymal stem cells}",
   Journal="J Anim Sci Technol",
   Year="2015",
   Volume="57",
   Pages="18"
}

@Article{soneson2018bias,
    Author="Soneson, C.  and Robinson, M. D. ",
    Title="Bias, robustness and scalability in single-cell differential expression analysis",
    Journal="Nat. Methods",
    URL = {https://www.ncbi.nlm.nih.gov/pubmed/29481549},
    Year="2018",
    Volume="15",
    Number="4",
    Pages="255--261",
    Month="Apr"
}

@Article{soneson2015differential,
    Author="Soneson, C.  and Love, M. I.  and Robinson, M. D. ",
    Title="{{D}ifferential analyses for {R}{N}{A}-seq: transcript-level estimates improve gene-level inferences}",
    Journal="F1000Res",
    Year="2015",
    Volume="4",
    Pages="1521"
}

@article{stoeckius2017hashing,
    author = {Stoeckius, M. and Zheng, S. and Houck-Loomis, B. and Hao, S. and Yeung, B. and Smibert, P. and Satija, R.},
    title = {Cell "hashing" with barcoded antibodies enables multiplexing and doublet detection for single cell genomics},
    year = {2017},
    doi = {10.1101/237693},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2017/12/21/237693},
    eprint = {https://www.biorxiv.org/content/early/2017/12/21/237693.full.pdf},
    journal = {bioRxiv}
}

@Article{kang2018multiplexed,
    Author="Kang, H. M.  and Subramaniam, M.  and Targ, S.  and Nguyen, M.  and Maliskova, L.  and McCarthy, E.  and Wan, E.  and Wong, S.  and Byrnes, L.  and Lanata, C. M.  and Gate, R. E.  and Mostafavi, S.  and Marson, A.  and Zaitlen, N.  and Criswell, L. A.  and Ye, C. J. ",
    Title="{{M}ultiplexed droplet single-cell {R}{N}{A}-sequencing using natural genetic variation}",
    Journal="Nat. Biotechnol.",
    Year="2018",
    Volume="36",
    Number="1",
    Pages="89--94",
    Month="Jan"
}

@Article{dahlin2018single,
    Author="Dahlin, J. S.  and Hamey, F. K.  and Pijuan-Sala, B.  and Shepherd, M.  and Lau, W. W. Y.  and Nestorowa, S.  and Weinreb, C.  and Wolock, S.  and Hannah, R.  and Diamanti, E.  and Kent, D. G.  and Gottgens, B.  and Wilson, N. K. ",
    Title="{{A} single-cell hematopoietic landscape resolves 8 lineage trajectories and defects in {K}it mutant mice}",
    Journal="Blood",
    Year="2018",
    Volume="131",
    Number="21",
    Pages="e1-e11",
    Month="May"
}

@Article{bach2017differentiation,
    Author="Bach, K.  and Pensa, S.  and Grzelak, M.  and Hadfield, J.  and Adams, D. J.  and Marioni, J. C.  and Khaled, W. T. ",
    Title="{{D}ifferentiation dynamics of mammary epithelial cells revealed by single-cell {R}{N}{A} sequencing}",
    Journal="Nat Commun",
    Year="2017",
    Volume="8",
    Number="1",
    Pages="2128",
    Month="Dec"
}

@article{tian2018scpipe,
  title={scPipe: A flexible R/Bioconductor preprocessing pipeline for single-cell RNA-sequencing data},
  author={Tian, Luyi and Su, Shian and Dong, Xueyi and Amann-Zalcenstein, Daniela and Biben, Christine and Seidi, Azadeh and Hilton, Douglas J and Naik, Shalin H and Ritchie, Matthew E},
  journal={PLoS computational biology},
  volume={14},
  number={8},
  pages={e1006361},
  year={2018},
  publisher={Public Library of Science},
  URL={https://www.ncbi.nlm.nih.gov/pubmed/30096152}
}

@article{kiselev2017sc3,
  title={SC3: consensus clustering of single-cell RNA-seq data},
  author={Kiselev, Vladimir Yu and Kirschner, Kristina and Schaub, Michael T and Andrews, Tallulah and Yiu, Andrew and Chandra, Tamir and Natarajan, Kedar N and Reik, Wolf and Barahona, Mauricio and Green, Anthony R and others},
  journal={Nature methods},
  volume={14},
  number={5},
  pages={483},
  year={2017},
  publisher={Nature Publishing Group},
  URL={https://www.ncbi.nlm.nih.gov/pubmed/28346451}
}

@ARTICLE{McInnes2018-hy,
  title         = "UMAP: Uniform Manifold Approximation and Projection for
                   Dimension Reduction",
  author        = "McInnes, Leland and Healy, John and Melville, James",
  abstract      = "UMAP (Uniform Manifold Approximation and Projection) is a
                   novel manifold learning technique for dimension reduction.
                   UMAP is constructed from a theoretical framework based in
                   Riemannian geometry and algebraic topology. The result is a
                   practical scalable algorithm that applies to real world
                   data. The UMAP algorithm is competitive with t-SNE for
                   visualization quality, and arguably preserves more of the
                   global structure with superior run time performance.
                   Furthermore, UMAP has no computational restrictions on
                   embedding dimension, making it viable as a general purpose
                   dimension reduction technique for machine learning.",
  month         =  feb,
  year          =  2018,
  url           = "http://arxiv.org/abs/1802.03426",
  archivePrefix = "arXiv",
  eprint        = "1802.03426",
  primaryClass  = "stat.ML",
  arxivid       = "1802.03426"
}

@article{lun2017overcoming,
  title={Overcoming confounding plate effects in differential expression analyses of single-cell RNA-seq data},
  author={Lun, Aaron TL and Marioni, John C},
  journal={Biostatistics},
  volume={18},
  number={3},
  pages={451--464},
  year={2017},
  publisher={Oxford University Press}
}

@Article{lun2017testing,
    Author="Lun, A. T. L.  and Richard, A. C.  and Marioni, J. C. ",
    Title="{{T}esting for differential abundance in mass cytometry data}",
    Journal="Nat. Methods",
    Year="2017",
    Volume="14",
    Number="7",
    Pages="707--709",
    Month="Jul"
}

@Article{chen2016reads,
    Author="Chen, Y.  and Lun, A. T.  and Smyth, G. K. ",
    Title="{{F}rom reads to genes to pathways: differential expression analysis of {R}{N}{A}-{S}eq experiments using {R}subread and the edge{R} quasi-likelihood pipeline}",
    Journal="F1000Res",
    Year="2016",
    Volume="5",
    Pages="1438"
}

@Article{phipson2016robust,
    Author="Phipson, B. and Lee, S. and Majewski, I. J. and Alexander, W. S. and Smyth, G. K.",
    year = 2016,
    title = "Robust hyperparameter estimation protects against hypervariable genes and improves power to detect differential expression",
    Journal="Ann. Appl. Stat.",
    Volume=10,
    Number=2,
    Pages="946-963"
}

@Article{aran2019reference,
   Author="Aran, D.  and Looney, A. P.  and Liu, L.  and Wu, E.  and Fong, V.  and Hsu, A.  and Chak, S.  and Naikawadi, R. P.  and Wolters, P. J.  and Abate, A. R.  and Butte, A. J.  and Bhattacharya, M. ",
   Title="Reference-based analysis of lung single-cell sequencing reveals a transitional profibrotic macrophage",
   Journal="Nat. Immunol.",
   Year="2019",
   Volume="20",
   Number="2",
   Pages="163--172",
   Month="02"
}

@article{Benayoun2019,
author = {Benayoun, B{\'{e}}r{\'{e}}nice A. and Pollina, Elizabeth A. and Singh, Param Priya and Mahmoudi, Salah and Harel, Itamar and Casey, Kerriann M. and Dulken, Ben W. and Kundaje, Anshul and Brunet, Anne},
doi = {10.1101/gr.240093.118},
journal = {Genome Research},
volume = {29},
pages = {697-709},
title = {Remodeling of epigenome and transcriptome landscapes with aging in mice reveals widespread induction of inflammatory responses},
year = {2019}
}

@article{ImmGenRef,
    author = {Heng, Tracy S.P. and Painter, Michio W. and Elpek, Kutlu and Lukacs-Kornek, Veronika and Mauermann, Nora and Turley, Shannon J. and Koller, Daphne and Kim, Francis S. and Wagers, Amy J. and Asinovski, Natasha and Davis, Scott and Fassett, Marlys and Feuerer, Markus and Gray, Daniel H.D. and Haxhinasto, Sokol and Hill, Jonathan A. and Hyatt, Gordon and Laplace, Catherine and Leatherbee, Kristen and Mathis, Diane and Benoist, Christophe and Jianu, Radu and Laidlaw, David H. and Best, J. Adam and Knell, Jamie and Goldrath, Ananda W. and Jarjoura, Jessica and Sun, Joseph C. and Zhu, Yanan and Lanier, Lewis L. and Ergun, Ayla and Li, Zheng and Collins, James J. and Shinton, Susan A. and Hardy, Richard R. and Friedline, Randall and Sylvia, Katelyn and Kang, Joonsoo},
    journal = {Nature Immunology},
    volume = {9},
    issue = {10},
    pages = {1091-4},
    doi = {10.1038/ni1008-1091},
    title = {The immunological genome project: Networks of gene expression in immune cells},
    year = {2008}
}

@article{choi2018haemopedia,
  title={Haemopedia RNA-seq: a database of gene expression during haematopoiesis in mice and humans},
  author={Choi, Jarny and Baldwin, Tracey M and Wong, Mae and Bolden, Jessica E and Fairfax, Kirsten A and Lucas, Erin C and Cole, Rebecca and Biben, Christine and Morgan, Clare and Ramsay, Kerry A and others},
  journal={Nucleic acids research},
  volume={47},
  number={D1},
  pages={D780--D785},
  year={2018},
  publisher={Oxford University Press}
}

@Article{mccarthy2009treat,
    Author="McCarthy, D. J.  and Smyth, G. K. ",
    Title="Testing significance relative to a fold-change threshold is a TREAT",
    Journal="Bioinformatics",
    Year="2009",
    Volume="25",
    Number="6",
    Pages="765--771",
    Month="Mar"
}

@article{hpcaRef,
author = {Mabbott, Neil A. and Baillie, J. K. and Brown, Helen and Freeman, Tom C. and Hume, David A.},
doi = {10.1186/1471-2164-14-632},
journal = {BMC Genomics},
volume = {14},
title = "An expression atlas of human primary cells: Inference of gene function from coexpression networks",
year = {2013}
}

@article{blueprintRef,
    author = {Martens, Joost H A and Stunnenberg, Hendrik G.},
    doi = {10.3324/haematol.2013.094243},
    journal = {Haematologica},
    title = "BLUEPRINT: Mapping human blood cell epigenomes",
    year = {2013},
    volume = {98},
    pages = {1487-1489}
}

@article{encodeRef,
author = {{The ENCODE Project Consortium}},
doi = {10.1038/nature11247},
issn = {14764687},
journal = {Nature},
title = "An integrated encyclopedia of DNA elements in the human genome",
year = {2012}
}

@article{lun2016step,
  title={A step-by-step workflow for low-level analysis of single-cell {RNA-seq} data},
  author={Lun, A. T. L. and McCarthy, D. J. and Marioni, J. C.},
  journal ={F1000Res.},
  volume={5},
  month={aug},
  year={2016}
}

@article{Pijuan-Sala2019-yz,
  title    = {A single-cell molecular map of mouse gastrulation and early organogenesis},
  author   = {Pijuan-Sala, Blanca and Griffiths, Jonathan A and Guibentif, Carolina and Hiscock, Tom W and Jawaid, Wajid and Calero-Nieto, Fernando J and Mulas, Carla and Ibarra-Soria Shankar and Simons, Benjamin D and Nichols, Jennifer and Marioni, John C and G{\"o}ttgens, Berthold},
  journal  = {Nature},
  volume   =  {566},
  number   =  {7745},
  month    =  feb,
  year     =  2019,
  url      = {http://dx.doi.org/10.1038/s41586-019-0933-9},
  pmid     = {30787436},
  doi      = {10.1038/s41586-019-0933-9},
  pmc      = {PMC6522369}
}

@article{diceRef,
  title = "Impact of Genetic Polymorphisms on Human Immune Cell Gene Expression",
  volume = 175,
  issn = {0092-8674},
  url = {http://www.sciencedirect.com/science/article/pii/S009286741831331X},
  doi = {10.1016/j.cell.2018.10.022},
  number = {6},
  journal = {Cell},
  author = {Schmiedel, Benjamin J. and Singh, Divya and Madrigal, Ariel and Valdovino-Gonzalez, Alan G. and White, Brandie M. and Zapardiel-Gonzalo, Jose and Ha, Brendan and Altay, Gokmen and Greenbaum, Jason A. and McVicker, Graham and Seumois, Grégory and Rao, Anjana and Kronenberg, Mitchell and Peters, Bjoern and Vijayanand, Pandurangan},
  month = "Nov",
  year = {2018},
  pages = {1701--1715.e16},
}

@article{nhRef,
  title = "Densely interconnected transcriptional circuits control cell states in human hematopoiesis",
  volume = {144},
  issn = {1097-4172},
  doi = {10.1016/j.cell.2011.01.004},
  number = {2},
  journal = {Cell},
  author = {Novershtern, Noa and Subramanian, Aravind and Lawton, Lee N. and Mak, Raymond H. and Haining, W. Nicholas and McConkey, Marie E. and Habib, Naomi and Yosef, Nir and Chang, Cindy Y. and Shay, Tal and Frampton, Garrett M. and Drake, Adam C. B. and Leskov, Ilya and Nilsson, Bjorn and Preffer, Fred and Dombkowski, David and Evans, John W. and Liefeld, Ted and Smutko, John S. and Chen, Jianzhu and Friedman, Nir and Young, Richard A. and Golub, Todd R. and Regev, Aviv and Ebert, Benjamin L.},
  month = "Jan",
  year = {2011},
  pmid = {21241896},
  pmcid = {PMC3049864},
  pages = {296--309}
}

@article{monaco_immuneRef,
  title = "RNA-Seq Signatures Normalized by mRNA Abundance Allow Absolute Deconvolution of Human Immune Cell Types",
  volume = {26},
  issn = {2211-1247},
  doi = {10.1016/j.celrep.2019.01.041},
  number = {6},
  journal = {Cell Reports},
  author = {Monaco, Gianni and Lee, Bernett and Xu, Weili and Mustafah, Seri and Hwang, You Yi and Carré, Christophe and Burdin, Nicolas and Visan, Lucian and Ceccarelli, Michele and Poidinger, Michael and Zippelius, Alfred and Pedro de Magalhães, João and Larbi, Anis},
  month = "Feb",
  year = {2019},
  pmid = {30726743},
  pmcid = {PMC6367568},
  pages = {1627--1640.e7}
}

@article{crowell2019discovery,
	author = {Crowell, H. L. and Soneson, C. and Germain, P.-L. and Calini, D. and Collin, L. and Raposo, C. and Malhotra, D. and Robinson, M. D.},
	title = {On the discovery of population-specific state transitions from multi-sample multi-condition single-cell RNA sequencing data},
	year = {2019},
	doi = {10.1101/713412},
	journal = {bioRxiv}
}

@article{finak2015mast,
  author = {Finak, G. and McDavid, A. and Yajima, M. and Deng, J. and Gersuk, V. and Shalek, A. K. and Slichter, C. K. and Miller, H. W. and McElrath, M. J. and Prlic, M.and Linsley, P. S. and Gottardo, R.},
  title = {MAST: a flexible statistical framework for assessing transcriptional changes and characterizing heterogeneity in single-cell RNA sequencing data},
  journal = {Genome Biol.},
  year = {2015},
  volume = {16}
}

@article{young10soupx,
  title={SoupX removes ambient RNA contamination from droplet based single cell RNA sequencing data. bioRxiv. 2018},
  author={Young, MD and Behjati, S},
  journal={DOI},
  volume={10},
  pages={303727}
}

@article{hill1973diversity,
  title={Diversity and evenness: a unifying notation and its consequences},
  author={Hill, Mark O},
  journal={Ecology},
  volume={54},
  number={2},
  pages={427--432},
  year={1973},
  publisher={Wiley Online Library}
}

@article{chao2014rarefaction,
  title={Rarefaction and extrapolation with Hill numbers: a framework for sampling and estimation in species diversity studies},
  author={Chao, Anne and Gotelli, Nicholas J and Hsieh, TC and Sander, Elizabeth L and Ma, KH and Colwell, Robert K and Ellison, Aaron M},
  journal={Ecological monographs},
  volume={84},
  number={1},
  pages={45--67},
  year={2014},
  publisher={Wiley Online Library}
}

@article{chao2015unveiling,
  title={Unveiling the species-rank abundance distribution by generalizing the Good-Turing sample coverage theory},
  author={Chao, Anne and Hsieh, TC and Chazdon, Robin L and Colwell, Robert K and Gotelli, Nicholas J},
  journal={Ecology},
  volume={96},
  number={5},
  pages={1189--1201},
  year={2015},
  publisher={Wiley Online Library}
}

@Article{richard2018tcell,
   Author="Richard, A. C.  and Lun, A. T. L.  and Lau, W. W. Y.  and Gottgens, B.  and Marioni, J. C.  and Griffiths, G. M. ",
   Title="{{T} cell cytolytic capacity is independent of initial stimulation strength}",
   Journal="Nat. Immunol.",
   Year="2018",
   Volume="19",
   Number="8",
   Pages="849--858",
   Month="08"
}

@article{venturi2007methods,
  title={Methods for comparing the diversity of samples of the T cell receptor repertoire},
  author={Venturi, Vanessa and Kedzierska, Katherine and Turner, Stephen J and Doherty, Peter C and Davenport, Miles P},
  journal={Journal of immunological methods},
  volume={321},
  number={1-2},
  pages={182--195},
  year={2007},
  publisher={Elsevier}
}

@Article{georgiou2014promise,
   Author="Georgiou, G.  and Ippolito, G. C.  and Beausang, J.  and Busse, C. E.  and Wardemann, H.  and Quake, S. R. ",
   Title="The promise and challenge of high-throughput sequencing of the antibody repertoire",
   Journal="Nat. Biotechnol.",
   Year="2014",
   Volume="32",
   Number="2",
   Pages="158--168",
   Month="Feb"
}

@Article{rosati2017overview,
   Author="Rosati, E.  and Dowds, C. M.  and Liaskou, E.  and Henriksen, E. K. K.  and Karlsen, T. H.  and Franke, A. ",
   Title="Overview of methodologies for T-cell receptor repertoire analysis",
   Journal="BMC Biotechnol.",
   Year="2017",
   Volume="17",
   Number="1",
   Pages="61",
   Month="07"
}
</script>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
