<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

  <!--radix_placeholder_meta_tags-->
  <title>C057: Preprocessing the Cooney (C057) memory CD4+ T-cell data set</title>
  
  
  
  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2020-07-14"/>
  <meta property="article:created" itemprop="dateCreated" content="2020-07-14"/>
  <meta name="article:author" content="Peter Hickey"/>
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="C057: Preprocessing the Cooney (C057) memory CD4+ T-cell data set"/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="C057"/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="C057: Preprocessing the Cooney (C057) memory CD4+ T-cell data set"/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Computational and analytical challenges in single-cell transcriptomics;citation_publication_date=2015;citation_volume=16;citation_author=O. Stegle;citation_author=S. A. Teichmann;citation_author=J. C. Marioni"/>
  <meta name="citation_reference" content="citation_title=A scaling normalization method for differential expression analysis of rna-seq data;citation_publication_date=2010;citation_volume=11;citation_author=M. D. Robinson;citation_author=A. Oshlack"/>
  <meta name="citation_reference" content="citation_publication_date=2016;citation_volume=17;citation_author=A. T. Lun;citation_author=K. Bach;citation_author=J. C. Marioni"/>
  <meta name="citation_reference" content="citation_title=Independent filtering increases detection power for high-throughput experiments;citation_publication_date=2010;citation_volume=107;citation_author=R. Bourgon;citation_author=R. Gentleman;citation_author=W. Huber"/>
  <meta name="citation_reference" content="citation_title=UMAP: Uniform manifold approximation and projection for dimension reduction;citation_publication_date=2018;citation_author=Leland McInnes;citation_author=John Healy;citation_author=James Melville"/>
  <meta name="citation_reference" content="citation_title=A step-by-step workflow for low-level analysis of single-cell &lt;span class=&quot;nocase&quot;&gt;RNA-seq&lt;/span&gt; data;citation_publication_date=2016;citation_volume=5;citation_author=A. T. L. Lun;citation_author=D. J. McCarthy;citation_author=J. C. Marioni"/>
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","date","output","editor_options","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["Preprocessing the Cooney (C057) memory CD4+ T-cell data set"]},{"type":"NULL"},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url","affiliation","affiliation_url"]}},"value":[{"type":"character","attributes":{},"value":["Peter Hickey"]},{"type":"character","attributes":{},"value":["https://peterhickey.org"]},{"type":"character","attributes":{},"value":["Single Cell Open Research Endeavour (SCORE), WEHI"]},{"type":"character","attributes":{},"value":["https://www.wehi.edu.au/people/shalin-naik/3310/score"]}]}]},{"type":"character","attributes":{},"value":["2020-07-14"]},{"type":"character","attributes":{},"value":["distill::distill_article"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["chunk_output_type"]}},"value":[{"type":"character","attributes":{},"value":["console"]}]},{"type":"character","attributes":{},"value":["ref.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #455a64;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative;}
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  </style>
  
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for table of contents */
  
  .d-toc {
    color: rgba(0,0,0,0.8);
    font-size: 0.8em;
    line-height: 1em;
  }
  
  .d-toc-header {
    font-size: 0.6rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.5);
    text-transform: uppercase;
    margin-top: 0;
    margin-bottom: 1.3em;
  }
  
  .d-toc a {
    border-bottom: none;
  }
  
  .d-toc ul {
    padding-left: 0;
  }
  
  .d-toc li>ul {
    padding-top: 0.8em;
    padding-left: 16px;
    margin-bottom: 0.6em;
  }
  
  .d-toc ul,
  .d-toc li {
    list-style-type: none;
  }
  
  .d-toc li {
    margin-bottom: 0.9em;
  }
  
  .d-toc-separator {
    margin-top: 20px;
    margin-bottom: 2em;
  }
  
  .d-article-with-toc {
    border-top: none;
    padding-top: 0;
  }
  
  
  
  /* Tweak code blocks (note that this CSS is repeated above in an injection
     into the d-code shadow dom) */
  
  d-code {
    overflow-x: auto !important;
  }
  
  pre.d-code code.d-code {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  pre.text-output {
  
    font-size: 12px;
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  @media(min-width: 768px) {
  
  d-code {
    overflow-x: visible !important;
  }
  
  pre.d-code code.d-code  {
      padding-left: 18px;
      font-size: 14px;
  }
  pre.text-output {
    font-size: 14px;
  }
  }
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .custom p {
    margin-bottom: 0.5em;
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // create d-bibliography
    var bibliography = $('<d-bibliography></d-bibliography>');
    $('#distill-bibliography').wrap(bibliography);
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace citations with <d-cite>
    $('.citation').each(function(i, val) {
      appendix = true;
      var cites = $(this).attr('data-cites').split(" ");
      var dt_cite = $('<d-cite></d-cite>');
      dt_cite.attr('key', cites.join());
      $(this).replaceWith(dt_cite);
    });
    // remove refs
    $('#refs').remove();
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-toc a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // replace code blocks with d-code
    $('pre>code').each(function(i, val) {
      var code = $(this);
      var pre = code.parent();
      var clz = "";
      var language = pre.attr('class');
      if (language) {
        // map unknown languages to "clike" (without this they just dissapear)
        if ($.inArray(language, ["bash", "clike", "css", "go", "html",
                                 "javascript", "js", "julia", "lua", "markdown",
                                 "markup", "mathml", "python", "svg", "xml"]) == -1)
          language = "clike";
        language = ' language="' + language + '"';
        var dt_code = $('<d-code block' + language + clz + '></d-code>');
        dt_code.text(code.text());
        pre.replaceWith(dt_code);
      } else {
        code.addClass('text-output').unwrap().changeElementType('pre');
      }
    });
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('d-code, pre.text-output, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // table of contents
      if (have_authors) // adjust border if we are in authors
        $('.d-toc').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // inject pre code styles (can't do this with a global stylesheet b/c a shadow root is used)
      $('d-code').each(function(i, val) {
        var style = document.createElement('style');
        style.innerHTML = 'pre code { padding-left: 10px; font-size: 12px; border-left: 2px solid rgba(0,0,0,0.1); } ' +
                          '@media(min-width: 768px) { pre code { padding-left: 18px; font-size: 14px; } }';
        if (this.shadowRoot)
          this.shadowRoot.appendChild(style);
      });
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-toc-header').remove();
    $('.d-toc').remove();
    $('.d-toc-separator').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Preprocessing the Cooney (C057) memory CD4+ T-cell data set","authors":[{"author":"Peter Hickey","authorURL":"https://peterhickey.org","affiliation":"Single Cell Open Research Endeavour (SCORE), WEHI","affiliationURL":"https://www.wehi.edu.au/people/shalin-naik/3310/score"}],"publishedDate":"2020-07-14T00:00:00.000+10:00","citationText":"Hickey, 2020"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">C057</a>
<a href="index.html">Introduction</a>
<a href="C057_Cooney.demultiplex.html">Demultiplexing</a>
<a href="C057_Cooney.preprocess.html">Preprocessing</a>
<a href="C057_Cooney.cell_selection.html">Selecting biologically relevant cells</a>
<a href="C057_Cooney.annotate.html">Annotating cells</a>
<a href="C057_Cooney.multi-sample_comparisons.html">Differential analyses</a>
<a href="C057_Cooney.repertoire-seq.html">Immune repertoire analysis</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Supplementary files
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="../output/FastQC/">FastQC reports</a>
<a href="../output/CellRanger/">CellRanger reports</a>
<a href="../output/marker_genes/">Marker gene tables</a>
<a href="../output/DEGs/">Differentially expressed genes</a>
<a href="glimma-plots/">Glimma plots</a>
</div>
</div>
</div>
<div class="nav-right">
<a href="https://github.com/WEHISCORE/C057_Cooney">
<i class="fa fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Preprocessing the Cooney (C057) memory CD4+ T-cell data set</h1>

</div>

<div class="d-byline">
  Peter Hickey <a href="https://peterhickey.org" class="uri">https://peterhickey.org</a> (Single Cell Open Research Endeavour (SCORE), WEHI)<a href="https://www.wehi.edu.au/people/shalin-naik/3310/score" class="uri">https://www.wehi.edu.au/people/shalin-naik/3310/score</a>
  
<br/>2020-07-14
</div>

<div class="d-article">
<h3 class="d-toc-header">Table of Contents</h3>
<nav class="d-toc" id="TOC">
<ul>
<li><a href="#setting-up-the-data">Setting up the data</a></li>
<li><a href="#quality-control-of-cells">Quality control of cells</a></li>
<li><a href="#examining-gene-level-metrics">Examining gene-level metrics</a></li>
<li><a href="#normalization-of-cell-specific-biases">Normalization of cell-specific biases</a></li>
<li><a href="#feature-selection">Feature selection</a></li>
<li><a href="#dimensionality-reduction">Dimensionality reduction</a></li>
<li><a href="#concluding-remarks">Concluding remarks</a></li>
<li><a href="#additional-information">Additional information</a></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</nav>
<hr class="d-toc-separator"/>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="setting-up-the-data">Setting up the data</h1>
<p>We start from the demultiplexed <em>SingleCellExperiment</em> object created in <a href="C057_Cooney.demultiplex.html">‘Demultiplexing the Cooney (C057) memory CD4+ T-cell data set’</a>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="incorporating-cell-based-annotation">Incorporating cell-based annotation</h2>
<p>Cell-based annotations are included in the <em>colData</em> of the <em>SingleCellExperiment</em>. We store the counts from the hashtag features as an ‘alternative experiment’.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="summary">Summary</h3>
<p>This dataset contains samples from 6 samples, along with ‘doublets’ and ‘ambiguous’ samples. All samples are human memory CD4+ T cells from ‘humanised mice’. Humanised mice are an immunodeficient strain and injected with human stem cells as pups which then differentiate into human hematopoietic cells as the mice mature. Mice were infected or not infected with HTLV, with each condition done in biological triplicate (1 mouse / replicate).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:experiment-by-sample"></span>
<img src="C057_Cooney.preprocess_files/experiment-by-sample-1.png" alt="Breakdown of the samples" width="624" />
<p class="caption">
Figure 1: Breakdown of the samples
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:experiment-by-treatment"></span>
<img src="C057_Cooney.preprocess_files/experiment-by-treatment-1.png" alt="Breakdown of the two treatments." width="624" />
<p class="caption">
Figure 2: Breakdown of the two treatments.
</p>
</div>
</div>
<h2 id="incorporating-gene-based-annotation">Incorporating gene-based annotation</h2>
<p>Having quantified gene expression against the Ensembl gene annotation, we have Ensembl-style identifiers for the genes. These identifiers are used as they are unambiguous and highly stable. However, they are difficult to interpret compared to the gene symbols which are more commonly used in the literature. Given the Ensembl identifiers, we obtain the corresponding gene symbols using annotation packages available through Bioconductor. Henceforth, we will use gene symbols (where available) to refer to genes in our analysis and otherwise use the Ensembl-style gene identifiers<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="quality-control-of-cells">Quality control of cells</h1>
<h2 id="defining-the-quality-control-metrics">Defining the quality control metrics</h2>
<p>Low-quality cells need to be removed to ensure that technical effects do not distort downstream analysis results. We use several quality control (QC) metrics to measure the quality of the cells:</p>
<ul>
<li><code>sum</code>: This measures the library size of the cells, which is the total sum of counts across both genes and spike-in transcripts. We want cells to have high library sizes as this means more RNA has been successfully captured during library preparation.</li>
<li><code>detected</code>: This is the number of expressed features<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a> in each cell. Cells with few expressed features are likely to be of poor quality, as the diverse transcript population has not been successful captured.</li>
<li><code>subsets_Mt_percent</code>: This measures the proportion of reads which are mapped to mitochondrial RNA. If there is a higher than expected proportion of mitochondrial RNA this is often symptomatic of a cell which is under stress and is therefore of low quality and will not be used for the analysis.</li>
</ul>
<p>In summary, we aim to identify cells with low library sizes, few expressed genes, and very high percentages of spike-in transcripts (when available) or mitochondrial DNA.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="visualizing-the-qc-metrics">Visualizing the QC metrics</h2>
<p>Figure <a href="#fig:qcplot-by-sample">3</a> shows that the vast majority of samples are good-quality; the library sizes are in the thousands<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a>, we observe around two thousand expressed genes per cell, and moderate mitochondrial percentages per cell. As we would expect, the <code>doublet</code> droplets have larger library sizes and more genes detected. The <code>ambiguous</code> droplets have notably smaller library sizes and fewer genes detected.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:qcplot-by-sample"></span>
<img src="C057_Cooney.preprocess_files/qcplot-by-sample-1.png" alt="Distributions of various QC metrics for all cells in the dataset. This includes the library sizes, number of genes detected, and percentage of reads mapped to mitochondrial genes." width="624" />
<p class="caption">
Figure 3: Distributions of various QC metrics for all cells in the dataset. This includes the library sizes, number of genes detected, and percentage of reads mapped to mitochondrial genes.
</p>
</div>
</div>
<p>It is also valuable to examine how the other QC metrics behave with respect to each other (Figure <a href="#fig:qcbiplot">4</a>). Generally, they will be in rough agreement, i.e., cells with low total counts will also have low numbers of expressed features and mitochondrial proportions. Clear discrepancies may correspond to technical differences between batches of cells or genuine biological differences in RNA content.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:qcbiplot"></span>
<img src="C057_Cooney.preprocess_files/qcbiplot-1.png" alt="Behaviour of each QC metric compared to the total number of expressed features. Each point represents a cell in the data set." width="960" />
<p class="caption">
Figure 4: Behaviour of each QC metric compared to the total number of expressed features. Each point represents a cell in the data set.
</p>
</div>
</div>
<h2 id="identifying-outliers-by-each-metric">Identifying outliers by each metric</h2>
<p>In calling cells from empty droplets, we have already removed cells with very low library sizes or (by association) low numbers of expressed genes. Thus, further filtering on these metrics is not strictly necessary.</p>
<aside>
See <a href="../docs/C057_Cooney.demultiplex.html"><code>docs/C057_Cooney.demultiplex.html</code></a> for removal of empty droplets.
</aside>
<p>Filtering on the mitochondrial proportion provides the most additional benefit in this situation and so we seek to identify droplets with unusually large mitochondrial proportions (i.e. outliers). Outlier thresholds are defined based on the median absolute deviation (MADs) from the median value of the metric across all cells. Here, we opt to use sample-specific thresholds to account for sample-specific differences<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The following table summarises the QC cutoffs:</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-8">Table 1: </span>Sample-specific QC metric cutoffs</caption>
<thead>
<tr class="header">
<th style="text-align: left;">batch</th>
<th style="text-align: right;">%mito</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ambiguous</td>
<td style="text-align: right;">10.4</td>
</tr>
<tr class="even">
<td style="text-align: left;">doublet</td>
<td style="text-align: right;">10.4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">infected_1</td>
<td style="text-align: right;">11.2</td>
</tr>
<tr class="even">
<td style="text-align: left;">infected_2</td>
<td style="text-align: right;">10.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">infected_3</td>
<td style="text-align: right;">10.7</td>
</tr>
<tr class="even">
<td style="text-align: left;">uninfected_1</td>
<td style="text-align: right;">10.5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">uninfected_2</td>
<td style="text-align: right;">10.1</td>
</tr>
<tr class="even">
<td style="text-align: left;">uninfected_3</td>
<td style="text-align: right;">9.8</td>
</tr>
</tbody>
</table>
</div>
<p>The vast majority of cells are retained for all samples.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-9">Table 2: </span>Number of samples removed by each QC step and the number of samples remaining.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">batch</th>
<th style="text-align: right;">ByMito</th>
<th style="text-align: right;">Remaining</th>
<th style="text-align: right;">PercRemaining</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ambiguous</td>
<td style="text-align: right;">48</td>
<td style="text-align: right;">3168</td>
<td style="text-align: right;">98.5</td>
</tr>
<tr class="even">
<td style="text-align: left;">doublet</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">427</td>
<td style="text-align: right;">97.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">infected_3</td>
<td style="text-align: right;">54</td>
<td style="text-align: right;">1455</td>
<td style="text-align: right;">96.4</td>
</tr>
<tr class="even">
<td style="text-align: left;">infected_2</td>
<td style="text-align: right;">57</td>
<td style="text-align: right;">1226</td>
<td style="text-align: right;">95.6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">uninfected_2</td>
<td style="text-align: right;">109</td>
<td style="text-align: right;">2212</td>
<td style="text-align: right;">95.3</td>
</tr>
<tr class="even">
<td style="text-align: left;">uninfected_3</td>
<td style="text-align: right;">80</td>
<td style="text-align: right;">1584</td>
<td style="text-align: right;">95.2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">uninfected_1</td>
<td style="text-align: right;">101</td>
<td style="text-align: right;">1929</td>
<td style="text-align: right;">95.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">infected_1</td>
<td style="text-align: right;">76</td>
<td style="text-align: right;">1404</td>
<td style="text-align: right;">94.9</td>
</tr>
</tbody>
</table>
</div>
<h3 id="checking-for-removal-of-biologically-relevant-subpopulations">Checking for removal of biologically relevant subpopulations</h3>
<p>The biggest practical concern during QC is whether an entire cell type is inadvertently discarded. There is always some risk of this occurring as the QC metrics are never fully independent of biological state. We can diagnose cell type loss by looking for systematic differences in gene expression between the discarded and retained cells.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:qc-md-plot">5</a> shows the result of this analysis, highlighting that the systematically upregulated genes are mitochondrial transcripts and that those systematically downregulated genes are largely ribosomal protein genes. This suggests that the QC step did not inadvertently filter out an entire biologically relevant subpopulation.</p>
<aside>
An interactive version of Figure <a href="#fig:qc-md-plot">5</a> is available from <a href="../output/glimma-plots/qc-md-plot.html"><code>output/glimma-plots/qc-md-plot.html</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:qc-md-plot"></span>
<img src="C057_Cooney.preprocess_files/qc-md-plot-1.png" alt="Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate $|logFC| = 1" width="624" />
<p class="caption">
Figure 5: Log-fold change in expression in the discarded cells compared to the retained cells. Each point represents a gene with mitochondrial transcripts in blue and ribosomal protein genes in orange. Dashed red lines indicate $|logFC| = 1
</p>
</div>
</div>
<p>If the discarded pool is enriched for a certain cell type, we should observe increased expression of the corresponding marker genes.</p>
<p>Another concern is whether the cells removed during QC preferentially derive from particular experimental groups. Reassuringly, Figure <a href="#fig:barplot-highlighting-outliers">6</a> shows that this is not the case.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:barplot-highlighting-outliers"></span>
<img src="C057_Cooney.preprocess_files/barplot-highlighting-outliers-1.png" alt="Droplets removed during QC, stratified by `Sample`. " width="624" />
<p class="caption">
Figure 6: Droplets removed during QC, stratified by <code>Sample</code>.
</p>
</div>
</div>
<p>Finally, Figure <a href="#fig:qcplot-highlighting-outliers">7</a> compares the QC metrics of the discarded and retained droplets.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:qcplot-highlighting-outliers"></span>
<img src="C057_Cooney.preprocess_files/qcplot-highlighting-outliers-1.png" alt="Distribution of QC metrics for each plate in the dataset. Each point represents a cell and is colored according to whether it was discarded during the QC process. Note that a cell will only be kept if it passes the relevant threshold for all four QC metrics." width="624" />
<p class="caption">
Figure 7: Distribution of QC metrics for each plate in the dataset. Each point represents a cell and is colored according to whether it was discarded during the QC process. Note that a cell will only be kept if it passes the relevant threshold for all four QC metrics.
</p>
</div>
</div>
<h2 id="summary-1">Summary</h2>
<p>We had already removed droplets that have unusually small library sizes or number of genes detected by the process of identifying empty droplets. We have now further removed droplets whose mitochondrial proportions we deem to be an outlier.</p>
<p>For the time being, we opt to retain the <code>Unknown</code> droplets because these may allow us to identify a subpopulation of cells that are not captured by hashing.</p>
<aside>
See <a href="../docs/C057_Cooney.annotate.html"><code>docs/C057_Cooney.annotate.html</code></a> for removal of <code>Unknown</code> droplets.
</aside>
<p>To conclude, Figure <a href="#fig:qcplot-post-outlier-removal">8</a> shows that following QC that most samples have similar QC metrics, as is to be expected, and Figure <a href="#fig:experiment-by-sample-postqc">9</a> summarises the experimental design following QC.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:qcplot-post-outlier-removal"></span>
<img src="C057_Cooney.preprocess_files/qcplot-post-outlier-removal-1.png" alt="Distributions of various QC metrics for all cells in the dataset passing QC. This includes the library sizes and proportion of reads mapped to mitochondrial genes." width="624" />
<p class="caption">
Figure 8: Distributions of various QC metrics for all cells in the dataset passing QC. This includes the library sizes and proportion of reads mapped to mitochondrial genes.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:experiment-by-sample-postqc"></span>
<img src="C057_Cooney.preprocess_files/experiment-by-sample-postqc-1.png" alt="Breakdown of the samples following QC." width="624" />
<p class="caption">
Figure 9: Breakdown of the samples following QC.
</p>
</div>
</div>
<h1 id="examining-gene-level-metrics">Examining gene-level metrics</h1>
<h2 id="inspecting-the-most-highly-expressed-genes">Inspecting the most highly expressed genes</h2>
<p>Figure <a href="#fig:topgenes">10</a> shows the most highly expressed genes across the cell population in the combined data set. Some of them are mitochondrial genes, matching what we’ve already seen in the QC metrics, and ribosomal protein genes, which commonly are amongst the most highly expressed genes in scRNA-seq data.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:topgenes"></span>
<img src="C057_Cooney.preprocess_files/topgenes-1.png" alt="Percentage of total counts assigned to the top 50 most highly-abundant features in the combined data set. For each feature, each bar represents the percentage assigned to that feature for a single cell, while the circle represents the average across all cells." width="624" />
<p class="caption">
Figure 10: Percentage of total counts assigned to the top 50 most highly-abundant features in the combined data set. For each feature, each bar represents the percentage assigned to that feature for a single cell, while the circle represents the average across all cells.
</p>
</div>
</div>
<p>Figure <a href="#fig:topgenes-filtered">11</a> shows the most highly expressed endogenous genes after excluding the mitochondrial and ribosomal protein genes.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:topgenes-filtered"></span>
<img src="C057_Cooney.preprocess_files/topgenes-filtered-1.png" alt="Percentage of total counts assigned to the top 50 most highly-abundant features in the combined data set. For each feature, each bar represents the percentage assigned to that feature for a single cell, while the circle represents the average across all cells. Bars are coloured by the total number of expressed features in each cell, while circles are coloured according to whether the feature is labelled as a control feature." width="624" />
<p class="caption">
Figure 11: Percentage of total counts assigned to the top 50 most highly-abundant features in the combined data set. For each feature, each bar represents the percentage assigned to that feature for a single cell, while the circle represents the average across all cells. Bars are coloured by the total number of expressed features in each cell, while circles are coloured according to whether the feature is labelled as a control feature.
</p>
</div>
</div>
<h2 id="filtering-out-low-abundance-genes">Filtering out low-abundance genes</h2>
<p>Low-abundance genes are problematic as zero or near-zero counts do not contain much information for reliable statistical inference <span class="citation" data-cites="bourgon2010independent">(Bourgon, Gentleman, and Huber <a href="#ref-bourgon2010independent" role="doc-biblioref">2010</a>)</span>. These genes typically do not provide enough evidence to reject the null hypothesis during testing, yet they still increase the severity of the multiple testing correction. In addition, the discreteness of the counts may interfere with statistical procedures, e.g., by compromising the accuracy of continuous approximations. Thus, low-abundance genes are often removed in many RNA-seq analysis pipelines before the application of downstream methods.</p>
<p>The ‘optimal’ choice of filtering strategy depends on the downstream application. A more aggressive filter is usually required to remove discreteness (e.g., for normalization) compared to that required for removing underpowered tests. For hypothesis testing, the filter statistic should also be independent of the test statistic under the null hypothesis. Thus, we (or the relevant function) will filter at each step as needed, rather than applying a single filter for the entire analysis.</p>
<p>Several metrics can be used to define low-abundance genes. The most obvious is the average count for each gene, computed across all cells in the data set. We typically observe a peak of moderately expressed genes following a plateau of lowly expressed genes (Figure <a href="#fig:abhist">12</a>).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:abhist"></span>
<img src="C057_Cooney.preprocess_files/abhist-1.png" alt="Histogram of log-average counts for all genes in the combined data set." width="624" />
<p class="caption">
Figure 12: Histogram of log-average counts for all genes in the combined data set.
</p>
</div>
</div>
<p>We remove 12936 genes that are not expressed in any cell. Such genes provide no information and would be removed by any filtering strategy. We retain 20602 for downstream analysis.</p>
<h1 id="normalization-of-cell-specific-biases">Normalization of cell-specific biases</h1>
<p>Read counts are subject to differences in capture efficiency and sequencing depth between cells <span class="citation" data-cites="stegle2015computational">(Stegle, Teichmann, and Marioni <a href="#ref-stegle2015computational" role="doc-biblioref">2015</a>)</span>. Normalization is required to eliminate these cell-specific biases prior to downstream quantitative analyses. This is often done by assuming that most genes are not differentially expressed (DE) between cells. Any systematic difference in count size across the non-DE majority of genes between two cells is assumed to represent bias and is removed by scaling. More specifically, ‘size factors’ are calculated that represent the extent to which counts should be scaled in each library.</p>
<h2 id="using-the-deconvolution-method-to-deal-with-zero-counts">Using the deconvolution method to deal with zero counts</h2>
<p>Size factors can be computed with several different approaches, e.g., using the <code>estimateSizeFactorsFromMatrix</code> function in the <em><a href="https://bioconductor.org/packages/3.11/DESeq2">DESeq2</a></em> package <span class="citation" data-cites="anders2010differential love2014moderated">(Anders and Huber <a href="#ref-anders2010differential" role="doc-biblioref">2010</a>; Love, Huber, and Anders <a href="#ref-love2014moderated" role="doc-biblioref">2014</a>)</span>, or with the <code>calcNormFactors</code> function <span class="citation" data-cites="robinson2010scaling">(Robinson and Oshlack <a href="#ref-robinson2010scaling" role="doc-biblioref">2010</a>)</span> in the <em><a href="https://bioconductor.org/packages/3.11/edgeR">edgeR</a></em> package. However, single-cell data can be problematic for these bulk data-based methods due to the dominance of low and zero counts. To overcome this, we pool counts from many cells to increase the count size for accurate size factor estimation <span class="citation" data-cites="lun2016pooling">(A. T. Lun, Bach, and Marioni <a href="#ref-lun2016pooling" role="doc-biblioref">2016</a>)</span>. Pool-based size factors are then ‘deconvolved’ into cell-based factors for cell-specific normalization. This removes scaling biases associated with cell-specific differences in capture efficiency, sequencing depth and composition biases.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>We check that the size factors are roughly aligned with the total library sizes (Figure <a href="#fig:normplot">13</a>). Deviations from the diagonal correspond to composition biases due to differential expression between cell subpopulations.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:normplot"></span>
<img src="C057_Cooney.preprocess_files/normplot-1.png" alt="Size factors from deconvolution, plotted against library sizes for all cells in each dataset. Axes are shown on a log-scale." width="624" />
<p class="caption">
Figure 13: Size factors from deconvolution, plotted against library sizes for all cells in each dataset. Axes are shown on a log-scale.
</p>
</div>
</div>
<h2 id="applying-the-size-factors-to-normalize-gene-expression">Applying the size factors to normalize gene expression</h2>
<p>The count data are used to compute normalized log-expression values for use in downstream analyses. Each value is defined as the log<sub>2</sub>-ratio of each count to the size factor for the corresponding cell, after adding a prior count of 1 to avoid undefined values at zero counts. Division by the size factor ensures that any cell-specific biases are removed.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="feature-selection">Feature selection</h1>
<h2 id="motivation">Motivation</h2>
<p>We often use scRNA-seq data in exploratory analyses to characterize heterogeneity across cells. Procedures like dimensionality reduction and clustering compare cells based on their gene expression profiles, which involves aggregating per-gene differences into a single (dis)similarity metric between a pair of cells. The choice of genes to use in this calculation has a major impact on the behaviour of the metric and the performance of downstream methods. We want to select genes that contain useful information about the biology of the system while removing genes that contain random noise. This aims to preserve interesting biological structure without the variance that obscures that structure. It also reduces the size of the dataset to improve computational efficiency of later steps.</p>
<h2 id="quantifying-per-gene-variation">Quantifying per-gene variation</h2>
<h3 id="variance-of-the-log-counts">Variance of the log-counts</h3>
<p>The simplest approach to quantifying per-gene variation is to simply compute the variance of the log-normalized expression values (referred to as “log-counts” for simplicity) for each gene across all cells in the population <span class="citation" data-cites="lun2016step">(A. T. L. Lun, McCarthy, and Marioni <a href="#ref-lun2016step" role="doc-biblioref">2016</a>)</span>. This has an advantage in that the feature selection is based on the same log-values that are used for later downstream steps. In particular, genes with the largest variances in log-values will contribute the most to the Euclidean distances between cells. By using log-values here, we ensure that our quantitative definition of heterogeneity is consistent throughout the entire analysis.</p>
<p>Calculation of the per-gene variance is simple but feature selection requires modelling of the mean-variance relationship.</p>
<h3 id="quantifying-technical-noise">Quantifying technical noise</h3>
<p>To account for the mean-variance relationship, we fit a trend to the variance with under the assumption of Poisson variation. UMI counts typically exhibit near-Poisson variation if we only consider technical noise from library preparation and sequencing.</p>
<h3 id="accounting-for-blocking-factors">Accounting for blocking factors</h3>
<p>Data containing multiple batches will often exhibit batch effects. We are usually not interested in highly variable genes (HVGs) that are driven by batch effects. Rather, we want to focus on genes that are highly variable within each batch. This is naturally achieved by performing trend fitting and variance decomposition separately for each batch.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The use of a batch-specific trend fit is useful as it accommodates differences in the mean-variance trends between batches. This is especially important if batches exhibit systematic technical differences, e.g., differences in coverage.</p>
<p>Figure <a href="#fig:trendplots">14</a> visualizes the quality of the batch-specific trend fits and highlights the need for batch-specific estimates of these fits. The analysis of each batch yields estimates of the biological and technical components for each gene, which are averaged across batches.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:trendplots"></span>
<img src="C057_Cooney.preprocess_files/trendplots-1.png" alt="Variance in the data set as a function of the mean after blocking. Each plot represents the results for a single batch, each point represents a gene (black) and the blue line represents the Poisson trend." width="624" />
<p class="caption">
Figure 14: Variance in the data set as a function of the mean after blocking. Each plot represents the results for a single batch, each point represents a gene (black) and the blue line represents the Poisson trend.
</p>
</div>
</div>
<h2 id="selecting-highly-variable-genes-hvgs">Selecting highly variable genes (HVGs)</h2>
<p>Once we have quantified the per-gene variation, the next step is to select the subset of HVGs to use in downstream analyses. A larger subset will reduce the risk of discarding interesting biological signal by retaining more potentially relevant genes, at the cost of increasing noise from irrelevant genes that might obscure said signal. It is difficult to determine the optimal trade-off for any given application as noise in one context may be useful signal in another. For example, heterogeneity in T cell activation responses is an interesting phenomena but may be irrelevant noise in studies that only care about distinguishing the major immunophenotypes.</p>
<p>We opt to only remove the obviously uninteresting genes with variances below the trend. By doing so, we avoid the need to make any judgement calls regarding what level of variation is interesting enough to retain. This approach represents one extreme of the bias-variance trade-off where bias is minimized at the cost of maximizing noise.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="summary-2">Summary</h2>
<p>We are left with 15073 HVGs by this approach<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a>. Figure <a href="#fig:hvgs">15</a> shows the expression of the top-10 HVGs.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:hvgs"></span>
<img src="C057_Cooney.preprocess_files/hvgs-1.png" alt="Violin plots of normalized log-expression values for the top-10 HVGs. Each point represents the log-expression value in a single cell." width="624" />
<p class="caption">
Figure 15: Violin plots of normalized log-expression values for the top-10 HVGs. Each point represents the log-expression value in a single cell.
</p>
</div>
</div>
<h1 id="dimensionality-reduction">Dimensionality reduction</h1>
<p>Many scRNA-seq analysis procedures involve comparing cells based on their expression values across multiple genes. In these applications, each individual gene represents a dimension of the data. If we had a scRNA-seq data set with two genes, we could make a two-dimensional plot where each axis represents the expression of one gene and each point in the plot represents a cell. Dimensionality reduction extends this idea to data sets with thousands of genes where each cell’s expression profile defines its location in the high-dimensional expression space.</p>
<p>As the name suggests, dimensionality reduction aims to reduce the number of separate dimensions in the data. This is possible because different genes are correlated if they are affected by the same biological process. Thus, we do not need to store separate information for individual genes, but can instead compress multiple features into a single dimension. This reduces computational work in downstream analyses, as calculations only need to be performed for a few dimensions rather than thousands of genes; reduces noise by averaging across multiple genes to obtain a more precise representation of the patterns in the data; and enables effective plotting of the data, for those of us who are not capable of visualizing more than 3 dimensions.</p>
<h2 id="principal-component-analysis">Principal component analysis</h2>
<p>Principal components analysis (PCA) discovers axes in high-dimensional space that capture the largest amount of variation. In PCA, the first axis (or “principal component”, PC) is chosen such that it captures the greatest variance across cells. The next PC is chosen such that it is orthogonal to the first and captures the greatest remaining amount of variation, and so on.</p>
<p>By definition, the top PCs capture the dominant factors of heterogeneity in the data set. Thus, we can perform dimensionality reduction by restricting downstream analyses to the top PCs. This strategy is simple, highly effective and widely used throughout the data sciences.</p>
<p>When applying PCA to scRNA-seq data, our assumption is that biological processes affect multiple genes in a coordinated manner. This means that the earlier PCs are likely to represent biological structure as more variation can be captured by considering the correlated behaviour of many genes. By comparison, random technical or biological noise is expected to affect each gene independently. There is unlikely to be an axis that can capture random variation across many genes, meaning that noise should mostly be concentrated in the later PCs. This motivates the use of the earlier PCs in our downstream analyses, which concentrates the biological signal to simultaneously reduce computational work and remove noise.</p>
<p>We perform the PCA on the log-normalized expression values. PCA is generally robust to random noise but an excess of it may cause the earlier PCs to capture noise instead of biological structure. This effect can be avoided - or at least mitigated - by restricting the PCA to a subset of HVGs, as done in <a href="#feature-selection">Feature selection</a>.</p>
<p>The choice of the number of PCs is a decision that is analogous to the choice of the number of HVGs to use. Using more PCs will avoid discarding biological signal in later PCs, at the cost of retaining more noise.</p>
<p>We use the strategy of retaining all PCs until the percentage of total variation explained reaches some threshold. We derive a suitable value for this threshold by calculating the proportion of variance in the data that is attributed to the biological component. This is done using the the variance modelling results from <a href="#quantifying-per-gene-variation">Quantifying per-gene variation</a>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>This retains 38 dimensions, which represents the lower bound on the number of PCs required to retain all biological variation. Any fewer PCs will definitely discard some aspect of biological signal<a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a>. This approach provides a reasonable choice when we want to retain as much signal as possible while still removing some noise.</p>
<h2 id="dimensionality-reduction-for-visualization">Dimensionality reduction for visualization</h2>
<p>We use the uniform manifold approximation and projection (UMAP) method <span class="citation" data-cites="McInnes2018-hy">(McInnes, Healy, and Melville <a href="#ref-McInnes2018-hy" role="doc-biblioref">2018</a>)</span> to perform further dimensionality reduction for visualization.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:umap">16</a> visualizes the droplets using the UMAP co-ordinates.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:umap"></span>
<img src="C057_Cooney.preprocess_files/umap-1.png" alt="UMAP plot of the dataset. Each point represents a droplets and is coloured by `Sample`. Each panel highlights droplets from a particular sample." width="624" />
<p class="caption">
Figure 16: UMAP plot of the dataset. Each point represents a droplets and is coloured by <code>Sample</code>. Each panel highlights droplets from a particular sample.
</p>
</div>
</div>
<h2 id="summary-3">Summary</h2>
<p>Although Figure <a href="#fig:umap">16</a> is only a preliminary summary of the data, there are few points worth highlighting:</p>
<ul>
<li>In terms of UMAP co-ordinates, there are some visual differences between <code>Infected</code> and <code>Uninfected</code> samples that are consistent across replicates.</li>
<li>The <code>Unknown</code> droplets are somewhat localised on the UMAP plot but these regions also contain droplets confidently assigned to a sample.</li>
</ul>
<h1 id="concluding-remarks">Concluding remarks</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The processed <em>SingleCellExperiment</em> object is available (see <a href="../data/SCEs/C057_Cooney.preprocessed.SCE.rds"><code>data/SCEs/C057_Cooney.preprocessed.SCE.rds</code></a>). This will be used in downstream analyses, e.g., selecting biologically relevant cells.</p>
<h1 id="additional-information" class="appendix">Additional information</h1>
<p>The following are available on request:</p>
<ul>
<li>Full CSV tables of any data presented.</li>
<li>PDF/PNG files of any static plots.</li>
</ul>
<h1 id="session-info" class="appendix">Session info</h1>
<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
─ Session info ─────────────────────────────────────────────────────
 setting  value                       
 version  R version 4.0.0 (2020-04-24)
 os       CentOS Linux 7 (Core)       
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Australia/Melbourne         
 date     2020-07-14                  

─ Packages ─────────────────────────────────────────────────────────
 ! package                * version  date       lib source        
 P AnnotationDbi          * 1.50.0   2020-04-27 [?] Bioconductor  
 P AnnotationFilter       * 1.12.0   2020-04-27 [?] Bioconductor  
 P AnnotationHub          * 2.20.0   2020-04-27 [?] Bioconductor  
 P askpass                  1.1      2019-01-13 [?] CRAN (R 4.0.0)
 P assertthat               0.2.1    2019-03-21 [?] CRAN (R 4.0.0)
 P backports                1.1.8    2020-06-17 [?] CRAN (R 4.0.0)
 P beeswarm                 0.2.3    2016-04-25 [?] CRAN (R 4.0.0)
 P Biobase                * 2.48.0   2020-04-27 [?] Bioconductor  
 P BiocFileCache          * 1.12.0   2020-04-27 [?] Bioconductor  
 P BiocGenerics           * 0.34.0   2020-04-27 [?] Bioconductor  
 P BiocManager              1.30.10  2019-11-16 [?] CRAN (R 4.0.0)
 P BiocNeighbors            1.6.0    2020-04-27 [?] Bioconductor  
 P BiocParallel           * 1.22.0   2020-04-27 [?] Bioconductor  
 P BiocSingular             1.4.0    2020-04-27 [?] Bioconductor  
 P BiocStyle              * 2.16.0   2020-04-27 [?] Bioconductor  
 P BiocVersion              3.11.1   2019-11-13 [?] Bioconductor  
 P biomaRt                  2.44.1   2020-06-17 [?] Bioconductor  
 P Biostrings               2.56.0   2020-04-27 [?] Bioconductor  
 P bit                      1.1-15.2 2020-02-10 [?] CRAN (R 4.0.0)
 P bit64                    0.9-7    2017-05-08 [?] CRAN (R 4.0.0)
 P bitops                   1.0-6    2013-08-17 [?] CRAN (R 4.0.0)
 P blob                     1.2.1    2020-01-20 [?] CRAN (R 4.0.0)
 P cli                      2.0.2    2020-02-28 [?] CRAN (R 4.0.0)
 P codetools                0.2-16   2018-12-24 [3] CRAN (R 4.0.0)
 P colorspace               1.4-1    2019-03-18 [?] CRAN (R 4.0.0)
 P cowplot                * 1.0.0    2019-07-11 [?] CRAN (R 4.0.0)
 P crayon                   1.3.4    2017-09-16 [?] CRAN (R 4.0.0)
 P curl                     4.3      2019-12-02 [?] CRAN (R 4.0.0)
 P DBI                      1.1.0    2019-12-15 [?] CRAN (R 4.0.0)
 P dbplyr                 * 1.4.4    2020-05-27 [?] CRAN (R 4.0.0)
 P DelayedArray           * 0.14.0   2020-04-27 [?] Bioconductor  
 P DelayedMatrixStats       1.10.0   2020-04-27 [?] Bioconductor  
 P digest                   0.6.25   2020-02-23 [?] CRAN (R 4.0.0)
 P distill                  0.8      2020-06-04 [?] CRAN (R 4.0.0)
 P dplyr                    1.0.0    2020-05-29 [?] CRAN (R 4.0.0)
 P dqrng                    0.2.1    2019-05-17 [?] CRAN (R 4.0.0)
 P edgeR                  * 3.30.3   2020-06-02 [?] Bioconductor  
 P ellipsis                 0.3.1    2020-05-15 [?] CRAN (R 4.0.0)
 P ensembldb              * 2.12.1   2020-05-06 [?] Bioconductor  
 P evaluate                 0.14     2019-05-28 [?] CRAN (R 4.0.0)
 P fansi                    0.4.1    2020-01-08 [?] CRAN (R 4.0.0)
 P farver                   2.0.3    2020-01-16 [?] CRAN (R 4.0.0)
 P fastmap                  1.0.1    2019-10-08 [?] CRAN (R 4.0.0)
 P generics                 0.0.2    2018-11-29 [?] CRAN (R 4.0.0)
 P GenomeInfoDb           * 1.24.2   2020-06-15 [?] Bioconductor  
 P GenomeInfoDbData         1.2.3    2020-05-18 [?] Bioconductor  
 P GenomicAlignments        1.24.0   2020-04-27 [?] Bioconductor  
 P GenomicFeatures        * 1.40.0   2020-04-27 [?] Bioconductor  
 P GenomicRanges          * 1.40.0   2020-04-27 [?] Bioconductor  
 P ggbeeswarm               0.6.0    2017-08-07 [?] CRAN (R 4.0.0)
 P ggplot2                * 3.3.2    2020-06-19 [?] CRAN (R 4.0.0)
 P Glimma                 * 1.16.0   2020-04-27 [?] Bioconductor  
 P glue                     1.4.1    2020-05-13 [?] CRAN (R 4.0.0)
 P gridExtra                2.3      2017-09-09 [?] CRAN (R 4.0.0)
 P gtable                   0.3.0    2019-03-25 [?] CRAN (R 4.0.0)
 P here                   * 0.1      2017-05-28 [?] CRAN (R 4.0.0)
 P highr                    0.8      2019-03-20 [?] CRAN (R 4.0.0)
 P hms                      0.5.3    2020-01-08 [?] CRAN (R 4.0.0)
 P htmltools                0.5.0    2020-06-16 [?] CRAN (R 4.0.0)
 P httpuv                   1.5.4    2020-06-06 [?] CRAN (R 4.0.0)
 P httr                     1.4.1    2019-08-05 [?] CRAN (R 4.0.0)
 P igraph                   1.2.5    2020-03-19 [?] CRAN (R 4.0.0)
 P interactiveDisplayBase   1.26.3   2020-06-02 [?] Bioconductor  
 P IRanges                * 2.22.2   2020-05-21 [?] Bioconductor  
 P irlba                    2.3.3    2019-02-05 [?] CRAN (R 4.0.0)
 P janitor                * 2.0.1    2020-04-12 [?] CRAN (R 4.0.0)
 P jsonlite                 1.6.1    2020-02-02 [?] CRAN (R 4.0.0)
 P knitr                    1.29     2020-06-23 [?] CRAN (R 4.0.0)
 P labeling                 0.3      2014-08-23 [?] CRAN (R 4.0.0)
 P later                    1.1.0.1  2020-06-05 [?] CRAN (R 4.0.0)
 P lattice                  0.20-41  2020-04-02 [?] CRAN (R 4.0.0)
 P lazyeval                 0.2.2    2019-03-15 [?] CRAN (R 4.0.0)
 P lifecycle                0.2.0    2020-03-06 [?] CRAN (R 4.0.0)
 P limma                  * 3.44.3   2020-06-12 [?] Bioconductor  
 P locfit                   1.5-9.4  2020-03-25 [?] CRAN (R 4.0.0)
 P lubridate                1.7.9    2020-06-08 [?] CRAN (R 4.0.0)
 P magrittr                 1.5      2014-11-22 [?] CRAN (R 4.0.0)
 P Matrix                 * 1.2-18   2019-11-27 [?] CRAN (R 4.0.0)
 P matrixStats            * 0.56.0   2020-03-13 [?] CRAN (R 4.0.0)
 P memoise                  1.1.0    2017-04-21 [?] CRAN (R 4.0.0)
 P mime                     0.9      2020-02-04 [?] CRAN (R 4.0.0)
 P msigdbr                * 7.1.1    2020-05-14 [?] CRAN (R 4.0.0)
 P munsell                  0.5.0    2018-06-12 [?] CRAN (R 4.0.0)
 P openssl                  1.4.1    2019-07-18 [?] CRAN (R 4.0.0)
 P pillar                   1.4.4    2020-05-05 [?] CRAN (R 4.0.0)
 P pkgconfig                2.0.3    2019-09-22 [?] CRAN (R 4.0.0)
 P prettyunits              1.1.1    2020-01-24 [?] CRAN (R 4.0.0)
 P progress                 1.2.2    2019-05-16 [?] CRAN (R 4.0.0)
 P promises                 1.1.1    2020-06-09 [?] CRAN (R 4.0.0)
 P ProtGenerics             1.20.0   2020-04-27 [?] Bioconductor  
 P purrr                    0.3.4    2020-04-17 [?] CRAN (R 4.0.0)
 P R6                       2.4.1    2019-11-12 [?] CRAN (R 4.0.0)
 P rappdirs                 0.3.1    2016-03-28 [?] CRAN (R 4.0.0)
 P Rcpp                     1.0.4.6  2020-04-09 [?] CRAN (R 4.0.0)
 P RcppAnnoy                0.0.16   2020-03-08 [?] CRAN (R 4.0.0)
 P RCurl                    1.98-1.2 2020-04-18 [?] CRAN (R 4.0.0)
 P rlang                    0.4.6    2020-05-02 [?] CRAN (R 4.0.0)
 P rmarkdown              * 2.3      2020-06-18 [?] CRAN (R 4.0.0)
 P rprojroot                1.3-2    2018-01-03 [?] CRAN (R 4.0.0)
 P Rsamtools                2.4.0    2020-04-27 [?] Bioconductor  
 P RSpectra                 0.16-0   2019-12-01 [?] CRAN (R 4.0.0)
 P RSQLite                  2.2.0    2020-01-07 [?] CRAN (R 4.0.0)
 P rsvd                     1.0.3    2020-02-17 [?] CRAN (R 4.0.0)
 P rtracklayer              1.48.0   2020-04-27 [?] Bioconductor  
 P S4Vectors              * 0.26.1   2020-05-16 [?] Bioconductor  
 P scales                   1.1.1    2020-05-11 [?] CRAN (R 4.0.0)
 P scater                 * 1.16.1   2020-05-28 [?] Bioconductor  
 P scran                  * 1.16.0   2020-04-27 [?] Bioconductor  
 P sessioninfo              1.1.1    2018-11-05 [?] CRAN (R 4.0.0)
 P shiny                    1.5.0    2020-06-23 [?] CRAN (R 4.0.0)
 P SingleCellExperiment   * 1.10.1   2020-04-28 [?] Bioconductor  
 P snakecase                0.11.0   2019-05-25 [?] CRAN (R 4.0.0)
 P statmod                  1.4.34   2020-02-17 [?] CRAN (R 4.0.0)
 P stringi                  1.4.6    2020-02-17 [?] CRAN (R 4.0.0)
 P stringr                  1.4.0    2019-02-10 [?] CRAN (R 4.0.0)
 P SummarizedExperiment   * 1.18.1   2020-04-30 [?] Bioconductor  
 P tibble                   3.0.1    2020-04-20 [?] CRAN (R 4.0.0)
 P tidyselect               1.1.0    2020-05-11 [?] CRAN (R 4.0.0)
 P uwot                   * 0.1.8    2020-03-16 [?] CRAN (R 4.0.0)
 P vctrs                    0.3.1    2020-06-05 [?] CRAN (R 4.0.0)
 P vipor                    0.4.5    2017-03-22 [?] CRAN (R 4.0.0)
 P viridis                  0.5.1    2018-03-29 [?] CRAN (R 4.0.0)
 P viridisLite              0.3.0    2018-02-01 [?] CRAN (R 4.0.0)
 P withr                    2.2.0    2020-04-20 [?] CRAN (R 4.0.0)
 P xfun                     0.15     2020-06-21 [?] CRAN (R 4.0.0)
 P XML                      3.99-0.3 2020-01-20 [?] CRAN (R 4.0.0)
 P xtable                   1.8-4    2019-04-21 [?] CRAN (R 4.0.0)
 P XVector                  0.28.0   2020-04-27 [?] Bioconductor  
 P yaml                     2.2.1    2020-02-01 [?] CRAN (R 4.0.0)
 P zlibbioc                 1.34.0   2020-04-27 [?] Bioconductor  

[1] /stornext/General/data/user_managed/grpu_mritchie_1/hickey/SCORE/C057_Cooney/renv/library/R-4.0/x86_64-pc-linux-gnu
[2] /tmp/Rtmp8K9ryC/renv-system-library
[3] /stornext/System/data/apps/R/R-4.0.0/lib64/R/library

 P ── Loaded and on-disk path mismatch.</code></pre>
</div>
</details>
<div id="refs" class="references" role="doc-bibliography">
<div id="ref-anders2010differential">
<p>Anders, S., and W. Huber. 2010. “Differential Expression Analysis for Sequence Count Data.” <em>Genome Biol.</em> 11 (10): R106. <a href="https://www.ncbi.nlm.nih.gov/pubmed/20979621">https://www.ncbi.nlm.nih.gov/pubmed/20979621</a>.</p>
</div>
<div id="ref-bourgon2010independent">
<p>Bourgon, R., R. Gentleman, and W. Huber. 2010. “Independent Filtering Increases Detection Power for High-Throughput Experiments.” <em>Proc. Natl. Acad. Sci. U.S.A.</em> 107 (21): 9546–51. <a href="https://www.ncbi.nlm.nih.gov/pubmed/20460310">https://www.ncbi.nlm.nih.gov/pubmed/20460310</a>.</p>
</div>
<div id="ref-love2014moderated">
<p>Love, M. I., W. Huber, and S. Anders. 2014. “Moderated Estimation of Fold Change and Dispersion for Rna-Seq Data with Deseq2.” <em>Genome Biol.</em> 15 (12): 550. <a href="https://www.ncbi.nlm.nih.gov/pubmed/25516281">https://www.ncbi.nlm.nih.gov/pubmed/25516281</a>.</p>
</div>
<div id="ref-lun2016pooling">
<p>Lun, A. T., K. Bach, and J. C. Marioni. 2016. <em>Genome Biol.</em> 17 (April): 75. <a href="https://www.ncbi.nlm.nih.gov/pubmed/27122128">https://www.ncbi.nlm.nih.gov/pubmed/27122128</a>.</p>
</div>
<div id="ref-lun2016step">
<p>Lun, A. T. L., D. J. McCarthy, and J. C. Marioni. 2016. “A Step-by-Step Workflow for Low-Level Analysis of Single-Cell RNA-seq Data.” <em>F1000Res.</em> 5 (August).</p>
</div>
<div id="ref-McInnes2018-hy">
<p>McInnes, Leland, John Healy, and James Melville. 2018. “UMAP: Uniform Manifold Approximation and Projection for Dimension Reduction,” February. <a href="http://arxiv.org/abs/1802.03426">http://arxiv.org/abs/1802.03426</a>.</p>
</div>
<div id="ref-robinson2010scaling">
<p>Robinson, M. D., and A. Oshlack. 2010. “A Scaling Normalization Method for Differential Expression Analysis of Rna-Seq Data.” <em>Genome Biol.</em> 11 (3): R25. <a href="https://www.ncbi.nlm.nih.gov/pubmed/20196867">https://www.ncbi.nlm.nih.gov/pubmed/20196867</a>.</p>
</div>
<div id="ref-stegle2015computational">
<p>Stegle, O., S. A. Teichmann, and J. C. Marioni. 2015. “Computational and Analytical Challenges in Single-Cell Transcriptomics.” <em>Nat. Rev. Genet.</em> 16 (3): 133–45. <a href="https://www.ncbi.nlm.nih.gov/pubmed/25628217">https://www.ncbi.nlm.nih.gov/pubmed/25628217</a>.</p>
</div>
</div>
<section class="footnotes" role="doc-endnotes">
<hr />
<ol>
<li id="fn1" role="doc-endnote"><p>Some care is taken to account for missing and duplicate gene symbols; missing symbols are replaced with the Ensembl identifier and duplicated symbols are concatenated with the (unique) Ensembl identifiers.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2" role="doc-endnote"><p>The number of expressed features refers to the number of genes which have non-zero counts (i.e. they have been identified in the cell at least once)<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3" role="doc-endnote"><p>This is consistent with the use of UMI counts rather than read counts, as each transcript molecule can only produce one UMI count but can yield many reads after fragmentation.<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4" role="doc-endnote"><p>It is important to note that we only using droplets assigned to a sample (i.e. we ignore <code>Unknown</code> samples) for the calculation of these thresholds.<a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5" role="doc-endnote"><p>This is a large number of HVGs but the results don’t qualitatively change by reducing this to, say, the top-1000 HVGs.<a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn6" role="doc-endnote"><p>Note that the converse is not true, i.e., there is no guarantee that the retained PCs capture all of the signal, which is only generally possible if no dimensionality reduction is performed at all.<a href="#fnref6" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom"></div>
<script id="distill-bibliography" type="text/bibtex">
@Article{stegle2015computational,
    Author="Stegle, O.  and Teichmann, S. A.  and Marioni, J. C. ",
    Title="Computational and analytical challenges in single-cell transcriptomics",
    Journal="Nat. Rev. Genet.",
    Year="2015",
    Volume="16",
    Number="3",
    Pages="133--145",
    Month="Mar",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/25628217}
}

@Article{islam2014quantitative,
    Author="Islam, S.  and Zeisel, A.  and Joost, S.  and La Manno, G.  and Zajac, P.  and Kasper, M.  and Lonnerberg, P.  and Linnarsson, S. ",
    Title="Quantitative single-cell RNA-seq with unique molecular identifiers",
    Journal="Nat. Methods",
    Year="2014",
    Volume="11",
    Number="2",
    Pages="163--166",
    Month="Feb"
}

@Article{huber2015orchestrating,
    Author="Huber, W.  and Carey, V. J.  and Gentleman, R.  and Anders, S.  and Carlson, M.  and Carvalho, B. S.  and Bravo, H. C.  and Davis, S.  and Gatto, L.  and Girke, T.  and Gottardo, R.  and Hahne, F.  and Hansen, K. D.  and Irizarry, R. A.  and Lawrence, M.  and Love, M. I.  and MacDonald, J.  and Obenchain, V.  and Oles, A. K.  and Pag\`es, H.  and Reyes, A.  and Shannon, P.  and Smyth, G. K.  and Tenenbaum, D.  and Waldron, L.  and Morgan, M. ",
    Title="{{O}rchestrating high-throughput genomic analysis with {B}ioconductor}",
    Journal="Nat. Methods",
    Year="2015",
    Volume="12",
    Number="2",
    Pages="115--121",
    Month="Feb"
}

@Article{brennecke2013accounting,
    Author="Brennecke, P.  and Anders, S.  and Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Zhang, X.  and Proserpio, V.  and Baying, B.  and Benes, V.  and Teichmann, S. A.  and Marioni, J. C.  and Heisler, M. G. ",
    Title="{{A}ccounting for technical noise in single-cell {R}{N}{A}-seq experiments}",
    Journal="Nat. Methods",
    Year="2013",
    Volume="10",
    Number="11",
    Pages="1093--1095",
    Month="Nov"
}

@Article{marinov2014singlecell,
    Author="Marinov, G. K.  and Williams, B. A.  and McCue, K.  and Schroth, G. P.  and Gertz, J.  and Myers, R. M.  and Wold, B. J. ",
    Title="{{F}rom single-cell to cell-pool transcriptomes: stochasticity in gene expression and {R}{N}{A} splicing}",
    Journal="Genome Res.",
    Year="2014",
    Volume="24",
    Number="3",
    Pages="496--510",
    Month="Mar"
}

@Article{vallejos2015basics,
    Author="Vallejos, C. A.  and Marioni, J. C.  and Richardson, S. ",
    Title="{{B}{A}{S}i{C}{S}: Bayesian analysis of single-cell sequencing data}",
    Journal="PLoS Comput. Biol.",
    Year="2015",
    Volume="11",
    Number="6",
    Pages="e1004333",
    Month="Jun"
}

@Article{trapnell2014dynamics,
    Author="Trapnell, C.  and Cacchiarelli, D.  and Grimsby, J.  and Pokharel, P.  and Li, S.  and Morse, M.  and Lennon, N. J.  and Livak, K. J.  and Mikkelsen, T. S.  and Rinn, J. L. ",
    Title="{{T}he dynamics and regulators of cell fate decisions are revealed by pseudotemporal ordering of single cells}",
    Journal="Nat. Biotechnol.",
    Year="2014",
    Volume="32",
    Number="4",
    Pages="381--386",
    Month="Apr"
}

@Article{kolod2015singlecell,
    Author="Ko\l{}odziejczyk, A. A.  and Kim, J. K.  and Tsang, J. C.  and Ilicic, T.  and Henriksson, J.  and Natarajan, K. N.  and Tuck, A. C.  and Gao, X.  and Buhler, M.  and Liu, P.  and Marioni, J. C.  and Teichmann, S. A. ",
    Title="{{S}ingle cell {R}{N}{A}-sequencing of pluripotent states unlocks modular transcriptional variation}",
    Journal="Cell Stem Cell",
    Year="2015",
    Volume="17",
    Number="4",
    Pages="471--485",
    Month="Oct"
}

@Article{anders2010differential,
    Author="Anders, S.  and Huber, W. ",
    Title="Differential expression analysis for sequence count data",
    Journal="Genome Biol.",
    Year="2010",
    Volume="11",
    Number="10",
    Pages="R106",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20979621}
}

@Article{robinson2010scaling,
    Author="Robinson, M. D.  and Oshlack, A. ",
    Title="A scaling normalization method for differential expression analysis of RNA-seq data",
    Journal="Genome Biol.",
    Year="2010",
    Volume="11",
    Number="3",
    Pages="R25",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20196867}
}

@Article{islam2011characterization,
    Author="Islam, S.  and Kjallquist, U.  and Moliner, A.  and Zajac, P.  and Fan, J. B.  and Lonnerberg, P.  and Linnarsson, S. ",
    Title="{{C}haracterization of the single-cell transcriptional landscape by highly multiplex {R}{N}{A}-seq}",
    Journal="Genome Res.",
    Year="2011",
    Volume="21",
    Number="7",
    Pages="1160--1167",
    Month="Jul"
}

@Article{wilson2015combined,
    Author="Wilson, N. K.  and Kent, D. G.  and Buettner, F.  and Shehata, M.  and Macaulay, I. C.  and Calero-Nieto, F. J.  and Sanchez Castillo, M.  and Oedekoven, C. A.  and Diamanti, E.  and Schulte, R.  and Ponting, C. P.  and Voet, T.  and Caldas, C.  and Stingl, J.  and Green, A. R.  and Theis, F. J.  and Gottgens, B. ",
    Title="{{C}ombined single-cell functional and gene expression analysis resolves heterogeneity within stem cell populations}",
    Journal="Cell Stem Cell",
    Year="2015",
    Volume="16",
    Number="6",
    Pages="712--724",
    Month="Jun"
}

@Article{picelli2014fulllength,
    Author="Picelli, S.  and Faridani, O. R.  and Bjorklund, A. K.  and Winberg, G.  and Sagasser, S.  and Sandberg, R. ",
    Title="{{F}ull-length {R}{N}{A}-seq from single cells using {S}mart-seq2}",
    Journal="Nat Protoc",
    Year="2014",
    Volume="9",
    Number="1",
    Pages="171--181",
    Month="Jan"
}

@Article{pollen2014lowcoverage,
    Author="Pollen, A. A.  and Nowakowski, T. J.  and Shuga, J.  and Wang, X.  and Leyrat, A. A.  and Lui, J. H.  and Li, N.  and Szpankowski, L.  and Fowler, B.  and Chen, P.  and Ramalingam, N.  and Sun, G.  and Thu, M.  and Norris, M.  and Lebofsky, R.  and Toppani, D.  and Kemp, D. W.  and Wong, M.  and Clerkson, B.  and Jones, B. N.  and Wu, S.  and Knutsson, L.  and Alvarado, B.  and Wang, J.  and Weaver, L. S.  and May, A. P.  and Jones, R. C.  and Unger, M. A.  and Kriegstein, A. R.  and West, J. A. ",
    Title="{{L}ow-coverage single-cell m{R}{N}{A} sequencing reveals cellular heterogeneity and activated signaling pathways in developing cerebral cortex}",
    Journal="Nat. Biotechnol.",
    Year="2014",
    Volume="32",
    Number="10",
    Pages="1053--1058",
    Month="Oct"
}

@Article{love2015rnaseq,
    Author="Love, M. I.  and Anders, S.  and Kim, V.  and Huber, W. ",
    Title="{{R}{N}{A}-{S}eq workflow: gene-level exploratory analysis and differential expression}",
    Journal="F1000Res",
    Year="2015",
    Volume="4",
    Pages="1070"
}

@Article{liao2014featurecounts,
    Author="Liao, Y.  and Smyth, G. K.  and Shi, W. ",
    Title="{feature{C}ounts: an efficient general purpose program for assigning sequence reads to genomic features}",
    Journal="Bioinformatics",
    Year="2014",
    Volume="30",
    Number="7",
    Pages="923--930",
    Month="Apr"
}

@Article{liao2013subread,
    Author="Liao, Y.  and Smyth, G. K.  and Shi, W. ",
    Title="{{T}he {S}ubread aligner: fast, accurate and scalable read mapping by seed-and-vote}",
    Journal="Nucleic Acids Res.",
    Year="2013",
    Volume="41",
    Number="10",
    Pages="e108",
    Month="May"
}

@Article{law2014voom,
    Author="Law, C. W.  and Chen, Y.  and Shi, W.  and Smyth, G. K. ",
    Title="voom: Precision weights unlock linear model analysis tools for RNA-seq read counts",
    Journal="Genome Biol.",
    Year="2014",
    Volume="15",
    Number="2",
    Pages="R29",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/24485249}
}

@Article{love2014moderated,
    Author="Love, M. I.  and Huber, W.  and Anders, S. ",
    Title="Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2",
    Journal="Genome Biol.",
    Year="2014",
    Volume="15",
    Number="12",
    Pages="550",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/25516281}
}

@article{van2008visualizing,
    title={Visualizing data using t-SNE},
    author={Van der Maaten, L. and Hinton, G.},
    journal={J. Mach. Learn. Res.},
    volume={9},
    number={2579-2605},
    pages={85},
    year={2008},
    URL={http://www.jmlr.org/papers/v9/vandermaaten08a.html}
}

@Article{scialdone2015computational,
    Author="Scialdone, A.  and Natarajan, K. N.  and Saraiva, L. R.  and Proserpio, V.  and Teichmann, S. A.  and Stegle, O.  and Marioni, J. C.  and Buettner, F. ",
    Title="Computational assignment of cell-cycle stage from single-cell transcriptome data",
    Journal="Methods",
    Year="2015",
    Volume="85",
    Pages="54--61",
    Month="Sep",
    URL = {https://www.ncbi.nlm.nih.gov/pubmed/26142758},
}

@Article{kim2015characterizing,
    Author="Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Illicic, T.  and Teichmann, S. A.  and Marioni, J. C. ",
    Title="{{C}haracterizing noise structure in single-cell {R}{N}{A}-seq distinguishes genuine from technical stochastic allelic expression}",
    Journal="Nat. Commun.",
    Year="2015",
    Volume="6",
    Pages="8687"
}

@Article{mccarthy2012differential,
    Author="McCarthy, D. J.  and Chen, Y.  and Smyth, G. K. ",
    Title="{{D}ifferential expression analysis of multifactor {R}{N}{A}-{S}eq experiments with respect to biological variation}",
    Journal="Nucleic Acids Res.",
    Year="2012",
    Volume="40",
    Number="10",
    Pages="4288--4297",
    Month="May"
}

@Article{angel1991role,
    Author="Angel, P.  and Karin, M. ",
    Title="{{T}he role of {J}un, {F}os and the {A}{P}-1 complex in cell-proliferation and transformation}",
    Journal="Biochim. Biophys. Acta",
    Year="1991",
    Volume="1072",
    Number="2-3",
    Pages="129--157",
    Month="Dec"
}

@Article{langfelder2008defining,
    Author="Langfelder, P.  and Zhang, B.  and Horvath, S. ",
    Title="{{D}efining clusters from a hierarchical cluster tree: the {D}ynamic {T}ree {C}ut package for {R}}",
    Journal="Bioinformatics",
    Year="2008",
    Volume="24",
    Number="5",
    Pages="719--720",
    Month="Mar"
}

@Article{ilicic2016classification,
    Author="Ilicic, T.  and Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Bagger, F. O.  and McCarthy, D. J.  and Marioni, J. C.  and Teichmann, S. A. ",
    Title="{{C}lassification of low quality cells from single-cell {R}{N}{A}-seq data}",
    Journal="Genome Biol.",
    Year="2016",
    Volume="17",
    Number="1",
    Pages="29"
}

@Article{heng2008immunological,
    Author="Heng, T. S.  and Painter, M. W.  and Elpek, K.  and Lukacs-Kornek, V.  and Mauermann, N.  and Turley, S. J.  and Koller, D.  and Kim, F. S.  and Wagers, A. J.  and Asinovski, N.  and Davis, S.  and Fassett, M.  and Feuerer, M.  and Gray, D. H.  and Haxhinasto, S.  and Hill, J. A.  and Hyatt, G.  and Laplace, C.  and Leatherbee, K.  and Mathis, D.  and Benoist, C.  and Jianu, R.  and Laidlaw, D. H.  and Best, J. A.  and Knell, J.  and Goldrath, A. W.  and Jarjoura, J.  and Sun, J. C.  and Zhu, Y.  and Lanier, L. L.  and Ergun, A.  and Li, Z.  and Collins, J. J.  and Shinton, S. A.  and Hardy, R. R.  and Friedline, R.  and Sylvia, K.  and Kang, J. ",
    Title="{{T}he {I}mmunological {G}enome {P}roject: networks of gene expression in immune cells}",
    Journal="Nat. Immunol.",
    Year="2008",
    Volume="9",
    Number="10",
    Pages="1091--1094",
    Month="Oct"
}

@Article{zeisel2015brain,
    Author="Zeisel, A.  and Munoz-Manchado, A. B.  and Codeluppi, S.  and Lonnerberg, P.  and La Manno, G.  and Jureus, A.  and Marques, S.  and Munguba, H.  and He, L.  and Betsholtz, C.  and Rolny, C.  and Castelo-Branco, G.  and Hjerling-Leffler, J.  and Linnarsson, S. ",
    Title="{{B}rain structure. {C}ell types in the mouse cortex and hippocampus revealed by single-cell {R}{N}{A}-seq}",
    Journal="Science",
    Year="2015",
    Volume="347",
    Number="6226",
    Pages="1138--1142",
    Month="Mar"
}

@Article{robinson2010edgeR,
    Author="Robinson, M. D.  and McCarthy, D. J.  and Smyth, G. K. ",
    Title="{edge{R}: a {B}ioconductor package for differential expression analysis of digital gene expression data}",
    Journal="Bioinformatics",
    Year="2010",
    Volume="26",
    Number="1",
    Pages="139--140",
    Month="Jan"
}

@Article{ritchie2015limma,
    Author="Ritchie, M. E.  and Phipson, B.  and Wu, D.  and Hu, Y.  and Law, C. W.  and Shi, W.  and Smyth, G. K. ",
    Title="{limma powers differential expression analyses for {R}{N}{A}-sequencing and microarray studies}",
    Journal="Nucleic Acids Res.",
    Year="2015",
    Volume="43",
    Number="7",
    Pages="e47",
    Month="Apr"
}

@Article{julia2015sincell,
    Author="Julia, M.  and Telenti, A.  and Rausell, A. ",
    Title="{{S}incell: an {R}/{B}ioconductor package for statistical assessment of cell-state hierarchies from single-cell {R}{N}{A}-seq}",
    Journal="Bioinformatics",
    Year="2015",
    Volume="31",
    Number="20",
    Pages="3380--3382",
    Month="Oct"
}

@Article{buettner2015computational,
    Author="Buettner, F.  and Natarajan, K. N.  and Casale, F. P.  and Proserpio, V.  and Scialdone, A.  and Theis, F. J.  and Teichmann, S. A.  and Marioni, J. C.  and Stegle, O. ",
    Title="{{C}omputational analysis of cell-to-cell heterogeneity in single-cell {R}{N}{A}-sequencing data reveals hidden subpopulations of cells}",
    Journal="Nat. Biotechnol.",
    Year="2015",
    Volume="33",
    Number="2",
    Pages="155--160",
    Month="Feb"
}

@Article{mahata2014singlecell,
    Author="Mahata, B.  and Zhang, X.  and Ko\l{}odziejczyk, A. A.  and Proserpio, V.  and Haim-Vilmovsky, L.  and Taylor, A. E.  and Hebenstreit, D.  and Dingler, F. A.  and Moignard, V.  and Gottgens, B.  and Arlt, W.  and McKenzie, A. N.  and Teichmann, S. A. ",
    Title="{{S}ingle-cell {R}{N}{A} sequencing reveals {T} helper cells synthesizing steroids de novo to contribute to immune homeostasis}",
    Journal="Cell Rep.",
    Year="2014",
    Volume="7",
    Number="4",
    Pages="1130--1142",
    Month="May"
}

@Article{phipson2010permutation,
    Author="Phipson, B.  and Smyth, G. K. ",
    Title="{{P}ermutation {P}-values should never be zero: calculating exact {P}-values when permutations are randomly drawn}",
    Journal="Stat. Appl. Genet. Mol. Biol.",
    Year="2010",
    Volume="9",
    Pages="Article 39"
}

@incollection{chen2014differential,
    title="{Differential expression analysis of complex RNA-seq experiments using edgeR}",
    author={Chen, Y. and Lun, A. T. L. and Smyth, G. K.},
    booktitle={Statistical analysis of next generation sequencing data},
    editor={Datta, S. and Nettleton, D.},
    pages={51--74},
    year={2014},
    publisher={Springer}
}

@article{hicks2015widespread,
    author = {Hicks, S. C. and Teng, M. and Irizarry, R. A.},
    title = "On the widespread and critical impact of systematic bias and batch effects in single-cell RNA-Seq data",
    year = {2015},
    doi = {10.1101/025528},
    publisher = {Cold Spring Harbor Labs Journals},
    URL = {http://biorxiv.org/content/early/2015/08/25/025528},
    eprint = {http://biorxiv.org/content/early/2015/08/25/025528.full.pdf},
    journal = {bioRxiv}
}

@Article{leng2015oscope,
    Author="Leng, N.  and Chu, L. F.  and Barry, C.  and Li, Y.  and Choi, J.  and Li, X.  and Jiang, P.  and Stewart, R. M.  and Thomson, J. A.  and Kendziorski, C. ",
    Title="{{O}scope identifies oscillatory genes in unsynchronized single-cell {R}{N}{A}-seq experiments}",
    Journal="Nat. Methods",
    Year="2015",
    Volume="12",
    Number="10",
    Pages="947--950",
    Month="Oct"
}

@Article{fan2016characterizing,
    Author="Fan, J.  and Salathia, N.  and Liu, R.  and Kaeser, G. E.  and Yung, Y. C.  and Herman, J. L.  and Kaper, F.  and Fan, J. B.  and Zhang, K.  and Chun, J.  and Kharchenko, P. V. ",
    Title="{{C}haracterizing transcriptional heterogeneity through pathway and gene set overdispersion analysis}",
    Journal="Nat. Methods",
    Year="2016",
    Volume="13",
    Number="3",
    Pages="241--244",
    Month="Mar"
}

@Article{lund2012detecting,
    Author="Lund, S. P.  and Nettleton, D.  and McCarthy, D. J.  and Smyth, G. K. ",
    Title="{{D}etecting differential expression in {R}{N}{A}-sequence data using quasi-likelihood with shrunken dispersion estimates}",
    Journal="Stat Appl Genet Mol Biol",
    Year="2012",
    Volume="11",
    Number="5",
    Pages=" "
}

@Article{angerer2016destiny,
    Author="Angerer, P.  and Haghverdi, L.  and Buttner, M.  and Theis, F. J.  and Marr, C.  and Buettner, F. ",
    Title="{destiny: diffusion maps for large-scale single-cell data in {R}}",
    Journal="Bioinformatics",
    Year="2016",
    Volume="32",
    Number="8",
    Pages="1241--1243",
}

@Article{lun2016pooling,
    Author="Lun, A. T.  and Bach, K.  and Marioni, J. C. ",
    Title="Pooling across cells to normalize single-cell RNA sequencing data with many zero counts}",
    Journal="Genome Biol.",
    Year="2016",
    Volume="17",
    Pages="75",
    Month="Apr",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/27122128}
}

@Article{zhu2006gata3,
    Author="Zhu, J. and Yamane, H. and Cote-Sierra, J. and Guo, L. and Paul, W. E.",
    Title="{{G}{A}{T}{A}-3 promotes {T}h2 responses through three different mechanisms: induction of {T}h2 cytokine production, selective growth of {T}h2 cells and inhibition of {T}h1 cell-specific factors}",
    Journal="Cell Res.",
    Year="2006",
    Volume="16",
    Number="1",
    Pages="3--10",
    Month="Jan"
}

@article{berger1996bioequivalence,
    author = "Berger, R. L. and Hsu, J. C.",
    doi = "10.1214/ss/1032280304",
    fjournal = "Statistical Science",
    journal = "Statist. Sci.",
    month = "11",
    number = "4",
    pages = "283--319",
    publisher = "The Institute of Mathematical Statistics",
    title = "Bioequivalence trials, intersection-union tests and equivalence confidence sets",
    url = "http://dx.doi.org/10.1214/ss/1032280304",
    volume = "11",
    year = "1996"
}

@Article{santos2015cyclebase,
    Author="Santos, A.  and Wernersson, R.  and Jensen, L. J. ",
    Title="{{C}yclebase 3.0: a multi-organism database on cell-cycle regulation and phenotypes}",
    Journal="Nucleic Acids Res.",
    Year="2015",
    Volume="43",
    Number="Database issue",
    Pages="D1140--1144",
    Month="Jan"
}

@article {tung2016batch,
    Author="Tung, P. Y.  and Blischak, J. D.  and Hsiao, C. J.  and Knowles, D. A.  and Burnett, J. E.  and Pritchard, J. K.  and Gilad, Y. ",
    Title="Batch effects and the effective design of single-cell gene expression studies",
    Journal="Sci. Rep.",
    Year="2017",
    Volume="7",
    Pages="39921",
    Month="Jan"
}

@Article{bray2016near,
    Author="Bray, N. L.  and Pimentel, H.  and Melsted, P.  and Pachter, L. ",
    Title="{{N}ear-optimal probabilistic {R}{N}{A}-seq quantification}",
    Journal="Nat. Biotechnol.",
    Year="2016",
    Volume="34",
    Number="5",
    Pages="525--527",
    Month="May"
}

@article{patro2015accurate,
    author = {Patro, Rob and Duggal, Geet and Kingsford, Carl},
    title = {Accurate, fast, and model-aware transcript expression quantification with Salmon},
    year = {2015},
    doi = {10.1101/021592},
    publisher = {Cold Spring Harbor Labs Journals},
    journal = {bioRxiv}
}

@Article{ji2016tscan,
    Author="Ji, Z.  and Ji, H. ",
    Title="{{T}{S}{C}{A}{N}: {P}seudo-time reconstruction and evaluation in single-cell {R}{N}{A}-seq analysis}",
    Journal="Nucleic Acids Res.",
    Year="2016",
    Volume="44",
    Number="13",
    Pages="e117",
    Month="Jul"
}

@Article{klein2015droplet,
    Author="Klein, A. M.  and Mazutis, L.  and Akartuna, I.  and Tallapragada, N.  and Veres, A.  and Li, V.  and Peshkin, L.  and Weitz, D. A.  and Kirschner, M. W. ",
    Title="{{D}roplet barcoding for single-cell transcriptomics applied to embryonic stem cells}",
    Journal="Cell",
    Year="2015",
    Volume="161",
    Number="5",
    Pages="1187--1201",
    Month="May"
}

@Article{macosko2015highly,
    Author="Macosko, E. Z.  and Basu, A.  and Satija, R.  and Nemesh, J.  and Shekhar, K.  and Goldman, M.  and Tirosh, I.  and Bialas, A. R.  and Kamitaki, N.  and Martersteck, E. M.  and Trombetta, J. J.  and Weitz, D. A.  and Sanes, J. R.  and Shalek, A. K.  and Regev, A.  and McCarroll, S. A. ",
    Title="{{H}ighly parallel genome-wide expression profiling of individual cells using nanoliter droplets}",
    Journal="Cell",
    Year="2015",
    Volume="161",
    Number="5",
    Pages="1202--1214",
    Month="May"
}

@Article{bourgon2010independent,
    Author="Bourgon, R.  and Gentleman, R.  and Huber, W. ",
    Title="Independent filtering increases detection power for high-throughput experiments",
    Journal="Proc. Natl. Acad. Sci. U.S.A.",
    Year="2010",
    Volume="107",
    Number="21",
    Pages="9546--9551",
    Month="May",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20460310}
}

@Article{conboy2007cell,
    Author="Conboy, C. M.  and Spyrou, C.  and Thorne, N. P.  and Wade, E. J.  and Barbosa-Morais, N. L.  and Wilson, M. D.  and Bhattacharjee, A.  and Young, R. A.  and Tavare, S.  and Lees, J. A.  and Odom, D. T. ",
    Title="Cell cycle genes are the evolutionarily conserved targets of the E2F4 transcription factor",
    Journal="PLoS ONE",
    Year="2007",
    Volume="2",
    Number="10",
    Pages="e1061",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/17957245}
}

@Article{bertoli2013control,
    Author="Bertoli, C.  and Skotheim, J. M.  and de Bruin, R. A. ",
    Title="Control of cell cycle transcription during G1 and S phases",
    Journal="Nat. Rev. Mol. Cell Biol.",
    Year="2013",
    Volume="14",
    Number="8",
    Pages="518--528",
    Month="Aug",
    URL={{C}ontrol of cell cycle transcription during {G}1 and {S} phases}
}

@Article{chen2016from,
    Author="Chen, Y.  and Lun, A. T.  and Smyth, G. K. ",
    Title="{{F}rom reads to genes to pathways: differential expression analysis of {R}{N}{A}-{S}eq experiments using {R}subread and the edge{R} quasi-likelihood pipeline}",
    Journal="F1000Res",
    Year="2016",
    Volume="5",
    Pages="1438"
}

@Article{coller2006new,
    Author="Coller, H. A.  and Sang, L.  and Roberts, J. M. ",
    Title="{{A} new description of cellular quiescence}",
    Journal="PLoS Biol.",
    Year="2006",
    Volume="4",
    Number="3",
    Pages="e83",
    Month="Mar"
}

@Article{mccarthy2017scater,
    Author="McCarthy, D. J.  and Campbell, K. R.  and Lun, A. T.  and Wills, Q. F. ",
    Title="{{S}cater: pre-processing, quality control, normalization and visualization of single-cell {R}{N}{A}-seq data in {R}}",
    Journal="Bioinformatics",
    Year="2017",
    Volume="33",
    Number="8",
    Pages="1179--1186",
    Month="Apr"
}

@article{simes1986improved,
    author = {Simes, R. J.},
    title = {An improved Bonferroni procedure for multiple tests of significance},
    volume = {73},
    number = {3},
    pages = {751--754},
    year = {1986},
    journal = {Biometrika}
}

@Article{xu2015identification,
    Author="Xu, C.  and Su, Z. ",
    Title="Identification of cell types from single-cell transcriptomes using a novel clustering method",
    Journal="Bioinformatics",
    Year="2015",
    Volume="31",
    Number="12",
    Pages="1974--1980",
    Month="Jun"
}

@Article{lun2017assessing,
    Author="Lun, A. T. L.  and Calero-Nieto, F. J.  and Haim-Vilmovsky, L.  and Gottgens, B.  and Marioni, J. C. ",
    Title="Assessing the reliability of spike-in normalization for analyses of single-cell RNA sequencing data",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="11",
    Pages="1795--1806",
    Month="Nov"
}

@Article{yang2016comparative,
    Author="Yang, Z.  and Algesheimer, R.  and Tessone, C. J. ",
    Title="{A comparative analysis of community detection algorithms on artificial networks}",
    Journal="Sci. Rep.",
    Year="2016",
    Volume="6",
    Pages="30750",
    Month="Aug"
}

@Article{smith2017umitools,
    Author="Smith, T.  and Heger, A.  and Sudbery, I. ",
    Title="{{U}{M}{I}-tools: modeling sequencing errors in {U}nique {M}olecular {I}dentifiers to improve quantification accuracy}",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="3",
    Pages="491--499",
    Month="Mar"
}

@Article{tasic2016adult,
    Author="Tasic, B.  and Menon, V.  and Nguyen, T. N.  and Kim, T. K.  and Jarsky, T.  and Yao, Z.  and Levi, B.  and Gray, L. T.  and Sorensen, S. A.  and Dolbeare, T.  and Bertagnolli, D.  and Goldy, J.  and Shapovalova, N.  and Parry, S.  and Lee, C.  and Smith, K.  and Bernard, A.  and Madisen, L.  and Sunkin, S. M.  and Hawrylycz, M.  and Koch, C.  and Zeng, H. ",
    Title="{{A}dult mouse cortical cell taxonomy revealed by single cell transcriptomics}",
    Journal="Nat. Neurosci.",
    Year="2016",
    Volume="19",
    Number="2",
    Pages="335--346",
    Month="Feb"
}

@Article{wajapeyee2010senescence,
    Author="Wajapeyee, N.  and Wang, S. Z.  and Serra, R. W.  and Solomon, P. D.  and Nagarajan, A.  and Zhu, X.  and Green, M. R. ",
    Title="{{S}enescence induction in human fibroblasts and hematopoietic progenitors by leukemogenic fusion proteins}",
    Journal="Blood",
    Year="2010",
    Volume="115",
    Number="24",
    Pages="5057--5060",
    Month="Jun"
}

@Article{zeng2012largescale,
    Author="Zeng, H.  and Shen, E. H.  and Hohmann, J. G.  and Oh, S. W.  and Bernard, A.  and Royall, J. J.  and Glattfelder, K. J.  and Sunkin, S. M.  and Morris, J. A.  and Guillozet-Bongaarts, A. L.  and Smith, K. A.  and Ebbert, A. J.  and Swanson, B.  and Kuan, L.  and Page, D. T.  and Overly, C. C.  and Lein, E. S.  and Hawrylycz, M. J.  and Hof, P. R.  and Hyde, T. M.  and Kleinman, J. E.  and Jones, A. R. ",
    Title="{{L}arge-scale cellular-resolution gene profiling in human neocortex reveals species-specific molecular signatures}",
    Journal="Cell",
    Year="2012",
    Volume="149",
    Number="2",
    Pages="483--496",
    Month="Apr"
}

@Article{grun2016denovo,
    Author="Grun, D.  and Muraro, M. J.  and Boisset, J. C.  and Wiebrands, K.  and Lyubimova, A.  and Dharmadhikari, G.  and van den Born, M.  and van Es, J.  and Jansen, E.  and Clevers, H.  and de Koning, E. J. P.  and van Oudenaarden, A. ",
    Title="{{D}e {N}ovo {P}rediction of {S}tem {C}ell {I}dentity using {S}ingle-{C}ell {T}ranscriptome {D}ata}",
    Journal="Cell Stem Cell",
    Year="2016",
    Volume="19",
    Number="2",
    Pages="266--277",
    Month="08"
}

@article{haghverdi2018batch,
    Author="Haghverdi, L.  and Lun, A. T. L.  and Morgan, M. D.  and Marioni, J. C. ",
    Title="Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors",
    Journal="Nat. Biotechnol.",
    Year="2018",
    Month="Apr"
}

@Article{muraro2016singlecell,
    Author="Muraro, M. J.  and Dharmadhikari, G.  and Grun, D.  and Groen, N.  and Dielen, T.  and Jansen, E.  and van Gurp, L.  and Engelse, M. A.  and Carlotti, F.  and de Koning, E. J.  and van Oudenaarden, A. ",
    Title="{{A} {S}ingle-{C}ell {T}ranscriptome {A}tlas of the {H}uman {P}ancreas}",
    Journal="Cell Syst",
    Year="2016",
    Volume="3",
    Number="4",
    Pages="385--394",
    Month="Oct"
}

@Article{segerstolpe2016singlecell,
    Author="Segerstolpe, A.  and Palasantza, A.  and Eliasson, P.  and Andersson, E. M.  and Andreasson, A. C.  and Sun, X.  and Picelli, S.  and Sabirsh, A.  and Clausen, M.  and Bjursell, M. K.  and Smith, D. M.  and Kasper, M.  and Ammala, C.  and Sandberg, R. ",
    Title="{{S}ingle-{C}ell {T}ranscriptome {P}rofiling of {H}uman {P}ancreatic {I}slets in {H}ealth and {T}ype 2 {D}iabetes}",
    Journal="Cell Metab.",
    Year="2016",
    Volume="24",
    Number="4",
    Pages="593--607",
    Month="Oct"
}

@Article{horn1965rationale,
    author="Horn, J. L.",
    title="A rationale and test for the number of factors in factor analysis",
    journal="Psychometrika",
    year="1965",
    month="Jun",
    day="01",
    volume="30",
    number="2",
    pages="179--185",
    issn="1860-0980",
    doi="10.1007/BF02289447",
    url="https://doi.org/10.1007/BF02289447"
}

@Article{buja1992remarks,
    Author="Buja, A.  and Eyuboglu, N. ",
    Title="{{R}emarks on {P}arallel {A}nalysis}",
    Journal="Multivariate Behav Res",
    Year="1992",
    Volume="27",
    Number="4",
    Pages="509--540",
    Month="Oct"
}

@Article{zheng2017massively,
    Author="Zheng, G. X.  and Terry, J. M.  and Belgrader, P.  and Ryvkin, P.  and Bent, Z. W.  and Wilson, R.  and Ziraldo, S. B.  and Wheeler, T. D.  and McDermott, G. P.  and Zhu, J.  and Gregory, M. T.  and Shuga, J.  and Montesclaros, L.  and Underwood, J. G.  and Masquelier, D. A.  and Nishimura, S. Y.  and Schnall-Levin, M.  and Wyatt, P. W.  and Hindson, C. M.  and Bharadwaj, R.  and Wong, A.  and Ness, K. D.  and Beppu, L. W.  and Deeg, H. J.  and McFarland, C.  and Loeb, K. R.  and Valente, W. J.  and Ericson, N. G.  and Stevens, E. A.  and Radich, J. P.  and Mikkelsen, T. S.  and Hindson, B. J.  and Bielas, J. H. ",
    Title="{{M}assively parallel digital transcriptional profiling of single cells}",
    Journal="Nat Commun",
    Year="2017",
    Volume="8",
    Pages="14049",
    Month="Jan"
}

@article{lun2018distinguishing,
    author = {Lun, A. and Riesenfeld, S. and Andrews, T. and Dao, T. P. and Gomes, T. and {{participants in the 1\textsuperscript{st} Human Cell Atlas Jamboree}} and Marioni, J.},
    title = {Distinguishing cells from empty droplets in droplet-based single-cell RNA sequencing data},
    year = {2018},
    doi = {10.1101/234872},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2018/04/04/234872},
    eprint = {https://www.biorxiv.org/content/early/2018/04/04/234872.full.pdf},
    journal = {bioRxiv}
}

@article {wolf2017graph,
    author = {Wolf, F. Alexander and Hamey, Fiona and Plass, Mireya and Solana, Jordi and Dahlin, Joakim S. and Gottgens, Berthold and Rajewsky, Nikolaus and Simon, Lukas and Theis, Fabian J.},
    title = {Graph abstraction reconciles clustering with trajectory inference through a topology preserving map of single cells},
    year = {2017},
    doi = {10.1101/208819},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2017/10/25/208819},
    eprint = {https://www.biorxiv.org/content/early/2017/10/25/208819.full.pdf},
    journal = {bioRxiv}
}

@Article{lawlor2017singlecell,
    Author="Lawlor, N.  and George, J.  and Bolisetty, M.  and Kursawe, R.  and Sun, L.  and Sivakamasundari, V.  and Kycia, I.  and Robson, P.  and Stitzel, M. L. ",
    Title="{{S}ingle-cell transcriptomes identify human islet cell signatures and reveal cell-type-specific expression changes in type 2 diabetes}",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="2",
    Pages="208--222",
    Month="02"
}

@Article{guimaraes2016patterns,
   Author="Guimaraes, J. C.  and Zavolan, M. ",
   Title="{{P}atterns of ribosomal protein expression specify normal and malignant human cells}",
   Journal="Genome Biol.",
   Year="2016",
   Volume="17",
   Number="1",
   Pages="236",
   Month="11"
}

@Article{glare2002betaactin,
   Author="Glare, E. M.  and Divjak, M.  and Bailey, M. J.  and Walters, E. H. ",
   Title="{beta-{A}ctin and {G}{A}{P}{D}{H} housekeeping gene expression in asthmatic airways is variable and not suitable for normalising m{R}{N}{A} levels}",
   Journal="Thorax",
   Year="2002",
   Volume="57",
   Number="9",
   Pages="765--770",
   Month="Sep"
}

@Article{nazari2015gapdh,
   Author="Nazari, F.  and Parham, A.  and Maleki, A. F. ",
   Title="{{G}{A}{P}{D}{H}, \beta-actin and \beta-microglobulin, as three common reference genes, are not reliable for gene expression studies in equine adipose- and marrow-derived mesenchymal stem cells}",
   Journal="J Anim Sci Technol",
   Year="2015",
   Volume="57",
   Pages="18"
}

@Article{soneson2018bias,
    Author="Soneson, C.  and Robinson, M. D. ",
    Title="Bias, robustness and scalability in single-cell differential expression analysis",
    Journal="Nat. Methods",
    URL = {https://www.ncbi.nlm.nih.gov/pubmed/29481549},
    Year="2018",
    Volume="15",
    Number="4",
    Pages="255--261",
    Month="Apr"
}

@Article{soneson2015differential,
    Author="Soneson, C.  and Love, M. I.  and Robinson, M. D. ",
    Title="{{D}ifferential analyses for {R}{N}{A}-seq: transcript-level estimates improve gene-level inferences}",
    Journal="F1000Res",
    Year="2015",
    Volume="4",
    Pages="1521"
}

@article{stoeckius2017hashing,
    author = {Stoeckius, M. and Zheng, S. and Houck-Loomis, B. and Hao, S. and Yeung, B. and Smibert, P. and Satija, R.},
    title = {Cell "hashing" with barcoded antibodies enables multiplexing and doublet detection for single cell genomics},
    year = {2017},
    doi = {10.1101/237693},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2017/12/21/237693},
    eprint = {https://www.biorxiv.org/content/early/2017/12/21/237693.full.pdf},
    journal = {bioRxiv}
}

@Article{kang2018multiplexed,
    Author="Kang, H. M.  and Subramaniam, M.  and Targ, S.  and Nguyen, M.  and Maliskova, L.  and McCarthy, E.  and Wan, E.  and Wong, S.  and Byrnes, L.  and Lanata, C. M.  and Gate, R. E.  and Mostafavi, S.  and Marson, A.  and Zaitlen, N.  and Criswell, L. A.  and Ye, C. J. ",
    Title="{{M}ultiplexed droplet single-cell {R}{N}{A}-sequencing using natural genetic variation}",
    Journal="Nat. Biotechnol.",
    Year="2018",
    Volume="36",
    Number="1",
    Pages="89--94",
    Month="Jan"
}

@Article{dahlin2018single,
    Author="Dahlin, J. S.  and Hamey, F. K.  and Pijuan-Sala, B.  and Shepherd, M.  and Lau, W. W. Y.  and Nestorowa, S.  and Weinreb, C.  and Wolock, S.  and Hannah, R.  and Diamanti, E.  and Kent, D. G.  and Gottgens, B.  and Wilson, N. K. ",
    Title="{{A} single-cell hematopoietic landscape resolves 8 lineage trajectories and defects in {K}it mutant mice}",
    Journal="Blood",
    Year="2018",
    Volume="131",
    Number="21",
    Pages="e1-e11",
    Month="May"
}

@Article{bach2017differentiation,
    Author="Bach, K.  and Pensa, S.  and Grzelak, M.  and Hadfield, J.  and Adams, D. J.  and Marioni, J. C.  and Khaled, W. T. ",
    Title="{{D}ifferentiation dynamics of mammary epithelial cells revealed by single-cell {R}{N}{A} sequencing}",
    Journal="Nat Commun",
    Year="2017",
    Volume="8",
    Number="1",
    Pages="2128",
    Month="Dec"
}

@article{tian2018scpipe,
  title={scPipe: A flexible R/Bioconductor preprocessing pipeline for single-cell RNA-sequencing data},
  author={Tian, Luyi and Su, Shian and Dong, Xueyi and Amann-Zalcenstein, Daniela and Biben, Christine and Seidi, Azadeh and Hilton, Douglas J and Naik, Shalin H and Ritchie, Matthew E},
  journal={PLoS computational biology},
  volume={14},
  number={8},
  pages={e1006361},
  year={2018},
  publisher={Public Library of Science},
  URL={https://www.ncbi.nlm.nih.gov/pubmed/30096152}
}

@article{kiselev2017sc3,
  title={SC3: consensus clustering of single-cell RNA-seq data},
  author={Kiselev, Vladimir Yu and Kirschner, Kristina and Schaub, Michael T and Andrews, Tallulah and Yiu, Andrew and Chandra, Tamir and Natarajan, Kedar N and Reik, Wolf and Barahona, Mauricio and Green, Anthony R and others},
  journal={Nature methods},
  volume={14},
  number={5},
  pages={483},
  year={2017},
  publisher={Nature Publishing Group},
  URL={https://www.ncbi.nlm.nih.gov/pubmed/28346451}
}

@ARTICLE{McInnes2018-hy,
  title         = "UMAP: Uniform Manifold Approximation and Projection for
                   Dimension Reduction",
  author        = "McInnes, Leland and Healy, John and Melville, James",
  abstract      = "UMAP (Uniform Manifold Approximation and Projection) is a
                   novel manifold learning technique for dimension reduction.
                   UMAP is constructed from a theoretical framework based in
                   Riemannian geometry and algebraic topology. The result is a
                   practical scalable algorithm that applies to real world
                   data. The UMAP algorithm is competitive with t-SNE for
                   visualization quality, and arguably preserves more of the
                   global structure with superior run time performance.
                   Furthermore, UMAP has no computational restrictions on
                   embedding dimension, making it viable as a general purpose
                   dimension reduction technique for machine learning.",
  month         =  feb,
  year          =  2018,
  url           = "http://arxiv.org/abs/1802.03426",
  archivePrefix = "arXiv",
  eprint        = "1802.03426",
  primaryClass  = "stat.ML",
  arxivid       = "1802.03426"
}

@article{lun2017overcoming,
  title={Overcoming confounding plate effects in differential expression analyses of single-cell RNA-seq data},
  author={Lun, Aaron TL and Marioni, John C},
  journal={Biostatistics},
  volume={18},
  number={3},
  pages={451--464},
  year={2017},
  publisher={Oxford University Press}
}

@Article{lun2017testing,
    Author="Lun, A. T. L.  and Richard, A. C.  and Marioni, J. C. ",
    Title="{{T}esting for differential abundance in mass cytometry data}",
    Journal="Nat. Methods",
    Year="2017",
    Volume="14",
    Number="7",
    Pages="707--709",
    Month="Jul"
}

@Article{chen2016reads,
    Author="Chen, Y.  and Lun, A. T.  and Smyth, G. K. ",
    Title="{{F}rom reads to genes to pathways: differential expression analysis of {R}{N}{A}-{S}eq experiments using {R}subread and the edge{R} quasi-likelihood pipeline}",
    Journal="F1000Res",
    Year="2016",
    Volume="5",
    Pages="1438"
}

@Article{phipson2016robust,
    Author="Phipson, B. and Lee, S. and Majewski, I. J. and Alexander, W. S. and Smyth, G. K.",
    year = 2016,
    title = "Robust hyperparameter estimation protects against hypervariable genes and improves power to detect differential expression",
    Journal="Ann. Appl. Stat.",
    Volume=10,
    Number=2,
    Pages="946-963"
}

@Article{aran2019reference,
   Author="Aran, D.  and Looney, A. P.  and Liu, L.  and Wu, E.  and Fong, V.  and Hsu, A.  and Chak, S.  and Naikawadi, R. P.  and Wolters, P. J.  and Abate, A. R.  and Butte, A. J.  and Bhattacharya, M. ",
   Title="Reference-based analysis of lung single-cell sequencing reveals a transitional profibrotic macrophage",
   Journal="Nat. Immunol.",
   Year="2019",
   Volume="20",
   Number="2",
   Pages="163--172",
   Month="02"
}

@article{Benayoun2019,
author = {Benayoun, B{\'{e}}r{\'{e}}nice A. and Pollina, Elizabeth A. and Singh, Param Priya and Mahmoudi, Salah and Harel, Itamar and Casey, Kerriann M. and Dulken, Ben W. and Kundaje, Anshul and Brunet, Anne},
doi = {10.1101/gr.240093.118},
journal = {Genome Research},
volume = {29},
pages = {697-709},
title = {Remodeling of epigenome and transcriptome landscapes with aging in mice reveals widespread induction of inflammatory responses},
year = {2019}
}

@article{ImmGenRef,
    author = {Heng, Tracy S.P. and Painter, Michio W. and Elpek, Kutlu and Lukacs-Kornek, Veronika and Mauermann, Nora and Turley, Shannon J. and Koller, Daphne and Kim, Francis S. and Wagers, Amy J. and Asinovski, Natasha and Davis, Scott and Fassett, Marlys and Feuerer, Markus and Gray, Daniel H.D. and Haxhinasto, Sokol and Hill, Jonathan A. and Hyatt, Gordon and Laplace, Catherine and Leatherbee, Kristen and Mathis, Diane and Benoist, Christophe and Jianu, Radu and Laidlaw, David H. and Best, J. Adam and Knell, Jamie and Goldrath, Ananda W. and Jarjoura, Jessica and Sun, Joseph C. and Zhu, Yanan and Lanier, Lewis L. and Ergun, Ayla and Li, Zheng and Collins, James J. and Shinton, Susan A. and Hardy, Richard R. and Friedline, Randall and Sylvia, Katelyn and Kang, Joonsoo},
    journal = {Nature Immunology},
    volume = {9},
    issue = {10},
    pages = {1091-4},
    doi = {10.1038/ni1008-1091},
    title = {The immunological genome project: Networks of gene expression in immune cells},
    year = {2008}
}

@article{choi2018haemopedia,
  title={Haemopedia RNA-seq: a database of gene expression during haematopoiesis in mice and humans},
  author={Choi, Jarny and Baldwin, Tracey M and Wong, Mae and Bolden, Jessica E and Fairfax, Kirsten A and Lucas, Erin C and Cole, Rebecca and Biben, Christine and Morgan, Clare and Ramsay, Kerry A and others},
  journal={Nucleic acids research},
  volume={47},
  number={D1},
  pages={D780--D785},
  year={2018},
  publisher={Oxford University Press}
}

@Article{mccarthy2009treat,
    Author="McCarthy, D. J.  and Smyth, G. K. ",
    Title="Testing significance relative to a fold-change threshold is a TREAT",
    Journal="Bioinformatics",
    Year="2009",
    Volume="25",
    Number="6",
    Pages="765--771",
    Month="Mar"
}

@article{hpcaRef,
author = {Mabbott, Neil A. and Baillie, J. K. and Brown, Helen and Freeman, Tom C. and Hume, David A.},
doi = {10.1186/1471-2164-14-632},
journal = {BMC Genomics},
volume = {14},
title = "An expression atlas of human primary cells: Inference of gene function from coexpression networks",
year = {2013}
}

@article{blueprintRef,
    author = {Martens, Joost H A and Stunnenberg, Hendrik G.},
    doi = {10.3324/haematol.2013.094243},
    journal = {Haematologica},
    title = "BLUEPRINT: Mapping human blood cell epigenomes",
    year = {2013},
    volume = {98},
    pages = {1487-1489}
}

@article{encodeRef,
author = {{The ENCODE Project Consortium}},
doi = {10.1038/nature11247},
issn = {14764687},
journal = {Nature},
title = "An integrated encyclopedia of DNA elements in the human genome",
year = {2012}
}

@article{lun2016step,
  title={A step-by-step workflow for low-level analysis of single-cell {RNA-seq} data},
  author={Lun, A. T. L. and McCarthy, D. J. and Marioni, J. C.},
  journal ={F1000Res.},
  volume={5},
  month={aug},
  year={2016}
}

@article{Pijuan-Sala2019-yz,
  title    = {A single-cell molecular map of mouse gastrulation and early organogenesis},
  author   = {Pijuan-Sala, Blanca and Griffiths, Jonathan A and Guibentif, Carolina and Hiscock, Tom W and Jawaid, Wajid and Calero-Nieto, Fernando J and Mulas, Carla and Ibarra-Soria Shankar and Simons, Benjamin D and Nichols, Jennifer and Marioni, John C and G{\"o}ttgens, Berthold},
  journal  = {Nature},
  volume   =  {566},
  number   =  {7745},
  month    =  feb,
  year     =  2019,
  url      = {http://dx.doi.org/10.1038/s41586-019-0933-9},
  pmid     = {30787436},
  doi      = {10.1038/s41586-019-0933-9},
  pmc      = {PMC6522369}
}

@article{diceRef,
  title = "Impact of Genetic Polymorphisms on Human Immune Cell Gene Expression",
  volume = 175,
  issn = {0092-8674},
  url = {http://www.sciencedirect.com/science/article/pii/S009286741831331X},
  doi = {10.1016/j.cell.2018.10.022},
  number = {6},
  journal = {Cell},
  author = {Schmiedel, Benjamin J. and Singh, Divya and Madrigal, Ariel and Valdovino-Gonzalez, Alan G. and White, Brandie M. and Zapardiel-Gonzalo, Jose and Ha, Brendan and Altay, Gokmen and Greenbaum, Jason A. and McVicker, Graham and Seumois, Grégory and Rao, Anjana and Kronenberg, Mitchell and Peters, Bjoern and Vijayanand, Pandurangan},
  month = "Nov",
  year = {2018},
  pages = {1701--1715.e16},
}

@article{nhRef,
  title = "Densely interconnected transcriptional circuits control cell states in human hematopoiesis",
  volume = {144},
  issn = {1097-4172},
  doi = {10.1016/j.cell.2011.01.004},
  number = {2},
  journal = {Cell},
  author = {Novershtern, Noa and Subramanian, Aravind and Lawton, Lee N. and Mak, Raymond H. and Haining, W. Nicholas and McConkey, Marie E. and Habib, Naomi and Yosef, Nir and Chang, Cindy Y. and Shay, Tal and Frampton, Garrett M. and Drake, Adam C. B. and Leskov, Ilya and Nilsson, Bjorn and Preffer, Fred and Dombkowski, David and Evans, John W. and Liefeld, Ted and Smutko, John S. and Chen, Jianzhu and Friedman, Nir and Young, Richard A. and Golub, Todd R. and Regev, Aviv and Ebert, Benjamin L.},
  month = "Jan",
  year = {2011},
  pmid = {21241896},
  pmcid = {PMC3049864},
  pages = {296--309}
}

@article{monaco_immuneRef,
  title = "RNA-Seq Signatures Normalized by mRNA Abundance Allow Absolute Deconvolution of Human Immune Cell Types",
  volume = {26},
  issn = {2211-1247},
  doi = {10.1016/j.celrep.2019.01.041},
  number = {6},
  journal = {Cell Reports},
  author = {Monaco, Gianni and Lee, Bernett and Xu, Weili and Mustafah, Seri and Hwang, You Yi and Carré, Christophe and Burdin, Nicolas and Visan, Lucian and Ceccarelli, Michele and Poidinger, Michael and Zippelius, Alfred and Pedro de Magalhães, João and Larbi, Anis},
  month = "Feb",
  year = {2019},
  pmid = {30726743},
  pmcid = {PMC6367568},
  pages = {1627--1640.e7}
}

@article{crowell2019discovery,
	author = {Crowell, H. L. and Soneson, C. and Germain, P.-L. and Calini, D. and Collin, L. and Raposo, C. and Malhotra, D. and Robinson, M. D.},
	title = {On the discovery of population-specific state transitions from multi-sample multi-condition single-cell RNA sequencing data},
	year = {2019},
	doi = {10.1101/713412},
	journal = {bioRxiv}
}

@article{finak2015mast,
  author = {Finak, G. and McDavid, A. and Yajima, M. and Deng, J. and Gersuk, V. and Shalek, A. K. and Slichter, C. K. and Miller, H. W. and McElrath, M. J. and Prlic, M.and Linsley, P. S. and Gottardo, R.},
  title = {MAST: a flexible statistical framework for assessing transcriptional changes and characterizing heterogeneity in single-cell RNA sequencing data},
  journal = {Genome Biol.},
  year = {2015},
  volume = {16}
}

@article{young10soupx,
  title={SoupX removes ambient RNA contamination from droplet based single cell RNA sequencing data. bioRxiv. 2018},
  author={Young, MD and Behjati, S},
  journal={DOI},
  volume={10},
  pages={303727}
}

@article{hill1973diversity,
  title={Diversity and evenness: a unifying notation and its consequences},
  author={Hill, Mark O},
  journal={Ecology},
  volume={54},
  number={2},
  pages={427--432},
  year={1973},
  publisher={Wiley Online Library}
}

@article{chao2014rarefaction,
  title={Rarefaction and extrapolation with Hill numbers: a framework for sampling and estimation in species diversity studies},
  author={Chao, Anne and Gotelli, Nicholas J and Hsieh, TC and Sander, Elizabeth L and Ma, KH and Colwell, Robert K and Ellison, Aaron M},
  journal={Ecological monographs},
  volume={84},
  number={1},
  pages={45--67},
  year={2014},
  publisher={Wiley Online Library}
}

@article{chao2015unveiling,
  title={Unveiling the species-rank abundance distribution by generalizing the Good-Turing sample coverage theory},
  author={Chao, Anne and Hsieh, TC and Chazdon, Robin L and Colwell, Robert K and Gotelli, Nicholas J},
  journal={Ecology},
  volume={96},
  number={5},
  pages={1189--1201},
  year={2015},
  publisher={Wiley Online Library}
}

@Article{richard2018tcell,
   Author="Richard, A. C.  and Lun, A. T. L.  and Lau, W. W. Y.  and Gottgens, B.  and Marioni, J. C.  and Griffiths, G. M. ",
   Title="{{T} cell cytolytic capacity is independent of initial stimulation strength}",
   Journal="Nat. Immunol.",
   Year="2018",
   Volume="19",
   Number="8",
   Pages="849--858",
   Month="08"
}
</script>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
