<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

  <!--radix_placeholder_meta_tags-->
  <title>C057: Annotating the Cooney (C057) memory CD4+ T-cell data set after ignoring 'unknown' samples</title>
  
  
  
  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2019-12-11"/>
  <meta property="article:created" itemprop="dateCreated" content="2019-12-11"/>
  <meta name="article:author" content="Peter Hickey"/>
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="C057: Annotating the Cooney (C057) memory CD4+ T-cell data set after ignoring &#39;unknown&#39; samples"/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="C057"/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="C057: Annotating the Cooney (C057) memory CD4+ T-cell data set after ignoring &#39;unknown&#39; samples"/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Computational assignment of cell-cycle stage from single-cell transcriptome data;citation_publication_date=2015;citation_volume=85;citation_author=A. Scialdone;citation_author=K. N. Natarajan;citation_author=L. R. Saraiva;citation_author=V. Proserpio;citation_author=S. A. Teichmann;citation_author=O. Stegle;citation_author=J. C. Marioni;citation_author=F. Buettner"/>
  <meta name="citation_reference" content="citation_title=Bioequivalence trials, intersection-union tests and equivalence confidence sets;citation_publication_date=1996;citation_publisher=The Institute of Mathematical Statistics;citation_volume=11;citation_doi=10.1214/ss/1032280304;citation_author=R. L. Berger;citation_author=J. C. Hsu"/>
  <meta name="citation_reference" content="citation_title=An improved bonferroni procedure for multiple tests of significance;citation_publication_date=1986;citation_volume=73;citation_author=R. J. Simes"/>
  <meta name="citation_reference" content="citation_title=Identification of cell types from single-cell transcriptomes using a novel clustering method;citation_publication_date=2015;citation_volume=31;citation_author=C. Xu;citation_author=Z. Su"/>
  <meta name="citation_reference" content="citation_title=&lt;span class=&quot;nocase&quot;&gt;Massively parallel digital transcriptional profiling of single cells&lt;/span&gt;;citation_publication_date=2017;citation_volume=8;citation_author=G. X. Zheng;citation_author=J. M. Terry;citation_author=P. Belgrader;citation_author=P. Ryvkin;citation_author=Z. W. Bent;citation_author=R. Wilson;citation_author=S. B. Ziraldo;citation_author=T. D. Wheeler;citation_author=G. P. McDermott;citation_author=J. Zhu;citation_author=M. T. Gregory;citation_author=J. Shuga;citation_author=L. Montesclaros;citation_author=J. G. Underwood;citation_author=D. A. Masquelier;citation_author=S. Y. Nishimura;citation_author=M. Schnall-Levin;citation_author=P. W. Wyatt;citation_author=C. M. Hindson;citation_author=R. Bharadwaj;citation_author=A. Wong;citation_author=K. D. Ness;citation_author=L. W. Beppu;citation_author=H. J. Deeg;citation_author=C. McFarland;citation_author=K. R. Loeb;citation_author=W. J. Valente;citation_author=N. G. Ericson;citation_author=E. A. Stevens;citation_author=J. P. Radich;citation_author=T. S. Mikkelsen;citation_author=B. J. Hindson;citation_author=J. H. Bielas"/>
  <meta name="citation_reference" content="citation_title=Bias, robustness and scalability in single-cell differential expression analysis;citation_publication_date=2018;citation_volume=15;citation_author=C. Soneson;citation_author=M. D. Robinson"/>
  <meta name="citation_reference" content="citation_title=Cell &quot;hashing&quot; with barcoded antibodies enables multiplexing and doublet detection for single cell genomics;citation_publication_date=2017;citation_publisher=Cold Spring Harbor Laboratory;citation_doi=10.1101/237693;citation_author=M. Stoeckius;citation_author=S. Zheng;citation_author=B. Houck-Loomis;citation_author=S. Hao;citation_author=B. Yeung;citation_author=P. Smibert;citation_author=R. Satija"/>
  <meta name="citation_reference" content="citation_title=&lt;span class=&quot;nocase&quot;&gt;A single-cell hematopoietic landscape resolves 8 lineage trajectories and defects in Kit mutant mice&lt;/span&gt;;citation_publication_date=2018;citation_volume=131;citation_author=J. S. Dahlin;citation_author=F. K. Hamey;citation_author=B. Pijuan-Sala;citation_author=M. Shepherd;citation_author=W. W. Y. Lau;citation_author=S. Nestorowa;citation_author=C. Weinreb;citation_author=S. Wolock;citation_author=R. Hannah;citation_author=E. Diamanti;citation_author=D. G. Kent;citation_author=B. Gottgens;citation_author=N. K. Wilson"/>
  <meta name="citation_reference" content="citation_title=&lt;span class=&quot;nocase&quot;&gt;Differentiation dynamics of mammary epithelial cells revealed by single-cell RNA sequencing&lt;/span&gt;;citation_publication_date=2017;citation_volume=8;citation_author=K. Bach;citation_author=S. Pensa;citation_author=M. Grzelak;citation_author=J. Hadfield;citation_author=D. J. Adams;citation_author=J. C. Marioni;citation_author=W. T. Khaled"/>
  <meta name="citation_reference" content="citation_title=Reference-based analysis of lung single-cell sequencing reveals a transitional profibrotic macrophage;citation_publication_date=2019;citation_volume=20;citation_author=D. Aran;citation_author=A. P. Looney;citation_author=L. Liu;citation_author=E. Wu;citation_author=V. Fong;citation_author=A. Hsu;citation_author=S. Chak;citation_author=R. P. Naikawadi;citation_author=P. J. Wolters;citation_author=A. R. Abate;citation_author=A. J. Butte;citation_author=M. Bhattacharya"/>
  <meta name="citation_reference" content="citation_title=&lt;span class=&quot;nocase&quot;&gt;Testing significance relative to a fold-change threshold is a TREAT&lt;/span&gt;;citation_publication_date=2009;citation_volume=25;citation_author=D. J. McCarthy;citation_author=G. K. Smyth"/>
  <meta name="citation_reference" content="citation_title=&lt;span class=&quot;nocase&quot;&gt;An expression atlas of human primary cells: Inference of gene function from coexpression networks&lt;/span&gt;;citation_publication_date=2013;citation_volume=14;citation_doi=10.1186/1471-2164-14-632;citation_author=Neil A. Mabbott;citation_author=J. K. Baillie;citation_author=Helen Brown;citation_author=Tom C. Freeman;citation_author=David A. Hume"/>
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","date","output","editor_options","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["Annotating the Cooney (C057) memory CD4+ T-cell data set after ignoring 'unknown' samples"]},{"type":"NULL"},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url","affiliation","affiliation_url"]}},"value":[{"type":"character","attributes":{},"value":["Peter Hickey"]},{"type":"character","attributes":{},"value":["https://peterhickey.org"]},{"type":"character","attributes":{},"value":["Single Cell Open Research Endeavour (SCORE), WEHI"]},{"type":"character","attributes":{},"value":["https://www.wehi.edu.au/people/shalin-naik/3310/score"]}]}]},{"type":"character","attributes":{},"value":["2019-12-11"]},{"type":"character","attributes":{},"value":["radix::radix_article"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["chunk_output_type"]}},"value":[{"type":"character","attributes":{},"value":["console"]}]},{"type":"character","attributes":{},"value":["ref.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #455a64;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative;}
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  </style>
  
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for table of contents */
  
  .d-toc {
    color: rgba(0,0,0,0.8);
    font-size: 0.8em;
    line-height: 1em;
  }
  
  .d-toc-header {
    font-size: 0.6rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.5);
    text-transform: uppercase;
    margin-top: 0;
    margin-bottom: 1.3em;
  }
  
  .d-toc a {
    border-bottom: none;
  }
  
  .d-toc ul {
    padding-left: 0;
  }
  
  .d-toc li>ul {
    padding-top: 0.8em;
    padding-left: 16px;
    margin-bottom: 0.6em;
  }
  
  .d-toc ul,
  .d-toc li {
    list-style-type: none;
  }
  
  .d-toc li {
    margin-bottom: 0.9em;
  }
  
  .d-toc-separator {
    margin-top: 20px;
    margin-bottom: 2em;
  }
  
  .d-article-with-toc {
    border-top: none;
    padding-top: 0;
  }
  
  
  
  /* Tweak code blocks (note that this CSS is repeated above in an injection
     into the d-code shadow dom) */
  
  d-code {
    overflow-x: auto !important;
  }
  
  pre.d-code code.d-code {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  pre.text-output {
  
    font-size: 12px;
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  @media(min-width: 768px) {
  
  d-code {
    overflow-x: visible !important;
  }
  
  pre.d-code code.d-code  {
      padding-left: 18px;
      font-size: 14px;
  }
  pre.text-output {
    font-size: 14px;
  }
  }
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // create d-bibliography
    var bibliography = $('<d-bibliography></d-bibliography>');
    $('#distill-bibliography').wrap(bibliography);
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace citations with <d-cite>
    $('.citation').each(function(i, val) {
      appendix = true;
      var cites = $(this).attr('data-cites').split(" ");
      var dt_cite = $('<d-cite></d-cite>');
      dt_cite.attr('key', cites.join());
      $(this).replaceWith(dt_cite);
    });
    // remove refs
    $('#refs').remove();
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-toc a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // replace code blocks with d-code
    $('pre>code').each(function(i, val) {
      var code = $(this);
      var pre = code.parent();
      var clz = "";
      var language = pre.attr('class');
      if (language) {
        // map unknown languages to "clike" (without this they just dissapear)
        if ($.inArray(language, ["bash", "clike", "css", "go", "html",
                                 "javascript", "js", "julia", "lua", "markdown",
                                 "markup", "mathml", "python", "svg", "xml"]) == -1)
          language = "clike";
        language = ' language="' + language + '"';
        var dt_code = $('<d-code block' + language + clz + '></d-code>');
        dt_code.text(code.text());
        pre.replaceWith(dt_code);
      } else {
        code.addClass('text-output').unwrap().changeElementType('pre');
      }
    });
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('d-code, pre.text-output, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // table of contents
      if (have_authors) // adjust border if we are in authors
        $('.d-toc').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // inject pre code styles (can't do this with a global stylesheet b/c a shadow root is used)
      $('d-code').each(function(i, val) {
        var style = document.createElement('style');
        style.innerHTML = 'pre code { padding-left: 10px; font-size: 12px; border-left: 2px solid rgba(0,0,0,0.1); } ' +
                          '@media(min-width: 768px) { pre code { padding-left: 18px; font-size: 14px; } }';
        if (this.shadowRoot)
          this.shadowRoot.appendChild(style);
      });
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-toc-header').remove();
    $('.d-toc').remove();
    $('.d-toc-separator').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
  <script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Annotating the Cooney (C057) memory CD4+ T-cell data set after ignoring 'unknown' samples","authors":[{"author":"Peter Hickey","authorURL":"https://peterhickey.org","affiliation":"Single Cell Open Research Endeavour (SCORE), WEHI","affiliationURL":"https://www.wehi.edu.au/people/shalin-naik/3310/score"}],"publishedDate":"2019-12-11T00:00:00.000+11:00","citationText":"Hickey, 2019"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">C057</a>
<a href="index.html">Introduction</a>
<a href="C057_Cooney.demultiplex.html">Demultiplexing</a>
<a href="C057_Cooney.preprocess.html">Preprocessing</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Dimensionality reduction
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="C057_Cooney.merge.html">All samples</a>
<a href="C057_Cooney_ignoring_unknowns.merge.html">Ignoring 'unknown' samples</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Annotating cells
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="C057_Cooney.annotate.html">All samples</a>
<a href="C057_Cooney_ignoring_unknowns.annotate.html">Ignoring 'unknown' samples</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Differential analyses
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="C057_Cooney.multi-sample_comparisons.html">All samples</a>
<a href="C057_Cooney_ignorning_unknowns.multi-sample_comparisons.html">Ignoring 'unknown' samples</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Supplementary files
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="../output/FastQC/">FastQC reports</a>
<a href="../output/scPipe/">scPipe reports</a>
<a href="../output/marker_genes/">Marker gene tables</a>
<a href="../output/DEGs/">Differentially expressed genes</a>
</div>
</div>
</div>
<div class="nav-right">
<a href="https://github.com/WEHISCORE/C057_Cooney">
<i class="fa fa-github"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Annotating the Cooney (C057) memory CD4+ T-cell data set after ignoring ‘unknown’ samples</h1>

</div>

<div class="d-byline">
  Peter Hickey <a href="https://peterhickey.org" class="uri">https://peterhickey.org</a> (Single Cell Open Research Endeavour (SCORE), WEHI)<a href="https://www.wehi.edu.au/people/shalin-naik/3310/score" class="uri">https://www.wehi.edu.au/people/shalin-naik/3310/score</a>
  
<br/>2019-12-11
</div>

<div class="d-article">
<h3 class="d-toc-header">Table of Contents</h3>
<nav class="d-toc" id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#preparing-the-data">Preparing the data</a></li>
<li><a href="#clustering-cells">Clustering cells</a></li>
<li><a href="#doublet-detection">Doublet detection</a></li>
<li><a href="#marker-gene-detection">Marker gene detection</a></li>
<li><a href="#using-marker-genes-to-perform-cell-type-annotation">Using marker genes to perform cell type annotation</a></li>
<li><a href="#automated-cell-type-labelling-with-singler">Automated cell type labelling with <em><span>SingleR</span></em></a></li>
<li><a href="#cell-cycle-assignment">Cell cycle assignment</a></li>
<li><a href="#concluding-remarks">Concluding remarks</a></li>
<li><a href="#additional-information">Additional information</a></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</nav>
<hr class="d-toc-separator"/>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="introduction">Introduction</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>scRNA-seq has enabled an unprecedented level of granularity in characterizing gene expression changes in disease models. The classification of clusters by cell type relies heavily on known marker genes and the annotation of clusters is typically performed manually, although (semi)automated methods are becoming available.</p>
<p>Here, we perform clustering and marker gene detection, which can be used for manual annotation of clusters, along with an automated annotation method. In practice, some combination of manual and automated classification methods are usually necessary to properly annotate an scRNA-seq data set.</p>
<h1 id="preparing-the-data">Preparing the data</h1>
<p>We start from the preprocessed <em>SingleCellExperiment</em> object created in <a href="C057_Cooney_ignoring_unknowns.merge.html">‘Merging the Cooney (C057) memory CD4+ T-cell data set after ignoring the ’unknown’ samples’</a>.</p>
<h1 id="clustering-cells">Clustering cells</h1>
<p>Clustering is an unsupervised learning procedure that is used in scRNA-seq data analysis to empirically define groups of cells with similar expression profiles. Its primary purpose is to summarize the data in a digestible format for human interpretation. This allows us to describe population heterogeneity in terms of discrete labels that are easily understood, rather than attempting to comprehend the high-dimensional manifold on which the cells truly reside. After annotation based on marker genes, the clusters can be treated as proxies for more abstract biological concepts such as cell types or states. Clustering is thus a critical step for extracting biological insights from scRNA-seq data.</p>
<h2 id="clusters-vs.cell-types">Clusters vs. cell types</h2>
<p>It is worth stressing the distinction between clusters and cell types. The former is an empirical construct while the latter is a biological truth (albeit a vaguely defined one). For this reason, questions like “what is the true number of clusters?” are usually meaningless. We can define as many clusters as we like, with whatever algorithm we like - each clustering will represent its own partitioning of the high-dimensional expression space, and is as “real” as any other clustering.</p>
<p>A more relevant question is “how well do the clusters approximate the cell types?” Unfortunately, this is difficult to answer given the context-dependent interpretation of biological truth. Some analysts will be satisfied with resolution of the major cell types; other analysts may want resolution of subtypes; and others still may require resolution of different states (e.g., metabolic activity, stress) within those subtypes. Two clusterings can also be highly inconsistent yet both valid, simply partitioning the cells based on different aspects of biology. Indeed, asking for an unqualified “best” clustering is akin to asking for the best magnification on a microscope without any context.</p>
<p>It is helpful to realize that clustering, like a microscope, is simply a tool to explore the data. We can zoom in and out by changing the resolution of the clustering parameters, and we can experiment with different clustering algorithms to obtain alternative perspectives of the data. This iterative approach is entirely permissible for data exploration, which constitutes the majority of all scRNA-seq data analysis.</p>
<h2 id="graph-based-clustering">Graph-based clustering</h2>
<p>We build a shared nearest neighbour graph <span class="citation" data-cites="xu2015identification">(Xu and Su <a href="#ref-xu2015identification">2015</a>)</span> and use the Louvain algorithm to identify clusters. We build the graph using the principal components.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>There are 10 clusters detected, summarised in Figure <a href="#fig:cluster-barplot">1</a>.</p>
<div class="layout-chunk" data-layout="l-page">
<div class="figure"><span id="fig:cluster-barplot"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 1: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<h3 id="assessing-cluster-separation">Assessing cluster separation</h3>
<p>We look at the ratio of the observed and expected edge weights to confirm that the clusters are modular. Figure <a href="#fig:clustermod">2</a> indicates some cluster overlap, with the large-ish non-zero off-diagonal entries.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:clustermod"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/clustermod-1.png" alt="Heatmap of the log~10~-ratio of the total weight between nodes in the same cluster or in different clusters, relative to the total weight expected under a null model of random links." width="624" />
<p class="caption">
Figure 2: Heatmap of the log<sub>10</sub>-ratio of the total weight between nodes in the same cluster or in different clusters, relative to the total weight expected under a null model of random links.
</p>
</div>
</div>
<h2 id="visual-inspection-of-clustering-results">Visual inspection of clustering results</h2>
<p>Figure <a href="#fig:clusterplot-tsne">3</a> shows the cluster labels on a t-SNE plot to confirm that a sensible partitioning was generated.</p>
<div class="layout-chunk" data-layout="l-page">
<div class="figure"><span id="fig:clusterplot-tsne"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/clusterplot-tsne-1.png" alt="t-SNE plot of the t-SNE data, where each point represents a cell and is coloured according to the legend." width="624" />
<p class="caption">
Figure 3: t-SNE plot of the t-SNE data, where each point represents a cell and is coloured according to the legend.
</p>
</div>
</div>
<h1 id="doublet-detection">Doublet detection</h1>
<h2 id="overview">Overview</h2>
<p>In single-cell RNA sequencing (scRNA-seq) experiments, doublets are artifactual libraries generated from two cells. They typically arise due to errors in cell sorting or capture, especially in droplet-based protocols <span class="citation" data-cites="zheng2017massively">(Zheng et al. <a href="#ref-zheng2017massively">2017</a>)</span> involving thousands of cells. Doublets are obviously undesirable when the aim is to characterize populations at the single-cell level. In particular, doublets can be mistaken for intermediate populations or transitory states that do not actually exist. Thus, it is desirable to identify and remove doublet libraries so that they do not compromise interpretation of the results.</p>
<p>Several experimental strategies are available for doublet removal. One approach, used here, is to mark a subset of cells (e.g., all cells from one sample) with an antibody conjugated to a different oligonucleotide <span class="citation" data-cites="zheng2017massively">(Zheng et al. <a href="#ref-zheng2017massively">2017</a>, <span class="citation" data-cites="stoeckius2017hashing">@stoeckius2017hashing</span>)</span>. Upon pooling, libraries that are observed to have different oligonucleotides are considered to be doublets and removed. This approaches can be highly effective but made more difficult for this dataset by one of the samples being labelled with all five HTOs (see <a href="C057_Cooney_demultiplex.html">‘Demultiplexing the Cooney (C057) memory CD4+ T-cell data set’</a>).</p>
<p>A more general approach is to infer doublets from the expression profiles alone <span class="citation" data-cites="dahlin2018single">(Dahlin et al. <a href="#ref-dahlin2018single">2018</a>)</span>. Here, we will apply two purely computational approaches for detecting doublets from scRNA-seq data. The main difference between these two methods is whether or not they need cluster information beforehand.</p>
<h2 id="doublet-detection-with-clusters">Doublet detection with clusters</h2>
<p>The <code>doubletCluster()</code> function identifes from <em><a href="https://bioconductor.org/packages/3.9/scran">scran</a></em> clusters with expression profiles lying between two other clusters <span class="citation" data-cites="bach2017differentiation">(Bach et al. <a href="#ref-bach2017differentiation">2017</a>)</span>. We consider every possible triplet of clusters consisting of a query cluster and two putative “source” clusters. Under the null hypothesis that the query consists of doublets from the two sources, we then compute a number of statistics:</p>
<ul>
<li><code>N</code>: The number of genes that are differentially expressed in the same direction in the query cluster compared to both of the source clusters. Such genes would be unique markers for the query cluster and provide evidence against the null hypothesis. Clusters with few unique genes are more likely to be composed of doublets.</li>
<li><code>lib.size</code> fields: The ratio of the median library size in each source to the median library size in the query. Doublet libraries are generated from a larger initial pool of RNA compared to libraries for single cells, and thus the former should have larger library sizes. Library size ratios much greater than unity are inconsistent with a doublet identity for the query.</li>
<li><code>prop</code>: The proportion of cells in the query cluster should also be reasonable - typically less than 5% of all cells, depending on how many cells were loaded onto the 10X Genomics device.</li>
</ul>
<p>For each query cluster, the best pair of sources is identified based solely on the lowest <code>N</code><a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>. Clusters are then ranked by <code>N</code> in the output of <code>doubletClusters()</code>, allowing us to prioritize high-risk clusters that require careful investigation.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-7">Table 1: </span>Results of running <code>doubletClusters()</code>.</caption>
<thead>
<tr class="header">
<th></th>
<th style="text-align: left;">source1</th>
<th style="text-align: left;">source2</th>
<th style="text-align: right;">N</th>
<th style="text-align: left;">best</th>
<th style="text-align: right;">p.value</th>
<th style="text-align: right;">lib.size1</th>
<th style="text-align: right;">lib.size2</th>
<th style="text-align: right;">prop</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>5</td>
<td style="text-align: left;">6</td>
<td style="text-align: left;">3</td>
<td style="text-align: right;">91</td>
<td style="text-align: left;">ALOX5AP</td>
<td style="text-align: right;">1.848914e-16</td>
<td style="text-align: right;">0.8486670</td>
<td style="text-align: right;">0.8164586</td>
<td style="text-align: right;">0.078457178</td>
</tr>
<tr class="even">
<td>6</td>
<td style="text-align: left;">9</td>
<td style="text-align: left;">5</td>
<td style="text-align: right;">151</td>
<td style="text-align: left;">GNLY</td>
<td style="text-align: right;">1.093345e-37</td>
<td style="text-align: right;">0.8200472</td>
<td style="text-align: right;">1.1783185</td>
<td style="text-align: right;">0.198521968</td>
</tr>
<tr class="odd">
<td>2</td>
<td style="text-align: left;">8</td>
<td style="text-align: left;">3</td>
<td style="text-align: right;">152</td>
<td style="text-align: left;">GZMA</td>
<td style="text-align: right;">5.013146e-17</td>
<td style="text-align: right;">1.2971842</td>
<td style="text-align: right;">0.7515427</td>
<td style="text-align: right;">0.187487346</td>
</tr>
<tr class="even">
<td>1</td>
<td style="text-align: left;">9</td>
<td style="text-align: left;">8</td>
<td style="text-align: right;">182</td>
<td style="text-align: left;">AL138963.3</td>
<td style="text-align: right;">9.798285e-35</td>
<td style="text-align: right;">2.2958821</td>
<td style="text-align: right;">4.6489578</td>
<td style="text-align: right;">0.037963150</td>
</tr>
<tr class="odd">
<td>7</td>
<td style="text-align: left;">9</td>
<td style="text-align: left;">8</td>
<td style="text-align: right;">215</td>
<td style="text-align: left;">C12orf75</td>
<td style="text-align: right;">5.029059e-31</td>
<td style="text-align: right;">0.8306815</td>
<td style="text-align: right;">1.6820565</td>
<td style="text-align: right;">0.128973476</td>
</tr>
<tr class="even">
<td>3</td>
<td style="text-align: left;">9</td>
<td style="text-align: left;">8</td>
<td style="text-align: right;">235</td>
<td style="text-align: left;">CRIP2</td>
<td style="text-align: right;">1.170472e-44</td>
<td style="text-align: right;">0.8523971</td>
<td style="text-align: right;">1.7260287</td>
<td style="text-align: right;">0.102348654</td>
</tr>
<tr class="odd">
<td>4</td>
<td style="text-align: left;">9</td>
<td style="text-align: left;">8</td>
<td style="text-align: right;">336</td>
<td style="text-align: left;">FOXP3</td>
<td style="text-align: right;">1.084348e-48</td>
<td style="text-align: right;">0.8325959</td>
<td style="text-align: right;">1.6859329</td>
<td style="text-align: right;">0.091718971</td>
</tr>
<tr class="even">
<td>8</td>
<td style="text-align: left;">10</td>
<td style="text-align: left;">1</td>
<td style="text-align: right;">592</td>
<td style="text-align: left;">GZMA</td>
<td style="text-align: right;">2.920343e-28</td>
<td style="text-align: right;">1.4640494</td>
<td style="text-align: right;">0.2151020</td>
<td style="text-align: right;">0.006074104</td>
</tr>
<tr class="odd">
<td>9</td>
<td style="text-align: left;">5</td>
<td style="text-align: left;">1</td>
<td style="text-align: right;">961</td>
<td style="text-align: left;">PTPRC</td>
<td style="text-align: right;">4.201859e-47</td>
<td style="text-align: right;">1.4368911</td>
<td style="text-align: right;">0.4355624</td>
<td style="text-align: right;">0.017007491</td>
</tr>
<tr class="even">
<td>10</td>
<td style="text-align: left;">9</td>
<td style="text-align: left;">8</td>
<td style="text-align: right;">1349</td>
<td style="text-align: left;">ZWINT</td>
<td style="text-align: right;">2.832052e-289</td>
<td style="text-align: right;">0.3373170</td>
<td style="text-align: right;">0.6830370</td>
<td style="text-align: right;">0.151447661</td>
</tr>
</tbody>
</table>
</div>
<h3 id="summary">Summary</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Based on the results of running <code>doubletClusters()</code>, none of the clusters stand out as being an obvious doublet cluster.</p>
<h2 id="doublet-detection-by-simulation">Doublet detection by simulation</h2>
<p>The other doublet detection strategy involves in silico simulation of doublets from the single-cell expression profiles <span class="citation" data-cites="dahlin2018single">(Dahlin et al. <a href="#ref-dahlin2018single">2018</a>)</span>. This is performed using the <code>doubletCells()</code> function from the <em><a href="https://bioconductor.org/packages/3.9/scran">scran</a></em>, which will:</p>
<ol type="1">
<li>Simulate thousands of doublets by adding together two randomly chosen single-cell profiles.</li>
<li>For each original cell, compute the density of simulated doublets in the surrounding neighbourhood.</li>
<li>For each original cell, compute the density of other observed cells in the neighbourhood.</li>
<li>Return the ratio between the two densities as a “doublet score” for each cell.</li>
</ol>
<p>This approach assumes that the simulated doublets are good approximations for real doublets. The use of random selection accounts for the relative abundances of different subpopulations, which affect the likelihood of their involvement in doublets; and the calculation of a ratio avoids high scores for non-doublet cells in highly abundant subpopulations.</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.00000 0.04367 0.18083 0.35031 0.44264 9.39015 </code></pre>
</div>
<p>Figures <a href="#fig:tsne-doublet-score">4</a> and <a href="#fig:doublet-score">5</a> reveal no concentration of the highest doublets scores in a single cluster(s) of cells.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:tsne-doublet-score"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/tsne-doublet-score-1.png" alt="t-SNE plot of the data set. Each point is a cell coloured according to its doublet density." width="624" />
<p class="caption">
Figure 4: t-SNE plot of the data set. Each point is a cell coloured according to its doublet density.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:doublet-score"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/doublet-score-1.png" alt="Distribution of doublet scores for each cluster in the data set. Each point is a cell." width="624" />
<p class="caption">
Figure 5: Distribution of doublet scores for each cluster in the data set. Each point is a cell.
</p>
</div>
</div>
<h3 id="summary-1">Summary</h3>
<p>The advantage of <code>doubletCells()</code> is that it does not depend on clusters, reducing the sensitivity of the results to clustering quality. The downside is that it requires some strong assumptions about how doublets form, such as the combining proportions and the sampling from pure subpopulations. In particular, <code>doubletCells()</code> treats the library size of each cell as an accurate proxy for its total RNA content. If this is not true, the simulation will not combine expression profiles from different cells in the correct proportions. This means that the simulated doublets will be systematically shifted away from the real doublets, resulting in doublet scores that are too low.</p>
<p>Simply removing cells with high doublet scores will not be sufficient to eliminate real doublets from the data set. In some cases, only a subset of the cells in the putative doublet cluster actually have high scores, and removing these would still leave enough cells in that cluster to mislead downstream analyses. In fact, even defining a threshold on the doublet score is difficult as the interpretation of the score is relative. There is no general definition for a fixed threshold above which libraries are to be considered doublets.</p>
<p>We recommend interpreting the <code>doubletCells()</code> scores in the context of cluster annotation. All cells from a cluster with a large average doublet score should be considered suspect, and close neighbours of problematic clusters should also be treated with caution. In contrast, a cluster containing a small proportion of high-scoring cells is probably safe provided that any interesting results are not being driven by those cells (e.g., checking that DE in an interesting gene is not driven solely by cells with high doublet scores). While clustering is still required, this approach is more robust than <code>doubletClusters()</code> to the quality of the clustering as the scores are computed on a per-cell basis.</p>
<h2 id="summary-2">Summary</h2>
<p>Based on the results of running <code>doubletClusters()</code> and <code>doubletCells()</code>, none of the clusters stand out as being an obvious doublet cluster. This does not mean that there aren’t doublets; in fact, in <a href="C057_Cooney_demultiplex.html">‘Demultiplexing the Cooney (C057) memory CD4+ T-cell data set’</a> we see evidence of samples with 2 HTOs outside of the ‘unknown’ cells HTO cluster.</p>
<h1 id="marker-gene-detection">Marker gene detection</h1>
<p>To interpret our clustering results, we identify the genes that drive separation between clusters. These marker genes allow us to assign biological meaning to each cluster based on their functional annotation. In the most obvious case, the marker genes for each cluster are <em>a priori</em> associated with particular cell types, allowing us to treat the clustering as a proxy for cell type identity. The same principle can be applied to more subtle differences in activation status or differentiation state.</p>
<p>Identification of marker genes is usually based around the retrospective detection of differential expression between clusters<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>. Genes that are more strongly DE are more likely to have driven cluster separation in the first place. The top DE genes are likely to be good candidate markers as they can effectively distinguish between cells in different clusters.</p>
<h2 id="using-pairwise-t-tests">Using pairwise <span class="math inline">\(t\)</span>-tests</h2>
<h3 id="standard-application">Standard application</h3>
<p>The Welch <span class="math inline">\(t\)</span>-test is an obvious choice of statistical method to test for differences in expression between clusters. It is quickly computed and has good statistical properties for large numbers of cells <span class="citation" data-cites="soneson2018bias">(Soneson and Robinson <a href="#ref-soneson2018bias">2018</a>)</span>. We perform intra-samples comparisons by blocking on the sample of origin to avoid confounding effects from library-to-library variability or differential expression between samples<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>.</p>
<p>For each cluster, the DE results of the relevant comparisons are consolidated into a single output table. This allows a set of marker genes to be easily defined by taking the top DE genes from each pairwise comparison between clusters. Other statistics are also reported for each gene, including the adjusted p-values<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a> and the log-fold changes relative to every other cluster.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>To demonstrate, we use cluster 3 as our cluster of interest for this section. The relevant table contains log<sub>2</sub>-fold changes of expression in cluster 3 over each other cluster, along with several statistics obtained by combining <span class="math inline">\(p\)</span>-values <span class="citation" data-cites="simes1986improved">(Simes <a href="#ref-simes1986improved">1986</a>)</span> across the pairwise comparisons involving 3.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-12">Table 2: </span>First 10 marker genes for cluster 3 across comparisons.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">Gene</th>
<th style="text-align: right;">Top</th>
<th style="text-align: right;">p.value</th>
<th style="text-align: right;">FDR</th>
<th style="text-align: right;">logFC.1</th>
<th style="text-align: right;">logFC.2</th>
<th style="text-align: right;">logFC.4</th>
<th style="text-align: right;">logFC.5</th>
<th style="text-align: right;">logFC.6</th>
<th style="text-align: right;">logFC.7</th>
<th style="text-align: right;">logFC.8</th>
<th style="text-align: right;">logFC.9</th>
<th style="text-align: right;">logFC.10</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">STMN1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">-0.5407872</td>
<td style="text-align: right;">0.1210641</td>
<td style="text-align: right;">0.0009979</td>
<td style="text-align: right;">-0.0670370</td>
<td style="text-align: right;">0.1492848</td>
<td style="text-align: right;">0.0989509</td>
<td style="text-align: right;">0.1725786</td>
<td style="text-align: right;">0.0583456</td>
<td style="text-align: right;">-3.0792004</td>
</tr>
<tr class="even">
<td style="text-align: left;">S100A10</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1.0236310</td>
<td style="text-align: right;">1.7908263</td>
<td style="text-align: right;">1.1890605</td>
<td style="text-align: right;">0.2607279</td>
<td style="text-align: right;">1.1199929</td>
<td style="text-align: right;">0.9990700</td>
<td style="text-align: right;">2.7025669</td>
<td style="text-align: right;">2.3720942</td>
<td style="text-align: right;">1.1881696</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ANXA1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1.0530266</td>
<td style="text-align: right;">1.2680893</td>
<td style="text-align: right;">2.6783170</td>
<td style="text-align: right;">0.0931940</td>
<td style="text-align: right;">0.7790602</td>
<td style="text-align: right;">0.6194267</td>
<td style="text-align: right;">2.3889415</td>
<td style="text-align: right;">1.4559686</td>
<td style="text-align: right;">1.1437464</td>
</tr>
<tr class="even">
<td style="text-align: left;">GZMA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">-1.1084046</td>
<td style="text-align: right;">-0.6662801</td>
<td style="text-align: right;">0.1124929</td>
<td style="text-align: right;">-1.1872692</td>
<td style="text-align: right;">-2.5496281</td>
<td style="text-align: right;">-2.0672566</td>
<td style="text-align: right;">0.4375723</td>
<td style="text-align: right;">-0.9655642</td>
<td style="text-align: right;">-1.0139455</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CRIP2</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1.2976420</td>
<td style="text-align: right;">1.3374026</td>
<td style="text-align: right;">1.0349228</td>
<td style="text-align: right;">0.3884179</td>
<td style="text-align: right;">1.2625536</td>
<td style="text-align: right;">0.9808907</td>
<td style="text-align: right;">1.7996009</td>
<td style="text-align: right;">1.4990843</td>
<td style="text-align: right;">1.2902847</td>
</tr>
<tr class="even">
<td style="text-align: left;">CCL5</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">-0.9542304</td>
<td style="text-align: right;">-0.0491921</td>
<td style="text-align: right;">-0.2393143</td>
<td style="text-align: right;">-0.6350060</td>
<td style="text-align: right;">-2.5983714</td>
<td style="text-align: right;">-0.8404632</td>
<td style="text-align: right;">-0.0867154</td>
<td style="text-align: right;">-0.6550402</td>
<td style="text-align: right;">-0.4873579</td>
</tr>
<tr class="odd">
<td style="text-align: left;">RPL30</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.4036053</td>
<td style="text-align: right;">-0.0380198</td>
<td style="text-align: right;">0.3782729</td>
<td style="text-align: right;">0.3188572</td>
<td style="text-align: right;">0.3838818</td>
<td style="text-align: right;">0.3019607</td>
<td style="text-align: right;">0.1297383</td>
<td style="text-align: right;">1.8344132</td>
<td style="text-align: right;">0.7312181</td>
</tr>
<tr class="even">
<td style="text-align: left;">PDLIM1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.7866119</td>
<td style="text-align: right;">0.7335502</td>
<td style="text-align: right;">0.8844840</td>
<td style="text-align: right;">0.3162534</td>
<td style="text-align: right;">0.9046133</td>
<td style="text-align: right;">0.7870959</td>
<td style="text-align: right;">1.0450514</td>
<td style="text-align: right;">0.9840771</td>
<td style="text-align: right;">0.7166936</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HLA-DRB5</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">-0.7513633</td>
<td style="text-align: right;">-0.0629336</td>
<td style="text-align: right;">-0.5026730</td>
<td style="text-align: right;">-1.0243036</td>
<td style="text-align: right;">-0.9334601</td>
<td style="text-align: right;">-0.4578900</td>
<td style="text-align: right;">-0.2801242</td>
<td style="text-align: right;">-0.6569262</td>
<td style="text-align: right;">-0.7282321</td>
</tr>
<tr class="even">
<td style="text-align: left;">HMGN2</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">-0.3801497</td>
<td style="text-align: right;">-0.2377400</td>
<td style="text-align: right;">-0.2432392</td>
<td style="text-align: right;">-0.1211472</td>
<td style="text-align: right;">-0.0662938</td>
<td style="text-align: right;">-0.0443311</td>
<td style="text-align: right;">-0.2959010</td>
<td style="text-align: right;">0.2496630</td>
<td style="text-align: right;">-2.4361599</td>
</tr>
</tbody>
</table>
</div>
<p>Of particular interest is the <code>Top</code> field, which contains the highest rank for each gene across all pairwise comparisons involving cluster 3. The set of genes with <code>Top</code> values of 1 contains the gene with the lowest <span class="math inline">\(p\)</span>-value from each comparison. Similarly, the set of genes with <code>Top</code> values less than or equal to 10 contains the top 10 genes from each comparison.</p>
<p>We use the <code>Top</code> value to identify a set of genes that is guaranteed to distinguish cluster 3 from any other cluster. Figure <a href="#fig:heat-logFC">6</a> plots the log-fold changes for the top 10 genes from each pairwise comparison.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:heat-logFC"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-logFC-1.png" alt="Heatmap of log-fold changes for the top 10 marker genes for cluster 3 over all other clusters. Each column is a cluster, each row a gene. Colours are capped at -5 and 5 to preserve dynamic range." width="624" />
<p class="caption">
Figure 6: Heatmap of log-fold changes for the top 10 marker genes for cluster 3 over all other clusters. Each column is a cluster, each row a gene. Colours are capped at -5 and 5 to preserve dynamic range.
</p>
</div>
</div>
<p>Complementing Figure <a href="#fig:heat-logFC">6</a>, Figure <a href="#fig:heat-logExp">7</a> plots the log-expression values of each cells in each cluster for the top 5 genes</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:heat-logExp"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-logExp-1.png" alt="Heatmap of log-expression values in each sample for the top 10 marker genes in cluster 3. Each column is a sample, each row a gene. Colours are capped at -5 and 5 to preserve dynamic range." width="624" />
<p class="caption">
Figure 7: Heatmap of log-expression values in each sample for the top 10 marker genes in cluster 3. Each column is a sample, each row a gene. Colours are capped at -5 and 5 to preserve dynamic range.
</p>
</div>
</div>
<p>Similar figures showing the top 10 marker genes for each cluster are available in <a href="../output/marker_genes/ignoring_unknowns/union_any/heatmaps.union_any.pdf"><code>output/marker_genes/ignoring_unknowns/union_any/heatmaps.union_any.pdf</code></a>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>We intentionally use pairwise comparisons between clusters rather than comparing each cluster to the average of all other cells. The latter approach is sensitive to the population composition, potentially resulting in wildly different sets of markers when cell type abundances change in different contexts. In the worst case, the presence of a single dominant subpopulation will drive the selection of top markers for every other cluster, pushing out useful genes that can resolve the various minor subpopulations. Moreover, pairwise comparisons naturally provide more information to interpret of the utility of a marker, e.g., by providing log-fold changes to indicate which clusters are distinguished by this gene.</p>
<h3 id="using-the-log-fold-change">Using the log-fold change</h3>
<p>The differential expression analysis in <a href="#standard-application">Standard application</a> considers both up- and downregulated genes to be potential markers. However, downregulated genes are less appealing as markers as it is more difficult to interpret and experimentally validate an absence of expression. To focus on upregulated markers, we can instead perform a one-sided <span class="math inline">\(t\)</span>-test to identify genes that are upregulated in each cluster compared to the others.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The <span class="math inline">\(t\)</span>-test also allows us to specify a non-zero log-fold change as the null hypothesis. This allows us to consider the magnitude of the log-fold change in our <span class="math inline">\(p\)</span>-value calculations, in a manner that is more rigorous than simply filtering directly on the log-fold changes<a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>. For example, we can test for genes with log-fold changes that are significantly greater than 1.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Combining these two settings yield a more focused set of candidate marker genes that are ‘upregulated with a log-fold change &gt; 1’ in cluster 3 (Figures <a href="#fig:heat-focused-logFC">8</a> and <a href="#fig:heat-focused-logExp">9</a>).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:heat-focused-logFC"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-focused-logFC-1.png" alt="Heatmap of log-fold changes for cluster 3 over all other clusters. Colours are capped at -5 and 5 to preserve dynamic range." width="624" />
<p class="caption">
Figure 8: Heatmap of log-fold changes for cluster 3 over all other clusters. Colours are capped at -5 and 5 to preserve dynamic range.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:heat-focused-logExp"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-focused-logExp-1.png" alt="Heatmap of log-expression values in each sample for the top 10 'upregulated with a log-fold change &gt; 1' marker genes in cluster 3. Each column is a sample, each row a gene. Colours are capped at -5 and 5 to preserve dynamic range." width="624" />
<p class="caption">
Figure 9: Heatmap of log-expression values in each sample for the top 10 ‘upregulated with a log-fold change &gt; 1’ marker genes in cluster 3. Each column is a sample, each row a gene. Colours are capped at -5 and 5 to preserve dynamic range.
</p>
</div>
</div>
<p>Similar figures showing the top 10 upregulated and ‘upregulated with a log-fold change &gt; 1’ marker genes for each cluster are available in <a href="../output/marker_genes/ignoring_unknowns/union_up/heatmaps.union_up.pdf"><code>output/marker_genes/ignoring_unknowns/union_up/heatmaps.union_up.pdf</code></a>. and <a href="../output/marker_genes/ignoring_unknowns/union_up_lfc1/heatmaps.union_up_lfc1.pdf"><code>output/marker_genes/ignoring_unknowns/union_up_lfc1/heatmaps.union_up_lfc1.pdf</code></a>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Of course, this increased stringency is not without cost. If only upregulated genes are requested, any cluster defined by downregulation of a marker gene will not contain that gene among the top set of features in its table. This is occasionally relevant for subtypes or other states that are distinguished by high versus low expression of particular genes. Similarly, setting an excessively high log-fold change threshold may discard otherwise useful genes. For example, a gene upregulated in a small proportion of cells of a cluster will have a small log-fold change but can still be an effective marker if the focus is on specificity rather than sensitivity.</p>
<h3 id="finding-cluster-specific-markers">Finding cluster-specific markers</h3>
<p>The genes detected in <a href="#standard-application">Standard application</a> and <a href="#using-the-log-fold-change">Using the log-fold change</a> will give a high ranking to genes that are differentially expressed in any pairwise comparison. This is because a gene only needs a very low <span class="math inline">\(p\)</span>-value in a single pairwise comparison to achieve a low <code>Top</code> value. A more stringent approach would only consider genes that are differentially expressed in all pairwise comparisons involving the cluster of interest. To achieve this, we use an intersection-union test <span class="citation" data-cites="berger1996bioequivalence">(Berger and Hsu <a href="#ref-berger1996bioequivalence">1996</a>)</span> where the combined <span class="math inline">\(p\)</span>-value for each gene is the maximum of the <span class="math inline">\(p\)</span>-values from all pairwise comparisons. A gene will only achieve a low combined <span class="math inline">\(p\)</span>-value if it is strongly DE in all comparisons to other clusters.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>This strategy will only report genes that are highly specific to the cluster of interest. When it works, it can be highly effective as it generates a small focused set of candidate markers. However, any gene that is expressed at the same level in two or more clusters will simply not be detected. This is likely to discard many interesting genes, especially if the clusters are finely resolved with weak separation. To give a concrete example, consider a mixed population of CD4<sup>+</sup>-only, CD8<sup>+</sup>-only, double-positive and double-negative T cells. Using this strategy, neither <em>Cd4</em> or <em>Cd8</em> would be detected as subpopulation-specific markers because each gene is expressed in two subpopulations. In comparison, the strategy in <a href="#standard-application">Standard application</a> will detect both of these genes as they will be DE between at least one pair of subpopulations.</p>
<p>Figures <a href="#fig:heat-focused-logFC">8</a> and <a href="#fig:heat-focused-logExp">9</a> show examples of this strategy for cluster 3.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:heat-uniquely-up-logFC"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-uniquely-up-logFC-1.png" alt="Heatmap of log-fold changes for cluster 3 over all other clusters. Colours are capped at -5 and 5 to preserve dynamic range." width="624" />
<p class="caption">
Figure 10: Heatmap of log-fold changes for cluster 3 over all other clusters. Colours are capped at -5 and 5 to preserve dynamic range.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:heat-uniquely-up-logExp"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-uniquely-up-logExp-1.png" alt="Heatmap of log-expression values in each sample for the top 10 uniquely upregulated marker genes in cluster 3. Each column is a sample, each row a gene. Colours are capped at -5 and 5 to preserve dynamic range." width="624" />
<p class="caption">
Figure 11: Heatmap of log-expression values in each sample for the top 10 uniquely upregulated marker genes in cluster 3. Each column is a sample, each row a gene. Colours are capped at -5 and 5 to preserve dynamic range.
</p>
</div>
</div>
<p>Similar figures showing the top 100 uniquely upregulated marker genes for each cluster are available in <a href="../output/marker_genes/ignoring_unknowns/uniquely_up/heatmaps.uniquely_up.pdf"><code>output/marker_genes/ignoring_unknowns/uniquely_up/heatmaps.uniquely_up.pdf</code></a>. Recall that the stringency of the ‘cluster-specific marker’ criteria is such that a cluster may only have a small number of statistically significant marker genes or indeed none at all. Therefore, these plots include an row annotation highlighting if the gene is a statisticially significant cluster-specific marker (i.e. <span class="math inline">\(FDR &lt; 0.05\)</span>).</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="summary-3">Summary</h2>
<p>The tables of marker genes are available as CSV files in <a href="../output/marker_genes/ignoring_unknowns"><code>output/marker_genes/ignoring_unknowns</code></a> and are named as follows:</p>
<ul>
<li><code>union_any</code>: Marker genes from <a href="#standard-application">Standard application</a> whereby marker genes are formed from the <strong>union</strong> of genes differentially expressed in <strong>any</strong> direction.</li>
<li><code>union_up</code>: Marker genes from <a href="#using-the-log-fold-change">Using the log-fold change</a> whereby marker genes are formed from the <strong>union</strong> of genes differentially expressed that are <strong>up</strong>regulated.</li>
<li><code>union_up_lfc1</code>: Marker genes from <a href="#using-the-log-fold-change">Using the log-fold change</a> whereby marker genes are formed from the <strong>union</strong> of genes differentially expressed that are <strong>up</strong>regulated with a log-fold change of at least 1 (<strong>lfc1</strong>).</li>
<li><code>uniquely_up</code>: Marker genes from <a href="#finding-cluster-specific-markers">Finding cluster-specific markers</a> whereby marker genes are cluster-specific (<strong>uniquely</strong>) differentially expressed genes that are <strong>up</strong>regulated.</li>
</ul>
<h1 id="using-marker-genes-to-perform-cell-type-annotation">Using marker genes to perform cell type annotation</h1>
<p>Regardless of the strategy employed, <a href="#standard-application">Standard application</a>, <a href="#using-the-log-fold-change">Using the log-fold change</a> or [Finding cluster-speicific markers], the result is a gene list that needs to be examined to try to identify the cell type. As such, they can be analysed like any other gene list, just as you might a list of differentially expressed genes from a bulk RNA-seq experiment. This includes: - Staring at heatmaps and spreadsheets, such as those included with this report, and googling some of the more informative genes, such as the cluster-specific marker genes. - Gene set enrichment analysis, e.g., using GO or KEGG terms. This can require a fair amount of manual work and knowledge of the biological system. In the next section, we will complement this rather tedious but necessary manual work with an automated approach to cell type labelling.</p>
<h1 id="automated-cell-type-labelling-with-singler">Automated cell type labelling with <em><a href="https://bioconductor.org/packages/3.9/SingleR">SingleR</a></em></h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><em><a href="https://bioconductor.org/packages/3.9/SingleR">SingleR</a></em> is an automatic annotation method for scRNAseq data <span class="citation" data-cites="aran2019reference">(Aran et al. <a href="#ref-aran2019reference">2019</a>)</span>. Given a reference data set of samples<a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a> with known labels, it labels new cells from a test data set based on similarity to the reference set. Specifically, for each test cell:</p>
<ol type="1">
<li>We compute the Spearman correlation between its expression profile and that of each reference sample.</li>
<li>We define the per-label score as a fixed quantile (by default, 0.8) of the distribution of correlations.</li>
<li>We repeat this for all labels and we take the label with the highest score as the annotation for this cell.</li>
<li>Finally, we perform a fine-tuning step:</li>
</ol>
<ul>
<li>The reference data set is subsetted to only include labels with scores close to the maximum.</li>
<li>Scores are recomputed using only marker genes for the subset of labels.</li>
<li>This is iterated until one label remains.</li>
</ul>
<p>For visualization purposes we use the normalized scores, which are the scores linearly adjusted for each cell so that the smallest score is 0 and the largest score is 1. This is followed by cubing of the adjusted scores to improve dynamic range near 1. Automatic annotation provides a convenient way of transferring biological knowledge across data sets. In this manner, the burden of interpreting clusters and defining marker genes only has to be done once (i.e. for the reference set). I use <em><a href="https://bioconductor.org/packages/3.9/SingleR">SingleR</a></em> to annotate the data set against 5 reference data sets:</p>
<ul>
<li><code>HumanPrimaryCellAtlasData</code>: This reference set entails 713 microarray samples of the Human Primary Cell Atlas (HPCA) <span class="citation" data-cites="hpcaRef">(Mabbott et al. <a href="#ref-hpcaRef">2013</a>)</span>.</li>
<li><code>BlueprintEncodeData</code>: This reference set entails 259 RNA-seq samples of pure stroma and immune cells as generated and supplied by Blueprint and ENCODE.</li>
<li><code>DatabaseImmuneCellExpressionData</code>: This reference set entails 1561 bulk RNA-seq samples of sorted cell populations from the DICE database.</li>
<li><code>MonacoImmuneData</code>: This reference set entails 114 bulk RNA-seq samples of sorted immune cell populations that can be found in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107011">GSE107011</a>.</li>
<li><code>NovershternHematopoieticData</code>: This reference set entails 211 bulk human microarray samples of sorted hematopoietic cell populations that can be found in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE24759">GSE24759</a></li>
</ul>
<p><em><a href="https://bioconductor.org/packages/3.9/SingleR">SingleR</a></em> can be used to annotate at the cell-level or the cluster-level. The latter is generally used to speed-up calculations, which is not necessary for this sized data set.</p>
<p>Therefore, I have annotated at the cell-level to obtain a higher resolution analysis.</p>
<h2 id="humanprimarycellatlasdata"><code>HumanPrimaryCellAtlasData</code></h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The <code>HumanPrimaryCellAtlasData</code> reference set entails 713 microarray samples of the Human Primary Cell Atlas (HPCA) <span class="citation" data-cites="hpcaRef">(Mabbott et al. <a href="#ref-hpcaRef">2013</a>)</span>. These 713 samples were processed and normalized as described in <span class="citation" data-cites="aran2019reference">(Aran et al. <a href="#ref-aran2019reference">2019</a>)</span> and each sample has been assigned to one of 37 main cell types and 157 subtypes. Each sample has a ‘main’ and a ‘fine’ cell type label, summarised below.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-24">Table 3: </span>Summary of main cell types in the <code>HumanPrimaryCellAtlasData</code> reference set.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Astrocyte</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.3%</td>
</tr>
<tr class="even">
<td style="text-align: left;">B_cell</td>
<td style="text-align: right;">26</td>
<td style="text-align: left;">3.6%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BM</td>
<td style="text-align: right;">7</td>
<td style="text-align: left;">1.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">BM &amp; Prog.</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Chondrocytes</td>
<td style="text-align: right;">8</td>
<td style="text-align: left;">1.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">CMP</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.3%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: right;">88</td>
<td style="text-align: left;">12.3%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Embryonic_stem_cells</td>
<td style="text-align: right;">17</td>
<td style="text-align: left;">2.4%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial_cells</td>
<td style="text-align: right;">64</td>
<td style="text-align: left;">9.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Epithelial_cells</td>
<td style="text-align: right;">16</td>
<td style="text-align: left;">2.2%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Erythroblast</td>
<td style="text-align: right;">8</td>
<td style="text-align: left;">1.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fibroblasts</td>
<td style="text-align: right;">10</td>
<td style="text-align: left;">1.4%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gametocytes</td>
<td style="text-align: right;">5</td>
<td style="text-align: left;">0.7%</td>
</tr>
<tr class="even">
<td style="text-align: left;">GMP</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.3%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Hepatocytes</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">0.4%</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSC_-G-CSF</td>
<td style="text-align: right;">10</td>
<td style="text-align: left;">1.4%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSC_CD34+</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">0.8%</td>
</tr>
<tr class="even">
<td style="text-align: left;">iPS_cells</td>
<td style="text-align: right;">42</td>
<td style="text-align: left;">5.9%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: right;">25</td>
<td style="text-align: left;">3.5%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophage</td>
<td style="text-align: right;">90</td>
<td style="text-align: left;">12.6%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">MEP</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.3%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: right;">60</td>
<td style="text-align: left;">8.4%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">MSC</td>
<td style="text-align: right;">9</td>
<td style="text-align: left;">1.3%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Myelocyte</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.3%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neuroepithelial_cell</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neurons</td>
<td style="text-align: right;">16</td>
<td style="text-align: left;">2.2%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neutrophil</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">0.4%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: right;">18</td>
<td style="text-align: left;">2.5%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK_cell</td>
<td style="text-align: right;">5</td>
<td style="text-align: left;">0.7%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Osteoblasts</td>
<td style="text-align: right;">15</td>
<td style="text-align: left;">2.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Platelets</td>
<td style="text-align: right;">5</td>
<td style="text-align: left;">0.7%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pre-B_cell_CD34-</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.3%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Pro-B_cell_CD34+</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.3%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pro-Myelocyte</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.3%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Smooth_muscle_cells</td>
<td style="text-align: right;">16</td>
<td style="text-align: left;">2.2%</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cells</td>
<td style="text-align: right;">68</td>
<td style="text-align: left;">9.5%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tissue_stem_cells</td>
<td style="text-align: right;">55</td>
<td style="text-align: left;">7.7%</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["label.main"],"name":[1],"type":["chr"],"align":["left"]},{"label":["label.fine"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"Astrocyte","2":"Astrocyte:Embryonic_stem_cell-derived","3":"2"},{"1":"B_cell","2":"B_cell","3":"4"},{"1":"B_cell","2":"B_cell:CXCR4-_centrocyte","3":"4"},{"1":"B_cell","2":"B_cell:CXCR4+_centroblast","3":"4"},{"1":"B_cell","2":"B_cell:Germinal_center","3":"3"},{"1":"B_cell","2":"B_cell:immature","3":"2"},{"1":"B_cell","2":"B_cell:Memory","3":"3"},{"1":"B_cell","2":"B_cell:Naive","3":"3"},{"1":"B_cell","2":"B_cell:Plasma_cell","3":"3"},{"1":"BM","2":"BM","3":"7"},{"1":"BM & Prog.","2":"BM","3":"1"},{"1":"Chondrocytes","2":"Chondrocytes:MSC-derived","3":"8"},{"1":"CMP","2":"CMP","3":"2"},{"1":"DC","2":"DC:monocyte-derived","3":"19"},{"1":"DC","2":"DC:monocyte-derived:A._fumigatus_germ_tubes_6h","3":"2"},{"1":"DC","2":"DC:monocyte-derived:AEC-conditioned","3":"5"},{"1":"DC","2":"DC:monocyte-derived:AM580","3":"3"},{"1":"DC","2":"DC:monocyte-derived:anti-DC-SIGN_2h","3":"3"},{"1":"DC","2":"DC:monocyte-derived:antiCD40/VAF347","3":"2"},{"1":"DC","2":"DC:monocyte-derived:CD40L","3":"3"},{"1":"DC","2":"DC:monocyte-derived:Galectin-1","3":"3"},{"1":"DC","2":"DC:monocyte-derived:immature","3":"20"},{"1":"DC","2":"DC:monocyte-derived:LPS","3":"6"},{"1":"DC","2":"DC:monocyte-derived:mature","3":"11"},{"1":"DC","2":"DC:monocyte-derived:Poly(IC)","3":"3"},{"1":"DC","2":"DC:monocyte-derived:rosiglitazone","3":"3"},{"1":"DC","2":"DC:monocyte-derived:rosiglitazone/AGN193109","3":"2"},{"1":"DC","2":"DC:monocyte-derived:Schuler_treatment","3":"3"},{"1":"Embryonic_stem_cells","2":"Embryonic_stem_cells","3":"17"},{"1":"Endothelial_cells","2":"Endothelial_cells:blood_vessel","3":"8"},{"1":"Endothelial_cells","2":"Endothelial_cells:HUVEC","3":"16"},{"1":"Endothelial_cells","2":"Endothelial_cells:HUVEC:B._anthracis_LT","3":"2"},{"1":"Endothelial_cells","2":"Endothelial_cells:HUVEC:Borrelia_burgdorferi","3":"2"},{"1":"Endothelial_cells","2":"Endothelial_cells:HUVEC:FPV-infected","3":"3"},{"1":"Endothelial_cells","2":"Endothelial_cells:HUVEC:H5N1-infected","3":"3"},{"1":"Endothelial_cells","2":"Endothelial_cells:HUVEC:IFNg","3":"1"},{"1":"Endothelial_cells","2":"Endothelial_cells:HUVEC:IL-1b","3":"3"},{"1":"Endothelial_cells","2":"Endothelial_cells:HUVEC:PR8-infected","3":"3"},{"1":"Endothelial_cells","2":"Endothelial_cells:HUVEC:Serum_Amyloid_A","3":"6"},{"1":"Endothelial_cells","2":"Endothelial_cells:HUVEC:VEGF","3":"3"},{"1":"Endothelial_cells","2":"Endothelial_cells:lymphatic","3":"7"},{"1":"Endothelial_cells","2":"Endothelial_cells:lymphatic:KSHV","3":"4"},{"1":"Endothelial_cells","2":"Endothelial_cells:lymphatic:TNFa_48h","3":"3"},{"1":"Epithelial_cells","2":"Epithelial_cells:bladder","3":"6"},{"1":"Epithelial_cells","2":"Epithelial_cells:bronchial","3":"10"},{"1":"Erythroblast","2":"Erythroblast","3":"8"},{"1":"Fibroblasts","2":"Fibroblasts:breast","3":"6"},{"1":"Fibroblasts","2":"Fibroblasts:foreskin","3":"4"},{"1":"Gametocytes","2":"Gametocytes:oocyte","3":"3"},{"1":"Gametocytes","2":"Gametocytes:spermatocyte","3":"2"},{"1":"GMP","2":"GMP","3":"2"},{"1":"Hepatocytes","2":"Hepatocytes","3":"3"},{"1":"HSC_-G-CSF","2":"HSC_-G-CSF","3":"10"},{"1":"HSC_CD34+","2":"HSC_CD34+","3":"6"},{"1":"iPS_cells","2":"iPS_cells:adipose_stem_cell-derived:lentiviral","3":"3"},{"1":"iPS_cells","2":"iPS_cells:adipose_stem_cell-derived:minicircle-derived","3":"3"},{"1":"iPS_cells","2":"iPS_cells:adipose_stem_cells","3":"3"},{"1":"iPS_cells","2":"iPS_cells:CRL2097_foreskin","3":"3"},{"1":"iPS_cells","2":"iPS_cells:CRL2097_foreskin-derived:d20_hepatic_diff","3":"3"},{"1":"iPS_cells","2":"iPS_cells:CRL2097_foreskin-derived:undiff.","3":"3"},{"1":"iPS_cells","2":"iPS_cells:fibroblast-derived:Direct_del._reprog","3":"2"},{"1":"iPS_cells","2":"iPS_cells:fibroblast-derived:Retroviral_transf","3":"1"},{"1":"iPS_cells","2":"iPS_cells:fibroblasts","3":"1"},{"1":"iPS_cells","2":"iPS_cells:foreskin_fibrobasts","3":"1"},{"1":"iPS_cells","2":"iPS_cells:iPS:minicircle-derived","3":"5"},{"1":"iPS_cells","2":"iPS_cells:PDB_1lox-17Puro-10","3":"1"},{"1":"iPS_cells","2":"iPS_cells:PDB_1lox-17Puro-5","3":"1"},{"1":"iPS_cells","2":"iPS_cells:PDB_1lox-21Puro-20","3":"1"},{"1":"iPS_cells","2":"iPS_cells:PDB_1lox-21Puro-26","3":"1"},{"1":"iPS_cells","2":"iPS_cells:PDB_2lox-17","3":"1"},{"1":"iPS_cells","2":"iPS_cells:PDB_2lox-21","3":"1"},{"1":"iPS_cells","2":"iPS_cells:PDB_2lox-22","3":"1"},{"1":"iPS_cells","2":"iPS_cells:PDB_2lox-5","3":"1"},{"1":"iPS_cells","2":"iPS_cells:PDB_fibroblasts","3":"1"},{"1":"iPS_cells","2":"iPS_cells:skin_fibroblast","3":"2"},{"1":"iPS_cells","2":"iPS_cells:skin_fibroblast-derived","3":"3"},{"1":"Keratinocytes","2":"Keratinocytes","3":"3"},{"1":"Keratinocytes","2":"Keratinocytes:IFNg","3":"2"},{"1":"Keratinocytes","2":"Keratinocytes:IL19","3":"3"},{"1":"Keratinocytes","2":"Keratinocytes:IL1b","3":"2"},{"1":"Keratinocytes","2":"Keratinocytes:IL20","3":"3"},{"1":"Keratinocytes","2":"Keratinocytes:IL22","3":"3"},{"1":"Keratinocytes","2":"Keratinocytes:IL24","3":"3"},{"1":"Keratinocytes","2":"Keratinocytes:IL26","3":"3"},{"1":"Keratinocytes","2":"Keratinocytes:KGF","3":"3"},{"1":"Macrophage","2":"Macrophage:Alveolar","3":"4"},{"1":"Macrophage","2":"Macrophage:Alveolar:B._anthacis_spores","3":"3"},{"1":"Macrophage","2":"Macrophage:monocyte-derived","3":"26"},{"1":"Macrophage","2":"Macrophage:monocyte-derived:IFNa","3":"9"},{"1":"Macrophage","2":"Macrophage:monocyte-derived:IL-4/cntrl","3":"5"},{"1":"Macrophage","2":"Macrophage:monocyte-derived:IL-4/Dex/cntrl","3":"5"},{"1":"Macrophage","2":"Macrophage:monocyte-derived:IL-4/Dex/TGFb","3":"10"},{"1":"Macrophage","2":"Macrophage:monocyte-derived:IL-4/TGFb","3":"5"},{"1":"Macrophage","2":"Macrophage:monocyte-derived:M-CSF","3":"2"},{"1":"Macrophage","2":"Macrophage:monocyte-derived:M-CSF/IFNg","3":"2"},{"1":"Macrophage","2":"Macrophage:monocyte-derived:M-CSF/IFNg/Pam3Cys","3":"2"},{"1":"Macrophage","2":"Macrophage:monocyte-derived:M-CSF/Pam3Cys","3":"2"},{"1":"Macrophage","2":"Macrophage:monocyte-derived:S._aureus","3":"15"},{"1":"MEP","2":"MEP","3":"2"},{"1":"Monocyte","2":"Monocyte","3":"27"},{"1":"Monocyte","2":"Monocyte:anti-FcgRIIB","3":"2"},{"1":"Monocyte","2":"Monocyte:CD14+","3":"3"},{"1":"Monocyte","2":"Monocyte:CD16-","3":"7"},{"1":"Monocyte","2":"Monocyte:CD16+","3":"6"},{"1":"Monocyte","2":"Monocyte:CXCL4","3":"2"},{"1":"Monocyte","2":"Monocyte:F._tularensis_novicida","3":"6"},{"1":"Monocyte","2":"Monocyte:leukotriene_D4","3":"4"},{"1":"Monocyte","2":"Monocyte:MCSF","3":"2"},{"1":"Monocyte","2":"Monocyte:S._typhimurium_flagellin","3":"1"},{"1":"MSC","2":"MSC","3":"9"},{"1":"Myelocyte","2":"Myelocyte","3":"2"},{"1":"Neuroepithelial_cell","2":"Neuroepithelial_cell:ESC-derived","3":"1"},{"1":"Neurons","2":"Neurons:adrenal_medulla_cell_line","3":"6"},{"1":"Neurons","2":"Neurons:ES_cell-derived_neural_precursor","3":"6"},{"1":"Neurons","2":"Neurons:Schwann_cell","3":"4"},{"1":"Neutrophil","2":"Neutrophil","3":"3"},{"1":"Neutrophils","2":"Neutrophil","3":"3"},{"1":"Neutrophils","2":"Neutrophil:commensal_E._coli_MG1655","3":"2"},{"1":"Neutrophils","2":"Neutrophil:GM-CSF_IFNg","3":"4"},{"1":"Neutrophils","2":"Neutrophil:inflam","3":"4"},{"1":"Neutrophils","2":"Neutrophil:LPS","3":"4"},{"1":"Neutrophils","2":"Neutrophil:uropathogenic_E._coli_UTI89","3":"1"},{"1":"NK_cell","2":"NK_cell","3":"1"},{"1":"NK_cell","2":"NK_cell:CD56hiCD62L+","3":"1"},{"1":"NK_cell","2":"NK_cell:IL2","3":"3"},{"1":"Osteoblasts","2":"Osteoblasts","3":"9"},{"1":"Osteoblasts","2":"Osteoblasts:BMP2","3":"6"},{"1":"Platelets","2":"Platelets","3":"5"},{"1":"Pre-B_cell_CD34-","2":"Pre-B_cell_CD34-","3":"2"},{"1":"Pro-B_cell_CD34+","2":"Pro-B_cell_CD34+","3":"2"},{"1":"Pro-Myelocyte","2":"Pro-Myelocyte","3":"2"},{"1":"Smooth_muscle_cells","2":"Smooth_muscle_cells:bronchial","3":"3"},{"1":"Smooth_muscle_cells","2":"Smooth_muscle_cells:bronchial:vit_D","3":"3"},{"1":"Smooth_muscle_cells","2":"Smooth_muscle_cells:umbilical_vein","3":"2"},{"1":"Smooth_muscle_cells","2":"Smooth_muscle_cells:vascular","3":"5"},{"1":"Smooth_muscle_cells","2":"Smooth_muscle_cells:vascular:IL-17","3":"3"},{"1":"T_cells","2":"T_cell:CCR10-CLA+1,25(OH)2_vit_D3/IL-12","3":"1"},{"1":"T_cells","2":"T_cell:CCR10+CLA+1,25(OH)2_vit_D3/IL-12","3":"1"},{"1":"T_cells","2":"T_cell:CD4+","3":"12"},{"1":"T_cells","2":"T_cell:CD4+_central_memory","3":"5"},{"1":"T_cells","2":"T_cell:CD4+_effector_memory","3":"4"},{"1":"T_cells","2":"T_cell:CD4+_Naive","3":"6"},{"1":"T_cells","2":"T_cell:CD8+","3":"16"},{"1":"T_cells","2":"T_cell:CD8+_Central_memory","3":"3"},{"1":"T_cells","2":"T_cell:CD8+_effector_memory","3":"4"},{"1":"T_cells","2":"T_cell:CD8+_effector_memory_RA","3":"4"},{"1":"T_cells","2":"T_cell:CD8+_naive","3":"4"},{"1":"T_cells","2":"T_cell:effector","3":"4"},{"1":"T_cells","2":"T_cell:gamma-delta","3":"2"},{"1":"T_cells","2":"T_cell:Treg:Naive","3":"2"},{"1":"Tissue_stem_cells","2":"Tissue_stem_cells:adipose-derived_MSC_AM3","3":"2"},{"1":"Tissue_stem_cells","2":"Tissue_stem_cells:BM_MSC","3":"8"},{"1":"Tissue_stem_cells","2":"Tissue_stem_cells:BM_MSC:BMP2","3":"12"},{"1":"Tissue_stem_cells","2":"Tissue_stem_cells:BM_MSC:osteogenic","3":"8"},{"1":"Tissue_stem_cells","2":"Tissue_stem_cells:BM_MSC:TGFb3","3":"11"},{"1":"Tissue_stem_cells","2":"Tissue_stem_cells:CD326-CD56+","3":"3"},{"1":"Tissue_stem_cells","2":"Tissue_stem_cells:dental_pulp","3":"6"},{"1":"Tissue_stem_cells","2":"Tissue_stem_cells:iliac_MSC","3":"3"},{"1":"Tissue_stem_cells","2":"Tissue_stem_cells:lipoma-derived_MSC","3":"2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<h3 id="main-label">Main label</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Annotating against the main labels, we obtain the following cell type labels.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-27">Table 4: </span>Cell label assignments using the main labels of the <code>HumanPrimaryCellAtlasData</code> reference data.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">T_cells</td>
<td style="text-align: right;">9853</td>
<td style="text-align: left;">99.7%</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK_cell</td>
<td style="text-align: right;">12</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B_cell</td>
<td style="text-align: right;">7</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pre-B_cell_CD34-</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocyte</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pro-B_cell_CD34+</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
</tbody>
</table>
</div>
<p>Heatmaps of the normalized scores are shown, with and without clustering by these normalized scores, in Figures <a href="#fig:heat-hpcad-main-clustered">12</a> and <a href="#fig:heat-hpcad-main">13</a>.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-hpcad-main-clustered"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-hpcad-main-clustered-1.png" alt="Heatmap of normalized scores using the main labels from the `HumanPrimaryCellAtlasData` reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores." width="624" />
<p class="caption">
Figure 12: Heatmap of normalized scores using the main labels from the <code>HumanPrimaryCellAtlasData</code> reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-hpcad-main"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-hpcad-main-1.png" alt="Heatmap of normalized scores using the main labels from the `HumanPrimaryCellAtlasData` reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels." width="624" />
<p class="caption">
Figure 13: Heatmap of normalized scores using the main labels from the <code>HumanPrimaryCellAtlasData</code> reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels.
</p>
</div>
</div>
<p>Figure <a href="#fig:tsne-hpcad-main">14</a> overlays the normalized scores for the 10 labels with the largest maximum scores on the t-SNE plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:tsne-hpcad-main"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/tsne-hpcad-main-1.png" alt="t-SNE overlaid with normalized scores using the main labels from the `HumanPrimaryCellAtlasData` reference set. Only the 10 labels with the largest maximum scores are shown." width="624" />
<p class="caption">
Figure 14: t-SNE overlaid with normalized scores using the main labels from the <code>HumanPrimaryCellAtlasData</code> reference set. Only the 10 labels with the largest maximum scores are shown.
</p>
</div>
</div>
<h3 id="fine-label">Fine label</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Annotating against the fine labels, we obtain the following cell type labels.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-29">Table 5: </span>Cell label assignments using the fine labels of the <code>HumanPrimaryCellAtlasData</code> reference data.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">T_cell:CD4+_central_memory</td>
<td style="text-align: right;">4530</td>
<td style="text-align: left;">45.9%</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cell:CD4+_effector_memory</td>
<td style="text-align: right;">4139</td>
<td style="text-align: left;">41.9%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cell:gamma-delta</td>
<td style="text-align: right;">616</td>
<td style="text-align: left;">6.2%</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cell:CD8+</td>
<td style="text-align: right;">321</td>
<td style="text-align: left;">3.2%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cell:CD4+_Naive</td>
<td style="text-align: right;">147</td>
<td style="text-align: left;">1.5%</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cell:CD4+</td>
<td style="text-align: right;">89</td>
<td style="text-align: left;">0.9%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cell:Treg:Naive</td>
<td style="text-align: right;">21</td>
<td style="text-align: left;">0.2%</td>
</tr>
<tr class="even">
<td style="text-align: left;">B_cell</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cell:CD8+_effector_memory</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">B_cell:immature</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B_cell:Memory</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">B_cell:Naive</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSC_-G-CSF</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK_cell:IL2</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Pre-B_cell_CD34-</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pro-B_cell_CD34+</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T_cell:CD8+_naive</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">T_cell:effector</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
</tbody>
</table>
</div>
<p>Heatmaps of the normalized scores are shown, with and without clustering by these normalized scores, in Figures <a href="#fig:heat-hpcad-fine-clustered">15</a> and <a href="#fig:heat-hpcad-fine">16</a>.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-hpcad-fine-clustered"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-hpcad-fine-clustered-1.png" alt="Heatmap of normalized scores using the fine labels from the `HumanPrimaryCellAtlasData` reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores." width="624" />
<p class="caption">
Figure 15: Heatmap of normalized scores using the fine labels from the <code>HumanPrimaryCellAtlasData</code> reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-hpcad-fine"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-hpcad-fine-1.png" alt="Heatmap of normalized scores using the fine labels from the `HumanPrimaryCellAtlasData` reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels." width="624" />
<p class="caption">
Figure 16: Heatmap of normalized scores using the fine labels from the <code>HumanPrimaryCellAtlasData</code> reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels.
</p>
</div>
</div>
<p>Figure <a href="#fig:tsne-hpcad-fine">17</a> overlays the normalized scores for the 10 labels with the largest maximum scores on the t-SNE plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:tsne-hpcad-fine"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/tsne-hpcad-fine-1.png" alt="t-SNE overlaid with normalized scores using the fine labels from the `HumanPrimaryCellAtlasData` reference set. Only the 10 labels with the largest maximum scores are shown." width="624" />
<p class="caption">
Figure 17: t-SNE overlaid with normalized scores using the fine labels from the <code>HumanPrimaryCellAtlasData</code> reference set. Only the 10 labels with the largest maximum scores are shown.
</p>
</div>
</div>
<h2 id="blueprintencodedata"><code>BlueprintEncodeData</code></h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The <code>BlueprintEncodeData</code> reference set entails 259 RNA-seq samples of pure stroma and immune cells as generated and supplied by Blueprint and ENCODE. Each sample has a ‘main’ and a ‘fine’ cell type label, summarised below.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-31">Table 6: </span>Summary of main cell types in the <code>BlueprintEncodeData</code> reference set.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Adipocytes</td>
<td style="text-align: right;">11</td>
<td style="text-align: left;">4.2%</td>
</tr>
<tr class="even">
<td style="text-align: left;">B-cells</td>
<td style="text-align: right;">8</td>
<td style="text-align: left;">3.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T-cells</td>
<td style="text-align: right;">14</td>
<td style="text-align: left;">5.4%</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T-cells</td>
<td style="text-align: right;">5</td>
<td style="text-align: left;">1.9%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Chondrocytes</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.8%</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.4%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Endothelial cells</td>
<td style="text-align: right;">26</td>
<td style="text-align: left;">10.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Eosinophils</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.4%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Epithelial cells</td>
<td style="text-align: right;">18</td>
<td style="text-align: left;">6.9%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Erythrocytes</td>
<td style="text-align: right;">7</td>
<td style="text-align: left;">2.7%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Fibroblasts</td>
<td style="text-align: right;">20</td>
<td style="text-align: left;">7.7%</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSC</td>
<td style="text-align: right;">38</td>
<td style="text-align: left;">14.7%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Keratinocytes</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.8%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Macrophages</td>
<td style="text-align: right;">25</td>
<td style="text-align: left;">9.7%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Melanocytes</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">1.5%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mesangial cells</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.8%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: right;">16</td>
<td style="text-align: left;">6.2%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Myocytes</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">1.5%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neurons</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">1.5%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: right;">23</td>
<td style="text-align: left;">8.9%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">1.2%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pericytes</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.8%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Skeletal muscle</td>
<td style="text-align: right;">7</td>
<td style="text-align: left;">2.7%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Smooth muscle</td>
<td style="text-align: right;">16</td>
<td style="text-align: left;">6.2%</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["label.main"],"name":[1],"type":["chr"],"align":["left"]},{"label":["label.fine"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"Adipocytes","2":"Adipocytes","3":"7"},{"1":"Adipocytes","2":"Astrocytes","3":"2"},{"1":"Adipocytes","2":"Preadipocytes","3":"2"},{"1":"B-cells","2":"Class-switched memory B-cells","3":"1"},{"1":"B-cells","2":"Memory B-cells","3":"1"},{"1":"B-cells","2":"naive B-cells","3":"2"},{"1":"B-cells","2":"Plasma cells","3":"4"},{"1":"CD4+ T-cells","2":"CD4+ T-cells","3":"11"},{"1":"CD4+ T-cells","2":"CD4+ Tcm","3":"1"},{"1":"CD4+ T-cells","2":"CD4+ Tem","3":"1"},{"1":"CD4+ T-cells","2":"Tregs","3":"1"},{"1":"CD8+ T-cells","2":"CD8+ T-cells","3":"3"},{"1":"CD8+ T-cells","2":"CD8+ Tcm","3":"1"},{"1":"CD8+ T-cells","2":"CD8+ Tem","3":"1"},{"1":"Chondrocytes","2":"Chondrocytes","3":"2"},{"1":"DC","2":"DC","3":"1"},{"1":"Endothelial cells","2":"Endothelial cells","3":"18"},{"1":"Endothelial cells","2":"mv Endothelial cells","3":"8"},{"1":"Eosinophils","2":"Eosinophils","3":"1"},{"1":"Epithelial cells","2":"Epithelial cells","3":"18"},{"1":"Erythrocytes","2":"Erythrocytes","3":"7"},{"1":"Fibroblasts","2":"Fibroblasts","3":"20"},{"1":"HSC","2":"CLP","3":"5"},{"1":"HSC","2":"CMP","3":"11"},{"1":"HSC","2":"GMP","3":"3"},{"1":"HSC","2":"HSC","3":"6"},{"1":"HSC","2":"Megakaryocytes","3":"5"},{"1":"HSC","2":"MEP","3":"4"},{"1":"HSC","2":"MPP","3":"4"},{"1":"Keratinocytes","2":"Keratinocytes","3":"2"},{"1":"Macrophages","2":"Macrophages","3":"18"},{"1":"Macrophages","2":"Macrophages M1","3":"3"},{"1":"Macrophages","2":"Macrophages M2","3":"4"},{"1":"Melanocytes","2":"Melanocytes","3":"4"},{"1":"Mesangial cells","2":"Mesangial cells","3":"2"},{"1":"Monocytes","2":"Monocytes","3":"16"},{"1":"Myocytes","2":"Myocytes","3":"4"},{"1":"Neurons","2":"Neurons","3":"4"},{"1":"Neutrophils","2":"Neutrophils","3":"23"},{"1":"NK cells","2":"NK cells","3":"3"},{"1":"Pericytes","2":"Pericytes","3":"2"},{"1":"Skeletal muscle","2":"Skeletal muscle","3":"7"},{"1":"Smooth muscle","2":"Smooth muscle","3":"16"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<h3 id="main-label-1">Main label</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Annotating against the main labels, we obtain the following cell type labels.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-34">Table 7: </span>Cell label assignments using the main labels of the <code>BlueprintEncodeData</code> reference data.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">CD8+ T-cells</td>
<td style="text-align: right;">5269</td>
<td style="text-align: left;">53.3%</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T-cells</td>
<td style="text-align: right;">4590</td>
<td style="text-align: left;">46.5%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: right;">10</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">B-cells</td>
<td style="text-align: right;">7</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DC</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
</tbody>
</table>
</div>
<p>Heatmaps of the normalized scores are shown, with and without clustering by these normalized scores, in Figures <a href="#fig:heat-bed-main-clustered">18</a> and <a href="#fig:heat-bed-main">19</a>.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-bed-main-clustered"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-bed-main-clustered-1.png" alt="Heatmap of normalized scores using the main labels from the `BlueprintEncodeData` reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores." width="624" />
<p class="caption">
Figure 18: Heatmap of normalized scores using the main labels from the <code>BlueprintEncodeData</code> reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-bed-main"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-bed-main-1.png" alt="Heatmap of normalized scores using the main labels from the `BlueprintEncodeData` reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels." width="624" />
<p class="caption">
Figure 19: Heatmap of normalized scores using the main labels from the <code>BlueprintEncodeData</code> reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels.
</p>
</div>
</div>
<p>Figure <a href="#fig:tsne-bed-main">20</a> overlays the normalized scores for the 10 labels with the largest maximum scores on the t-SNE plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:tsne-bed-main"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/tsne-bed-main-1.png" alt="t-SNE overlaid with normalized scores using the main labels from the `BlueprintEncodeData` reference set. Only the 10 labels with the largest maximum scores are shown." width="624" />
<p class="caption">
Figure 20: t-SNE overlaid with normalized scores using the main labels from the <code>BlueprintEncodeData</code> reference set. Only the 10 labels with the largest maximum scores are shown.
</p>
</div>
</div>
<h3 id="fine-label-1">Fine label</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Annotating against the fine labels, we obtain the following cell type labels.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-36">Table 8: </span>Cell label assignments using the fine labels of the <code>BlueprintEncodeData</code> reference data.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">CD4+ Tem</td>
<td style="text-align: right;">3217</td>
<td style="text-align: left;">32.6%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tregs</td>
<td style="text-align: right;">2786</td>
<td style="text-align: left;">28.2%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ Tcm</td>
<td style="text-align: right;">1586</td>
<td style="text-align: left;">16.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ Tcm</td>
<td style="text-align: right;">1469</td>
<td style="text-align: left;">14.9%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T-cells</td>
<td style="text-align: right;">392</td>
<td style="text-align: left;">4.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ Tem</td>
<td style="text-align: right;">392</td>
<td style="text-align: left;">4.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8+ T-cells</td>
<td style="text-align: right;">21</td>
<td style="text-align: left;">0.2%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">naive B-cells</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">DC</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Memory B-cells</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
</tbody>
</table>
</div>
<p>Heatmaps of the normalized scores are shown, with and without clustering by these normalized scores, in Figures <a href="#fig:heat-bed-fine-clustered">21</a> and <a href="#fig:heat-bed-fine">22</a>.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-bed-fine-clustered"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-bed-fine-clustered-1.png" alt="Heatmap of normalized scores using the fine labels from the `BlueprintEncodeData` reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores." width="624" />
<p class="caption">
Figure 21: Heatmap of normalized scores using the fine labels from the <code>BlueprintEncodeData</code> reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-bed-fine"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-bed-fine-1.png" alt="Heatmap of normalized scores using the fine labels from the `BlueprintEncodeData` reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels." width="624" />
<p class="caption">
Figure 22: Heatmap of normalized scores using the fine labels from the <code>BlueprintEncodeData</code> reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels.
</p>
</div>
</div>
<p>Figure <a href="#fig:tsne-bed-fine">23</a> overlays the normalized scores for the 10 labels with the largest maximum scores on the t-SNE plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:tsne-bed-fine"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/tsne-bed-fine-1.png" alt="t-SNE overlaid with normalized scores using the fine labels from the `BlueprintEncodeData` reference set. Only the 10 labels with the largest maximum scores are shown." width="624" />
<p class="caption">
Figure 23: t-SNE overlaid with normalized scores using the fine labels from the <code>BlueprintEncodeData</code> reference set. Only the 10 labels with the largest maximum scores are shown.
</p>
</div>
</div>
<h2 id="databaseimmunecellexpressiondata"><code>DatabaseImmuneCellExpressionData</code></h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The <code>DatabaseImmuneCellExpressionData</code> reference set entails 1561 bulk RNA-seq samples of sorted cell populations from the DICE database. Each sample has a ‘main’ and a ‘fine’ cell type label, summarised below.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-38">Table 9: </span>Summary of main cell types in the <code>DatabaseImmuneCellExpressionData</code> reference set.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: right;">106</td>
<td style="text-align: left;">6.8%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: right;">211</td>
<td style="text-align: left;">13.5%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">105</td>
<td style="text-align: left;">6.7%</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: right;">933</td>
<td style="text-align: left;">59.8%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD8+</td>
<td style="text-align: right;">206</td>
<td style="text-align: left;">13.2%</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["label.main"],"name":[1],"type":["chr"],"align":["left"]},{"label":["label.fine"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"B cells","2":"B cells, naive","3":"106"},{"1":"Monocytes","2":"Monocytes, CD14+","3":"106"},{"1":"Monocytes","2":"Monocytes, CD16+","3":"105"},{"1":"NK cells","2":"NK cells","3":"105"},{"1":"T cells, CD4+","2":"T cells, CD4+, memory TREG","3":"104"},{"1":"T cells, CD4+","2":"T cells, CD4+, naive","3":"103"},{"1":"T cells, CD4+","2":"T cells, CD4+, naive TREG","3":"104"},{"1":"T cells, CD4+","2":"T cells, CD4+, naive, stimulated","3":"102"},{"1":"T cells, CD4+","2":"T cells, CD4+, TFH","3":"104"},{"1":"T cells, CD4+","2":"T cells, CD4+, Th1","3":"104"},{"1":"T cells, CD4+","2":"T cells, CD4+, Th1_17","3":"104"},{"1":"T cells, CD4+","2":"T cells, CD4+, Th17","3":"104"},{"1":"T cells, CD4+","2":"T cells, CD4+, Th2","3":"104"},{"1":"T cells, CD8+","2":"T cells, CD8+, naive","3":"104"},{"1":"T cells, CD8+","2":"T cells, CD8+, naive, stimulated","3":"102"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<h3 id="main-label-2">Main label</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Annotating against the main labels, we obtain the following cell type labels.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-41">Table 10: </span>Cell label assignments using the main labels of the <code>DatabaseImmuneCellExpressionData</code> reference data.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+</td>
<td style="text-align: right;">9756</td>
<td style="text-align: left;">98.8%</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">97</td>
<td style="text-align: left;">1.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: right;">9</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD8+</td>
<td style="text-align: right;">9</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: right;">7</td>
<td style="text-align: left;">0.1%</td>
</tr>
</tbody>
</table>
</div>
<p>Heatmaps of the normalized scores are shown, with and without clustering by these normalized scores, in Figures <a href="#fig:heat-diced-main-clustered">24</a> and <a href="#fig:heat-diced-main">25</a>.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-diced-main-clustered"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-diced-main-clustered-1.png" alt="Heatmap of normalized scores using the main labels from the `DatabaseImmuneCellExpressionData` reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores." width="624" />
<p class="caption">
Figure 24: Heatmap of normalized scores using the main labels from the <code>DatabaseImmuneCellExpressionData</code> reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-diced-main"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-diced-main-1.png" alt="Heatmap of normalized scores using the main labels from the `DatabaseImmuneCellExpressionData` reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels." width="624" />
<p class="caption">
Figure 25: Heatmap of normalized scores using the main labels from the <code>DatabaseImmuneCellExpressionData</code> reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels.
</p>
</div>
</div>
<p>Figure <a href="#fig:tsne-diced-main">26</a> overlays the normalized scores for the 10 labels with the largest maximum scores on the t-SNE plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:tsne-diced-main"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/tsne-diced-main-1.png" alt="t-SNE overlaid with normalized scores using the main labels from the `DatabaseImmuneCellExpressionData` reference set. Only the 10 labels with the largest maximum scores are shown." width="624" />
<p class="caption">
Figure 26: t-SNE overlaid with normalized scores using the main labels from the <code>DatabaseImmuneCellExpressionData</code> reference set. Only the 10 labels with the largest maximum scores are shown.
</p>
</div>
</div>
<h3 id="fine-label-2">Fine label</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Annotating against the fine labels, we obtain the following cell type labels.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-43">Table 11: </span>Cell label assignments using the fine labels of the <code>DatabaseImmuneCellExpressionData</code> reference data.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, memory TREG</td>
<td style="text-align: right;">5031</td>
<td style="text-align: left;">50.9%</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, Th1</td>
<td style="text-align: right;">1509</td>
<td style="text-align: left;">15.3%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, Th2</td>
<td style="text-align: right;">1309</td>
<td style="text-align: left;">13.3%</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, Th17</td>
<td style="text-align: right;">1226</td>
<td style="text-align: left;">12.4%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, TFH</td>
<td style="text-align: right;">472</td>
<td style="text-align: left;">4.8%</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, Th1_17</td>
<td style="text-align: right;">211</td>
<td style="text-align: left;">2.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">54</td>
<td style="text-align: left;">0.5%</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD4+, naive, stimulated</td>
<td style="text-align: right;">20</td>
<td style="text-align: left;">0.2%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, naive</td>
<td style="text-align: right;">19</td>
<td style="text-align: left;">0.2%</td>
</tr>
<tr class="even">
<td style="text-align: left;">B cells, naive</td>
<td style="text-align: right;">7</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD4+, naive TREG</td>
<td style="text-align: right;">7</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes, CD16+</td>
<td style="text-align: right;">5</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells, CD8+, naive</td>
<td style="text-align: right;">5</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">T cells, CD8+, naive, stimulated</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes, CD14+</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
</tbody>
</table>
</div>
<p>Heatmaps of the normalized scores are shown, with and without clustering by these normalized scores, in Figures <a href="#fig:heat-diced-fine-clustered">27</a> and <a href="#fig:heat-diced-fine">28</a>.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-diced-fine-clustered"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-diced-fine-clustered-1.png" alt="Heatmap of normalized scores using the fine labels from the `DatabaseImmuneCellExpressionData` reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores." width="624" />
<p class="caption">
Figure 27: Heatmap of normalized scores using the fine labels from the <code>DatabaseImmuneCellExpressionData</code> reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-diced-fine"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-diced-fine-1.png" alt="Heatmap of normalized scores using the fine labels from the `DatabaseImmuneCellExpressionData` reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels." width="624" />
<p class="caption">
Figure 28: Heatmap of normalized scores using the fine labels from the <code>DatabaseImmuneCellExpressionData</code> reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels.
</p>
</div>
</div>
<p>Figure <a href="#fig:tsne-diced-fine">29</a> overlays the normalized scores for the 10 labels with the largest maximum scores on the t-SNE plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:tsne-diced-fine"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/tsne-diced-fine-1.png" alt="t-SNE overlaid with normalized scores using the fine labels from the `DatabaseImmuneCellExpressionData` reference set. Only the 10 labels with the largest maximum scores are shown." width="624" />
<p class="caption">
Figure 29: t-SNE overlaid with normalized scores using the fine labels from the <code>DatabaseImmuneCellExpressionData</code> reference set. Only the 10 labels with the largest maximum scores are shown.
</p>
</div>
</div>
<h2 id="monacoimmunedata"><code>MonacoImmuneData</code></h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The <code>MonacoImmuneData</code> reference set entails 114 bulk RNA-seq samples of sorted immune cell populations that can be found in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107011">GSE107011</a>. Each sample has a ‘main’ and a ‘fine’ cell type label, summarised below.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-45">Table 12: </span>Summary of main cell types in the <code>MonacoImmuneData</code> reference set.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: right;">16</td>
<td style="text-align: left;">14.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Basophils</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">3.5%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: right;">30</td>
<td style="text-align: left;">26.3%</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: right;">16</td>
<td style="text-align: left;">14.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Dendritic cells</td>
<td style="text-align: right;">8</td>
<td style="text-align: left;">7.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: right;">12</td>
<td style="text-align: left;">10.5%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Neutrophils</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">3.5%</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">3.5%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Plasmablasts</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">3.5%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Progenitors</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">3.5%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">T cells</td>
<td style="text-align: right;">12</td>
<td style="text-align: left;">10.5%</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["label.main"],"name":[1],"type":["chr"],"align":["left"]},{"label":["label.fine"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"B cells","2":"Bcell.Exhausted","3":"4"},{"1":"B cells","2":"Bcell.naive","3":"4"},{"1":"B cells","2":"Bcell.NonSwitchedMemory","3":"4"},{"1":"B cells","2":"Bcell.SwitchedMemory","3":"4"},{"1":"Basophils","2":"Basophil","3":"4"},{"1":"CD4+ T cells","2":"Tcell.CD4.naive","3":"4"},{"1":"CD4+ T cells","2":"Tcell.CD4.TerminalEffector","3":"2"},{"1":"CD4+ T cells","2":"Tcell.FollicularHelper","3":"4"},{"1":"CD4+ T cells","2":"Tcell.Th1","3":"4"},{"1":"CD4+ T cells","2":"Tcell.Th1.Th17","3":"4"},{"1":"CD4+ T cells","2":"Tcell.Th17","3":"4"},{"1":"CD4+ T cells","2":"Tcell.Th2","3":"4"},{"1":"CD4+ T cells","2":"Tcell.Treg","3":"4"},{"1":"CD8+ T cells","2":"Tcell.CD8.CentralMemory","3":"4"},{"1":"CD8+ T cells","2":"Tcell.CD8.EffectorMemory","3":"4"},{"1":"CD8+ T cells","2":"Tcell.CD8.naive","3":"4"},{"1":"CD8+ T cells","2":"Tcell.CD8.TerminalEffector","3":"4"},{"1":"Dendritic cells","2":"Dendritic.Myeloid","3":"4"},{"1":"Dendritic cells","2":"Dendritic.Plasmacytoid","3":"4"},{"1":"Monocytes","2":"Monocyte.Classical","3":"4"},{"1":"Monocytes","2":"Monocyte.Int","3":"4"},{"1":"Monocytes","2":"Monocyte.NonClassical","3":"4"},{"1":"Neutrophils","2":"Neutrophil","3":"4"},{"1":"NK cells","2":"NK","3":"4"},{"1":"Plasmablasts","2":"Plasmablast","3":"4"},{"1":"Progenitors","2":"Progenitor","3":"4"},{"1":"T cells","2":"Tcell.MAIT","3":"4"},{"1":"T cells","2":"Tcell.VD2neg.gd","3":"4"},{"1":"T cells","2":"Tcell.VD2pos.gd","3":"4"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<h3 id="main-label-3">Main label</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Annotating against the main labels, we obtain the following cell type labels.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-48">Table 13: </span>Cell label assignments using the main labels of the <code>MonacoImmuneData</code> reference data.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">T cells</td>
<td style="text-align: right;">5044</td>
<td style="text-align: left;">51.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: right;">4151</td>
<td style="text-align: left;">42.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: right;">656</td>
<td style="text-align: left;">6.6%</td>
</tr>
<tr class="even">
<td style="text-align: left;">B cells</td>
<td style="text-align: right;">10</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: right;">7</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">7</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Progenitors</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Dendritic cells</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
</tbody>
</table>
</div>
<p>Heatmaps of the normalized scores are shown, with and without clustering by these normalized scores, in Figures <a href="#fig:heat-mid-main-clustered">30</a> and <a href="#fig:heat-mid-main">31</a>.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-mid-main-clustered"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-mid-main-clustered-1.png" alt="Heatmap of normalized scores using the main labels from the `MonacoImmuneData` reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores." width="624" />
<p class="caption">
Figure 30: Heatmap of normalized scores using the main labels from the <code>MonacoImmuneData</code> reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-mid-main"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-mid-main-1.png" alt="Heatmap of normalized scores using the main labels from the `MonacoImmuneData` reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels." width="624" />
<p class="caption">
Figure 31: Heatmap of normalized scores using the main labels from the <code>MonacoImmuneData</code> reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels.
</p>
</div>
</div>
<p>Figure <a href="#fig:tsne-mid-main">32</a> overlays the normalized scores for the 10 labels with the largest maximum scores on the t-SNE plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:tsne-mid-main"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/tsne-mid-main-1.png" alt="t-SNE overlaid with normalized scores using the main labels from the `MonacoImmuneData` reference set. Only the 10 labels with the largest maximum scores are shown." width="624" />
<p class="caption">
Figure 32: t-SNE overlaid with normalized scores using the main labels from the <code>MonacoImmuneData</code> reference set. Only the 10 labels with the largest maximum scores are shown.
</p>
</div>
</div>
<h3 id="fine-label-3">Fine label</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Annotating against the fine labels, we obtain the following cell type labels.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-50">Table 14: </span>Cell label assignments using the fine labels of the <code>MonacoImmuneData</code> reference data.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Tcell.Th1</td>
<td style="text-align: right;">1823</td>
<td style="text-align: left;">18.5%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tcell.CD8.EffectorMemory</td>
<td style="text-align: right;">1568</td>
<td style="text-align: left;">15.9%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tcell.Th2</td>
<td style="text-align: right;">1461</td>
<td style="text-align: left;">14.8%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tcell.Treg</td>
<td style="text-align: right;">1358</td>
<td style="text-align: left;">13.7%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tcell.Th17</td>
<td style="text-align: right;">1042</td>
<td style="text-align: left;">10.5%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tcell.VD2pos.gd</td>
<td style="text-align: right;">840</td>
<td style="text-align: left;">8.5%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tcell.Th1.Th17</td>
<td style="text-align: right;">636</td>
<td style="text-align: left;">6.4%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tcell.MAIT</td>
<td style="text-align: right;">518</td>
<td style="text-align: left;">5.2%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tcell.CD8.CentralMemory</td>
<td style="text-align: right;">253</td>
<td style="text-align: left;">2.6%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tcell.FollicularHelper</td>
<td style="text-align: right;">105</td>
<td style="text-align: left;">1.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tcell.VD2neg.gd</td>
<td style="text-align: right;">100</td>
<td style="text-align: left;">1.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tcell.CD4.TerminalEffector</td>
<td style="text-align: right;">84</td>
<td style="text-align: left;">0.9%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tcell.CD4.naive</td>
<td style="text-align: right;">37</td>
<td style="text-align: left;">0.4%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tcell.CD8.TerminalEffector</td>
<td style="text-align: right;">25</td>
<td style="text-align: left;">0.3%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK</td>
<td style="text-align: right;">7</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocyte.Int</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Bcell.naive</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Bcell.Exhausted</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Bcell.NonSwitchedMemory</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Progenitor</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Bcell.SwitchedMemory</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Dendritic.Myeloid</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocyte.NonClassical</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tcell.CD8.naive</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
</tbody>
</table>
</div>
<p>Heatmaps of the normalized scores are shown, with and without clustering by these normalized scores, in Figures <a href="#fig:heat-mid-fine-clustered">33</a> and <a href="#fig:heat-mid-fine">34</a>.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-mid-fine-clustered"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-mid-fine-clustered-1.png" alt="Heatmap of normalized scores using the fine labels from the `MonacoImmuneData` reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores." width="624" />
<p class="caption">
Figure 33: Heatmap of normalized scores using the fine labels from the <code>MonacoImmuneData</code> reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-mid-fine"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-mid-fine-1.png" alt="Heatmap of normalized scores using the fine labels from the `MonacoImmuneData` reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels." width="624" />
<p class="caption">
Figure 34: Heatmap of normalized scores using the fine labels from the <code>MonacoImmuneData</code> reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels.
</p>
</div>
</div>
<p>Figure <a href="#fig:tsne-mid-fine">35</a> overlays the normalized scores for the 10 labels with the largest maximum scores on the t-SNE plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:tsne-mid-fine"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/tsne-mid-fine-1.png" alt="t-SNE overlaid with normalized scores using the fine labels from the `MonacoImmuneData` reference set. Only the 10 labels with the largest maximum scores are shown." width="624" />
<p class="caption">
Figure 35: t-SNE overlaid with normalized scores using the fine labels from the <code>MonacoImmuneData</code> reference set. Only the 10 labels with the largest maximum scores are shown.
</p>
</div>
</div>
<h2 id="novershternhematopoieticdata"><code>NovershternHematopoieticData</code></h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The <code>NovershternHematopoieticData</code> reference set entails 211 bulk human microarray samples of sorted hematopoietic cell populations that can be found in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE24759">GSE24759</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-52">Table 15: </span>Summary of main cell types in the <code>NovershternHematopoieticData</code> reference set.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: right;">29</td>
<td style="text-align: left;">13.7%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Basophils</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">2.8%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: right;">21</td>
<td style="text-align: left;">10.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: right;">24</td>
<td style="text-align: left;">11.4%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CMPs</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">1.9%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Dendritic cells</td>
<td style="text-align: right;">10</td>
<td style="text-align: left;">4.7%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Eosinophils</td>
<td style="text-align: right;">5</td>
<td style="text-align: left;">2.4%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Erythroid cells</td>
<td style="text-align: right;">33</td>
<td style="text-align: left;">15.6%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GMPs</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">1.9%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Granulocytes</td>
<td style="text-align: right;">13</td>
<td style="text-align: left;">6.2%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSCs</td>
<td style="text-align: right;">14</td>
<td style="text-align: left;">6.6%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Megakaryocytes</td>
<td style="text-align: right;">12</td>
<td style="text-align: left;">5.7%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">MEPs</td>
<td style="text-align: right;">9</td>
<td style="text-align: left;">4.3%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: right;">9</td>
<td style="text-align: left;">4.3%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">14</td>
<td style="text-align: left;">6.6%</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK T cells</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">1.9%</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["label.main"],"name":[1],"type":["chr"],"align":["left"]},{"label":["label.fine"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"B cells","2":"Early B-cells","3":"4"},{"1":"B cells","2":"Mature B-cells","3":"5"},{"1":"B cells","2":"Mature B-cells class able to switch","3":"5"},{"1":"B cells","2":"Mature B-cells class switched","3":"5"},{"1":"B cells","2":"Naïve B cells","3":"5"},{"1":"B cells","2":"Pro B-cells","3":"5"},{"1":"Basophils","2":"Basophils","3":"6"},{"1":"CD4+ T cells","2":"CD4+ Central Memory","3":"7"},{"1":"CD4+ T cells","2":"CD4+ Effector Memory","3":"7"},{"1":"CD4+ T cells","2":"Naive CD4+ T-cells","3":"7"},{"1":"CD8+ T cells","2":"CD8+ Central Memory","3":"7"},{"1":"CD8+ T cells","2":"CD8+ Effector Memory","3":"6"},{"1":"CD8+ T cells","2":"CD8+ Effector Memory RA","3":"4"},{"1":"CD8+ T cells","2":"Naive CD8+ T-cells","3":"7"},{"1":"CMPs","2":"Common myeloid progenitors","3":"4"},{"1":"Dendritic cells","2":"Myeloid Dendritic Cells","3":"5"},{"1":"Dendritic cells","2":"Plasmacytoid Dendritic Cells","3":"5"},{"1":"Eosinophils","2":"Eosinophils","3":"5"},{"1":"Erythroid cells","2":"Erythroid_CD34- CD71- GlyA+","3":"6"},{"1":"Erythroid cells","2":"Erythroid_CD34- CD71+ GlyA-","3":"7"},{"1":"Erythroid cells","2":"Erythroid_CD34- CD71+ GlyA+","3":"6"},{"1":"Erythroid cells","2":"Erythroid_CD34- CD71lo GlyA+","3":"7"},{"1":"Erythroid cells","2":"Erythroid_CD34+ CD71+ GlyA-","3":"7"},{"1":"GMPs","2":"Granulocyte/monocyte progenitors","3":"4"},{"1":"Granulocytes","2":"Colony Forming Unit-Granulocytes","3":"5"},{"1":"Granulocytes","2":"Granulocyte (Neutrophilic Metamyelocytes)","3":"4"},{"1":"Granulocytes","2":"Granulocyte (Neutrophils)","3":"4"},{"1":"HSCs","2":"Hematopoietic stem cell_CD38- CD34+","3":"4"},{"1":"HSCs","2":"Hematopoietic stem cells_CD133+ CD34dim","3":"10"},{"1":"Megakaryocytes","2":"Colony Forming Unit-Megakaryocytic","3":"5"},{"1":"Megakaryocytes","2":"Megakaryocytes","3":"7"},{"1":"MEPs","2":"Megakaryocyte/erythroid progenitors","3":"9"},{"1":"Monocytes","2":"Colony Forming Unit-Monocytes","3":"4"},{"1":"Monocytes","2":"Monocytes","3":"5"},{"1":"NK cells","2":"Mature NK cells_CD56- CD16- CD3-","3":"5"},{"1":"NK cells","2":"Mature NK cells_CD56- CD16+ CD3-","3":"4"},{"1":"NK cells","2":"Mature NK cells_CD56+ CD16+ CD3-","3":"5"},{"1":"NK T cells","2":"NK T cells","3":"4"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<h3 id="main-label-4">Main label</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Annotating against the main labels, we obtain the following cell type labels.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-55">Table 16: </span>Cell label assignments using the main labels of the <code>NovershternHematopoieticData</code> reference data.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.main</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">CD4+ T cells</td>
<td style="text-align: right;">6598</td>
<td style="text-align: left;">66.8%</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ T cells</td>
<td style="text-align: right;">3238</td>
<td style="text-align: left;">32.8%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NK T cells</td>
<td style="text-align: right;">24</td>
<td style="text-align: left;">0.2%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Granulocytes</td>
<td style="text-align: right;">9</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B cells</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Basophils</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Dendritic cells</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">NK cells</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
</tbody>
</table>
</div>
<p>Heatmaps of the normalized scores are shown, with and without clustering by these normalized scores, in Figures <a href="#fig:heat-nhd-main-clustered">36</a> and <a href="#fig:heat-nhd-main">37</a>.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-nhd-main-clustered"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-nhd-main-clustered-1.png" alt="Heatmap of normalized scores using the main labels from the `NovershternHematopoieticData` reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores." width="624" />
<p class="caption">
Figure 36: Heatmap of normalized scores using the main labels from the <code>NovershternHematopoieticData</code> reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-nhd-main"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-nhd-main-1.png" alt="Heatmap of normalized scores using the main labels from the `NovershternHematopoieticData` reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels." width="624" />
<p class="caption">
Figure 37: Heatmap of normalized scores using the main labels from the <code>NovershternHematopoieticData</code> reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels.
</p>
</div>
</div>
<p>Figure <a href="#fig:tsne-nhd-main">38</a> overlays the normalized scores for the 10 labels with the largest maximum scores on the t-SNE plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:tsne-nhd-main"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/tsne-nhd-main-1.png" alt="t-SNE overlaid with normalized scores using the main labels from the `NovershternHematopoieticData` reference set. Only the 10 labels with the largest maximum scores are shown." width="624" />
<p class="caption">
Figure 38: t-SNE overlaid with normalized scores using the main labels from the <code>NovershternHematopoieticData</code> reference set. Only the 10 labels with the largest maximum scores are shown.
</p>
</div>
</div>
<h3 id="fine-label-4">FIne label</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Annotating against the fine labels, we obtain the following cell type labels.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-57">Table 17: </span>Cell label assignments using the fine labels of the <code>NovershternHematopoieticData</code> reference data.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label.fine</th>
<th style="text-align: right;">n</th>
<th style="text-align: left;">percent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">CD4+ Effector Memory</td>
<td style="text-align: right;">6318</td>
<td style="text-align: left;">64.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ Effector Memory</td>
<td style="text-align: right;">1617</td>
<td style="text-align: left;">16.4%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD4+ Central Memory</td>
<td style="text-align: right;">1615</td>
<td style="text-align: left;">16.3%</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD8+ Central Memory</td>
<td style="text-align: right;">208</td>
<td style="text-align: left;">2.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8+ Effector Memory RA</td>
<td style="text-align: right;">74</td>
<td style="text-align: left;">0.7%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Naive CD4+ T-cells</td>
<td style="text-align: right;">18</td>
<td style="text-align: left;">0.2%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Colony Forming Unit-Granulocytes</td>
<td style="text-align: right;">8</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Naive CD8+ T-cells</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">0.1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Naïve B cells</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Granulocyte (Neutrophils)</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Monocytes</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Granulocyte (Neutrophilic Metamyelocytes)</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mature B-cells class able to switch</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mature NK cells_CD56- CD16- CD3-</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Myeloid Dendritic Cells</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pro B-cells</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">0.0%</td>
</tr>
</tbody>
</table>
</div>
<p>Heatmaps of the normalized scores are shown, with and without clustering by these normalized scores, in Figures <a href="#fig:heat-nhd-fine-clustered">39</a> and <a href="#fig:heat-nhd-fine">40</a>.</p>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-nhd-fine-clustered"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-nhd-fine-clustered-1.png" alt="Heatmap of normalized scores using the fine labels from the `NovershternHematopoieticData` reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores." width="624" />
<p class="caption">
Figure 39: Heatmap of normalized scores using the fine labels from the <code>NovershternHematopoieticData</code> reference set. Cells are columns, rows are labels. Columns are ordered by clustering the normalized scores.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body-outset">
<div class="figure"><span id="fig:heat-nhd-fine"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/heat-nhd-fine-1.png" alt="Heatmap of normalized scores using the fine labels from the `NovershternHematopoieticData` reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels." width="624" />
<p class="caption">
Figure 40: Heatmap of normalized scores using the fine labels from the <code>NovershternHematopoieticData</code> reference set. Cells are columns, rows are labels. Columns are ordered by and annotated by the column labels.
</p>
</div>
</div>
<p>Figure <a href="#fig:tsne-nhd-fine">41</a> overlays the normalized scores for the 10 labels with the largest maximum scores on the t-SNE plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:tsne-nhd-fine"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/tsne-nhd-fine-1.png" alt="t-SNE overlaid with normalized scores using the fine labels from the `NovershternHematopoieticData` reference set. Only the 10 labels with the largest maximum scores are shown." width="624" />
<p class="caption">
Figure 41: t-SNE overlaid with normalized scores using the fine labels from the <code>NovershternHematopoieticData</code> reference set. Only the 10 labels with the largest maximum scores are shown.
</p>
</div>
</div>
<h2 id="summary-4">Summary</h2>
<p>Any automated cell type annotation method is only as good as its reference set. Any cell type present in the test set but with no good match in the reference set will be misannotated to some degree. We will need to consider the cell types available in each reference set and the appropriateness of annotating embroyonic cells using reference sets that likely mostly comprise adult cells.</p>
<p><strong>This means that you will need to assess the appropriateness of the reference sets for your data set.</strong></p>
<h1 id="cell-cycle-assignment">Cell cycle assignment</h1>
<p>On occasion, it can be desirable to determine cell cycle activity from scRNA-seq data. In and of itself, the distribution of cells across phases of the cell cycle is not usually informative, but we can use this to determine if there are differences in cell cycle progression between subpopulations or across treatment conditions. Many of the key events in the cell cycle (e.g., passage through checkpoints) are post-translational and thus not directly visible in transcriptomic data; nonetheless, there are enough changes in expression that can be exploited to determine cell cycle phase.</p>
<h2 id="using-the-cyclins">Using the cyclins</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The cyclins control progression through the cell cycle and have well-characterized patterns of expression across cell cycle phases. The cyclins available in this dataset are as follows: CCNA2, CCNB1, CCND3, CCNB3, CCNE2, CCND1, CCND2, CCNA1, CCNB2, and CCNE1. Cyclin D is expressed throughout but peaks at G1; cyclin E is expressed highest in the G1/S transition; cyclin A is expressed across S and G2; and cyclin B is expressed highest in late G2 and mitosis. Inspection of the relative expression of cyclins across the population can often be sufficient to determine the relative cell cycle activity in each cluster.</p>
<p>For example, Figure <a href="#fig:cyclin-heatmap">42</a> shows that cluster 7 may be in G2/M phase while the other clusters are scattered across the later phases.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:cyclin-heatmap"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/cyclin-heatmap-1.png" alt="Heatmap of the log-normalized expression values of the cyclin genes in the dataset." width="624" />
<p class="caption">
Figure 42: Heatmap of the log-normalized expression values of the cyclin genes in the dataset.
</p>
</div>
</div>
<p>We can also use standard differential expression methods to look for upregulation of each cyclin, allowing us to determine if there are more cells in the corresponding phase of the cell cycle between subpopulations.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Here we use pairwise Wilcoxon rank sum tests to compute differential expression of cycline genes between each pair of clusters. The Wilcoxon test estimates effect sizes as overlap proportions (<code>overlap</code> in the tables below). Theoverlap proportion is defined as the probability that a randomly selected cell in <span class="math inline">\(A\)</span> has a greater expression value of <span class="math inline">\(X\)</span> than a randomly selected cell in <span class="math inline">\(B\)</span>. Overlap proportions near 0 (<span class="math inline">\(A\)</span> is lower than <span class="math inline">\(B\)</span>) or 1 (<span class="math inline">\(A\)</span> is higher than <span class="math inline">\(B\)</span>) indicate that the expression distributions are well-separated. The Wilcoxon rank sum test effectively tests for significant deviations from an overlap proportion of 0.5.</p>
<p>We can infer that cluster 10 likely has more cells in G2/M than the other clusters, based on higher expression of the cyclin B’s.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-60">Table 18: </span>Cyclin DE analysis comparing cluster 10 to all other clusters.</caption>
<thead>
<tr class="header">
<th></th>
<th style="text-align: right;">Top</th>
<th style="text-align: right;">p.value</th>
<th style="text-align: right;">FDR</th>
<th style="text-align: right;">overlap.1</th>
<th style="text-align: right;">overlap.2</th>
<th style="text-align: right;">overlap.3</th>
<th style="text-align: right;">overlap.4</th>
<th style="text-align: right;">overlap.5</th>
<th style="text-align: right;">overlap.6</th>
<th style="text-align: right;">overlap.7</th>
<th style="text-align: right;">overlap.8</th>
<th style="text-align: right;">overlap.9</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CCNA2</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.7852696</td>
<td style="text-align: right;">0.8400491</td>
<td style="text-align: right;">0.8381667</td>
<td style="text-align: right;">0.8370000</td>
<td style="text-align: right;">0.8510879</td>
<td style="text-align: right;">0.8561935</td>
<td style="text-align: right;">0.8349568</td>
<td style="text-align: right;">0.8495877</td>
<td style="text-align: right;">0.8455853</td>
</tr>
<tr class="even">
<td>CCNB2</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.8010589</td>
<td style="text-align: right;">0.8433536</td>
<td style="text-align: right;">0.8430028</td>
<td style="text-align: right;">0.8304018</td>
<td style="text-align: right;">0.8414312</td>
<td style="text-align: right;">0.8325311</td>
<td style="text-align: right;">0.8296425</td>
<td style="text-align: right;">0.8429903</td>
<td style="text-align: right;">0.8362781</td>
</tr>
<tr class="odd">
<td>CCNB1</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.7590119</td>
<td style="text-align: right;">0.7955155</td>
<td style="text-align: right;">0.7896423</td>
<td style="text-align: right;">0.7917150</td>
<td style="text-align: right;">0.7870604</td>
<td style="text-align: right;">0.7768951</td>
<td style="text-align: right;">0.7860920</td>
<td style="text-align: right;">0.7877734</td>
<td style="text-align: right;">0.7899288</td>
</tr>
<tr class="even">
<td>CCND3</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.6727991</td>
<td style="text-align: right;">0.6173327</td>
<td style="text-align: right;">0.5663354</td>
<td style="text-align: right;">0.7296600</td>
<td style="text-align: right;">0.5854522</td>
<td style="text-align: right;">0.6145773</td>
<td style="text-align: right;">0.4965627</td>
<td style="text-align: right;">0.8483686</td>
<td style="text-align: right;">0.7478225</td>
</tr>
<tr class="odd">
<td>CCNE2</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.6448484</td>
<td style="text-align: right;">0.6491561</td>
<td style="text-align: right;">0.6487729</td>
<td style="text-align: right;">0.6461555</td>
<td style="text-align: right;">0.6404418</td>
<td style="text-align: right;">0.6457139</td>
<td style="text-align: right;">0.6446241</td>
<td style="text-align: right;">0.6356759</td>
<td style="text-align: right;">0.6405659</td>
</tr>
<tr class="even">
<td>CCNE1</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.5421153</td>
<td style="text-align: right;">0.5325393</td>
<td style="text-align: right;">0.5296759</td>
<td style="text-align: right;">0.5296031</td>
<td style="text-align: right;">0.5562533</td>
<td style="text-align: right;">0.5632116</td>
<td style="text-align: right;">0.5301545</td>
<td style="text-align: right;">0.5301542</td>
<td style="text-align: right;">0.5535661</td>
</tr>
<tr class="odd">
<td>CCNA1</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.5156421</td>
<td style="text-align: right;">0.5108572</td>
<td style="text-align: right;">0.5105493</td>
<td style="text-align: right;">0.5088376</td>
<td style="text-align: right;">0.5271956</td>
<td style="text-align: right;">0.5322819</td>
<td style="text-align: right;">0.5092389</td>
<td style="text-align: right;">0.5120832</td>
<td style="text-align: right;">0.5196845</td>
</tr>
<tr class="even">
<td>CCND2</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.6123339</td>
<td style="text-align: right;">0.4196388</td>
<td style="text-align: right;">0.4454246</td>
<td style="text-align: right;">0.3448309</td>
<td style="text-align: right;">0.4146274</td>
<td style="text-align: right;">0.4669454</td>
<td style="text-align: right;">0.3929882</td>
<td style="text-align: right;">0.3581929</td>
<td style="text-align: right;">0.4130500</td>
</tr>
<tr class="odd">
<td>CCND1</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.5003418</td>
<td style="text-align: right;">0.4913207</td>
<td style="text-align: right;">0.5013283</td>
<td style="text-align: right;">0.4992528</td>
<td style="text-align: right;">0.4998177</td>
<td style="text-align: right;">0.5008459</td>
<td style="text-align: right;">0.4991252</td>
<td style="text-align: right;">0.4807816</td>
<td style="text-align: right;">0.4985815</td>
</tr>
<tr class="even">
<td>CCNB3</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.5011883</td>
<td style="text-align: right;">0.5011323</td>
<td style="text-align: right;">0.5012770</td>
<td style="text-align: right;">0.5010881</td>
<td style="text-align: right;">0.4998587</td>
<td style="text-align: right;">0.5008078</td>
<td style="text-align: right;">0.5006712</td>
<td style="text-align: right;">0.5009322</td>
<td style="text-align: right;">0.5003608</td>
</tr>
</tbody>
</table>
</div>
<p>Direct examination of cyclin expression is easily understood, interpreted and validated with other technologies. However, it is best suited for statements about relative cell cycle activity; for example, we would find it difficult to assign cell cycle phase in Figure <a href="#fig:cyclin-heatmap">42</a> without the presence of clusters spanning all phases to provide benchmarks for “high” and “low” expression of each cyclin. We also assume that cyclin expression is not affected by biological processes other than the cell cycle, which may be a strong assumption in some cases, e.g., malignant cells.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="using-the-cyclone-classifier">Using the <code>cyclone()</code> classifier</h2>
<p>The prediction method described by <span class="citation" data-cites="scialdone2015computational">(Scialdone et al. <a href="#ref-scialdone2015computational">2015</a>)</span> is another approach for classifying cells into cell cycle phases. Using a reference dataset, we first compute the sign of the difference in expression between each pair of genes. Pairs with changes in the sign across cell cycle phases are chosen as markers. Cells in a test dataset can then be classified into the appropriate phase, based on whether the observed sign for each marker pair is consistent with one phase or another. This approach is implemented in the <code>cyclone()</code> function from the <code>R BiocStyle::Biocpkg(&quot;scran&quot;)</code> package.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The phase assignment result for each cell in the dataset is shown in Figure <a href="#fig:cyclone">43</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:cyclone"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/cyclone-1.png" alt="Cell cycle phase scores from applying the pair-based classifier on the dataset. Each point represents a cell, plotted according to its scores for G1 and G2/M phases, and coloured according to cluster." width="624" />
<p class="caption">
Figure 43: Cell cycle phase scores from applying the pair-based classifier on the dataset. Each point represents a cell, plotted according to its scores for G1 and G2/M phases, and coloured according to cluster.
</p>
</div>
</div>
<p>For each cell, a higher score for a phase corresponds to a higher probability that the cell is in that phase. We focus on the G1 and G2/M scores as these are the most informative for classification.</p>
<p>Cells are classified as being in G1 phase if the G1 score is above 0.5 and greater than the G2/M score; in G2/M phase if the G2/M score is above 0.5 and greater than the G1 score; and in S phase if neither score is above 0.5.</p>
<p>Figure <a href="#fig:clusters-by-cyclone-phase">44</a> shows an enrichment of cells in G2/M phase in cluster 9, with cells from the remaining clusters mostly classified as G1.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:clusters-by-cyclone-phase"></span>
<img src="C057_Cooney_ignoring_unknowns_annotate_files/clusters-by-cyclone-phase-1.png" alt="Breakdown of clusters by phase (as predicted using cyclone)." width="624" />
<p class="caption">
Figure 44: Breakdown of clusters by phase (as predicted using cyclone).
</p>
</div>
</div>
<p>The key assumption here is that the cell cycle is orthogonal to cell type and other aspects of cell behavior. However, it is not difficult to find routine violations of this assumption.</p>
<p>Thus, a healthy dose of skepticism is required when interpreting these assignments. Our hope is that any systematic assignment error is consistent across clusters and conditions such that they cancel out in comparisons of phase frequencies, which is the more interesting analysis anyway. Indeed, while the availability of absolute phase calls may be more appealing, it may not make much practical difference to the conclusions if the frequencies are ultimately interpreted in a relative sense (e.g., using a chi-squared test).</p>
<h1 id="concluding-remarks">Concluding remarks</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The processed <em>SingleCellExperiment</em> object is available (see <a href="../data/SCEs/C057_C057_Cooney_ignoring_unknowns.annotated.SCE.rdsCooney.annotated.SCE.rds"><code>data/SCEs/C057_Cooney_ignoring_unknowns.annotated.SCE.rds</code></a>). This will be used in downstream analyses, e.g., differential expression analysis between conditions within each cluster and differential abundance analyses between conditions for each cluster.</p>
<h1 id="additional-information" class="appendix">Additional information</h1>
<p>The following are available on request:</p>
<ul>
<li>Full CSV tables of any data presented.</li>
<li>PDF/PNG files of any static plots.</li>
</ul>
<h1 id="session-info" class="appendix">Session info</h1>
<p>The analysis and this document were prepared using the following software:</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /stornext/System/data/tools/openBLAS/openBLAS-0.3.6-gcc-9.1.0/lib/libopenblas_haswellp-r0.3.6.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils    
[7] datasets  methods   base     

other attached packages:
 [1] SingleR_0.99.7              scater_1.12.2              
 [3] pheatmap_1.0.12             cowplot_1.0.0              
 [5] ggplot2_3.2.1               janitor_1.2.0              
 [7] scran_1.12.1                BiocStyle_2.12.0           
 [9] rmarkdown_1.18              here_0.1                   
[11] SingleCellExperiment_1.6.0  SummarizedExperiment_1.14.1
[13] DelayedArray_0.10.0         BiocParallel_1.18.1        
[15] matrixStats_0.55.0          Biobase_2.44.0             
[17] GenomicRanges_1.36.1        GenomeInfoDb_1.20.0        
[19] IRanges_2.18.3              S4Vectors_0.22.1           
[21] BiocGenerics_0.30.0        

loaded via a namespace (and not attached):
 [1] ggbeeswarm_0.6.0              colorspace_1.4-1             
 [3] ellipsis_0.3.0                dynamicTreeCut_1.63-1        
 [5] rprojroot_1.3-2               XVector_0.24.0               
 [7] BiocNeighbors_1.3.3           bit64_0.9-7                  
 [9] interactiveDisplayBase_1.22.0 AnnotationDbi_1.46.1         
[11] Polychrome_1.2.3              knitr_1.26                   
[13] zeallot_0.1.0                 jsonlite_1.6                 
[15] colorblindr_0.1.0             dbplyr_1.4.2                 
[17] shiny_1.4.0                   BiocManager_1.30.10          
[19] compiler_3.6.1                httr_1.4.1                   
[21] dqrng_0.2.1                   backports_1.1.5              
[23] assertthat_0.2.1              Matrix_1.2-17                
[25] fastmap_1.0.1                 lazyeval_0.2.2               
[27] limma_3.40.6                  later_1.0.0                  
[29] BiocSingular_1.0.0            htmltools_0.4.0              
[31] tools_3.6.1                   rsvd_1.0.2                   
[33] igraph_1.2.4.2                gtable_0.3.0                 
[35] glue_1.3.1                    GenomeInfoDbData_1.2.1       
[37] dplyr_0.8.3                   rappdirs_0.3.1               
[39] Rcpp_1.0.3                    vctrs_0.2.0                  
[41] ExperimentHub_1.10.0          DelayedMatrixStats_1.6.1     
[43] xfun_0.11                     stringr_1.4.0                
[45] lifecycle_0.1.0               mime_0.7                     
[47] irlba_2.3.3                   statmod_1.4.32               
[49] AnnotationHub_2.16.1          edgeR_3.26.8                 
[51] zlibbioc_1.30.0               scales_1.1.0                 
[53] promises_1.1.0                RColorBrewer_1.1-2           
[55] yaml_2.2.0                    curl_4.3                     
[57] memoise_1.1.0                 gridExtra_2.3                
[59] distill_0.7                   stringi_1.4.3                
[61] RSQLite_2.1.4                 highr_0.8                    
[63] rlang_0.4.2                   pkgconfig_2.0.3              
[65] bitops_1.0-6                  evaluate_0.14                
[67] lattice_0.20-38               purrr_0.3.3                  
[69] labeling_0.3                  bit_1.1-14                   
[71] tidyselect_0.2.5              magrittr_1.5                 
[73] R6_2.4.1                      DBI_1.0.0                    
[75] pillar_1.4.2                  withr_2.1.2                  
[77] scatterplot3d_0.3-41          RCurl_1.95-4.12              
[79] tibble_2.1.3                  crayon_1.3.4                 
[81] BiocFileCache_1.8.0           viridis_0.5.1                
[83] locfit_1.5-9.1                grid_3.6.1                   
[85] blob_1.2.0                    digest_0.6.23                
[87] xtable_1.8-4                  tidyr_1.0.0                  
[89] httpuv_1.5.2                  munsell_0.5.0                
[91] beeswarm_0.2.3                viridisLite_0.3.0            
[93] vipor_0.4.5                  </code></pre>
</div>
<div id="refs" class="references">
<div id="ref-aran2019reference">
<p>Aran, D., A. P. Looney, L. Liu, E. Wu, V. Fong, A. Hsu, S. Chak, et al. 2019. “Reference-Based Analysis of Lung Single-Cell Sequencing Reveals a Transitional Profibrotic Macrophage.” <em>Nat. Immunol.</em> 20 (2): 163–72.</p>
</div>
<div id="ref-bach2017differentiation">
<p>Bach, K., S. Pensa, M. Grzelak, J. Hadfield, D. J. Adams, J. C. Marioni, and W. T. Khaled. 2017. “Differentiation dynamics of mammary epithelial cells revealed by single-cell RNA sequencing.” <em>Nat Commun</em> 8 (1): 2128.</p>
</div>
<div id="ref-berger1996bioequivalence">
<p>Berger, R. L., and J. C. Hsu. 1996. “Bioequivalence Trials, Intersection-Union Tests and Equivalence Confidence Sets.” <em>Statist. Sci.</em> 11 (4). The Institute of Mathematical Statistics: 283–319. <a href="https://doi.org/10.1214/ss/1032280304">https://doi.org/10.1214/ss/1032280304</a>.</p>
</div>
<div id="ref-dahlin2018single">
<p>Dahlin, J. S., F. K. Hamey, B. Pijuan-Sala, M. Shepherd, W. W. Y. Lau, S. Nestorowa, C. Weinreb, et al. 2018. “A single-cell hematopoietic landscape resolves 8 lineage trajectories and defects in Kit mutant mice.” <em>Blood</em> 131 (21): e1–e11.</p>
</div>
<div id="ref-hpcaRef">
<p>Mabbott, Neil A., J. K. Baillie, Helen Brown, Tom C. Freeman, and David A. Hume. 2013. “An expression atlas of human primary cells: Inference of gene function from coexpression networks.” <em>BMC Genomics</em> 14. <a href="https://doi.org/10.1186/1471-2164-14-632">https://doi.org/10.1186/1471-2164-14-632</a>.</p>
</div>
<div id="ref-mccarthy2009treat">
<p>McCarthy, D. J., and G. K. Smyth. 2009. “Testing significance relative to a fold-change threshold is a TREAT.” <em>Bioinformatics</em> 25 (6): 765–71.</p>
</div>
<div id="ref-scialdone2015computational">
<p>Scialdone, A., K. N. Natarajan, L. R. Saraiva, V. Proserpio, S. A. Teichmann, O. Stegle, J. C. Marioni, and F. Buettner. 2015. “Computational Assignment of Cell-Cycle Stage from Single-Cell Transcriptome Data.” <em>Methods</em> 85 (September): 54–61. <a href="https://www.ncbi.nlm.nih.gov/pubmed/26142758">https://www.ncbi.nlm.nih.gov/pubmed/26142758</a>.</p>
</div>
<div id="ref-simes1986improved">
<p>Simes, R. J. 1986. “An Improved Bonferroni Procedure for Multiple Tests of Significance.” <em>Biometrika</em> 73 (3): 751–54.</p>
</div>
<div id="ref-soneson2018bias">
<p>Soneson, C., and M. D. Robinson. 2018. “Bias, Robustness and Scalability in Single-Cell Differential Expression Analysis.” <em>Nat. Methods</em> 15 (4): 255–61. <a href="https://www.ncbi.nlm.nih.gov/pubmed/29481549">https://www.ncbi.nlm.nih.gov/pubmed/29481549</a>.</p>
</div>
<div id="ref-stoeckius2017hashing">
<p>Stoeckius, M., S. Zheng, B. Houck-Loomis, S. Hao, B. Yeung, P. Smibert, and R. Satija. 2017. “Cell &quot;Hashing&quot; with Barcoded Antibodies Enables Multiplexing and Doublet Detection for Single Cell Genomics.” <em>bioRxiv</em>. Cold Spring Harbor Laboratory. <a href="https://doi.org/10.1101/237693">https://doi.org/10.1101/237693</a>.</p>
</div>
<div id="ref-xu2015identification">
<p>Xu, C., and Z. Su. 2015. “Identification of Cell Types from Single-Cell Transcriptomes Using a Novel Clustering Method.” <em>Bioinformatics</em> 31 (12): 1974–80.</p>
</div>
<div id="ref-zheng2017massively">
<p>Zheng, G. X., J. M. Terry, P. Belgrader, P. Ryvkin, Z. W. Bent, R. Wilson, S. B. Ziraldo, et al. 2017. “Massively parallel digital transcriptional profiling of single cells.” <em>Nat Commun</em> 8 (January): 14049.</p>
</div>
</div>
<section class="footnotes">
<hr />
<ol>
<li id="fn1"><p>This means that any <code>lib.size</code> field above unity is not definitive evidence against a doublet identity for a query cluster, as there may be a pair of sources with slightly higher N but both lib.size* values below unity.<a href="#fnref1" class="footnote-back">↩</a></p></li>
<li id="fn2"><p>Differential expression analyses is performed on the original log-expression values.We do not use the MNN-corrected values here except to obtain the clusters to be characterized.<a href="#fnref2" class="footnote-back">↩</a></p></li>
<li id="fn3"><p>Intra-sample comparisons are robust to differences between libraries but assume that each pair of clusters is present in at least one library. When this is not the case, we cannot commute a logFC for that pair of clusters.<a href="#fnref3" class="footnote-back">↩</a></p></li>
<li id="fn4"><p>It must be stressed that the (adjusted) p-values computed here cannot be properly interpreted as measures of significance because the clusters have been empirically identified from the data. Nonetheless, they can still be used as way to rank ‘interesting’ genes.<a href="#fnref4" class="footnote-back">↩</a></p></li>
<li id="fn5"><p>Specifically, a simple threshold does not consider the variance and can enrich for genes that have both large log-fold changes and large variances <span class="citation" data-cites="mccarthy2009treat">(McCarthy and Smyth <a href="#ref-mccarthy2009treat">2009</a>)</span><a href="#fnref5" class="footnote-back">↩</a></p></li>
<li id="fn6"><p>This can be individual cells or bulk samples.<a href="#fnref6" class="footnote-back">↩</a></p></li>
</ol>
</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom"></div>
<script id="distill-bibliography" type="text/bibtex">
@Article{stegle2015computational,
    Author="Stegle, O.  and Teichmann, S. A.  and Marioni, J. C. ",
    Title="Computational and analytical challenges in single-cell transcriptomics",
    Journal="Nat. Rev. Genet.",
    Year="2015",
    Volume="16",
    Number="3",
    Pages="133--145",
    Month="Mar",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/25628217}
}

@Article{islam2014quantitative,
    Author="Islam, S.  and Zeisel, A.  and Joost, S.  and La Manno, G.  and Zajac, P.  and Kasper, M.  and Lonnerberg, P.  and Linnarsson, S. ",
    Title="Quantitative single-cell RNA-seq with unique molecular identifiers",
    Journal="Nat. Methods",
    Year="2014",
    Volume="11",
    Number="2",
    Pages="163--166",
    Month="Feb"
}

@Article{huber2015orchestrating,
    Author="Huber, W.  and Carey, V. J.  and Gentleman, R.  and Anders, S.  and Carlson, M.  and Carvalho, B. S.  and Bravo, H. C.  and Davis, S.  and Gatto, L.  and Girke, T.  and Gottardo, R.  and Hahne, F.  and Hansen, K. D.  and Irizarry, R. A.  and Lawrence, M.  and Love, M. I.  and MacDonald, J.  and Obenchain, V.  and Oles, A. K.  and Pag\`es, H.  and Reyes, A.  and Shannon, P.  and Smyth, G. K.  and Tenenbaum, D.  and Waldron, L.  and Morgan, M. ",
    Title="{{O}rchestrating high-throughput genomic analysis with {B}ioconductor}",
    Journal="Nat. Methods",
    Year="2015",
    Volume="12",
    Number="2",
    Pages="115--121",
    Month="Feb"
}

@Article{brennecke2013accounting,
    Author="Brennecke, P.  and Anders, S.  and Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Zhang, X.  and Proserpio, V.  and Baying, B.  and Benes, V.  and Teichmann, S. A.  and Marioni, J. C.  and Heisler, M. G. ",
    Title="{{A}ccounting for technical noise in single-cell {R}{N}{A}-seq experiments}",
    Journal="Nat. Methods",
    Year="2013",
    Volume="10",
    Number="11",
    Pages="1093--1095",
    Month="Nov"
}

@Article{marinov2014singlecell,
    Author="Marinov, G. K.  and Williams, B. A.  and McCue, K.  and Schroth, G. P.  and Gertz, J.  and Myers, R. M.  and Wold, B. J. ",
    Title="{{F}rom single-cell to cell-pool transcriptomes: stochasticity in gene expression and {R}{N}{A} splicing}",
    Journal="Genome Res.",
    Year="2014",
    Volume="24",
    Number="3",
    Pages="496--510",
    Month="Mar"
}

@Article{vallejos2015basics,
    Author="Vallejos, C. A.  and Marioni, J. C.  and Richardson, S. ",
    Title="{{B}{A}{S}i{C}{S}: Bayesian analysis of single-cell sequencing data}",
    Journal="PLoS Comput. Biol.",
    Year="2015",
    Volume="11",
    Number="6",
    Pages="e1004333",
    Month="Jun"
}

@Article{trapnell2014dynamics,
    Author="Trapnell, C.  and Cacchiarelli, D.  and Grimsby, J.  and Pokharel, P.  and Li, S.  and Morse, M.  and Lennon, N. J.  and Livak, K. J.  and Mikkelsen, T. S.  and Rinn, J. L. ",
    Title="{{T}he dynamics and regulators of cell fate decisions are revealed by pseudotemporal ordering of single cells}",
    Journal="Nat. Biotechnol.",
    Year="2014",
    Volume="32",
    Number="4",
    Pages="381--386",
    Month="Apr"
}

@Article{kolod2015singlecell,
    Author="Ko\l{}odziejczyk, A. A.  and Kim, J. K.  and Tsang, J. C.  and Ilicic, T.  and Henriksson, J.  and Natarajan, K. N.  and Tuck, A. C.  and Gao, X.  and Buhler, M.  and Liu, P.  and Marioni, J. C.  and Teichmann, S. A. ",
    Title="{{S}ingle cell {R}{N}{A}-sequencing of pluripotent states unlocks modular transcriptional variation}",
    Journal="Cell Stem Cell",
    Year="2015",
    Volume="17",
    Number="4",
    Pages="471--485",
    Month="Oct"
}

@Article{anders2010differential,
    Author="Anders, S.  and Huber, W. ",
    Title="Differential expression analysis for sequence count data",
    Journal="Genome Biol.",
    Year="2010",
    Volume="11",
    Number="10",
    Pages="R106",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20979621}
}

@Article{robinson2010scaling,
    Author="Robinson, M. D.  and Oshlack, A. ",
    Title="A scaling normalization method for differential expression analysis of RNA-seq data",
    Journal="Genome Biol.",
    Year="2010",
    Volume="11",
    Number="3",
    Pages="R25",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20196867}
}

@Article{islam2011characterization,
    Author="Islam, S.  and Kjallquist, U.  and Moliner, A.  and Zajac, P.  and Fan, J. B.  and Lonnerberg, P.  and Linnarsson, S. ",
    Title="{{C}haracterization of the single-cell transcriptional landscape by highly multiplex {R}{N}{A}-seq}",
    Journal="Genome Res.",
    Year="2011",
    Volume="21",
    Number="7",
    Pages="1160--1167",
    Month="Jul"
}

@Article{wilson2015combined,
    Author="Wilson, N. K.  and Kent, D. G.  and Buettner, F.  and Shehata, M.  and Macaulay, I. C.  and Calero-Nieto, F. J.  and Sanchez Castillo, M.  and Oedekoven, C. A.  and Diamanti, E.  and Schulte, R.  and Ponting, C. P.  and Voet, T.  and Caldas, C.  and Stingl, J.  and Green, A. R.  and Theis, F. J.  and Gottgens, B. ",
    Title="{{C}ombined single-cell functional and gene expression analysis resolves heterogeneity within stem cell populations}",
    Journal="Cell Stem Cell",
    Year="2015",
    Volume="16",
    Number="6",
    Pages="712--724",
    Month="Jun"
}

@Article{picelli2014fulllength,
    Author="Picelli, S.  and Faridani, O. R.  and Bjorklund, A. K.  and Winberg, G.  and Sagasser, S.  and Sandberg, R. ",
    Title="{{F}ull-length {R}{N}{A}-seq from single cells using {S}mart-seq2}",
    Journal="Nat Protoc",
    Year="2014",
    Volume="9",
    Number="1",
    Pages="171--181",
    Month="Jan"
}

@Article{pollen2014lowcoverage,
    Author="Pollen, A. A.  and Nowakowski, T. J.  and Shuga, J.  and Wang, X.  and Leyrat, A. A.  and Lui, J. H.  and Li, N.  and Szpankowski, L.  and Fowler, B.  and Chen, P.  and Ramalingam, N.  and Sun, G.  and Thu, M.  and Norris, M.  and Lebofsky, R.  and Toppani, D.  and Kemp, D. W.  and Wong, M.  and Clerkson, B.  and Jones, B. N.  and Wu, S.  and Knutsson, L.  and Alvarado, B.  and Wang, J.  and Weaver, L. S.  and May, A. P.  and Jones, R. C.  and Unger, M. A.  and Kriegstein, A. R.  and West, J. A. ",
    Title="{{L}ow-coverage single-cell m{R}{N}{A} sequencing reveals cellular heterogeneity and activated signaling pathways in developing cerebral cortex}",
    Journal="Nat. Biotechnol.",
    Year="2014",
    Volume="32",
    Number="10",
    Pages="1053--1058",
    Month="Oct"
}

@Article{love2015rnaseq,
    Author="Love, M. I.  and Anders, S.  and Kim, V.  and Huber, W. ",
    Title="{{R}{N}{A}-{S}eq workflow: gene-level exploratory analysis and differential expression}",
    Journal="F1000Res",
    Year="2015",
    Volume="4",
    Pages="1070"
}

@Article{liao2014featurecounts,
    Author="Liao, Y.  and Smyth, G. K.  and Shi, W. ",
    Title="{feature{C}ounts: an efficient general purpose program for assigning sequence reads to genomic features}",
    Journal="Bioinformatics",
    Year="2014",
    Volume="30",
    Number="7",
    Pages="923--930",
    Month="Apr"
}

@Article{liao2013subread,
    Author="Liao, Y.  and Smyth, G. K.  and Shi, W. ",
    Title="{{T}he {S}ubread aligner: fast, accurate and scalable read mapping by seed-and-vote}",
    Journal="Nucleic Acids Res.",
    Year="2013",
    Volume="41",
    Number="10",
    Pages="e108",
    Month="May"
}

@Article{law2014voom,
    Author="Law, C. W.  and Chen, Y.  and Shi, W.  and Smyth, G. K. ",
    Title="voom: Precision weights unlock linear model analysis tools for RNA-seq read counts",
    Journal="Genome Biol.",
    Year="2014",
    Volume="15",
    Number="2",
    Pages="R29",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/24485249}
}

@Article{love2014moderated,
    Author="Love, M. I.  and Huber, W.  and Anders, S. ",
    Title="Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2",
    Journal="Genome Biol.",
    Year="2014",
    Volume="15",
    Number="12",
    Pages="550",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/25516281}
}

@article{van2008visualizing,
    title={Visualizing data using t-SNE},
    author={Van der Maaten, L. and Hinton, G.},
    journal={J. Mach. Learn. Res.},
    volume={9},
    number={2579-2605},
    pages={85},
    year={2008},
    URL={http://www.jmlr.org/papers/v9/vandermaaten08a.html}
}

@Article{scialdone2015computational,
    Author="Scialdone, A.  and Natarajan, K. N.  and Saraiva, L. R.  and Proserpio, V.  and Teichmann, S. A.  and Stegle, O.  and Marioni, J. C.  and Buettner, F. ",
    Title="Computational assignment of cell-cycle stage from single-cell transcriptome data",
    Journal="Methods",
    Year="2015",
    Volume="85",
    Pages="54--61",
    Month="Sep",
    URL = {https://www.ncbi.nlm.nih.gov/pubmed/26142758},
}

@Article{kim2015characterizing,
    Author="Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Illicic, T.  and Teichmann, S. A.  and Marioni, J. C. ",
    Title="{{C}haracterizing noise structure in single-cell {R}{N}{A}-seq distinguishes genuine from technical stochastic allelic expression}",
    Journal="Nat. Commun.",
    Year="2015",
    Volume="6",
    Pages="8687"
}

@Article{mccarthy2012differential,
    Author="McCarthy, D. J.  and Chen, Y.  and Smyth, G. K. ",
    Title="{{D}ifferential expression analysis of multifactor {R}{N}{A}-{S}eq experiments with respect to biological variation}",
    Journal="Nucleic Acids Res.",
    Year="2012",
    Volume="40",
    Number="10",
    Pages="4288--4297",
    Month="May"
}

@Article{angel1991role,
    Author="Angel, P.  and Karin, M. ",
    Title="{{T}he role of {J}un, {F}os and the {A}{P}-1 complex in cell-proliferation and transformation}",
    Journal="Biochim. Biophys. Acta",
    Year="1991",
    Volume="1072",
    Number="2-3",
    Pages="129--157",
    Month="Dec"
}

@Article{langfelder2008defining,
    Author="Langfelder, P.  and Zhang, B.  and Horvath, S. ",
    Title="{{D}efining clusters from a hierarchical cluster tree: the {D}ynamic {T}ree {C}ut package for {R}}",
    Journal="Bioinformatics",
    Year="2008",
    Volume="24",
    Number="5",
    Pages="719--720",
    Month="Mar"
}

@Article{ilicic2016classification,
    Author="Ilicic, T.  and Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Bagger, F. O.  and McCarthy, D. J.  and Marioni, J. C.  and Teichmann, S. A. ",
    Title="{{C}lassification of low quality cells from single-cell {R}{N}{A}-seq data}",
    Journal="Genome Biol.",
    Year="2016",
    Volume="17",
    Number="1",
    Pages="29"
}

@Article{heng2008immunological,
    Author="Heng, T. S.  and Painter, M. W.  and Elpek, K.  and Lukacs-Kornek, V.  and Mauermann, N.  and Turley, S. J.  and Koller, D.  and Kim, F. S.  and Wagers, A. J.  and Asinovski, N.  and Davis, S.  and Fassett, M.  and Feuerer, M.  and Gray, D. H.  and Haxhinasto, S.  and Hill, J. A.  and Hyatt, G.  and Laplace, C.  and Leatherbee, K.  and Mathis, D.  and Benoist, C.  and Jianu, R.  and Laidlaw, D. H.  and Best, J. A.  and Knell, J.  and Goldrath, A. W.  and Jarjoura, J.  and Sun, J. C.  and Zhu, Y.  and Lanier, L. L.  and Ergun, A.  and Li, Z.  and Collins, J. J.  and Shinton, S. A.  and Hardy, R. R.  and Friedline, R.  and Sylvia, K.  and Kang, J. ",
    Title="{{T}he {I}mmunological {G}enome {P}roject: networks of gene expression in immune cells}",
    Journal="Nat. Immunol.",
    Year="2008",
    Volume="9",
    Number="10",
    Pages="1091--1094",
    Month="Oct"
}

@Article{zeisel2015brain,
    Author="Zeisel, A.  and Munoz-Manchado, A. B.  and Codeluppi, S.  and Lonnerberg, P.  and La Manno, G.  and Jureus, A.  and Marques, S.  and Munguba, H.  and He, L.  and Betsholtz, C.  and Rolny, C.  and Castelo-Branco, G.  and Hjerling-Leffler, J.  and Linnarsson, S. ",
    Title="{{B}rain structure. {C}ell types in the mouse cortex and hippocampus revealed by single-cell {R}{N}{A}-seq}",
    Journal="Science",
    Year="2015",
    Volume="347",
    Number="6226",
    Pages="1138--1142",
    Month="Mar"
}

@Article{robinson2010edgeR,
    Author="Robinson, M. D.  and McCarthy, D. J.  and Smyth, G. K. ",
    Title="{edge{R}: a {B}ioconductor package for differential expression analysis of digital gene expression data}",
    Journal="Bioinformatics",
    Year="2010",
    Volume="26",
    Number="1",
    Pages="139--140",
    Month="Jan"
}

@Article{ritchie2015limma,
    Author="Ritchie, M. E.  and Phipson, B.  and Wu, D.  and Hu, Y.  and Law, C. W.  and Shi, W.  and Smyth, G. K. ",
    Title="{limma powers differential expression analyses for {R}{N}{A}-sequencing and microarray studies}",
    Journal="Nucleic Acids Res.",
    Year="2015",
    Volume="43",
    Number="7",
    Pages="e47",
    Month="Apr"
}

@Article{julia2015sincell,
    Author="Julia, M.  and Telenti, A.  and Rausell, A. ",
    Title="{{S}incell: an {R}/{B}ioconductor package for statistical assessment of cell-state hierarchies from single-cell {R}{N}{A}-seq}",
    Journal="Bioinformatics",
    Year="2015",
    Volume="31",
    Number="20",
    Pages="3380--3382",
    Month="Oct"
}

@Article{buettner2015computational,
    Author="Buettner, F.  and Natarajan, K. N.  and Casale, F. P.  and Proserpio, V.  and Scialdone, A.  and Theis, F. J.  and Teichmann, S. A.  and Marioni, J. C.  and Stegle, O. ",
    Title="{{C}omputational analysis of cell-to-cell heterogeneity in single-cell {R}{N}{A}-sequencing data reveals hidden subpopulations of cells}",
    Journal="Nat. Biotechnol.",
    Year="2015",
    Volume="33",
    Number="2",
    Pages="155--160",
    Month="Feb"
}

@Article{mahata2014singlecell,
    Author="Mahata, B.  and Zhang, X.  and Ko\l{}odziejczyk, A. A.  and Proserpio, V.  and Haim-Vilmovsky, L.  and Taylor, A. E.  and Hebenstreit, D.  and Dingler, F. A.  and Moignard, V.  and Gottgens, B.  and Arlt, W.  and McKenzie, A. N.  and Teichmann, S. A. ",
    Title="{{S}ingle-cell {R}{N}{A} sequencing reveals {T} helper cells synthesizing steroids de novo to contribute to immune homeostasis}",
    Journal="Cell Rep.",
    Year="2014",
    Volume="7",
    Number="4",
    Pages="1130--1142",
    Month="May"
}

@Article{phipson2010permutation,
    Author="Phipson, B.  and Smyth, G. K. ",
    Title="{{P}ermutation {P}-values should never be zero: calculating exact {P}-values when permutations are randomly drawn}",
    Journal="Stat. Appl. Genet. Mol. Biol.",
    Year="2010",
    Volume="9",
    Pages="Article 39"
}

@incollection{chen2014differential,
    title="{Differential expression analysis of complex RNA-seq experiments using edgeR}",
    author={Chen, Y. and Lun, A. T. L. and Smyth, G. K.},
    booktitle={Statistical analysis of next generation sequencing data},
    editor={Datta, S. and Nettleton, D.},
    pages={51--74},
    year={2014},
    publisher={Springer}
}

@article{hicks2015widespread,
    author = {Hicks, S. C. and Teng, M. and Irizarry, R. A.},
    title = "On the widespread and critical impact of systematic bias and batch effects in single-cell RNA-Seq data",
    year = {2015},
    doi = {10.1101/025528},
    publisher = {Cold Spring Harbor Labs Journals},
    URL = {http://biorxiv.org/content/early/2015/08/25/025528},
    eprint = {http://biorxiv.org/content/early/2015/08/25/025528.full.pdf},
    journal = {bioRxiv}
}

@Article{leng2015oscope,
    Author="Leng, N.  and Chu, L. F.  and Barry, C.  and Li, Y.  and Choi, J.  and Li, X.  and Jiang, P.  and Stewart, R. M.  and Thomson, J. A.  and Kendziorski, C. ",
    Title="{{O}scope identifies oscillatory genes in unsynchronized single-cell {R}{N}{A}-seq experiments}",
    Journal="Nat. Methods",
    Year="2015",
    Volume="12",
    Number="10",
    Pages="947--950",
    Month="Oct"
}

@Article{fan2016characterizing,
    Author="Fan, J.  and Salathia, N.  and Liu, R.  and Kaeser, G. E.  and Yung, Y. C.  and Herman, J. L.  and Kaper, F.  and Fan, J. B.  and Zhang, K.  and Chun, J.  and Kharchenko, P. V. ",
    Title="{{C}haracterizing transcriptional heterogeneity through pathway and gene set overdispersion analysis}",
    Journal="Nat. Methods",
    Year="2016",
    Volume="13",
    Number="3",
    Pages="241--244",
    Month="Mar"
}

@Article{lund2012detecting,
    Author="Lund, S. P.  and Nettleton, D.  and McCarthy, D. J.  and Smyth, G. K. ",
    Title="{{D}etecting differential expression in {R}{N}{A}-sequence data using quasi-likelihood with shrunken dispersion estimates}",
    Journal="Stat Appl Genet Mol Biol",
    Year="2012",
    Volume="11",
    Number="5",
    Pages=" "
}

@Article{angerer2016destiny,
    Author="Angerer, P.  and Haghverdi, L.  and Buttner, M.  and Theis, F. J.  and Marr, C.  and Buettner, F. ",
    Title="{destiny: diffusion maps for large-scale single-cell data in {R}}",
    Journal="Bioinformatics",
    Year="2016",
    Volume="32",
    Number="8",
    Pages="1241--1243",
}

@Article{lun2016pooling,
    Author="Lun, A. T.  and Bach, K.  and Marioni, J. C. ",
    Title="Pooling across cells to normalize single-cell RNA sequencing data with many zero counts}",
    Journal="Genome Biol.",
    Year="2016",
    Volume="17",
    Pages="75",
    Month="Apr",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/27122128}
}

@Article{zhu2006gata3,
    Author="Zhu, J. and Yamane, H. and Cote-Sierra, J. and Guo, L. and Paul, W. E.",
    Title="{{G}{A}{T}{A}-3 promotes {T}h2 responses through three different mechanisms: induction of {T}h2 cytokine production, selective growth of {T}h2 cells and inhibition of {T}h1 cell-specific factors}",
    Journal="Cell Res.",
    Year="2006",
    Volume="16",
    Number="1",
    Pages="3--10",
    Month="Jan"
}

@article{berger1996bioequivalence,
    author = "Berger, R. L. and Hsu, J. C.",
    doi = "10.1214/ss/1032280304",
    fjournal = "Statistical Science",
    journal = "Statist. Sci.",
    month = "11",
    number = "4",
    pages = "283--319",
    publisher = "The Institute of Mathematical Statistics",
    title = "Bioequivalence trials, intersection-union tests and equivalence confidence sets",
    url = "http://dx.doi.org/10.1214/ss/1032280304",
    volume = "11",
    year = "1996"
}

@Article{santos2015cyclebase,
    Author="Santos, A.  and Wernersson, R.  and Jensen, L. J. ",
    Title="{{C}yclebase 3.0: a multi-organism database on cell-cycle regulation and phenotypes}",
    Journal="Nucleic Acids Res.",
    Year="2015",
    Volume="43",
    Number="Database issue",
    Pages="D1140--1144",
    Month="Jan"
}

@article {tung2016batch,
    Author="Tung, P. Y.  and Blischak, J. D.  and Hsiao, C. J.  and Knowles, D. A.  and Burnett, J. E.  and Pritchard, J. K.  and Gilad, Y. ",
    Title="Batch effects and the effective design of single-cell gene expression studies",
    Journal="Sci. Rep.",
    Year="2017",
    Volume="7",
    Pages="39921",
    Month="Jan"
}

@Article{bray2016near,
    Author="Bray, N. L.  and Pimentel, H.  and Melsted, P.  and Pachter, L. ",
    Title="{{N}ear-optimal probabilistic {R}{N}{A}-seq quantification}",
    Journal="Nat. Biotechnol.",
    Year="2016",
    Volume="34",
    Number="5",
    Pages="525--527",
    Month="May"
}

@article{patro2015accurate,
    author = {Patro, Rob and Duggal, Geet and Kingsford, Carl},
    title = {Accurate, fast, and model-aware transcript expression quantification with Salmon},
    year = {2015},
    doi = {10.1101/021592},
    publisher = {Cold Spring Harbor Labs Journals},
    journal = {bioRxiv}
}

@Article{ji2016tscan,
    Author="Ji, Z.  and Ji, H. ",
    Title="{{T}{S}{C}{A}{N}: {P}seudo-time reconstruction and evaluation in single-cell {R}{N}{A}-seq analysis}",
    Journal="Nucleic Acids Res.",
    Year="2016",
    Volume="44",
    Number="13",
    Pages="e117",
    Month="Jul"
}

@Article{klein2015droplet,
    Author="Klein, A. M.  and Mazutis, L.  and Akartuna, I.  and Tallapragada, N.  and Veres, A.  and Li, V.  and Peshkin, L.  and Weitz, D. A.  and Kirschner, M. W. ",
    Title="{{D}roplet barcoding for single-cell transcriptomics applied to embryonic stem cells}",
    Journal="Cell",
    Year="2015",
    Volume="161",
    Number="5",
    Pages="1187--1201",
    Month="May"
}

@Article{macosko2015highly,
    Author="Macosko, E. Z.  and Basu, A.  and Satija, R.  and Nemesh, J.  and Shekhar, K.  and Goldman, M.  and Tirosh, I.  and Bialas, A. R.  and Kamitaki, N.  and Martersteck, E. M.  and Trombetta, J. J.  and Weitz, D. A.  and Sanes, J. R.  and Shalek, A. K.  and Regev, A.  and McCarroll, S. A. ",
    Title="{{H}ighly parallel genome-wide expression profiling of individual cells using nanoliter droplets}",
    Journal="Cell",
    Year="2015",
    Volume="161",
    Number="5",
    Pages="1202--1214",
    Month="May"
}

@Article{bourgon2010independent,
    Author="Bourgon, R.  and Gentleman, R.  and Huber, W. ",
    Title="Independent filtering increases detection power for high-throughput experiments",
    Journal="Proc. Natl. Acad. Sci. U.S.A.",
    Year="2010",
    Volume="107",
    Number="21",
    Pages="9546--9551",
    Month="May",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20460310}
}

@Article{finak2015mast,
    Author="Finak, G.  and McDavid, A.  and Yajima, M.  and Deng, J.  and Gersuk, V.  and Shalek, A. K.  and Slichter, C. K.  and Miller, H. W.  and McElrath, M. J.  and Prlic, M.  and Linsley, P. S.  and Gottardo, R. ",
    Title="{{M}{A}{S}{T}: a flexible statistical framework for assessing transcriptional changes and characterizing heterogeneity in single-cell {R}{N}{A} sequencing data}",
    Journal="Genome Biol.",
    Year="2015",
    Volume="16",
    Pages="278"
}

@Article{conboy2007cell,
    Author="Conboy, C. M.  and Spyrou, C.  and Thorne, N. P.  and Wade, E. J.  and Barbosa-Morais, N. L.  and Wilson, M. D.  and Bhattacharjee, A.  and Young, R. A.  and Tavare, S.  and Lees, J. A.  and Odom, D. T. ",
    Title="Cell cycle genes are the evolutionarily conserved targets of the E2F4 transcription factor",
    Journal="PLoS ONE",
    Year="2007",
    Volume="2",
    Number="10",
    Pages="e1061",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/17957245}
}

@Article{bertoli2013control,
    Author="Bertoli, C.  and Skotheim, J. M.  and de Bruin, R. A. ",
    Title="Control of cell cycle transcription during G1 and S phases",
    Journal="Nat. Rev. Mol. Cell Biol.",
    Year="2013",
    Volume="14",
    Number="8",
    Pages="518--528",
    Month="Aug",
    URL={{C}ontrol of cell cycle transcription during {G}1 and {S} phases}
}

@Article{chen2016from,
    Author="Chen, Y.  and Lun, A. T.  and Smyth, G. K. ",
    Title="{{F}rom reads to genes to pathways: differential expression analysis of {R}{N}{A}-{S}eq experiments using {R}subread and the edge{R} quasi-likelihood pipeline}",
    Journal="F1000Res",
    Year="2016",
    Volume="5",
    Pages="1438"
}

@Article{coller2006new,
    Author="Coller, H. A.  and Sang, L.  and Roberts, J. M. ",
    Title="{{A} new description of cellular quiescence}",
    Journal="PLoS Biol.",
    Year="2006",
    Volume="4",
    Number="3",
    Pages="e83",
    Month="Mar"
}

@Article{mccarthy2017scater,
    Author="McCarthy, D. J.  and Campbell, K. R.  and Lun, A. T.  and Wills, Q. F. ",
    Title="{{S}cater: pre-processing, quality control, normalization and visualization of single-cell {R}{N}{A}-seq data in {R}}",
    Journal="Bioinformatics",
    Year="2017",
    Volume="33",
    Number="8",
    Pages="1179--1186",
    Month="Apr"
}

@article{simes1986improved,
    author = {Simes, R. J.},
    title = {An improved Bonferroni procedure for multiple tests of significance},
    volume = {73},
    number = {3},
    pages = {751--754},
    year = {1986},
    journal = {Biometrika}
}

@Article{xu2015identification,
    Author="Xu, C.  and Su, Z. ",
    Title="Identification of cell types from single-cell transcriptomes using a novel clustering method",
    Journal="Bioinformatics",
    Year="2015",
    Volume="31",
    Number="12",
    Pages="1974--1980",
    Month="Jun"
}

@Article{lun2017assessing,
    Author="Lun, A. T. L.  and Calero-Nieto, F. J.  and Haim-Vilmovsky, L.  and Gottgens, B.  and Marioni, J. C. ",
    Title="Assessing the reliability of spike-in normalization for analyses of single-cell RNA sequencing data",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="11",
    Pages="1795--1806",
    Month="Nov"
}

@Article{yang2016comparative,
    Author="Yang, Z.  and Algesheimer, R.  and Tessone, C. J. ",
    Title="{A comparative analysis of community detection algorithms on artificial networks}",
    Journal="Sci. Rep.",
    Year="2016",
    Volume="6",
    Pages="30750",
    Month="Aug"
}

@Article{smith2017umitools,
    Author="Smith, T.  and Heger, A.  and Sudbery, I. ",
    Title="{{U}{M}{I}-tools: modeling sequencing errors in {U}nique {M}olecular {I}dentifiers to improve quantification accuracy}",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="3",
    Pages="491--499",
    Month="Mar"
}

@Article{tasic2016adult,
    Author="Tasic, B.  and Menon, V.  and Nguyen, T. N.  and Kim, T. K.  and Jarsky, T.  and Yao, Z.  and Levi, B.  and Gray, L. T.  and Sorensen, S. A.  and Dolbeare, T.  and Bertagnolli, D.  and Goldy, J.  and Shapovalova, N.  and Parry, S.  and Lee, C.  and Smith, K.  and Bernard, A.  and Madisen, L.  and Sunkin, S. M.  and Hawrylycz, M.  and Koch, C.  and Zeng, H. ",
    Title="{{A}dult mouse cortical cell taxonomy revealed by single cell transcriptomics}",
    Journal="Nat. Neurosci.",
    Year="2016",
    Volume="19",
    Number="2",
    Pages="335--346",
    Month="Feb"
}

@Article{wajapeyee2010senescence,
    Author="Wajapeyee, N.  and Wang, S. Z.  and Serra, R. W.  and Solomon, P. D.  and Nagarajan, A.  and Zhu, X.  and Green, M. R. ",
    Title="{{S}enescence induction in human fibroblasts and hematopoietic progenitors by leukemogenic fusion proteins}",
    Journal="Blood",
    Year="2010",
    Volume="115",
    Number="24",
    Pages="5057--5060",
    Month="Jun"
}

@Article{zeng2012largescale,
    Author="Zeng, H.  and Shen, E. H.  and Hohmann, J. G.  and Oh, S. W.  and Bernard, A.  and Royall, J. J.  and Glattfelder, K. J.  and Sunkin, S. M.  and Morris, J. A.  and Guillozet-Bongaarts, A. L.  and Smith, K. A.  and Ebbert, A. J.  and Swanson, B.  and Kuan, L.  and Page, D. T.  and Overly, C. C.  and Lein, E. S.  and Hawrylycz, M. J.  and Hof, P. R.  and Hyde, T. M.  and Kleinman, J. E.  and Jones, A. R. ",
    Title="{{L}arge-scale cellular-resolution gene profiling in human neocortex reveals species-specific molecular signatures}",
    Journal="Cell",
    Year="2012",
    Volume="149",
    Number="2",
    Pages="483--496",
    Month="Apr"
}

@Article{grun2016denovo,
    Author="Grun, D.  and Muraro, M. J.  and Boisset, J. C.  and Wiebrands, K.  and Lyubimova, A.  and Dharmadhikari, G.  and van den Born, M.  and van Es, J.  and Jansen, E.  and Clevers, H.  and de Koning, E. J. P.  and van Oudenaarden, A. ",
    Title="{{D}e {N}ovo {P}rediction of {S}tem {C}ell {I}dentity using {S}ingle-{C}ell {T}ranscriptome {D}ata}",
    Journal="Cell Stem Cell",
    Year="2016",
    Volume="19",
    Number="2",
    Pages="266--277",
    Month="08"
}

@article{haghverdi2018batch,
    Author="Haghverdi, L.  and Lun, A. T. L.  and Morgan, M. D.  and Marioni, J. C. ",
    Title="Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors",
    Journal="Nat. Biotechnol.",
    Year="2018",
    Month="Apr"
}

@Article{muraro2016singlecell,
    Author="Muraro, M. J.  and Dharmadhikari, G.  and Grun, D.  and Groen, N.  and Dielen, T.  and Jansen, E.  and van Gurp, L.  and Engelse, M. A.  and Carlotti, F.  and de Koning, E. J.  and van Oudenaarden, A. ",
    Title="{{A} {S}ingle-{C}ell {T}ranscriptome {A}tlas of the {H}uman {P}ancreas}",
    Journal="Cell Syst",
    Year="2016",
    Volume="3",
    Number="4",
    Pages="385--394",
    Month="Oct"
}

@Article{segerstolpe2016singlecell,
    Author="Segerstolpe, A.  and Palasantza, A.  and Eliasson, P.  and Andersson, E. M.  and Andreasson, A. C.  and Sun, X.  and Picelli, S.  and Sabirsh, A.  and Clausen, M.  and Bjursell, M. K.  and Smith, D. M.  and Kasper, M.  and Ammala, C.  and Sandberg, R. ",
    Title="{{S}ingle-{C}ell {T}ranscriptome {P}rofiling of {H}uman {P}ancreatic {I}slets in {H}ealth and {T}ype 2 {D}iabetes}",
    Journal="Cell Metab.",
    Year="2016",
    Volume="24",
    Number="4",
    Pages="593--607",
    Month="Oct"
}

@Article{horn1965rationale,
    author="Horn, J. L.",
    title="A rationale and test for the number of factors in factor analysis",
    journal="Psychometrika",
    year="1965",
    month="Jun",
    day="01",
    volume="30",
    number="2",
    pages="179--185",
    issn="1860-0980",
    doi="10.1007/BF02289447",
    url="https://doi.org/10.1007/BF02289447"
}

@Article{buja1992remarks,
    Author="Buja, A.  and Eyuboglu, N. ",
    Title="{{R}emarks on {P}arallel {A}nalysis}",
    Journal="Multivariate Behav Res",
    Year="1992",
    Volume="27",
    Number="4",
    Pages="509--540",
    Month="Oct"
}

@Article{zheng2017massively,
    Author="Zheng, G. X.  and Terry, J. M.  and Belgrader, P.  and Ryvkin, P.  and Bent, Z. W.  and Wilson, R.  and Ziraldo, S. B.  and Wheeler, T. D.  and McDermott, G. P.  and Zhu, J.  and Gregory, M. T.  and Shuga, J.  and Montesclaros, L.  and Underwood, J. G.  and Masquelier, D. A.  and Nishimura, S. Y.  and Schnall-Levin, M.  and Wyatt, P. W.  and Hindson, C. M.  and Bharadwaj, R.  and Wong, A.  and Ness, K. D.  and Beppu, L. W.  and Deeg, H. J.  and McFarland, C.  and Loeb, K. R.  and Valente, W. J.  and Ericson, N. G.  and Stevens, E. A.  and Radich, J. P.  and Mikkelsen, T. S.  and Hindson, B. J.  and Bielas, J. H. ",
    Title="{{M}assively parallel digital transcriptional profiling of single cells}",
    Journal="Nat Commun",
    Year="2017",
    Volume="8",
    Pages="14049",
    Month="Jan"
}

@article{lun2018distinguishing,
    author = {Lun, A. and Riesenfeld, S. and Andrews, T. and Dao, T. P. and Gomes, T. and {{participants in the 1\textsuperscript{st} Human Cell Atlas Jamboree}} and Marioni, J.},
    title = {Distinguishing cells from empty droplets in droplet-based single-cell RNA sequencing data},
    year = {2018},
    doi = {10.1101/234872},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2018/04/04/234872},
    eprint = {https://www.biorxiv.org/content/early/2018/04/04/234872.full.pdf},
    journal = {bioRxiv}
}

@article {wolf2017graph,
    author = {Wolf, F. Alexander and Hamey, Fiona and Plass, Mireya and Solana, Jordi and Dahlin, Joakim S. and Gottgens, Berthold and Rajewsky, Nikolaus and Simon, Lukas and Theis, Fabian J.},
    title = {Graph abstraction reconciles clustering with trajectory inference through a topology preserving map of single cells},
    year = {2017},
    doi = {10.1101/208819},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2017/10/25/208819},
    eprint = {https://www.biorxiv.org/content/early/2017/10/25/208819.full.pdf},
    journal = {bioRxiv}
}

@Article{lawlor2017singlecell,
    Author="Lawlor, N.  and George, J.  and Bolisetty, M.  and Kursawe, R.  and Sun, L.  and Sivakamasundari, V.  and Kycia, I.  and Robson, P.  and Stitzel, M. L. ",
    Title="{{S}ingle-cell transcriptomes identify human islet cell signatures and reveal cell-type-specific expression changes in type 2 diabetes}",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="2",
    Pages="208--222",
    Month="02"
}

@Article{guimaraes2016patterns,
   Author="Guimaraes, J. C.  and Zavolan, M. ",
   Title="{{P}atterns of ribosomal protein expression specify normal and malignant human cells}",
   Journal="Genome Biol.",
   Year="2016",
   Volume="17",
   Number="1",
   Pages="236",
   Month="11"
}

@Article{glare2002betaactin,
   Author="Glare, E. M.  and Divjak, M.  and Bailey, M. J.  and Walters, E. H. ",
   Title="{beta-{A}ctin and {G}{A}{P}{D}{H} housekeeping gene expression in asthmatic airways is variable and not suitable for normalising m{R}{N}{A} levels}",
   Journal="Thorax",
   Year="2002",
   Volume="57",
   Number="9",
   Pages="765--770",
   Month="Sep"
}

@Article{nazari2015gapdh,
   Author="Nazari, F.  and Parham, A.  and Maleki, A. F. ",
   Title="{{G}{A}{P}{D}{H}, \beta-actin and \beta-microglobulin, as three common reference genes, are not reliable for gene expression studies in equine adipose- and marrow-derived mesenchymal stem cells}",
   Journal="J Anim Sci Technol",
   Year="2015",
   Volume="57",
   Pages="18"
}

@Article{soneson2018bias,
    Author="Soneson, C.  and Robinson, M. D. ",
    Title="Bias, robustness and scalability in single-cell differential expression analysis",
    Journal="Nat. Methods",
    URL = {https://www.ncbi.nlm.nih.gov/pubmed/29481549},
    Year="2018",
    Volume="15",
    Number="4",
    Pages="255--261",
    Month="Apr"
}

@Article{soneson2015differential,
    Author="Soneson, C.  and Love, M. I.  and Robinson, M. D. ",
    Title="{{D}ifferential analyses for {R}{N}{A}-seq: transcript-level estimates improve gene-level inferences}",
    Journal="F1000Res",
    Year="2015",
    Volume="4",
    Pages="1521"
}

@article{stoeckius2017hashing,
    author = {Stoeckius, M. and Zheng, S. and Houck-Loomis, B. and Hao, S. and Yeung, B. and Smibert, P. and Satija, R.},
    title = {Cell "hashing" with barcoded antibodies enables multiplexing and doublet detection for single cell genomics},
    year = {2017},
    doi = {10.1101/237693},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2017/12/21/237693},
    eprint = {https://www.biorxiv.org/content/early/2017/12/21/237693.full.pdf},
    journal = {bioRxiv}
}

@Article{kang2018multiplexed,
    Author="Kang, H. M.  and Subramaniam, M.  and Targ, S.  and Nguyen, M.  and Maliskova, L.  and McCarthy, E.  and Wan, E.  and Wong, S.  and Byrnes, L.  and Lanata, C. M.  and Gate, R. E.  and Mostafavi, S.  and Marson, A.  and Zaitlen, N.  and Criswell, L. A.  and Ye, C. J. ",
    Title="{{M}ultiplexed droplet single-cell {R}{N}{A}-sequencing using natural genetic variation}",
    Journal="Nat. Biotechnol.",
    Year="2018",
    Volume="36",
    Number="1",
    Pages="89--94",
    Month="Jan"
}

@Article{dahlin2018single,
    Author="Dahlin, J. S.  and Hamey, F. K.  and Pijuan-Sala, B.  and Shepherd, M.  and Lau, W. W. Y.  and Nestorowa, S.  and Weinreb, C.  and Wolock, S.  and Hannah, R.  and Diamanti, E.  and Kent, D. G.  and Gottgens, B.  and Wilson, N. K. ",
    Title="{{A} single-cell hematopoietic landscape resolves 8 lineage trajectories and defects in {K}it mutant mice}",
    Journal="Blood",
    Year="2018",
    Volume="131",
    Number="21",
    Pages="e1-e11",
    Month="May"
}

@Article{bach2017differentiation,
    Author="Bach, K.  and Pensa, S.  and Grzelak, M.  and Hadfield, J.  and Adams, D. J.  and Marioni, J. C.  and Khaled, W. T. ",
    Title="{{D}ifferentiation dynamics of mammary epithelial cells revealed by single-cell {R}{N}{A} sequencing}",
    Journal="Nat Commun",
    Year="2017",
    Volume="8",
    Number="1",
    Pages="2128",
    Month="Dec"
}

@article{tian2018scpipe,
  title={scPipe: A flexible R/Bioconductor preprocessing pipeline for single-cell RNA-sequencing data},
  author={Tian, Luyi and Su, Shian and Dong, Xueyi and Amann-Zalcenstein, Daniela and Biben, Christine and Seidi, Azadeh and Hilton, Douglas J and Naik, Shalin H and Ritchie, Matthew E},
  journal={PLoS computational biology},
  volume={14},
  number={8},
  pages={e1006361},
  year={2018},
  publisher={Public Library of Science},
  URL={https://www.ncbi.nlm.nih.gov/pubmed/30096152}
}

@article{kiselev2017sc3,
  title={SC3: consensus clustering of single-cell RNA-seq data},
  author={Kiselev, Vladimir Yu and Kirschner, Kristina and Schaub, Michael T and Andrews, Tallulah and Yiu, Andrew and Chandra, Tamir and Natarajan, Kedar N and Reik, Wolf and Barahona, Mauricio and Green, Anthony R and others},
  journal={Nature methods},
  volume={14},
  number={5},
  pages={483},
  year={2017},
  publisher={Nature Publishing Group},
  URL={https://www.ncbi.nlm.nih.gov/pubmed/28346451}
}

@ARTICLE{McInnes2018-hy,
  title         = "UMAP: Uniform Manifold Approximation and Projection for
                   Dimension Reduction",
  author        = "McInnes, Leland and Healy, John and Melville, James",
  abstract      = "UMAP (Uniform Manifold Approximation and Projection) is a
                   novel manifold learning technique for dimension reduction.
                   UMAP is constructed from a theoretical framework based in
                   Riemannian geometry and algebraic topology. The result is a
                   practical scalable algorithm that applies to real world
                   data. The UMAP algorithm is competitive with t-SNE for
                   visualization quality, and arguably preserves more of the
                   global structure with superior run time performance.
                   Furthermore, UMAP has no computational restrictions on
                   embedding dimension, making it viable as a general purpose
                   dimension reduction technique for machine learning.",
  month         =  feb,
  year          =  2018,
  url           = "http://arxiv.org/abs/1802.03426",
  archivePrefix = "arXiv",
  eprint        = "1802.03426",
  primaryClass  = "stat.ML",
  arxivid       = "1802.03426"
}

@article{lun2017overcoming,
  title={Overcoming confounding plate effects in differential expression analyses of single-cell RNA-seq data},
  author={Lun, Aaron TL and Marioni, John C},
  journal={Biostatistics},
  volume={18},
  number={3},
  pages={451--464},
  year={2017},
  publisher={Oxford University Press}
}

@Article{lun2017testing,
    Author="Lun, A. T. L.  and Richard, A. C.  and Marioni, J. C. ",
    Title="{{T}esting for differential abundance in mass cytometry data}",
    Journal="Nat. Methods",
    Year="2017",
    Volume="14",
    Number="7",
    Pages="707--709",
    Month="Jul"
}

@Article{chen2016reads,
    Author="Chen, Y.  and Lun, A. T.  and Smyth, G. K. ",
    Title="{{F}rom reads to genes to pathways: differential expression analysis of {R}{N}{A}-{S}eq experiments using {R}subread and the edge{R} quasi-likelihood pipeline}",
    Journal="F1000Res",
    Year="2016",
    Volume="5",
    Pages="1438"
}

@Article{phipson2016robust,
    Author="Phipson, B. and Lee, S. and Majewski, I. J. and Alexander, W. S. and Smyth, G. K.",
    year = 2016,
    title = "Robust hyperparameter estimation protects against hypervariable genes and improves power to detect differential expression",
    Journal="Ann. Appl. Stat.",
    Volume=10,
    Number=2,
    Pages="946-963"
}

@Article{aran2019reference,
   Author="Aran, D.  and Looney, A. P.  and Liu, L.  and Wu, E.  and Fong, V.  and Hsu, A.  and Chak, S.  and Naikawadi, R. P.  and Wolters, P. J.  and Abate, A. R.  and Butte, A. J.  and Bhattacharya, M. ",
   Title="Reference-based analysis of lung single-cell sequencing reveals a transitional profibrotic macrophage",
   Journal="Nat. Immunol.",
   Year="2019",
   Volume="20",
   Number="2",
   Pages="163--172",
   Month="02"
}

@article{Benayoun2019,
author = {Benayoun, B{\'{e}}r{\'{e}}nice A. and Pollina, Elizabeth A. and Singh, Param Priya and Mahmoudi, Salah and Harel, Itamar and Casey, Kerriann M. and Dulken, Ben W. and Kundaje, Anshul and Brunet, Anne},
doi = {10.1101/gr.240093.118},
journal = {Genome Research},
volume = {29},
pages = {697-709},
title = {Remodeling of epigenome and transcriptome landscapes with aging in mice reveals widespread induction of inflammatory responses},
year = {2019}
}

@article{ImmGenRef,
    author = {Heng, Tracy S.P. and Painter, Michio W. and Elpek, Kutlu and Lukacs-Kornek, Veronika and Mauermann, Nora and Turley, Shannon J. and Koller, Daphne and Kim, Francis S. and Wagers, Amy J. and Asinovski, Natasha and Davis, Scott and Fassett, Marlys and Feuerer, Markus and Gray, Daniel H.D. and Haxhinasto, Sokol and Hill, Jonathan A. and Hyatt, Gordon and Laplace, Catherine and Leatherbee, Kristen and Mathis, Diane and Benoist, Christophe and Jianu, Radu and Laidlaw, David H. and Best, J. Adam and Knell, Jamie and Goldrath, Ananda W. and Jarjoura, Jessica and Sun, Joseph C. and Zhu, Yanan and Lanier, Lewis L. and Ergun, Ayla and Li, Zheng and Collins, James J. and Shinton, Susan A. and Hardy, Richard R. and Friedline, Randall and Sylvia, Katelyn and Kang, Joonsoo},
    journal = {Nature Immunology},
    volume = {9},
    issue = {10},
    pages = {1091-4},
    doi = {10.1038/ni1008-1091},
    title = {The immunological genome project: Networks of gene expression in immune cells},
    year = {2008}
}

@article{choi2018haemopedia,
  title={Haemopedia RNA-seq: a database of gene expression during haematopoiesis in mice and humans},
  author={Choi, Jarny and Baldwin, Tracey M and Wong, Mae and Bolden, Jessica E and Fairfax, Kirsten A and Lucas, Erin C and Cole, Rebecca and Biben, Christine and Morgan, Clare and Ramsay, Kerry A and others},
  journal={Nucleic acids research},
  volume={47},
  number={D1},
  pages={D780--D785},
  year={2018},
  publisher={Oxford University Press}
}

@Article{mccarthy2009treat,
    Author="McCarthy, D. J.  and Smyth, G. K. ",
    Title="{Testing significance relative to a fold-change threshold is a TREAT}",
    Journal="Bioinformatics",
    Year="2009",
    Volume="25",
    Number="6",
    Pages="765--771",
    Month="Mar"
}

@article{hpcaRef,
author = {Mabbott, Neil A. and Baillie, J. K. and Brown, Helen and Freeman, Tom C. and Hume, David A.},
doi = {10.1186/1471-2164-14-632},
journal = {BMC Genomics},
volume = {14},
title = {{An expression atlas of human primary cells: Inference of gene function from coexpression networks}},
year = {2013}
}

@article{blueprintRef,
    author = {Martens, Joost H A and Stunnenberg, Hendrik G.},
    doi = {10.3324/haematol.2013.094243},
    journal = {Haematologica},
    title = {{BLUEPRINT: Mapping human blood cell epigenomes}},
    year = {2013},
    volume = {98},
    pages = {1487-1489}
}

@article{encodeRef,
author = {{The ENCODE Project Consortium}},
doi = {10.1038/nature11247},
issn = {14764687},
journal = {Nature},
title = {{An integrated encyclopedia of DNA elements in the human genome}},
year = {2012}
}

@article{young10soupx,
  title={SoupX removes ambient RNA contamination from droplet based single cell RNA sequencing data. bioRxiv. 2018},
  author={Young, MD and Behjati, S},
  journal={DOI},
  volume={10},
  pages={303727}
}
</script>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
