---
title: "Analysing the repertoire sequencing data from the Cooney (C057) memory CD4+ T-cell data set"
description: |
author:
  - name: Peter Hickey 
    url: https://peterhickey.org
    affiliation: Single Cell Open Research Endeavour (SCORE), WEHI
    affiliation_url: https://www.wehi.edu.au/people/shalin-naik/3310/score
date: "`r Sys.Date()`"
output: distill::distill_article
editor_options: 
  chunk_output_type: console
bibliography: ref.bib
---

```{r setup}
library(SingleCellExperiment)
library(here)
library(dplyr)
library(BiocParallel)
library(ggplot2)
library(cowplot)
library(scater)

source(here("code", "helper_functions.R"))

# NOTE: Using multiple cores seizes up my laptop. Can use more on unix box.
options("mc.cores" = ifelse(Sys.info()[["nodename"]] == "PC1331", 2L, 8L))
register(MulticoreParam(workers = getOption("mc.cores")))

knitr::opts_chunk$set(fig.path = "C057_Cooney.repertoire-seq_files/")
```

# Introduction

An organism’s immune repertoire is defined as the set of T and B cell subtypes that contain genetic diversity in the T cell receptor (TCR) components or immunoglobin chains, respectively.
This diversity is important for ensuring that the adaptive immune system can respond effectively to a wide range of antigens.
We can profile the immune repertoire by simply sequencing the relevant transcripts [@georgiou2014promise; @rosati2017overview], which can be combined with scRNA-seq technologies to achieve single-cell resolution.
This data can then be used to characterize an individual’s immune response based on the expansion of T or B cell clones, i.e., multiple cells with the same sequences for each TCR component or immunoglobulin chain.

Here, we have profiled the immune repertoire of the CD4+ T-cells using 10X Genomics' 'Single Cell V(D)J + 5′ Gene Expression' kit.
We aim to analyse the immune repertoire and integrate it with the RNA sequencing data.

# Preparing the data

## RNA-sequencing

For the RNA-sequencing data, start from the preprocessed *SingleCellExperiment* object created in ['Annotating the Cooney (C057) memory CD4+ T-cell data set'](C057_Cooney.annotate.html).

```{r}
# TODO: Use the 'annotated' data when it's ready.
# sce <- readRDS(here("data", "SCEs", "C057_Cooney.annotated.SCE.rds"))
sce <- readRDS(here("data", "SCEs", "C057_Cooney.cells_selected.SCE.rds"))

# data frames containing co-ordinates and factors for creating reduced 
# dimensionality plots.
umap_df <- cbind(
  data.frame(
    x = reducedDim(sce, "UMAP")[, 1],
    y = reducedDim(sce, "UMAP")[, 2]),
  as.data.frame(colData(sce)[, !colnames(colData(sce)) %in% c("TRA", "TRB")]))

# Some useful colours
sample_colours <- setNames(
  unique(sce$sample_colours),
  unique(names(sce$sample_colours)))
treatment_colours <- setNames(
  unique(sce$treatment_colours),
  unique(names(sce$treatment_colours)))
cluster_colours <- setNames(
  unique(sce$cluster_colours),
  unique(names(sce$cluster_colours)))
label_main_collapsed_colours <- setNames(
  unique(sce$label_main_collapsed_colours),
  unique(names(sce$label_main_collapsed_colours)))
label_fine_collapsed_colours <- setNames(
  unique(sce$label_fine_collapsed_colours),
  unique(names(sce$label_fine_collapsed_colours)))
```

## TCR-sequencing

CellRanger generates a file containing filtered TCR contig annotations from the set of cells that passed CellRanger's internal quality controls of the RNA-sequencing data ([`data/CellRanger/HTLV.filtered_contig_annotations.csv`](../data/CellRanger/HTLV.filtered_contig_annotations.csv)).

<aside>
I have further filtered this file such that it only contains annotations from the set of cells remaining at the end of ['Annotating the Cooney (C057) memory CD4+ T-cell data set'](C057_Cooney.annotate.html).
</aside>

Each row of the file contains information about a single TCR component sequence in one cell, broken down into the alleles of the V(D)J genes making up that component (`v_gene`, `d_gene`, `j_gene`) where possible.
The number of reads and UMIs supporting the set of allele assignments for a cell is also shown, though only the UMI count should be used for quantifying expression of a particular TCR sequence.
Each cell is assigned to a clonotype (`raw_clonotype_id`) based on the combination of the α-chain (TRA) and β-chain (TRB) sequences in that cell.
The first few rows of data for the TRA are shown below:

```{r}
example_barcode <- "AAACCTGAGCCAACAG-1"
stopifnot(nrow(sce$TRA[[example_barcode]]) == 2)
rmarkdown::paged_table(as.data.frame(unlist(head(sce$TRA), use.names = FALSE)))
```

<aside>
A challenge lies in the fact that each cell may have zero, one, or many TCR sequences.
For example, the `r example_barcode` barcode has two TRA sequences.
</aside>

# Basic diagnostics

## Motivation

We start by generating some basic cell-level diagnostics to assess the quality and utility of the TCR sequencing data.

## Analysis

Figure \@ref(fig:prop-expressing) plots the proportion of cells that express each of the α-chain and β-chain for each cluster.
`r round(100 * sum(lengths(sce$TRA) >= 1 & lengths(sce$TRB) >= 1) / ncol(sce), 1)`% of cells express both the α-chain and β-chain.
This is as we expect given we have shown that the selected cells are all likely to be (CD4+) T cells.

```{r prop-expressing, fig.cap = "UMAP plot of data with points coloured by cluster (left) and a barplot of the proportion of cells in each cluster that express at least one sequence of the TCR α or β-chains (right).", fig.asp = 1 / 3}
library(tibble)
tra_tbl <- tibble(
  cluster = factor(levels(sce$cluster), levels(sce$cluster)),
  n_cells = as.vector(table(sce$cluster)),
  prop = as.vector(table(sce$cluster[lengths(sce$TRA) > 0]) / n_cells),
  chain = "α")
trb_tbl <- tibble(
  cluster = factor(levels(sce$cluster), levels(sce$cluster)),
  n_cells = as.vector(table(sce$cluster)),
  prop = as.vector(table(sce$cluster[lengths(sce$TRB) > 0]) / n_cells),
  chain = "β")

plot_grid(
  plotUMAP(sce, colour_by = "cluster", point_alpha = 0) + 
    geom_point(aes(colour = colour_by), size = 0.05, alpha = 0.5) +
    scale_colour_manual(values = cluster_colours, name = "cluster") +
    guides(colour = FALSE, fill = FALSE) + 
    ggtitle("Cluster") +
    theme_cowplot(font_size = 8),
  ggplot(rbind(tra_tbl, trb_tbl)) + 
    geom_col(aes(x = cluster, y = prop, fill = cluster)) + 
    facet_grid(~ chain) +
    scale_fill_manual(values = cluster_colours, name = "cluster") +
    theme_cowplot(font_size = 8) +
    ylim(0, 1) +
    ylab("Proportion") +
    guides(fill = FALSE) +
    ggtitle("TCR component"),
  rel_widths = c(1, 2),
  nrow = 1)
```

We can refine this analysis to only consider the *productive* sequences, i.e., contigs that are likely to produce a functional protein.
Figure \@ref(fig:prop-expressing-productive) plots the proportion of cells that express productive versions of each of the α-chain and β-chain for each cluster.
`r round(100 * sum(any(sce$TRA[, "productive"] == "True") & any(sce$TRB[, "productive"] == "True")) / ncol(sce), 1)`% of cells express productive versions of both the α-chain and β-chain.

```{r prop-expressing-productive, fig.cap = "UMAP plot of data with points coloured by cluster (left) and a barplot of the proportion of cells in each cluster that express at least one **productive** sequence of the TCR α or β-chains (right).", fig.asp = 1 / 3}
tra_tbl <- tibble(
  cluster = factor(levels(sce$cluster), levels(sce$cluster)),
  n_cells = as.vector(table(sce$cluster)),
  prop = as.vector(
    table(
      sce$cluster[lengths(sce$TRA[sce$TRA[, "productive"] == "True"]) > 0]) /
      n_cells),
  chain = "α")
trb_tbl <- tibble(
  cluster = factor(levels(sce$cluster), levels(sce$cluster)),
  n_cells = as.vector(table(sce$cluster)),
  prop = as.vector(
    table(
      sce$cluster[lengths(sce$TRB[sce$TRB[, "productive"] == "True"]) > 0]) /
      n_cells),
  chain = "β")

plot_grid(
  plotUMAP(sce, colour_by = "cluster", point_alpha = 0) + 
    geom_point(aes(colour = colour_by), size = 0.05, alpha = 0.5) +
    scale_colour_manual(values = cluster_colours, name = "cluster") +
    guides(colour = FALSE, fill = FALSE) + 
    ggtitle("Cluster") +
    theme_cowplot(font_size = 8),
  ggplot(rbind(tra_tbl, trb_tbl)) + 
    geom_col(aes(x = cluster, y = prop, fill = cluster)) + 
    facet_grid(~ chain) +
    scale_fill_manual(values = cluster_colours, name = "cluster") +
    theme_cowplot(font_size = 8) +
    ylim(0, 1) +
    ylab("Proportion") +
    guides(fill = FALSE) +
    ggtitle("TCR component"),
  rel_widths = c(1, 2),
  nrow = 1)
```

For completeness, we also count the number of cells in each cluster that have multiple sequences for a component (Figure \@ref(fig:prop-expressing-multiple)).
The percentages are clearly lower but this phenomenon is still surprisingly common.

```{r prop-expressing-multiple, fig.cap = "UMAP plot of data with points coloured by cluster (left) and a barplot of the proportion of cells in each cluster that express two or more sequences of the TCR α or β-chains (right).", fig.asp = 1 / 3}
tra_tbl <- tibble(
  cluster = factor(levels(sce$cluster), levels(sce$cluster)),
  n_cells = as.vector(table(sce$cluster)),
  prop = as.vector(table(sce$cluster[lengths(sce$TRA) >= 2]) / n_cells),
  chain = "α")
trb_tbl <- tibble(
  cluster = factor(levels(sce$cluster), levels(sce$cluster)),
  n_cells = as.vector(table(sce$cluster)),
  prop = as.vector(table(sce$cluster[lengths(sce$TRB) >= 2]) / n_cells),
  chain = "β")

plot_grid(
  plotUMAP(sce, colour_by = "cluster", point_alpha = 0) + 
    geom_point(aes(colour = colour_by), size = 0.05, alpha = 0.5) +
    scale_colour_manual(values = cluster_colours, name = "cluster") +
    guides(colour = FALSE, fill = FALSE) + 
    ggtitle("Cluster") +
    theme_cowplot(font_size = 8),
  ggplot(rbind(tra_tbl, trb_tbl)) + 
    geom_col(aes(x = cluster, y = prop, fill = cluster)) + 
    facet_grid(~ chain) +
    scale_fill_manual(values = cluster_colours, name = "cluster") +
    theme_cowplot(font_size = 8) +
    ylim(0, 1) +
    ylab("Proportion") +
    guides(fill = FALSE) +
    ggtitle("TCR component"),
  rel_widths = c(1, 2),
  nrow = 1)
```

However, if we focus only those sequences deemed *productive*, then Figure \@ref(fig:prop-expressing-multiple-productive) shows that this percentage drops.

```{r prop-expressing-multiple-productive, fig.cap = "UMAP plot of data with points coloured by cluster (left) and a barplot of the proportion of cells in each cluster that express two or more **productive** sequences of the TCR α or β-chains (right).", fig.asp = 1 / 3}
tra_tbl <- tibble(
  cluster = factor(levels(sce$cluster), levels(sce$cluster)),
  n_cells = as.vector(table(sce$cluster)),
  prop = as.vector(
    table(
      sce$cluster[lengths(sce$TRA[sce$TRA[, "productive"] == "True"]) > 1]) /
      n_cells),
  chain = "α")

trb_tbl <- tibble(
  cluster = factor(levels(sce$cluster), levels(sce$cluster)),
  n_cells = as.vector(table(sce$cluster)),
  prop = as.vector(
    table(
      sce$cluster[lengths(sce$TRB[sce$TRB[, "productive"] == "True"]) > 1]) /
      n_cells),
  chain = "β")

plot_grid(
  plotUMAP(sce, colour_by = "cluster", point_alpha = 0) + 
    geom_point(aes(colour = colour_by), size = 0.05, alpha = 0.5) +
    scale_colour_manual(values = cluster_colours, name = "cluster") +
    guides(colour = FALSE, fill = FALSE) + 
    ggtitle("Cluster") +
    theme_cowplot(font_size = 8),
  ggplot(rbind(tra_tbl, trb_tbl)) + 
    geom_col(aes(x = cluster, y = prop, fill = cluster)) + 
    facet_grid(~ chain) +
    scale_fill_manual(values = cluster_colours, name = "cluster") +
    theme_cowplot(font_size = 8) +
    ylim(0, 1) +
    ylab("Proportion") +
    guides(fill = FALSE) +
    ggtitle("TCR component"),
  rel_widths = c(1, 2),
  nrow = 1)
```

## Summary

As we would expect of (CD4+) T-cells:

- `r round(100 * sum(lengths(sce$TRA) >= 1 & lengths(sce$TRB) >= 1) / ncol(sce), 1)`% of cells express both the α-chain and β-chain.
- `r round(100 * sum(any(sce$TRA[, "productive"] == "True") & any(sce$TRB[, "productive"] == "True")) / ncol(sce), 1)`% of cells express *productive* versions of both the α-chain and β-chain.
- Somewhat surprisingly, `r round(100 * sum(sapply(sce$TRA[, "productive"], function(x) sum(x == "True")) >= 2) / ncol(sce))`% and `r round(100 * sum(sapply(sce$TRB[, "productive"], function(x) sum(x == "True")) >= 2) / ncol(sce))`% of cells express multiple productive versions of the α-chain and β-chain, respectively.

**It is still not clear to me if a single T cell can really express multiple (productive) sequences for a TCR component.**
However, a quick Google search suggests that it is possible (e.g., [https://www.ncbi.nlm.nih.gov/pubmed/8211163](https://www.ncbi.nlm.nih.gov/pubmed/8211163); [https://www.jimmunol.org/content/202/3/637](https://www.jimmunol.org/content/202/3/637); [https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4701647/](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4701647/))^[An alternative interpretation is that these are really doublets and hence would likely have multiple distinct TCR components (although this assumes a higher doublet rate than we expect).].

# Quantifying clonotype expansion

## Motivation

Quantification of clonal expansion is the most obvious application of repertoire sequencing data.
Cells with the same T cell clonotype are assumed to target the same antigen, and any increase in the frequency of a clonotype provides evidence for T cell activation and proliferation upon stimulation by the corresponding antigen.
Thus, we can gain some insights into the immune activity of each sample by counting the frequency of clonotypes in each sample.

## Analysis

CellRanger includes a histogram of top-10 clonotypes by frequency in the dataset.
Only productive sequences are included in this analysis.
This histogram is reproduced in Figure \@ref(fig:cellranger) but is of limited use because it doesn't distinguish which sample each observation came from.

```{r cellranger, fig.cap = "Top-10 clonotype frequencies (all droplets in the 'cell selected' dataset). A clonotype is defined as a unique set of CDR3 nucleotide sequences. Only productive sequences are counted."}
rbind(unlist(sce$TRA), unlist(sce$TRB)) %>%
  as.data.frame() %>%
  dplyr::filter(productive == "True") %>%
  dplyr::distinct(barcode, raw_clonotype_id) %>%
  dplyr::group_by(raw_clonotype_id) %>%
  dplyr::count() %>%
  dplyr::arrange(desc(n)) %>%
  dplyr::ungroup() %>%
  dplyr::mutate(prop = n / sum(n)) %>%
  head(n = 10) %>%
  dplyr::mutate(clonotype = factor(
    gsub("clonotype", "", raw_clonotype_id),
    1:10)) %>%
  ggplot(aes(x = clonotype, y = prop)) +
  geom_col() + 
  xlab("Clonotype") +
  ylab("Proportion of cells") +
  theme_cowplot()
```

For this project, we are interested in stratifying the analysis by sample to look clonal expansions associated with the treatment.
Figure \@ref(fig:cellranger-faceted) stratifies by sample the histogram of the top-10 clonotypes by frequency.

```{r cellranger-faceted, fig.cap = "Top-10 clonotype frequencies by sample. A clonotype is defined as a unique set of CDR3 nucleotide sequences. Only productive sequences are counted."}
b <- dplyr::inner_join(
  as.data.frame(colData(sce)[, !colnames(colData(sce)) %in% c("TRA", "TRB")]),
  as.data.frame(rbind(unlist(sce$TRA), unlist(sce$TRB))),
  by = c("Barcode" = "barcode")) %>%
  dplyr::filter(productive == "True", Treatment != "Unknown") %>%
  dplyr::distinct(Sample, Barcode, raw_clonotype_id, Treatment) %>%
  dplyr::group_by(raw_clonotype_id, Sample, Treatment) %>%
  dplyr::count() %>%
  dplyr::arrange(desc(n)) %>%
  dplyr::group_by(Sample) %>%
  dplyr::mutate(prop = n / sum(n))

bb <- b %>% 
  dplyr::group_by(Sample) %>%
  dplyr::mutate(
    cumsum_prop = cumsum(prop),
    x = row_number(),
    xx = x / max(x))

bb %>% 
  dplyr::filter(x <= 10) %>%
  dplyr::arrange(Sample, x) %>%
  dplyr::mutate(raw_clonotype_id = factor(x)) %>%
  ggplot(aes(x = raw_clonotype_id, y = prop, fill = Sample)) +
  geom_col() + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) + 
  facet_wrap(~Sample, ncol = 3) + 
  xlab("Clonotype") +
  ylab("Proportion of cells") + 
  theme_cowplot() +
  scale_fill_manual(values = sample_colours) + 
  guides(fill = FALSE)
```

This shows an apparent clonal expansion in the `infected_2` and `infected_3` samples, which we want to investigate further.
When I started this analysis, I was not very familiar with the aims of TCR sequencing, the data it generated, or the methods used for its analysis.
I therefore started with a [Naive analysis] which has subsequently developed into a more [Sophisticated analysis] of TCR diversity.

### Naive analysis

Figure \@ref(fig:cumulative-clonotype-frequencies) plots the cumulative clonotype frequencies in each sample as an alternative to simply visualising the frequency of the top-10 clonotypes in each sample. 

```{r cumulative-clonotype-frequencies, fig.cap = "Cumulative clonotype frequencies by sample. Frequencies are computed within each sample."}
ggplot(bb) + 
  geom_step(
    aes(x = x, y = cumsum_prop, colour = Sample, lty = Treatment),
    lwd = 1) +
  scale_colour_manual(values = sample_colours) + 
  xlab("Clonotype") + 
  ylab("Proportion of cells") +
  cowplot::theme_cowplot()
```

This shows, for example, that more than 75% of cells from the `infected_3` sample share a very small number of clonotypes..
This figure also shows that fewer clonotypes are observed in the `Infected` samples.
However, this could arise simply because we sampled fewer infected cells.
We therefore need to account for the number of *potential* clonotypes in each sample, i.e. if we sampled $n$ cells there would be $n$ *potential* clonotypes.

Two possible estimates of the number of *potential* clonotypes in each sample are:

1. The number of cells from that sample (likely an overestimate as not all cells necessarily express a TCR).
2. The number of droplets with a TCR sequence from that sample (likely an underestimate as some of the cells with a TCR didn't have the TCR sequenced).

I opt to estimate the number of potential clonotypes by (1).
Fortunately, the table below shows that these two estimates do not differ much (they are roughly within 5% of each other) so the results will not be sensitive to the choice of estimate.

```{r}
knitr::kable(
  data.frame(
    Sample = levels(factor(sce$Sample)),
    est_1 = tabulate(factor(sce$Sample)),
    est_2 = tapply(bb$n, factor(bb$Sample), sum)),
  caption = "Two estimates of the number of *potential* clonotypes in each sample. 'est_1' = The number of cells from that sample (likely an overestimate as not all droplets necessarily express a TCR). 'est_2' = The number of cells with a TCR sequence from that sample (likely an underestimate as some of the cells with a TCR didn't have the TCR sequenced).")
```

Figure \@ref(fig:cumulative-potential-clonotype-frequencies) basically repeats Figure \@ref(fig:cumulative-clonotype-frequencies) but extends the x-axis out to the number of *potential* clonotypes.

```{r cumulative-potential-clonotype-frequencies, fig.cap = "Cumulative potential clonotype frequencies by sample. Frequencies are computed within each sample."}
z <- lapply(levels(factor(sce$Sample)), function(s) {
  a <- dplyr::filter(b, Sample == s)
  # NOTE: Opted to use the (1) estimator
  n <- c(a$n, rep(0, sum(sce$Sample == s) - length(a$n)))
  # n <- c(a$n, rep(0, sum(a$n) - length(a$n)))
  n
})

tmp <- data.frame(
  Sample = rep(levels(factor(sce$Sample)), lengths(z)),
  Treatment = rep(sub("\\_[0-9]", "", levels(factor(sce$Sample))), lengths(z)),
  n = unlist(z))

tmp %>%
  dplyr::group_by(Sample) %>%
  dplyr::mutate(
    x = seq_along(n),
    y = cumsum(n) / sum(n)) %>%
  ggplot(.) +
  geom_step(
    aes(x = x, y = y, colour = Sample, lty = Treatment),
    lwd = 1) +
  scale_colour_manual(values = sample_colours) + 
  xlab("Potential clonotypes") +
  ylab("Proportion of cells") +
  cowplot::theme_cowplot()
```

Finally, Figure \@ref(fig:cumulative-potential-clonotype-frequencies-normalized) normalises by the number of *potential* haplotypes in each sample to make the samples comparable.

```{r cumulative-potential-clonotype-frequencies-normalized, fig.cap = "Normalised cumulative potential clonotype frequencies by sample. Frequencies are computed within each sample."}
tmp %>%
  dplyr::group_by(Sample) %>%
  dplyr::mutate(
    x = seq_along(n) / length(n),
    y = cumsum(n) / sum(n)) %>%
  ggplot(.) +
  geom_step(
    aes(x = x, y = y, colour = Sample, lty = Treatment),
    lwd = 1) +
  scale_colour_manual(values = sample_colours) + 
  xlab("Proportion of potential clonotypes") +
  ylab("Proportion of cells") +
  cowplot::theme_cowplot()
```

From this, we see evidence that fewer clonotypes are found in infected cells than in uninfected cells.
We would like to quantify 
For example, the table below reports the proportion of potential clonotypes required to 'cover' 75% of cells in each sample and shows that this number is somewhat smaller in infected cells than in uninfected cells.

```{r}
tmp %>%
  dplyr::group_by(Sample) %>%
  dplyr::mutate(
    x = seq_along(n) / length(n),
    y = cumsum(n) / sum(n)) %>%
  dplyr::filter(y > 0.75) %>%
  dplyr::group_by(Sample) %>%
  dplyr::mutate(i = dplyr::row_number()) %>%
  dplyr::filter(i == 1) %>%
  dplyr::select(Sample, x) %>%
  dplyr::rename("Proportion of potential clonotypes" = x) %>%
  knitr::kable(caption = "Proportion of potential clonotypes required to 'cover' 75% of cells in each sample.", digits = 3)
```

### Sophisticated analysis

I use the `r BiocStyle::CRANpkg("alakazam")` R package to perform a more sophisticated diversity analysis.

<aside>
The caveat to this is that I don't understand the methodology as much as I might like.
</aside>

The authors of `r BiocStyle::CRANpkg("alakazam")` [write that](https://alakazam.readthedocs.io/en/stable/vignettes/Diversity-Vignette/#diversity-analysis):

> The clonal diversity of the repertoire can be analysed using the general form of the diversity index, as proposed by Hill[@hill1973diversity].
> [This is] coupled with re-sampling strategies to correct for variations in sequencing depth, as well as inference of complete clonal abundance distributions[@chao2014rarefaction,@chao2015unveiling].

```{r}
library(alakazam)

alakazam_df <- as.data.frame(unlist(sce$TRA)) %>%
  filter(productive == "True") %>%
  select(barcode, raw_clonotype_id) %>%
  mutate(seq_count = 1) %>%
  inner_join(
    as.data.frame(
      colData(sce)[, !colnames(colData(sce)) %in% c("TRA", "TRB")]),
    by = c("barcode" = "Barcode"))
```

We start by generating a clonal abundance curve for each `Sample` and each `Treatment` (Figure \@ref(fig:clonal-abundance)).

<aside>
These curves are similar in flavour to Figure \@ref(fig:cellranger-faceted), but more detailed and sophisticated.
</aside>

```{r clonal-abundance, fig.cap = "Rank abundance curve (with 95% CI) of the relative clonal abundance for each `Sample` (left) and `Treatment` (right). The 95% confidence interval is estimated via bootstrapping (B = 200).", fig.asp = 1 / 3}
sample_curve <- estimateAbundance(
  alakazam_df,
  group = "Sample",
  ci = 0.95,
  nboot = 200,
  clone = "raw_clonotype_id")
p1 <- plot(sample_curve, colors = sample_colours, silent = TRUE)

treatment_curve <- estimateAbundance(
  alakazam_df,
  group = "Treatment",
  ci = 0.95,
  nboot = 200,
  clone = "raw_clonotype_id")
p2 <- plot(treatment_curve, colors = treatment_colours, silent = TRUE)

plot_grid(p1, p2, ncol = 2)
```

Next, we generate a diversity curve for each `Sample` and each `Treatment`.
A diversity curve quantifies diversity as a smooth function ($D$) of a single parameter $q$[@hill1973diversity].
Special cases of the generalized diversity index correspond to the most popular diversity measures in ecology: species richness ($q = 0$), the exponential of the Shannon-Weiner index ($q$ approaches $1$), the inverse of the Simpson index ($q = 2$), and the reciprocal abundance of the largest clone ($q$ approaches $+\infty$). 
At $q = 0$ different clones weight equally, regardless of their size.
As the parameter $q$ increase from $0$ to $+\infty$ the diversity index ($D$) depends less on rare clones and more on common (abundant) ones, thus encompassing a range of definitions that can be visualized as a single curve^[Values of q < 0 are valid, but are generally not meaningful.].

Figure \@ref(fig:diversity) plots these diversity curves and shows that the `Uninfected` samples are 'more diverse' for a range of $q$.

```{r diversity, fig.cap = "Diveristy curve (with 95% CI) for each `Sample` (left) and `Treatment` (right). The 95% confidence interval is estimated via bootstrapping (B = 200).", fig.asp = 1 / 3}
sample_alpha_curve <- alphaDiversity(
  sample_curve,
  min_q = 0,
  max_q = 4,
  step_q = 0.1,
  ci = 0.95,
  nboot = 200)
p1 <- plot(sample_alpha_curve, colors = sample_colours, silent = TRUE)

treatment_alpha_curve <- alphaDiversity(
  treatment_curve,
  min_q = 0,
  max_q = 4,
  step_q = 0.1,
  ci = 0.95,
  nboot = 200)
p2 <- plot(treatment_alpha_curve, colors = treatment_colours, silent = TRUE)

plot_grid(p1, p2, ncol = 2)
```

Finally, significance testing across groups is performed using the delta of the bootstrap distributions between groups when running.

**TODO:** Not real confident how these methods work.

```{r}
filter(sample_alpha_curve@tests, q %in% c(0, 1, 2, 4))
```

# Summary

- Infected samples have fewer clonotypes (i.e. have less TCR diversity) than uninfected cells, even after normalising for cell numbers.
- Infected samples show evidence of clonal expansion (alternatively, of reduced polyclonality) compared to uninfected samples as evidenced by more cells sharing a smaller number of clonotypes.

# Additional information {.appendix}

The following are available on request:

- Full CSV tables of any data presented.
- PDF/PNG files of any static plots.

# Session info {.appendix}

<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>

```{r}
sessioninfo::session_info()
```

</details>
