---
title: "Annotating the Cooney (C057) memory CD4+ T-cell data set"
description: |
author:
  - name: Peter Hickey 
    url: https://peterhickey.org
    affiliation: Single Cell Open Research Endeavour (SCORE), WEHI
    affiliation_url: https://www.wehi.edu.au/people/shalin-naik/3310/score
date: "`r Sys.Date()`"
output: distill::distill_article
editor_options: 
  chunk_output_type: console
bibliography: ref.bib
---

# Setup

```{r}
library(SingleCellExperiment)
library(here)
library(cowplot)
library(patchwork)

sce <- readRDS(here("data/SCEs/C057_Cooney.cells_selected.SCE.rds"))

# data frames containing co-ordinates and factors for creating reduced
# dimensionality plots.
umap_df <- cbind(
  data.frame(
    x = reducedDim(sce, "UMAP")[, 1],
    y = reducedDim(sce, "UMAP")[, 2]),
  as.data.frame(colData(sce)[, !colnames(colData(sce)) %in% c("TRA", "TRB")]))

# Some useful colours
sample_colours <- setNames(
  unique(sce$sample_colours),
  unique(names(sce$sample_colours)))
treatment_colours <- setNames(
  unique(sce$treatment_colours),
  unique(names(sce$treatment_colours)))
cluster_colours <- setNames(
  unique(sce$cluster_colours),
  unique(names(sce$cluster_colours)))

# Some useful gene sets
mito_set <- rownames(sce)[which(rowData(sce)$CHR == "MT")]
ribo_set <- grep("^RP(S|L)", rownames(sce), value = TRUE)
# NOTE: A more curated approach for identifying ribosomal protein genes
#       (https://github.com/Bioconductor/OrchestratingSingleCellAnalysis-base/blob/ae201bf26e3e4fa82d9165d8abf4f4dc4b8e5a68/feature-selection.Rmd#L376-L380)
library(msigdbr)
c2_sets <- msigdbr(species = "Homo sapiens", category = "C2")
ribo_set <- union(
  ribo_set,
  c2_sets[c2_sets$gs_name == "KEGG_RIBOSOME", ]$human_gene_symbol)

source(here("code", "helper_functions.R"))
```

## Re-processing

This is a no-op if the same hvg set is used from cell selection.

```{r}
library(scran)
set.seed(1000)
var_fit <- modelGeneVarByPoisson(sce, block = sce$batch)
hvg <- getTopHVGs(var_fit, var.threshold = 0)
is_mito <- hvg %in% mito_set
is_ribo <- hvg %in% ribo_set
hvg <- hvg[!(is_mito | is_ribo)]

set.seed(1010)
var_fit.sample <- modelGeneVarByPoisson(sce, block = sce$Sample)
hvg.sample <- getTopHVGs(var_fit.sample, var.threshold = 0)
is_mito <- hvg.sample %in% mito_set
is_ribo <- hvg.sample %in% ribo_set
hvg.sample <- hvg.sample[!(is_mito | is_ribo)]

library(scater)
# NOTE: 
set.seed(11235)
sce <- denoisePCA(sce, var_fit, subset.row = hvg)
set.seed(8875)
sce <- runUMAP(sce, dimred = "PCA")
umap_df <- cbind(
  data.frame(
    x = reducedDim(sce, "UMAP")[, 1],
    y = reducedDim(sce, "UMAP")[, 2]),
  as.data.frame(colData(sce)[, !colnames(colData(sce)) %in% c("TRA", "TRB")]))
set.seed(8111)
snn_gr <- buildSNNGraph(sce, use.dimred = "PCA")
clusters <- igraph::cluster_louvain(snn_gr)
sce$cluster <- factor(clusters$membership)
umap_df$cluster <- sce$cluster
cluster_colours <- setNames(
  Polychrome::glasbey.colors(nlevels(sce$cluster) + 1)[-1],
  levels(sce$cluster))
sce$cluster_colours <- cluster_colours[sce$cluster]
```

# Diagnosing batch effects

- Samples were collected in a single capture, but there are some strong differences between Venetoclax and Control samples.
- Explore whether batch effect exists and how it might be corrected.

```{r}
table(cluster = sce$cluster, treatment = sce$Treatment)
```

```{r, fig.asp = 1/2}
uncorrected.p1 <- plotUMAP(sce, colour_by = "Treatment", point_size = 0.5) +
  scale_colour_manual(values = treatment_colours, name = "Treatment") +
  ggtitle("Uncorrected")
uncorrected.p2 <- plotUMAP(sce, colour_by = "cluster", point_size = 0.5) +
  scale_colour_manual(values = cluster_colours, name = "cluster") +
  ggtitle("Uncorrected")
uncorrected.p1 + uncorrected.p2
```

```{r, fig.asp = 1 / 3}
p1 <- ggplot(as.data.frame(colData(sce)[, c("cluster", "Sample")])) +
  geom_bar(
    aes(x = cluster, fill = Sample),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = sample_colours) +
  theme_cowplot(font_size = 8)
p2 <- ggplot(as.data.frame(colData(sce)[, c("cluster", "Treatment")])) +
  geom_bar(
    aes(x = cluster, fill = Treatment),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = treatment_colours) +
  theme_cowplot(font_size = 8)
p3 <- ggplot(as.data.frame(colData(sce)[, "cluster", drop = FALSE])) +
  geom_bar(aes(x = cluster, fill = cluster)) +
  coord_flip() +
  ylab("Number of cells") +
  scale_fill_manual(values = cluster_colours) +
  theme_cowplot(font_size = 8) +
  guides(fill = FALSE)
p1 + p2 + p3
```

## Linear regression with `rescaleBatches()`

```{r}
library(batchelor)
rescaled <- rescaleBatches(sce, batch = sce$Treatment)
library(scran)
set.seed(666)
rescaled <- denoisePCA(
  rescaled,
  var_fit,
  subset.row = hvg,
  assay.type = "corrected")
snn.gr <- buildSNNGraph(rescaled, use.dimred="PCA")
clusters.resc <- igraph::cluster_louvain(snn.gr)$membership
rescaled <- runUMAP(rescaled, dimred="PCA")
rescaled$batch <- factor(rescaled$batch)
rescaled$cluster <- factor(clusters.resc)
table(cluster=clusters.resc, treatment=rescaled$batch)
```

```{r, fig.asp = 1 / 2}
rescaled.p1 <- plotUMAP(rescaled, colour_by = "batch", point_size = 0.5) +
  scale_colour_manual(values = treatment_colours, name = "Treatment") +
  ggtitle("Rescaled")
rescaled.p2 <- plotUMAP(rescaled, colour_by = "cluster", point_size = 0.5) +
  ggtitle("Rescaled")
rescaled.p1 + rescaled.p2
```

```{r, fig.asp = 1 / 3}
p1 <- ggplot(as.data.frame(
  cbind(
    colData(sce)[, "Sample", drop = FALSE],
    colData(rescaled)[, "cluster", drop = FALSE]))) +
  geom_bar(
    aes(x = cluster, fill = Sample),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
    scale_fill_manual(values = sample_colours) +
    theme_cowplot(font_size = 8)
p2 <- ggplot(as.data.frame(
    cbind(
      colData(sce)[, "Treatment", drop = FALSE],
      colData(rescaled)[, "cluster", drop = FALSE]))) +
    geom_bar(
      aes(x = cluster, fill = Treatment),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = treatment_colours) +
    theme_cowplot(font_size = 8)
p3 <- ggplot(as.data.frame(colData(rescaled)[, "cluster", drop = FALSE])) +
  geom_bar(aes(x = cluster, fill = cluster)) +
  coord_flip() +
  ylab("Number of cells") +
  theme_cowplot(font_size = 8) +
  guides(fill = FALSE)
p1 + p2 + p3
```

## Linear regression with `regressBatches()`

```{r}
set.seed(10001)
residuals <- regressBatches(sce, batch = sce$Treatment, d = 50,
                            subset.row=hvg, correct.all=TRUE)

snn.gr <- buildSNNGraph(residuals, use.dimred="corrected")
clusters.resid <- igraph::cluster_louvain(snn.gr)$membership
table(Cluster=clusters.resid, treatment=residuals$batch)
```

```{r, fig.asp = 1 / 2}
residuals <- runUMAP(residuals, dimred="corrected")
residuals$batch <- factor(residuals$batch)
residuals$cluster <- factor(clusters.resid)
residuals.p1 <- plotUMAP(residuals, colour_by = "batch", point_size = 0.5) +
  scale_colour_manual(values = treatment_colours, name = "Treatment") +
  ggtitle("Residuals")
residuals.p2 <- plotUMAP(residuals, colour_by = "cluster", point_size = 0.5) +
  ggtitle("Residuals")
residuals.p1 + residuals.p2
```

```{r, fig.asp = 1 / 3}
p1 <- ggplot(as.data.frame(
    cbind(
      colData(sce)[, "Sample", drop = FALSE],
      colData(residuals)[, "cluster", drop = FALSE]))) +
    geom_bar(
      aes(x = cluster, fill = Sample),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = sample_colours) +
    theme_cowplot(font_size = 8)
p2 <- ggplot(as.data.frame(
    cbind(
      colData(sce)[, "Treatment", drop = FALSE],
      colData(residuals)[, "cluster", drop = FALSE]))) +
    geom_bar(
      aes(x = cluster, fill = Treatment),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = treatment_colours) +
    theme_cowplot(font_size = 8)
p3 <- ggplot(as.data.frame(colData(residuals)[, "cluster", drop = FALSE])) +
    geom_bar(aes(x = cluster, fill = cluster)) +
    coord_flip() +
    ylab("Number of cells") +
    theme_cowplot(font_size = 8) +
    guides(fill = FALSE)
p1 + p2 + p3
```


## MNN by treatment

```{r}
set.seed(1000101001)
mnn_treatment.out <- fastMNN(
  sce,
  batch = sce$Treatment,
  d = 50,
  k = 20,
  subset.row = hvg)
snn.gr <- buildSNNGraph(mnn_treatment.out, use.dimred="corrected")
clusters.mnn_treatment <- igraph::cluster_louvain(snn.gr)$membership
table(Cluster=clusters.mnn_treatment, treatment=mnn_treatment.out$batch)
```

```{r, fig.asp = 1 / 2}
mnn_treatment.out <- runUMAP(mnn_treatment.out, dimred="corrected")
mnn_treatment.out$batch <- factor(mnn_treatment.out$batch)
mnn_treatment.out$cluster <- factor(clusters.mnn_treatment)
mnn_treatment.p1 <- plotUMAP(
  mnn_treatment.out,
  colour_by = "batch",
  point_size = 0.5) +
  scale_colour_manual(values = treatment_colours, name = "Treatment") +
  ggtitle("MNN (treatment)")
mnn_treatment.p2 <- plotUMAP(
  mnn_treatment.out,
  colour_by = "cluster",
  point_size = 0.5) +
  ggtitle("MNN (treatment)")
mnn_treatment.p1 + mnn_treatment.p2
```

```{r}
metadata(mnn_treatment.out)$merge.info$lost.var
```

```{r, fig.asp = 1 / 3}
p1 <- ggplot(as.data.frame(
    cbind(
      colData(sce)[, "Sample", drop = FALSE],
      colData(mnn_treatment.out)[, "cluster", drop = FALSE]))) +
    geom_bar(
      aes(x = cluster, fill = Sample),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = sample_colours) +
    theme_cowplot(font_size = 8)
p2 <- ggplot(as.data.frame(
    cbind(
      colData(sce)[, "Treatment", drop = FALSE],
      colData(mnn_treatment.out)[, "cluster", drop = FALSE]))) +
    geom_bar(
      aes(x = cluster, fill = Treatment),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = treatment_colours) +
    theme_cowplot(font_size = 8)
p3 <- ggplot(
  as.data.frame(colData(mnn_treatment.out)[, "cluster", drop = FALSE])) +
  geom_bar(aes(x = cluster, fill = cluster)) +
  coord_flip() +
  ylab("Number of cells") +
  theme_cowplot(font_size = 8) +
  guides(fill = FALSE)
p1 + p2 + p3
```

##  MNN by sample

```{r}
set.seed(1000101001)
mnn_sample.out <- fastMNN(
  sce,
  batch = sce$Sample,
  d = 50,
  k = 20,
  subset.row = hvg.sample)
snn.gr <- buildSNNGraph(mnn_sample.out, use.dimred="corrected")
clusters.mnn_sample <- igraph::cluster_louvain(snn.gr)$membership
table(Cluster = clusters.mnn_sample, treatment = mnn_sample.out$batch)
```

```{r, fig.asp = 1}
mnn_sample.out <- runUMAP(mnn_sample.out, dimred="corrected")
mnn_sample.out$batch <- factor(mnn_sample.out$batch)
mnn_sample.out$cluster <- factor(clusters.mnn_sample)
mnn_sample.p1 <- plotUMAP(
  mnn_sample.out,
  colour_by = I(sce$Treatment),
  point_size = 0.5) +
  scale_colour_manual(values = treatment_colours, name = "Treatment") +
  ggtitle("MNN (sample)")
mnn_sample.p2 <- plotUMAP(
  mnn_sample.out,
  colour_by = "cluster",
  point_size = 0.5) +
  ggtitle("MNN (sample)")
mnn_sample.p3 <- plotUMAP(
  mnn_sample.out,
  colour_by = "batch",
  point_size = 0.5) +
  scale_colour_manual(values = sample_colours, name = "sample") +
  ggtitle("MNN (sample)")
mnn_sample.p1 + mnn_sample.p2 + mnn_sample.p3 + plot_layout(ncol = 2)
```

```{r}
round(metadata(mnn_sample.out)$merge.info$lost.var, 3)
```

```{r, fig.asp = 1 / 3}
p1 <- ggplot(as.data.frame(
  cbind(
    colData(sce)[, "Sample", drop = FALSE],
    colData(mnn_sample.out)[, "cluster", drop = FALSE]))) +
  geom_bar(
    aes(x = cluster, fill = Sample),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = sample_colours) +
  theme_cowplot(font_size = 8)
p2 <- ggplot(as.data.frame(
    cbind(
      colData(sce)[, "Treatment", drop = FALSE],
      colData(mnn_sample.out)[, "cluster", drop = FALSE]))) +
    geom_bar(
      aes(x = cluster, fill = Treatment),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = treatment_colours) +
    theme_cowplot(font_size = 8)
p3 <- ggplot(
  as.data.frame(colData(mnn_sample.out)[, "cluster", drop = FALSE])) +
    geom_bar(aes(x = cluster, fill = cluster)) +
    coord_flip() +
    ylab("Number of cells") +
    theme_cowplot(font_size = 8) +
    guides(fill = FALSE)
p1 + p2 + p3 + plot_layout(ncol = 3)
```

## Summary of data integration results 

```{r, fig.asp = 3 / 2}
uncorrected.p1 + rescaled.p1 + residuals.p1 + mnn_treatment.p1 +
  mnn_sample.p1 + plot_layout(ncol = 2, guides = "collect")
```

```{r, fig.asp = 3 / 2}
uncorrected.p2 + rescaled.p2 + residuals.p2 + mnn_treatment.p2 +
  mnn_sample.p2 + plot_layout(ncol = 2)
```

- Uncorrected data has strong treatment-specific differences (and presumably donor-specific differences).
- Correcting for treatment using any of the 4 approaches removes / dampens those treatment-specific differences.
- However, it's a little hard to interpret if this is a good or bad thing.
- Therefore, will just use uncorrected data in what follows while remembering that this means there will be very treatment-specific (and perhaps donor-specific) clusters.

# Clustering at different resolutions

- In all clusterings we observe an 'out group' on the right hand side.

```{r, fig.asp = 1}
set.seed(126)
snn_gr <- buildSNNGraph(sce, use.dimred = "PCA", k = 20)
clusters <- igraph::cluster_louvain(snn_gr)
sce$cluster_k20 <- factor(clusters$membership)
snn_gr <- buildSNNGraph(sce, use.dimred = "PCA", k = 50)
clusters <- igraph::cluster_louvain(snn_gr)
sce$cluster_k50 <- factor(clusters$membership)
snn_gr <- buildSNNGraph(sce, use.dimred = "PCA", k = 100)
clusters <- igraph::cluster_louvain(snn_gr)
sce$cluster_k100 <- factor(clusters$membership)

p1 <- plotUMAP(
  sce,
  colour_by = "cluster",
  point_size = 0.5,
  point_alpha = 1,
  text_by = "cluster") +
  scale_colour_manual(values = cluster_colours, name = "cluster") +
  ggtitle("Uncorrected: k = 10") +
  guides(colour = FALSE)
p2 <- plotUMAP(
  sce,
  colour_by = "cluster_k20",
  point_size = 0.5,
  point_alpha = 1,
  text_by = "cluster_k20") +
  ggtitle("Uncorrected: k = 20") +
  guides(colour = FALSE)
p3 <- plotUMAP(
  sce,
  colour_by = "cluster_k50",
  point_size = 0.5,
  point_alpha = 1,
  text_by = "cluster_k50") +
  ggtitle("Uncorrected: k = 50") +
  guides(colour = FALSE)
p4 <- plotUMAP(
  sce,
  colour_by = "cluster_k100",
  point_size = 0.5,
  point_alpha = 1,
  text_by = "cluster_k100") +
  ggtitle("Uncorrected: k = 100") +
  guides(colour = FALSE)
p1 + p2 + p3 + p4 + plot_layout(ncol = 2)
```

# Some clusters are readily annotated by uniquely upregulated markers


```{r, fig.asp = 1}
plotUMAP(
  sce,
  colour_by = "cluster",
  point_size = 0.5,
  point_alpha = 1,
  text_by = "cluster") +
  scale_colour_manual(values = cluster_colours, name = "cluster") +
  ggtitle("Uncorrected") +
  guides(colour = guide_legend(override.aes = list(size = 5)))
```

```{r, fig.asp = 1.5}
# NOTE: Not blocking on `Sample` because some clusters are so `Sample`- and 
#       `Treatment`-specific.
out <- pairwiseTTests(
  sce,
  sce$cluster,
  direction = "up",
  lfc = 0.5)
top_markers <- getTopMarkers(
  out$statistics,
  out$pairs,
  pairwise = FALSE,
  pval.type = "all")
features <- unlist(top_markers)
sce$libsize <- log10(sce$sum)
plotHeatmap(
  sce,
  features,
  order_columns_by = c("cluster", "libsize"),
  cluster_rows = FALSE,
  color = hcl.colors(101, "Blue-Red 3"),
  center = TRUE,
  zlim = c(-3, 3),
  column_annotation_colors = list(
    cluster = cluster_colours),
  annotation_row = data.frame(
    cluster = names(features),
    row.names = features),
  main = "Uniquely upregulated (logFC > 0.5) cluster marker genes")
```

# Cell cycle / proliferation of activated T-cells

```{r}
cycling_subset <- "8"
sce$cycling_subset <- ifelse(
  sce$cluster %in% cycling_subset,
  "Cycling",
  "Not cycling")
colLabels(sce) <- sce$cycling_subset
```

- There appears to be a cycling and non-cycling subset of clusters driving the initial clustering (cluster `r cycling_subset`)
- Cycling subset typified by upregulation of a large number of genes, larger library sizes, and cyclin expression
- **Cell cycle could really be proliferation of T-cells upon activation**

```{r, fig.asp = 1 / 2}
p1 <- plotUMAP(
  sce,
  colour_by = "cluster",
  text_by = "cluster",
  point_size = 0.5,
  point_alpha = 1) +
  scale_colour_manual(values = cluster_colours) +
  guides(colour = FALSE)
p2 <- plotUMAP(
  sce,
  colour_by = "cycling_subset",
  text_by = "cluster",
  point_size = 0.5,
  point_alpha = 1)
p1 + p2
```

```{r, fig.asp = 1 / 3}
p1 <- ggplot(as.data.frame(colData(sce)[, c("cycling_subset", "Sample")])) +
    geom_bar(
      aes(x = cycling_subset, fill = Sample),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = sample_colours) +
    theme_cowplot(font_size = 8)
p2 <- ggplot(as.data.frame(colData(sce)[, c("cycling_subset", "Treatment")])) +
    geom_bar(
      aes(x = cycling_subset, fill = Treatment),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = treatment_colours) +
    theme_cowplot(font_size = 8)
p3 <- ggplot(as.data.frame(colData(sce)[, "cycling_subset", drop = FALSE])) +
    geom_bar(aes(x = cycling_subset, fill = cycling_subset)) +
    coord_flip() +
    ylab("Number of cells") +
    theme_cowplot(font_size = 8) +
    guides(fill = FALSE)
p1 + p2 + p3
```

## Gene ontology analysis of cycling subset markers

- Marker genes for cluster associated with cell cycling (logFC > 0.5, FDR < 0.05)

```{r}
markers <- findMarkers(
  sce,
  sce$cluster,
  direction = "up",
  block = sce$Sample,
  lfc = 0.5,
  pval.type = "all")
cluster_8_markers <- markers[[cycling_subset]][
  markers[[cycling_subset]]$FDR < 0.05, ]
features <- rownames(cluster_8_markers)
unlist(features)
```

- GO analysis of above gene list

```{r}
library(limma)
go <- goana(
  de = unlist(rowData(sce)[features, "NCBI.ENTREZID"]),
  species = "Hs")
DT::datatable(topGO(go, number = 100)) %>%
  DT::formatSignif("P.DE")
```

## Cycling subset has larger library sizes

```{r, fig.asp = 1}
p3 <- plotColData(
  sce,
  "sum",
  x = "cluster",
  colour_by = "cycling_subset",
  point_size = 0.5,
  point_alpha = 1)
(p1 + p2) / p3 + plot_layout(guides = "collect")
```

## Expression of cycling subset markers and cyclins

Please see [`output/marker_genes/cycling_subset/`](../output/marker_genes/cycling_subset/) for heatmaps and spreadsheets of these marker gene lists.

```{r, fig.asp = 1.5}
cycling_subset_markers <- findMarkers(
  sce,
  sce$cycling_subset,
  direction = "up",
  block = sce$Sample,
  lfc = 0.5,
  row.data = flattenDF(rowData(sce)))

cyclin_genes <- grep("^CCN[ABDE][0-9]$", rowData(sce)$Symbol)
cyclin_genes <- sort(rownames(sce)[cyclin_genes])

features <- unlist(
  List(
    cycling_markers = head(
      rownames(cycling_subset_markers[["Not cycling"]]),
      25),
    not_cycling_markers = head(
      rownames(cycling_subset_markers[["Cycling"]]),
      25),
    cyclins = cyclin_genes))
plotHeatmap(
  sce,
  features,
  order_columns_by = c("cycling_subset", "cluster", "libsize"),
  cluster_rows = FALSE,
  color = hcl.colors(101, "Blue-Red 3"),
  center = TRUE,
  zlim = c(-3, 3),
  column_annotation_colors = list(cluster = cluster_colours),
  annotation_row = data.frame(
    GeneSet = names(features),
    row.names = features))
```

```{r}
outdir <- here("output", "marker_genes", "cycling_subset")
dir.create(outdir, recursive = TRUE)
createClusterMarkerOutputs(
  sce = sce,
  outdir = outdir,
  markers = cycling_subset_markers,
  k = 100,
  width = 6,
  height = 7.5)
```

## Using the `cyclone()` classifier

- Unsure how much I trust these results given the classifier was trained on quite a different dataset

```{r}
# NOTE: This was run and the results saved.
# hs.pairs <- readRDS(system.file("exdata", "human_cycle_markers.rds",
#                                 package="scran"))
#
# # Using Ensembl IDs to match up with the annotation in 'hs.pairs'.
# set.seed(100)
# assignments <- cyclone(
#   sce,
#   hs.pairs,
#   gene.names = rowData(sce)$ENSEMBL.GENEID)
assignments <- readRDS(here("data", "cyclone_assignments.rds"))
colData(sce) <- cbind(colData(sce), DataFrame(assignments$score))
sce$phases <- assignments$phases
```

```{r, fig.asp = 1}
par(mfrow = c(2, 2))
boxplot(
  sce$G1 ~ sce$cluster,
  ylab = "score",
  col = cluster_colours,
  xlab = "cluster",
  main = "Cyclone: G1")
boxplot(
  sce$G2M ~ sce$cluster,
  ylab = "score",
  col = cluster_colours,
  xlab = "cluster",
  main = "Cyclone: G2M")
boxplot(
  sce$S ~ sce$cluster,
  ylab = "score",
  col = cluster_colours,
  xlab = "cluster",
  main = "Cyclone: S")
```

```{r}
proportions(table(sce$cluster, assignments$phases), 1)
```

```{r}
proportions(table(sce$cycling_subset, sce$phases), 1)
```

```{r, fig.asp = 1}
p1 <- plotUMAP(
  sce,
  colour_by = "G1",
  text_by = "cluster",
  point_size = 0.5,
  point_alpha = 1)
p2 <- plotUMAP(
  sce,
  colour_by = "G2M",
  text_by = "cluster",
  point_size = 0.5,
  point_alpha = 1)
p3 <- plotUMAP(
    sce,
    colour_by = "S",
    text_by = "cluster",
    point_size = 0.5,
    point_alpha = 1)
p4 <- plotUMAP(
    sce,
    colour_by = "cluster",
    text_by = "cluster",
    point_size = 0.5,
    point_alpha = 1) +
  scale_colour_manual(values = cluster_colours, name = "cluster") 
p1 + p2 + p3 + p4 + plot_layout(ncol = 2)
```

# Subset analysis based on `cycling_subset`

- Using a fairly coarse clustering (`k = 50`)

```{r}
set.seed(9391)
list_of_sce <- quickSubCluster(
  sce,
  groups = sce$cycling_subset,
  prepFUN = function(x) {
    var_fit <- modelGeneVarByPoisson(x)
    hvg <- getTopHVGs(var_fit, var.threshold = 0)
    is_mito <- hvg %in% mito_set
    is_ribo <- hvg %in% ribo_set
    hvg <- hvg[!(is_mito | is_ribo)]
    # NOTE: Keep the original dimensionality reduction around for downstream
    #       plotting.
    reducedDimNames(x) <- paste0("original_", reducedDimNames(x))
    denoisePCA(x, var_fit, subset.row = hvg)
  },
  clusterFUN = function(x) {
    snn_gr <- buildSNNGraph(x, use.dimred = "PCA", k = 50)
    factor(igraph::cluster_louvain(snn_gr)$membership)
  })
# It's also useful to have per-sample UMAP representations.
set.seed(17127)
list_of_sce <- lapply(list_of_sce, runUMAP, dimred = "PCA")
```

```{r}
# It's also useful to have SingleR DICE with fine labels for subclusters
library(SingleR)
library(celldex)
ref <- DatabaseImmuneCellExpressionData()
labels_fine <- ref$label.fine
# NOTE: This code doesn't necessarily generalise beyond the DICE main labels.
label_fine_collapsed_colours <- setNames(
  c(
    Polychrome::glasbey.colors(nlevels(factor(labels_fine)) + 1)[-1],
    "orange"),
  c(levels(factor(labels_fine)), "other"))
list_of_sce <- lapply(list_of_sce, function(x) {
  pred_subcluster_fine <- SingleR(
    test = x,
    ref = ref,
    labels = labels_fine,
    clusters = x$subcluster)
  x$label_subcluster_fine <- factor(
    pred_subcluster_fine[x$subcluster, "pruned.labels"])
  x
})
list_of_sce <- lapply(list_of_sce, function(x) {
  pred_cell_fine <- SingleR(
    test = x,
    ref = ref,
    labels = labels_fine)
  x$label_cell_fine <- factor(pred_cell_fine$pruned.labels)
  x
})
```

```{r, fig.asp = 1 / 2}
wrap_plots(
  lapply(names(list_of_sce), function(n) {
    x <- list_of_sce[[n]]
    plotUMAP(x, colour_by = "subcluster", point_size = 0.5, point_alpha = 1) +
      ggtitle(n)
  }))
```

```{r, fig.asp = 1 / 2}
wrap_plots(
  lapply(names(list_of_sce), function(n) {
    x <- list_of_sce[[n]]
    plotUMAP(
      x,
      colour_by = "label_subcluster_fine",
      text_by = "subcluster",
      point_size = 0.5,
      point_alpha = 1) +
      ggtitle(n)
  })) + 
  plot_layout(guides = "collect")
```

```{r, fig.asp = 1 / 2}
wrap_plots(
  lapply(names(list_of_sce), function(n) {
    x <- list_of_sce[[n]]
    plotUMAP(
      x,
      colour_by = "label_cell_fine",
      text_by = "subcluster",
      point_size = 0.5,
      point_alpha = 1) +
      ggtitle(n)
  })) + 
  plot_layout(guides = "collect")
```

## `Cycling` subset analysis

Please see [`output/marker_genes/cycling_subset_subclusters/`](../output/marker_genes/cycling_subset_subclusters/) for heatmaps and spreadsheets of these marker gene lists.

- `Cycling` subset subclusters are strongly enriched for `phases`
- `Cycling` subset subclusters have strong marker genes (but these probably reflect more cell cycle and less cell type)
- `Cycling` subset subclusters don't appear to be `Treatment`-specific


```{r}
cycling_sce <- list_of_sce[["Cycling"]]
colLabels(cycling_sce) <- cycling_sce$subcluster
```

```{r, fig.asp = 1}
wrap_plots(
  plotlist = c(
    lapply(c("G1", "G2M", "S"), function(phase) {
      plotUMAP(
        cycling_sce,
        colour_by = phase,
        text_by = "subcluster",
        point_size = 0.5,
        point_alpha = 1)
    }),
    list(
      ggplot(
        as.data.frame(
          colData(cycling_sce)[, c("subcluster", "phases")])) +
        geom_bar(
          aes(x = subcluster, fill = phases),
          position = position_fill(reverse = TRUE)) +
        coord_flip() +
        ylab("Frequency") +
        theme_cowplot(font_size = 8)))) +
  plot_annotation(title = "Cycling cluster")
```

```{r, fig.asp = 1.5}
out <- pairwiseTTests(
  cycling_sce,
  cycling_sce$subcluster,
  direction = "up",
  block = cycling_sce$Sample)
top_markers <- getTopMarkers(
  out$statistics,
  out$pairs,
  pairwise = FALSE,
  pval.type = "all",
  n = 10)
features <- unlist(top_markers)
plotHeatmap(
  cycling_sce,
  features,
  order_columns_by = c("subcluster", "Treatment", "Sample"),
  cluster_rows = FALSE,
  color = hcl.colors(101, "Blue-Red 3"),
  center = TRUE,
  zlim = c(-3, 3),
  column_annotation_colors = list(
    Treatment = treatment_colours,
    Sample = sample_colours),
  annotation_row = data.frame(
    subcluster = names(features),
    row.names = features),
  main = "Cycling: Uniquely upregulated cluster marker genes")
```

```{r, fig.asp = 1 /3}
p1 <- ggplot(as.data.frame(colData(cycling_sce)[, c("subcluster", "Sample")])) +
    geom_bar(
      aes(x = subcluster, fill = Sample),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = sample_colours) +
    theme_cowplot(font_size = 8)
p2 <- ggplot(as.data.frame(colData(cycling_sce)[, c("subcluster", "Treatment")])) +
    geom_bar(
      aes(x = subcluster, fill = Treatment),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = treatment_colours) +
    theme_cowplot(font_size = 8)
p3 <- ggplot(as.data.frame(colData(cycling_sce)[, "subcluster", drop = FALSE])) +
    geom_bar(aes(x = subcluster, fill = subcluster)) +
    coord_flip() +
    ylab("Number of cells") +
    theme_cowplot(font_size = 8) +
    guides(fill = FALSE)

p1 + p2 + p3 + plot_annotation(title = "Cycling")
```

```{r, fig.asp = 1 / 2}
p1 <- plotUMAP(
  cycling_sce,
  colour_by = "subcluster",
  text_by = "subcluster",
  point_size = 0.5,
  point_alpha = 1)
p2 <- plotUMAP(
  cycling_sce,
  colour_by = "Treatment",
  text_by = "subcluster",
  point_size = 0.5,
  point_alpha = 1) +
  scale_colour_manual(values = treatment_colours, name = "Treatment")
p1 + p2 + plot_annotation("Cycling subset")
```

```{r}
markers <- findMarkers(
  cycling_sce,
  cycling_sce$subcluster,
  direction = "up",
  pval.type = "all",
  block = cycling_sce$Sample,
  row.data = flattenDF(rowData(cycling_sce)))
outdir <- here("output", "marker_genes", "cycling_subset_subcluster")
dir.create(outdir, recursive = TRUE)
createClusterMarkerOutputs(
  sce = cycling_sce,
  outdir = outdir,
  markers = markers,
  k = 100,
  width = 6,
  height = 7.5)
```

## `Not cycling` subset analysis

Please see [`output/marker_genes/not_cycling_subset_subclusters/`](../output/marker_genes/not_cycling_subset_subclusters/) for heatmaps and spreadsheets of these marker gene lists.

- `Not cycling` subset subclusters have little association with `phases` (or `G1`, `G2M`, or `S`)
- `Not cycling` subset subclusters have strong marker genes (likely reflecting cell type / state)
- `Not cycling` subset subclusters appear to be highly `Treatment`-specific

```{r}
not_cycling_sce <- list_of_sce[["Not cycling"]]
colLabels(not_cycling_sce) <- not_cycling_sce$subcluster
```

```{r, fig.asp = 1}
wrap_plots(
  plotlist = c(
    lapply(c("G1", "G2M", "S"), function(phase) {
      plotUMAP(
        not_cycling_sce,
        colour_by = phase,
        text_by = "subcluster",
        point_size = 0.5,
        point_alpha = 1)
    }),
    list(
      ggplot(
        as.data.frame(
          colData(not_cycling_sce)[, c("subcluster", "phases")])) +
        geom_bar(
          aes(x = subcluster, fill = phases),
          position = position_fill(reverse = TRUE)) +
        coord_flip() +
        ylab("Frequency") +
        theme_cowplot(font_size = 8)))) +
  plot_annotation(title = "Cycling cluster")
```

```{r, fig.asp = 1.5}
out <- pairwiseTTests(
  not_cycling_sce,
  not_cycling_sce$subcluster,
  direction = "up",
  block = not_cycling_sce$Sample)
top_markers <- getTopMarkers(
  out$statistics,
  out$pairs,
  pairwise = FALSE,
  pval.type = "all",
  n = 10)
features <- unlist(top_markers)
plotHeatmap(
  not_cycling_sce,
  features,
  order_columns_by = c("subcluster", "Treatment", "Sample"),
  cluster_rows = FALSE,
  color = hcl.colors(101, "Blue-Red 3"),
  center = TRUE,
  zlim = c(-3, 3),
  column_annotation_colors = list(
    Treatment = treatment_colours,
    Sample = sample_colours),
  annotation_row = data.frame(
    subcluster = names(features),
    row.names = features),
  main = "Cycling: Uniquely upregulated cluster marker genes")
```

```{r, fig.asp = 1 /3}
p1 <- ggplot(
  as.data.frame(colData(not_cycling_sce)[, c("subcluster", "Sample")])) +
    geom_bar(
      aes(x = subcluster, fill = Sample),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = sample_colours) +
    theme_cowplot(font_size = 8)
p2 <- ggplot(
  as.data.frame(colData(not_cycling_sce)[, c("subcluster", "Treatment")])) +
    geom_bar(
      aes(x = subcluster, fill = Treatment),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = treatment_colours) +
    theme_cowplot(font_size = 8)
p3 <- ggplot(
  as.data.frame(colData(not_cycling_sce)[, "subcluster", drop = FALSE])) +
    geom_bar(aes(x = subcluster, fill = subcluster)) +
    coord_flip() +
    ylab("Number of cells") +
    theme_cowplot(font_size = 8) +
    guides(fill = FALSE)

p1 + p2 + p3 + plot_annotation(title = "Cycling")
```

```{r, fig.asp = 1 / 2}
p1 <- plotUMAP(
  not_cycling_sce,
  colour_by = "subcluster",
  text_by = "subcluster",
  point_size = 0.5,
  point_alpha = 1)
p2 <- plotUMAP(
  not_cycling_sce,
  colour_by = "Treatment",
  text_by = "subcluster",
  point_size = 0.5,
  point_alpha = 1) +
  scale_colour_manual(values = treatment_colours, name = "Treatment")
p1 + p2 + plot_annotation("Cycling subset")
```

```{r}
markers <- findMarkers(
  not_cycling_sce,
  not_cycling_sce$subcluster,
  direction = "up",
  pval.type = "all",
  block = not_cycling_sce$Sample,
  row.data = flattenDF(rowData(not_cycling_sce)))
outdir <- here("output", "marker_genes", "not_cycling_subset_subcluster")
dir.create(outdir, recursive = TRUE)
createClusterMarkerOutputs(
  sce = not_cycling_sce,
  outdir = outdir,
  markers = markers,
  k = 100,
  width = 6,
  height = 7.5)
```

# Concluding remarks

```{r}
saveRDS(
  sce,
  here("data", "SCEs", "C057_Cooney.annotated.SCE.rds"),
  compress = "xz")
saveRDS(
  cycling_sce,
  here("data", "SCEs", "C057_Cooney.cycling.annotated.SCE.rds"),
  compress = "xz")
saveRDS(
  not_cycling_sce,
  here("data", "SCEs", "C057_Cooney.not_cycling.annotated.SCE.rds"),
  compress = "xz")
```

The processed *SingleCellExperiment* objects are available (see [`data/SCEs/C057_Cooney.cells_selected.SCE.rds`](../data/SCEs/C057_Cooney.cells_selected.SCE.rds), [`data/SCEs/C057_Cooney.cycling.annotated.SCE.rds`](../data/SCEs/C057_Cooney.cycling.annotated.SCE.rds), and [`data/SCEs/C057_Cooney.not_cycling.annotated.SCE.rds`](../data/SCEs/C057_Cooney.not_cycling.annotated.SCE.rds)).
These will be used in downstream analyses, e.g., differential expression analysis between conditions within each cluster and differential abundance analyses between conditions for each cluster.

# Additional information {.appendix}

The following are available on request:

- Full CSV tables of any data presented.
- PDF/PNG files of any static plots.

# Session info {.appendix}

<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>

```{r}
sessioninfo::session_info()
```

</details>

