---
title: "Multi-sample comparisons with the Cooney (C057) memory CD4+ T-cell data set"
description: |
author:
  - name: Peter Hickey 
    url: https://peterhickey.org
    affiliation: WEHI SCORE
    affiliation_url: https://www.wehi.edu.au/people/shalin-naik/3310/score
date: "`r Sys.Date()`"
output: distill::distill_article
editor_options: 
  chunk_output_type: console
bibliography: ref.bib
---

# Setup

```{r}
library(SingleCellExperiment)
library(here)
library(cowplot)
library(patchwork)
library(edgeR)
library(scater)
library(scran)

sce <- readRDS(here("data", "SCEs", "C057_Cooney.annotated.SCE.rds"))
cycling_sce <- readRDS(
  here("data", "SCEs", "C057_Cooney.cycling.annotated.SCE.rds"))
not_cycling_sce <- readRDS(
  here("data", "SCEs", "C057_Cooney.not_cycling.annotated.SCE.rds"))

# Some useful colours
sample_colours <- setNames(
  unique(sce$sample_colours),
  unique(names(sce$sample_colours)))
treatment_colours <- setNames(
  unique(sce$treatment_colours),
  unique(names(sce$treatment_colours)))
cluster_colours <- setNames(
  unique(sce$cluster_colours),
  unique(names(sce$cluster_colours)))

# Re-level Treatment
sce$Treatment <- relevel(sce$Treatment, "Uninfected")
cycling_sce$Treatment <- relevel(cycling_sce$Treatment, "Uninfected")
not_cycling_sce$Treatment <- relevel(not_cycling_sce$Treatment, "Uninfected")

source(here("code", "helper_functions.R"))
```

# Analysis based on `cycling_subset`

## DE

Please see [`output/DEGs/cycling_subset/`](../output/DEGs/cycling_subset/) for heatmaps and spreadsheets of these DEG lists, including results of GO and KEGG analyses using the `goana()` and `kegga()` functions from the `r BiocStyle::Biocpkg("limma")` package.
This directory also contains interactive `r BiocStyle::Biocpkg("Glimma")` plots of the differential expression results.
The `r BiocStyle::Biocpkg("Glimma")` plots show the pseudobulk data for that label.

<aside>
Note that the DEG list spreadsheets contain all labels and genes whereas the Glimma plots only contain labels and genes actually tested for DE. The column names in the Glimma tables may be abbreviated. There are two types of heatmaps: (1) showing the 'pseudobulk' expression and (2) showing the average single-cell expression.
</aside>

```{r}
# NOTE: Have to drop the complicated TRA and TRB DFrameList objects.
x <- sce
x$TRA <- NULL
x$TRB <- NULL
summed <- aggregateAcrossCells(
  x,
  id = colData(x)[, c("cycling_subset", "Sample")])
sizeFactors(summed) <- NULL
summed <- logNormCounts(summed)
colLabels(summed) <- summed$cycling_subset
colnames(summed) <- paste0(summed$cycling_subset, ".", summed$Sample)
```

```{r, fig.asp = 1 / 2}
summed <- runMDS(summed)
p1 <- plotMDS(
  summed,
  colour_by = "Treatment",
  shape_by = "cycling_subset",
  point_size = 3,
  point_alpha = 2) +
  scale_colour_manual(values = treatment_colours, name = "Treatment")
p2 <- plotMDS(
  summed,
  colour_by = "Sample",
  shape_by = "cycling_subset",
  point_size = 3,
  point_alpha = 2) +
  scale_colour_manual(values = sample_colours, name = "Treatment")
p1 + p2 + plot_layout(guides = "collect")
```

```{r}
summed_filt <- summed[, summed$ncells >= 10]

de_results <- pseudoBulkDGE(
  summed_filt,
  label = summed_filt$cycling_subset,
  design = ~Treatment,
  coef = "TreatmentInfected",
  condition = summed_filt$Treatment,
  include.intermediates = TRUE)
```

```{r}
metadata(de_results)$failed
```

```{r}
is.de <- decideTestsPerLabel(de_results, threshold = 0.05)
summarizeTestsPerLabel(is.de)
```

```{r}
outdir <- here("output", "DEGs", "cycling_subset")
dir.create(outdir, recursive = TRUE)
createDEGOutputs(
  outdir = outdir, 
  de_results = de_results, 
  summed_filt = summed_filt,
  sce = sce,
  fdr = 0.05)
```

```{r, message = FALSE, results = "hide"}
lapply(names(de_results), function(n) {
  se <- summed_filt[, summed_filt$label == n]
  x <- de_results[[n]]
  features <- rownames(x[which(x$FDR < 0.05), ])
  if (length(features) > 2) {
    plotHeatmap(
      se,
      features = head(features, 50),
      center = TRUE,
      zlim = c(-3, 3),
      symmetric = TRUE,
      color = hcl.colors(101, "Blue-Red 3"),
      order_columns_by = c("Treatment", "Sample"),
      column_annotation_colors = list(
        Treatment = treatment_colours,
        Sample = sample_colours),
      cluster_rows = TRUE,
      fontsize = 8,
      main = n,
      filename = file.path(outdir, paste0(n, ".pseudobulk_heatmap.pdf")))
    
    plotGroupedHeatmap(
      sce,
      features = head(features, 50),
      group = "Sample",
      columns = sce$cycling_subset == n,
      center = TRUE,
      symmetric = TRUE,
      color = hcl.colors(101, "Blue-Red 3"),
      cluster_rows = TRUE,
      fontsize = 8,
      main = n,
      cluster_cols = FALSE,
      filename = file.path(
        outdir,
        paste0(n, ".average_expression_heatmap.pdf")))
  }
})
```

## DA

- Can't use quasi-likelihood framework with only two labels.

```{r}
abundances <- table(sce$cycling_subset, sce$Sample)
abundances <- unclass(abundances)
abundances
```


```{r, fig.asp = 1 / 2}
p1 <- plotUMAP(
  sce, 
  colour_by = "cycling_subset", 
  text_by = "cycling_subset",
  point_size = 0.5,
  point_alpha = 1)
p2 <- plotUMAP(
  sce, 
  colour_by = "Treatment",
  point_size = 0.5,
  point_alpha = 1) +
  scale_colour_manual(values = treatment_colours, name = "Treatment")
p1 + p2
```

```{r, fig.asp = 1 / 3}
p3 <- ggplot(as.data.frame(colData(sce)[, c("cycling_subset", "Sample")])) +
    geom_bar(
      aes(x = cycling_subset, fill = Sample),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = sample_colours) +
    theme_cowplot(font_size = 8)
p4 <- ggplot(as.data.frame(colData(sce)[, c("cycling_subset", "Treatment")])) +
    geom_bar(
      aes(x = cycling_subset, fill = Treatment),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = treatment_colours) +
    theme_cowplot(font_size = 8)
p5 <- ggplot(as.data.frame(colData(sce)[, "cycling_subset", drop = FALSE])) +
    geom_bar(aes(x = cycling_subset, fill = cycling_subset)) +
    coord_flip() +
    ylab("Number of cells") +
    theme_cowplot(font_size = 8) +
    guides(fill = FALSE)
p3 + p4 + p5 + plot_layout(ncol = 3)
```

```{r}
extra.info <- colData(sce)[match(colnames(abundances), sce$Sample), ]
y.ab <- DGEList(abundances, samples = extra.info)

keep <- filterByExpr(y.ab, group = y.ab$samples$Treatment)
summary(keep)
y.ab <- y.ab[keep,]

design <- model.matrix(~Treatment, y.ab$samples)

y.ab <- estimateDisp(y.ab, design, trend = "none")
fit.ab <- glmFit(y.ab)
res <- glmLRT(fit.ab, coef = "TreatmentInfected")
summary(decideTests(res))
```

```{r}
topTags(res)
```

# Analysis based on subcluster labels in `Not cycling` subset

## DE

Please see [`output/DEGs/not_cycling_subset_subclusters/`](../output/DEGs/not_cycling_subset_subclusters/) for heatmaps and spreadsheets of these DEG lists, including results of GO and KEGG analyses using the `goana()` and `kegga()` functions from the `r BiocStyle::Biocpkg("limma")` package.
This directory also contains interactive `r BiocStyle::Biocpkg("Glimma")` plots of the differential expression results.
The `r BiocStyle::Biocpkg("Glimma")` plots show the pseudobulk data for that label.

<aside>
Note that the DEG list spreadsheets contain all labels and genes whereas the Glimma plots only contain labels and genes actually tested for DE. The column names in the Glimma tables may be abbreviated. There are two types of heatmaps: (1) showing the 'pseudobulk' expression and (2) showing the average single-cell expression.
</aside>

```{r}
# NOTE: Have to drop the complicated TRA and TRB DFrameList objects.
x <- not_cycling_sce
x$TRA <- NULL
x$TRB <- NULL
summed <- aggregateAcrossCells(
  x,
  id = colData(x)[, c("subcluster", "Sample")])
sizeFactors(summed) <- NULL
summed <- logNormCounts(summed)
colLabels(summed) <- summed$subcluster
colnames(summed) <- paste0(summed$subcluster, ".", summed$Sample)
```

```{r, fig.asp = 1 / 2}
summed <- runMDS(summed)
p1 <- plotMDS(
  summed,
  colour_by = "subcluster",
  shape_by = "Treatment",
  point_size = 3,
  point_alpha = 2)
p2 <- plotMDS(
  summed,
  colour_by = "subcluster",
  shape_by = "Sample",
  point_size = 3,
  point_alpha = 2)
p1 + p2 + plot_layout(guides = "collect")
```

```{r}
summed_filt <- summed[, summed$ncells >= 10]

de_results <- pseudoBulkDGE(
  summed_filt,
  label = summed_filt$subcluster,
  design = ~Treatment,
  coef = "TreatmentInfected",
  condition = summed_filt$Treatment,
  include.intermediates = TRUE)
```

```{r}
metadata(de_results)$failed
```

```{r}
is.de <- decideTestsPerLabel(de_results, threshold = 0.05)
summarizeTestsPerLabel(is.de)
```

```{r}
outdir <- here("output", "DEGs", "not_cycling_subset_subcluster")
dir.create(outdir, recursive = TRUE)
createDEGOutputs(
  outdir = outdir, 
  de_results = de_results, 
  summed_filt = summed_filt,
  sce = x,
  fdr = 0.05)
```

```{r, message = FALSE, results = "hide"}
lapply(names(de_results), function(n) {
  se <- summed_filt[, summed_filt$label == n]
  x <- de_results[[n]]
  features <- rownames(x[which(x$FDR < 0.05), ])
  if (length(features) > 2) {
    plotHeatmap(
      se,
      features = head(features, 50),
      center = TRUE,
      zlim = c(-3, 3),
      symmetric = TRUE,
      color = hcl.colors(101, "Blue-Red 3"),
      order_columns_by = c("Treatment", "Sample"),
      column_annotation_colors = list(
        Treatment = treatment_colours,
        Sample = sample_colours),
      cluster_rows = TRUE,
      fontsize = 8,
      main = n,
      filename = file.path(outdir, paste0(n, ".pseudobulk_heatmap.pdf")))
    
    plotGroupedHeatmap(
      not_cycling_sce,
      features = head(features, 50),
      group = "Sample",
      columns = not_cycling_sce$subcluster == n,
      center = TRUE,
      symmetric = TRUE,
      color = hcl.colors(101, "Blue-Red 3"),
      cluster_rows = TRUE,
      fontsize = 8,
      main = n,
      cluster_cols = FALSE,
      filename = file.path(
        outdir,
        paste0(n, ".average_expression_heatmap.pdf")))
  }
})
```

## DA

```{r}
abundances <- table(x$subcluster, x$Sample)
abundances <- unclass(abundances)
abundances
```

```{r, fig.asp = 1 / 2}
p1 <- plotUMAP(
  x,
  colour_by = "subcluster",
  text_by = "subcluster",
  point_size = 0.5,
  point_alpha = 1)
p2 <- plotUMAP(
  x,
  colour_by = "Treatment",
  point_size = 0.5,
  point_alpha = 1) +
  scale_colour_manual(values = treatment_colours)
p1 + p2
```

```{r, fig.asp = 1 / 3}
p3 <- ggplot(as.data.frame(colData(x)[, c("subcluster", "Sample")])) +
  geom_bar(
    aes(x = subcluster, fill = Sample),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = sample_colours) +
  theme_cowplot(font_size = 8)
p4 <- ggplot(as.data.frame(colData(x)[, c("subcluster", "Treatment")])) +
  geom_bar(
    aes(x = subcluster, fill = Treatment),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = treatment_colours) +
  theme_cowplot(font_size = 8)
p5 <- ggplot(as.data.frame(colData(x)[, "subcluster", drop = FALSE])) +
  geom_bar(aes(x = subcluster, fill = subcluster)) +
  coord_flip() +
  ylab("Number of cells") +
  theme_cowplot(font_size = 8) +
  guides(fill = FALSE)
p3 + p4 + p5 + plot_layout(ncol = 3)
```

```{r}
extra.info <- colData(sce)[match(colnames(abundances), sce$Sample), ]
y.ab <- DGEList(abundances, samples = extra.info)

keep <- filterByExpr(y.ab, group = y.ab$samples$Treatment)
summary(keep)
y.ab <- y.ab[keep,]

design <- model.matrix(~Treatment, y.ab$samples)

y.ab <- estimateDisp(y.ab, design, trend = "none")
fit.ab <- glmFit(y.ab)
res <- glmLRT(fit.ab, coef = "TreatmentInfected")
summary(decideTests(res))
```

```{r}
topTags(res)
```

# Analysis based on subcluster labels in `Cycling` subset

## DE

Please see [`output/DEGs/cycling_subset_subclusters/`](../output/DEGs/cycling_subset_subclusters/) for heatmaps and spreadsheets of these DEG lists, including results of GO and KEGG analyses using the `goana()` and `kegga()` functions from the `r BiocStyle::Biocpkg("limma")` package.
This directory also contains interactive `r BiocStyle::Biocpkg("Glimma")` plots of the differential expression results.
The `r BiocStyle::Biocpkg("Glimma")` plots show the pseudobulk data for that label.

<aside>
Note that the DEG list spreadsheets contain all labels and genes whereas the Glimma plots only contain labels and genes actually tested for DE. The column names in the Glimma tables may be abbreviated. There are two types of heatmaps: (1) showing the 'pseudobulk' expression and (2) showing the average single-cell expression.
</aside>

```{r}
# NOTE: Have to drop the complicated TRA and TRB DFrameList objects.
x <- cycling_sce
x$TRA <- NULL
x$TRB <- NULL
summed <- aggregateAcrossCells(
  x,
  id = colData(x)[, c("subcluster", "Sample")])
sizeFactors(summed) <- NULL
summed <- logNormCounts(summed)
colLabels(summed) <- summed$subcluster
colnames(summed) <- paste0(summed$subcluster, ".", summed$Sample)
```

```{r, fig.asp = 1 / 2}
summed <- runMDS(summed)
p1 <- plotMDS(
  summed,
  colour_by = "subcluster",
  shape_by = "Treatment",
  point_size = 3,
  point_alpha = 2)
p2 <- plotMDS(
  summed,
  colour_by = "subcluster",
  shape_by = "Sample",
  point_size = 3,
  point_alpha = 2)
p1 + p2 + plot_layout(guides = "collect")
```

```{r}
summed_filt <- summed[, summed$ncells >= 10]

de_results <- pseudoBulkDGE(
  summed_filt,
  label = summed_filt$subcluster,
  design = ~Treatment,
  coef = "TreatmentInfected",
  condition = summed_filt$Treatment,
  include.intermediates = TRUE)
```

```{r}
metadata(de_results)$failed
```

```{r}
is.de <- decideTestsPerLabel(de_results, threshold = 0.05)
summarizeTestsPerLabel(is.de)
```

```{r}
outdir <- here("output", "DEGs", "cycling_subset_subcluster")
dir.create(outdir, recursive = TRUE)
createDEGOutputs(
  outdir = outdir, 
  de_results = de_results, 
  summed_filt = summed_filt,
  sce = x,
  fdr = 0.05)
```

```{r, message = FALSE, results = "hide"}
lapply(names(de_results), function(n) {
  se <- summed_filt[, summed_filt$label == n]
  x <- de_results[[n]]
  features <- rownames(x[which(x$FDR < 0.05), ])
  if (length(features) > 2) {
    plotHeatmap(
      se,
      features = head(features, 50),
      center = TRUE,
      zlim = c(-3, 3),
      symmetric = TRUE,
      color = hcl.colors(101, "Blue-Red 3"),
      order_columns_by = c("Treatment", "Sample"),
      column_annotation_colors = list(
        Treatment = treatment_colours,
        Sample = sample_colours),
      cluster_rows = TRUE,
      fontsize = 8,
      main = n,
      filename = file.path(outdir, paste0(n, ".pseudobulk_heatmap.pdf")))
    
    plotGroupedHeatmap(
      cycling_sce,
      features = head(features, 50),
      group = "Sample",
      columns = cycling_sce$subcluster == n,
      center = TRUE,
      symmetric = TRUE,
      color = hcl.colors(101, "Blue-Red 3"),
      cluster_rows = TRUE,
      fontsize = 8,
      main = n,
      cluster_cols = FALSE,
      filename = file.path(
        outdir,
        paste0(n, ".average_expression_heatmap.pdf")))
  }
})
```

## DA

```{r}
abundances <- table(x$subcluster, x$Sample)
abundances <- unclass(abundances)
abundances
```

```{r, fig.asp = 1 / 2}
p1 <- plotUMAP(
  x,
  colour_by = "subcluster",
  text_by = "subcluster",
  point_size = 0.5,
  point_alpha = 1)
p2 <- plotUMAP(
  x,
  colour_by = "Treatment",
  point_size = 0.5,
  point_alpha = 1) +
  scale_colour_manual(values = treatment_colours)
p1 + p2
```

```{r, fig.asp = 1 / 3}
p3 <- ggplot(as.data.frame(colData(x)[, c("subcluster", "Sample")])) +
  geom_bar(
    aes(x = subcluster, fill = Sample),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = sample_colours) +
  theme_cowplot(font_size = 8)
p4 <- ggplot(as.data.frame(colData(x)[, c("subcluster", "Treatment")])) +
  geom_bar(
    aes(x = subcluster, fill = Treatment),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = treatment_colours) +
  theme_cowplot(font_size = 8)
p5 <- ggplot(as.data.frame(colData(x)[, "subcluster", drop = FALSE])) +
  geom_bar(aes(x = subcluster, fill = subcluster)) +
  coord_flip() +
  ylab("Number of cells") +
  theme_cowplot(font_size = 8) +
  guides(fill = FALSE)
p3 + p4 + p5 + plot_layout(ncol = 3)
```

```{r}
extra.info <- colData(sce)[match(colnames(abundances), sce$Sample), ]
y.ab <- DGEList(abundances, samples = extra.info)

keep <- filterByExpr(y.ab, group = y.ab$samples$Treatment)
summary(keep)
y.ab <- y.ab[keep,]

design <- model.matrix(~Treatment, y.ab$samples)

y.ab <- estimateDisp(y.ab, design, trend = "none")
fit.ab <- glmFit(y.ab)
res <- glmLRT(fit.ab, coef = "TreatmentInfected")
summary(decideTests(res))
```

```{r}
topTags(res)
```

# Concluding remarks

The [`output/DEGs/`](../output/DEGs/) directory contains CSV files summarising the differential expression results and PDFs showing the pseudobulk expression data for each DEG.

# Additional information {.appendix}

The following are available on request:

- Full CSV tables of any data presented.
- PDF/PNG files of any static plots.

# Session info {.appendix}

<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>

```{r}
sessioninfo::session_info()
```

</details>
