---
title: "Demultiplexing the Cooney (C057) memory CD4+ T-cell data set"
description: |
author:
  - name: Peter Hickey 
    url: https://peterhickey.org
    affiliation: WEHI SCORE
    affiliation_url: https://www.wehi.edu.au/people/shalin-naik/3310/score
date: "`r Sys.Date()`"
output: distill::distill_article
editor_options: 
  chunk_output_type: console
bibliography: ref.bib
---

```{r setup}
library(here)
library(magrittr)
library(BiocStyle)
library(dplyr)
library(janitor)
library(cowplot)

source(here("code", "helper_functions.R"))

# NOTE: Using >= 4 cores seizes up my laptop. Can use more on unix box.
options("mc.cores" = ifelse(Sys.info()[["nodename"]] == "PC1331", 2L, 3L))

knitr::opts_chunk$set(fig.path = "C057_Cooney.demultiplex_files/")
```

# Introduction

Cells were obtained from six samples.
These samples were processed a technique called using Cell Hashing, a method that enables sample multiplexing. 
Cell Hashing uses a series of oligo-tagged antibodies against ubiquitously expressed surface proteins with different barcodes to uniquely label cells from distinct samples, which can be subsequently pooled in one scRNA-seq run.
By sequencing these tags alongside the cellular transcriptome, we can assign each cell to its sample of origin, and robustly identify doublets originating from multiple samples.

In a typical Cell Hashing experiment, each sample is uniquely labelled with a single hashtag oligonucleotide (HTO).
However, in this experiment we needed to label 6 samples but only had 5 HTOs.
We decided to uniquely label five of the samples (three infected, two uninfected) with a single HTO and to label the sixth sample with all five HTOs.
This non-standard use of HTOs meant we could not rely on standard cell hashing demultiplexing routines, like that available in `r BiocStyle::CRANpkg("Seurat")`.

# Setting up the data

The count data were processed using **CellRanger** and the `r Biocpkg("DropletUtils")` R/Bioconductor packages.
The counts and their metadata are available in a *SingleCellExperiment* object available as [`data/SCEs/HTLV_GEX_HTO.CellRanger.SCE.rds`](../data/SCEs/HTLV_GEX_HTO.CellRanger.SCE.rds).
These are the *unfiltered* data, i.e. the data includes all potential genes and barcodes.

```{r}
sce <- readRDS(here("data", "SCEs", "C057_Cooney.CellRanger.SCE.rds"))
```

# Calling cells from empty droplets

An interesting aspect of droplet-based data is that we have no prior knowledge about which droplets (i.e. cell barcodes) actually contain cells, and which are empty. 
Thus, we need to call cells from empty droplets based on the observed expression profiles.
This is done separately for each 10x run.

Calling empty droplets is not entirely straightforward as empty droplets can contain ambient (i.e. extracellular) RNA that can be captured and sequenced. 
The distribution of total counts exhibits a sharp transition between barcodes with large and small total counts (Figure \@ref(fig:rankplot)), probably corresponding to cell-containing and empty droplets respectively.

```{r rankplot, fig.cap = "Total UMI count for each barcode in the dataset, plotted against its rank (in decreasing order of total counts). The inferred locations of the inflection (dark green dashed lines) and knee points (blue dashed lines) are also shown.", fig.asp = 1}
library(DropletUtils)
bcrank <- barcodeRanks(counts(sce))
# Only showing unique points for plotting speed.
uniq <- !duplicated(bcrank$rank)
plot(
  x = bcrank$rank[uniq], 
  y = bcrank$total[uniq], 
  log = "xy",
  xlab = "Rank", 
  ylab = "Total UMI count", 
  main = "C057_Cooney",
  cex.lab = 1.2,
  xlim = c(1, 500000),
  ylim = c(1, 50000))
abline(h = metadata(bcrank)$inflection, col = "darkgreen", lty = 2)
abline(h = metadata(bcrank)$knee, col = "dodgerblue", lty = 2)
legend(
  "bottomleft",
  legend = c("Inflection", "Knee"), 
  col = c("darkgreen", "dodgerblue"),
  lty = 2,
  cex = 1.2)
```

We use the `emptyDrops()` function from the `r Biocpkg("DropletUtils")` package to test whether the expression profile for each cell barcode is significantly different from the ambient RNA pool [@lun2018distinguishing].
This tends to be less conservative than the cell calling algorithm from the **CellRanger** pipeline, which often discards genuine cells with low RNA content (and thus low total counts).
Any significant deviation indicates that the barcode corresponds to a cell-containing droplet.
We call cells at a false discovery rate (FDR) of 0.1%, meaning that no more than 0.1% of our called barcodes should be empty droplets on average.

```{r}
empties <- readRDS(
  here("data", "emptyDrops", "C057_Cooney.emptyDrops.rds"))
mutate(as.data.frame(empties), Sample = rep("C057", nrow(empties))) %>%
  group_by(Sample) %>%
  summarise(
    keep = sum(FDR <= 0.001, na.rm = TRUE), 
    n = n(),
    remove = n - keep) %>%
  mutate(library = factor(Sample)) %>%
  select(library, keep, remove) %>%
  knitr::kable(
    caption = "Number of droplets kept and removed after filtering empty drops.")
sce_empties <- sce[, -which(empties$FDR <= 0.001)]
sce <- sce[, which(empties$FDR <= 0.001)]
```

# Demultiplexing with hashtag oligos (HTOs)

## Motivation


James provided me with a table linking HTOs to samples, shown below.

```{r}
hto_to_sample_df <- data.frame(
  HTO = c(paste0("human_", 1:5), "human_1-human_2-human_3-human_4-human_5"),
  Sample = c(paste0("infected_", 1:3), paste0("uninfected_", 1:3)),
  Treatment = c(rep("Infected", 3), rep("Uninfected", 3)),
  Replicate = rep(1:3, 2))

knitr::kable(hto_to_sample_df, caption = "Table linking HTOs to samples")
```

## Analysis

We only use the droplets identified as non-empty in [Calling cells from empty droplets].

```{r}
# Select cell barcodes detected by both RNA and HTOs
rna_bcs <- colSums(counts(sce)) > 0
hto_bcs <- colSums(counts(altExp(sce, "Antibody Capture"))) > 0
joint_bcs <- rna_bcs & hto_bcs
sce <- sce[, joint_bcs]
```

These samples were processed using 5 HTOs (`human_1`, `human_2`, `human_3`, `human_4`, and `human_5`).
Nonetheless, we ran **CellRanger** using a larger panel of HTOs (n = `r nrow(altExp(sce, "Antibody Capture"))`) to enable us to check for background contamination of HTOs.

Figure \@ref(fig:htos) plots the distribution of each HTO counts for the the 8 most frequently HTOs detected and shows that HTOs `human_1`, `human_2`, `human_3`, `human_4`, and `human_5` are clearly detected more frequently than the other barcodes, as we would hope.

```{r htos, fig.cap = "Distribution of HTO counts in the dataset. Only those HTOs with more detected in at least one cell are shown. The red vertical line denotes detecting the HTO 10 times in a sample.", fig.asp = 2 / 4}
par(mfrow = c(2, 4))
o <- order(rowSums(counts(altExp(sce, "Antibody Capture"))), decreasing = TRUE)
for (i in o[1:8]) {
  if (rowSums(counts(altExp(sce, "Antibody Capture")))[i] > 10) {
    plot(
      table(counts(altExp(sce, "Antibody Capture"))[i, ]),
      main = rownames(altExp(sce, "Antibody Capture"))[i],
      xlim = c(1, 10 ^ 5),
      log = "x",
      xlab = "Detections",
      ylab = "Frequency")
    abline(v = 10, col = "red")
  }
}
```

We therefore focus on assigning cells to just the `human_1`, `human_2`, `human_3`, `human_4`, and `human_5` HTOs, along with the combination of all five HTOs.

```{r}
altExp(sce, "Antibody Capture") <- altExp(
  sce, "Antibody Capture")[paste0("human_", 1:5), ]

sce$hto_sum <- colSums(counts(altExp(sce)))
ambient <- rowMeans(counts(altExp(sce_empties)))[rownames(altExp(sce))]

hto_low_counts_cutoff <- metadata(
  barcodeRanks(counts(altExp(sce_empties))))$inflection
```

To do so, we will cluster the (log normalized) HTO expression profiles and manually annotate the resulting clusters.
We perform median-based size normalization of the HTO count data.^[Median-based size factors can adjust for the composition bias that is inherent to the HTO counts; see [https://osca.bioconductor.org/integrating-with-protein-abundance.html#normalization-1](https://osca.bioconductor.org/integrating-with-protein-abundance.html#normalization-1) for details.].
This requires an estimate of the ambient expression of each HTO, which we estimate from the empty droplets.

<aside>
It's noteworthy that the ambient expression varies across the HTOs, with a 4-fold difference between the least and most abundant HTO.
</aside>

```{r}
as.data.frame(ambient) %>%
  knitr::kable(digits = 1, caption = "Estimated ambient counts for each HTO.")
```

```{r}
library(scater)
sf_amb <- medianSizeFactors(
  altExp(sce, "Antibody Capture"),
  reference = ambient)
# NOTE: Size factors should be positive.
sf_amb[sf_amb == 0] <- min(sf_amb[sf_amb > 0])
sizeFactors(altExp(sce, "Antibody Capture")) <- sf_amb
altExp(sce, "Antibody Capture") <- logNormCounts(
  altExp(sce, "Antibody Capture"))
```

We then apply graph-based clustering to the normalized log-expression of the HTOs to identify samples with similar HTO expression profiles.
We deliberately overcluster the data in order to obtain quite specific clusters.

```{r}
library(scran)
# NOTE: `d = NA` means no dimensionality reduction is performed.
# NOTE: `k = 5` was selected by trial and error.
g <- buildSNNGraph(altExp(sce, "Antibody Capture"), k = 5, d = NA)
clusters <- igraph::cluster_louvain(g)$membership
# NOTE: Orders cluster labels (1, ..., N) so that the biggest cluster is N and
#       the smallest is 1.
sce$hto_cluster <- factor(
  as.integer(factor(clusters, names(sort(table(clusters), decreasing = TRUE)))))
hto_cluster_colours <- setNames(
  Polychrome::glasbey.colors(nlevels(sce$hto_cluster) + 1)[-1],
  levels(sce$hto_cluster))
sce$hto_cluster_colours <- hto_cluster_colours[sce$hto_cluster]
```

Figure \@ref(fig:hto-heatmap) is a heatmap of the counts for each HTO in each droplet.
We can clearly see droplets corresponding to singlets from each of the six samples, i.e. those droplets with large counts for exactly one of the `human_1`, `human_2`, `human_3`, `human_4`, or `human_5` HTOs and droplets with large counts for all five HTOs.

```{r hto-heatmap, fig.cap = "Heatmap of HTO log2(counts + 1) for the droplets. Columns are ordered by cluster and then HTO library size. Each droplet (column) is annotated by the (log10) HTO library size (`sum`) and whether that is smaller than the a cutoff estimated from the empty droplets. Each HTO (row) is annotated by the (log10) average count of the HTO in the empty droplets (`ambient`)."}
library(pheatmap)
pheatmap(
  log2(counts(altExp(sce)) + 1)[
    , order(sce$hto_cluster, sce$hto_sum, decreasing = c(FALSE, TRUE))],
  color = viridisLite::inferno(100),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  annotation_col = data.frame(
    hto_cluster = sce$hto_cluster,
    sum = log10(sce$hto_sum),
    low = ifelse(sce$hto_sum < hto_low_counts_cutoff, "Low", "High"),
    row.names = colnames(sce)),
  annotation_row = data.frame(ambient = log10(ambient)),
  annotation_colors = list(
    hto_cluster = hto_cluster_colours,
    sum = viridisLite::viridis(101),
    low = c("Low" = "red", "High" = "white"),
    ambient = viridisLite::viridis(101)),
  show_colnames = FALSE,
  scale = "none",
  fontsize = 6)
```

On the basis of Figure \@ref(fig:hto-heatmap), we assign these singlets to their sample of origin:

```{r}
singlet_clusters <- as.character(c(1:6, 9:12))
singlet_cluster_to_hto_tbl <- tibble::tribble(
  ~hto_cluster, ~HTO,
  "1",          "human_3",
  "2",          "human_2",
  "3",          "human_5",
  "4",          "human_5",
  "5",          "human_4",
  "6",          "human_4",
  "9",          "human_1-human_2-human_3-human_4-human_5",
  "10",         "human_1",
  "11",         "human_1-human_2-human_3-human_4-human_5",
  "12",         "human_1")
stopifnot(identical(singlet_clusters, singlet_cluster_to_hto_tbl$hto_cluster))

left_join(singlet_cluster_to_hto_tbl, hto_to_sample_df) %>%
  knitr::kable(., caption = "Assignments of singlet clusters to samples.")
```

We then turn our attention to the remaining clusters.
Figure \@ref(fig:not-yet-assigned-hto-heatmap) shows that the largest remaining clusters contain droplets with 'low' library sizes, which are hard to confidently assign to a sample.

```{r not-yet-assigned-hto-heatmap, fig.cap = "Heatmap of HTO log2(counts + 1) for the droplets not yet assigned to a sample of origin. Columns are ordered by cluster and then HTO library size. Each droplet (column) is annotated by the (log10) HTO library size (`sum`) and whether that is smaller than the a cutoff estimated from the empty droplets. Each HTO (row) is annotated by the (log10) average count of the HTO in the empty droplets (`ambient`)."}
tmp_sce <- sce[, !sce$hto_cluster %in% singlet_clusters]
pheatmap(
  log2(counts(altExp(tmp_sce)) + 1)[
    , order(tmp_sce$hto_cluster, tmp_sce$hto_sum, decreasing = c(FALSE, TRUE))],
  color = viridisLite::inferno(100),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  annotation_col = data.frame(
    hto_cluster = sce$hto_cluster,
    sum = log10(sce$hto_sum),
    low = ifelse(sce$hto_sum < hto_low_counts_cutoff, "Low", "High"),
    row.names = colnames(sce)),
  annotation_row = data.frame(ambient = log10(ambient)),
  annotation_colors = list(
    # NOTE: Select 'active' colour levels.
    hto_cluster = hto_cluster_colours[
      levels(factor(tmp_sce$hto_cluster))],
    sum = viridisLite::viridis(101),
    low = c("Low" = "red", "High" = "white"),
    ambient = viridisLite::viridis(101)),
  show_colnames = FALSE,
  scale = "none",
  fontsize = 6)
```

On the basis of Figure \@ref(fig:not-yet-assigned-hto-heatmap), we opt for a conservative approach that labels droplets in the these clusters as `ambiguous` (i.e. we cannot confidently assign them to a sample of origin).

```{r}
ambiguous_clusters <- as.character(c(7:8, 13:16))
ambiguous_cluster_to_hto_tbl <- tibble::tribble(
    ~hto_cluster, ~HTO,
    "7",          "ambiguous",
    "8",          "ambiguous",
    "13",         "ambiguous",
    "14",         "ambiguous",
    "15",         "ambiguous",
    "16",         "ambiguous")
stopifnot(
  identical(ambiguous_clusters, ambiguous_cluster_to_hto_tbl$hto_cluster))

left_join(ambiguous_cluster_to_hto_tbl, hto_to_sample_df) %>%
  knitr::kable(., caption = "Clusters containing `ambiguous` droplets.")
```

We then turn our attention to the remaining clusters.
Figure \@ref(fig:mostly-doublets-hto-heatmap) shows that the remaining clusters largely contain droplets with 2 HTOs, which correspond to droplets containing two cells ('doublets'), as well as a small cluster of droplets that appear to contain all five HTOs.

```{r mostly-doublets-hto-heatmap, fig.cap = "Heatmap of HTO log2(counts + 1) for the droplets not yet assigned to a sample of origin. Columns are ordered by cluster and then HTO library size. Each droplet (column) is annotated by the (log10) HTO library size (`sum`) and whether that is smaller than the a cutoff estimated from the empty droplets. Each HTO (row) is annotated by the (log10) average count of the HTO in the empty droplets (`ambient`)."}
tmp_sce <- sce[, !sce$hto_cluster %in% c(singlet_clusters, ambiguous_clusters)]
pheatmap(
  log2(counts(altExp(tmp_sce)) + 1)[
    , order(tmp_sce$hto_cluster, tmp_sce$hto_sum, decreasing = c(FALSE, TRUE))],
  color = viridisLite::inferno(100),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  annotation_col = data.frame(
    hto_cluster = sce$hto_cluster,
    sum = log10(sce$hto_sum),
    low = ifelse(sce$hto_sum < hto_low_counts_cutoff, "Low", "High"),
    row.names = colnames(sce)),
  annotation_row = data.frame(ambient = log10(ambient)),
  annotation_colors = list(
    # NOTE: Select 'active' colour levels.
    hto_cluster = hto_cluster_colours[
      levels(factor(tmp_sce$hto_cluster))],
    sum = viridisLite::viridis(101),
    low = c("Low" = "red", "High" = "white"),
    ambient = viridisLite::viridis(101)),
  show_colnames = FALSE,
  scale = "none",
  fontsize = 6)
```

On the basis of Figure \@ref(fig:mostly-doublets-hto-heatmap), we assign droplets in these clusters accordingly:

```{r}
mostly_doublet_clusters <- as.character(17:26)
mostly_doublet_cluster_to_hto_tbl <- tibble::tribble(
    ~hto_cluster,  ~HTO,
    "17",          "doublet",
    "18",          "doublet",
    "19",          "doublet",
    "20",          "doublet",
    "21",          "doublet",
    "22",          "doublet",
    "23",          "doublet",
    "24",          "doublet",
    "25",          "doublet",
    "26",          "human_1-human_2-human_3-human_4-human_5")
stopifnot(
  identical(mostly_doublet_clusters,
            mostly_doublet_cluster_to_hto_tbl$hto_cluster))
    
left_join(mostly_doublet_cluster_to_hto_tbl, hto_to_sample_df) %>%
  knitr::kable(., caption = "Clusters containing doublets (and a cluster.")
```

We have now assigned all droplets to a sample of origin, or labelled them as ambiguous or a doublet, on the basis of the clustering.
The following table summarises this assignment of clusters.

```{r}
sample_map <- left_join(
  rbind(
    singlet_cluster_to_hto_tbl,
    ambiguous_cluster_to_hto_tbl, 
    mostly_doublet_cluster_to_hto_tbl),
  hto_to_sample_df) %>%
  mutate(
    hto_cluster = as.integer(hto_cluster),
    Sample = factor(
      case_when(
        HTO == "ambiguous" ~ "ambiguous",
        HTO == "doublet" ~ "doublet",
        TRUE ~ Sample),
      levels = c(
        paste0("infected_", 1:3), 
        paste0("uninfected_", 1:3),
        "doublet",
        "ambiguous"))) %>%
  arrange(hto_cluster)
knitr::kable(sample_map, caption = "HTO cluster assignment to samples.")

# NOTE: Drop old 'Sample' column (which is useless).
sce$Sample <- NULL
# Add new 'Sample' column to colData.
# NOTE: This is a clunky join that is necessary because colData(sce) contains 
#       `TRA` and `TRB` which are SplitDataFrameList objects and not easily 
#       coercible to columns of a dataframe (if at all).
colData(sce) <- cbind(
  colData(sce),
  sample_map[
    match(sce$hto_cluster, sample_map$hto_cluster),
    !colnames(sample_map) %in% "hto_cluster"])
```

Finally, Figure \@ref(fig:hto-heatmap-by-sample) is a heatmap of the counts for each HTO in each droplet ordered by sample.

```{r hto-heatmap-by-sample, fig.cap = "Heatmap of HTO log2(counts + 1) for the droplets. Columns are ordered by sample and then HTO library size. Each droplet (column) is annotated by assigned sample of origin (`Sample`), the (log10) HTO library size (`sum`), and whether that is smaller than the a cutoff estimated from the empty droplets. Each HTO (row) is annotated by the (log10) average count of the HTO in the empty droplets (`ambient`)."}
sample_colours <- setNames(
  palette.colors(n = nlevels(sce$Sample)),
  levels(sce$Sample))
sce$sample_colours <- sample_colours[sce$Sample]

pheatmap(
  log2(counts(altExp(sce)) + 1)[, order(sce$Sample, sce$hto_sum, decreasing = c(FALSE, TRUE))],
  color = viridisLite::inferno(100),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  annotation_col = data.frame(
    Sample = sce$Sample,
    sum = log10(sce$hto_sum),
    low = ifelse(sce$hto_sum < hto_low_counts_cutoff, "Low", "High"),
    row.names = colnames(sce)),
  annotation_row = data.frame(ambient = log10(ambient)),
  annotation_colors = list(
    Sample = sample_colours,
    sum = viridisLite::viridis(101),
    low = c("Low" = "red", "High" = "white"),
    ambient = viridisLite::viridis(101)),
  show_colnames = FALSE,
  scale = "none",
  fontsize = 6)
```

## Summary

James was targeting ~2000 cells / sample.
The table below shows that we largely hit that target for the uninfected samples but not so for the infected samples.
It also shows that the proportion of droplets that are `ambiguous` is larger than we would like.
These `ambiguous` droplets include singlets (and doublets) with insufficient HTO sequencing coverage, mis-assigned singlets from the cells labelled with all five HTOs, and samples comprising ambient RNAs and HTOs (aka the 'soup' [@young10soupx]).

```{r}
knitr::kable(
  as.data.frame(colData(sce)[, c("Sample", "Treatment", "Replicate")]) %>%
    dplyr::count(Sample, Treatment, Replicate),
  caption = "Tabulation of samples by treatment and replicate (mouse). The samples are unpaired.")
```

# Concluding remarks

```{r}
saveRDS(
  sce,
  here("data", "SCEs", "C057_Cooney.demultiplexed.SCE.rds"),
  compress = "xz")
```

The processed *SingleCellExperiment* object is available (see [`data/SCEs/C057_Cooney.demultiplexed.SCE.rds`](../data/SCEs/C057_Cooney.demultiplexed.SCE.rds)).
This will be used in downstream analyses, e.g., pre-processing.

# Additional information {.appendix}

The following are available on request:

- Full CSV tables of any data presented.
- PDF/PNG files of any static plots.

# Session info {.appendix}

<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>

```{r}
sessioninfo::session_info()
```

</details>
