---
title: "Analysis of Cooney (C057) memory CD4+ T-cell data set"
description: |
author:
  - name: Peter Hickey 
    url: https://peterhickey.org
    affiliation: Single Cell Open Research Endeavour (SCORE), WEHI
    affiliation_url: https://www.wehi.edu.au/people/shalin-naik/3310/score
date: "`r Sys.Date()`"
output: distill::distill_article
editor_options: 
  chunk_output_type: console
bibliography: ref.bib
---

```{r setup}
library(BiocStyle)
library(here)

source(here("code", "helper_functions.R"))
```

# Overview

This study uses 10x Genomics single cell V(D)J + 5′ gene expression to study human T-lymphotropic virus 1 (HTLV1).
HTLV1 infects CD4+ T-cells (like HIV).
Unlike HIV, HTLV1 doesn't kill cells but causes proliferation which ultimately can lead to adult T-cell leukemia. 
HTLV1 is endemic to central Australia (I.e. Aboriginal communities); there, it doesn’t often lead to leukeamia because people die to young, but does lead to bronchial disease and weakened immune cells. 

This study uses a humanised mouse model of infection.
James takes CD34+ human stem cells and injects these into NOD scid gamma (NSG) recipient mice.
He then adds HTLV1 infected CD4+ T-cells. 
The mouse model is far more aggressive than human the human disease because the NSG mice don't have a functional immune system.

This experiment used 6 samples (3 infected, 3 uninfected).
Each sample is a from a different mouse and all infected mice have been infected with virus from a single (human) donor.
With each sample, FACS was performed to enrich for memory CD4+ T-cells.
Cells from each sample were barcoded using the 'Cell Hashing' technique using a hashtag oligonucleotide (HTO), then pooled and run on a single 10X run.

After sequencing, expression was quantified by aligning, filtering, barcode counting, and counting the number of UMIs mapped to each gene using [**CellRanger**](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger).
[Version `3.1.0` of the **CellRanger** reference files](https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build#GRCh38mm10_3.1.0) was used, which maps against GRCh38 of the reference genome and quantifies expression using gene models from Ensembl release 93.
Count data for all endogeneous genes, as well as the HTOs and the VDJ regions, are available in this repository.

This analysis is adapted from the `r BiocStyle::Biocpkg("simpleSingleCell")` workflow as well as the [**Orchestrating Single-Cell Analysis with Bioconductor book**](https://osca.bioconductor.org).

# Analysis version information  {.appendix}

**R version**: `r R.version.string`

**Bioconductor version**: `r BiocManager::version()`
